{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## CHM in other contexts"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Cell state transition"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#### Data process and CHM calling"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "ExecuteTime": {
     "start_time": "2022-05-29T01:52:35.383Z"
    },
    "code_folding": [
     0,
     14,
     22,
     24,
     36,
     48,
     59,
     76,
     79,
     92,
     108,
     111,
     140,
     170,
     173,
     182,
     192,
     205,
     222,
     225,
     233,
     247,
     256,
     267,
     278,
     292,
     350,
     354,
     372,
     379,
     406,
     415,
     426,
     438,
     462,
     466,
     493,
     506,
     532,
     534,
     542,
     558,
     561,
     579,
     586,
     610,
     619,
     631,
     643,
     661,
     664,
     690,
     700
    ],
    "scrolled": true
   },
   "outputs": [],
   "source": [
    "%%bash\n",
    "dirPATH=/mnt/Storage/home/yanghui/imprinting/result.2021/Exploring\n",
    "pyPATH=/mnt/Storage/home/yanghui/scripts/python\n",
    "shPATH=/mnt/Storage/home/yanghui/scripts/shell\n",
    "binPATH=/mnt/Storage/home/yanghui/scripts/bin\n",
    "rPATH=/mnt/Storage/home/yanghui/scripts/R\n",
    "anPATH=/mnt/Storage/home/yanghui/annotations\n",
    "dataPATH_retinal=${dirPATH}/CHMsInOtherContexts/CellStateTransition/RetinalDevelopment/GSE87064_Mouse\n",
    "dataPATH_PAD=/mnt/Storage/home/yanghui/data.public/sat4C/GSE65618_deLaat_GR/processed\n",
    "dataPATH_potentialEnhancer=/mnt/Storage/home/yuzhaowei/projects/imprinting/data_mm10/public/DC_ATAC/mouse_ATAC.bed\n",
    "\n",
    "mkdir -p ${dirPATH}/CHMsInOtherContexts/CellStateTransition;cd ${dirPATH}/CHMsInOtherContexts/CellStateTransition\n",
    "# collect WGBS and H3K9me3 ChIP-seq data during cell state transition\n",
    "\n",
    "collectData(){\n",
    "    # D: Sep-23-2021 15:48 Thu\n",
    "    mkdir -p ${dirPATH}/CHMsInOtherContexts/CellStateTransition/DataCollection;cd ${dirPATH}/CHMsInOtherContexts/CellStateTransition/DataCollection\n",
    "#     python ${pyPATH}/getGEOtext.py gds_result_Mouse_WGBS_20210923.txt gds_result_Mouse_WGBS_20210923 # executed in 10m\n",
    "    python ${pyPATH}/getGEOtext.py gds_result_Mouse_H3K9me3_ChIPSeq_20210924.txt gds_result_Mouse_H3K9me3_ChIPSeq # executed in 20m 7s, finished 09:41:51 2021-09-24\n",
    "}\n",
    "# collectData\n",
    "\n",
    "retinal_development(){\n",
    "    # D: Sep-28-2021 22:09 Tue\n",
    "    WGBS(){\n",
    "        # bash ${shPATH}/WGBS_processing_publicData.sh ${dirPATH}/CHMsInOtherContexts/CellStateTransition/RetinalDevelopment/GSE87064_Mouse GEO GSM2320284 PAIRED E14.5.WGBS.rep1 mm10\n",
    "        bash ${shPATH}/WGBS_processing_publicData.sh ${dirPATH}/CHMsInOtherContexts/CellStateTransition/RetinalDevelopment/GSE87064_Mouse GEO GSM2320285 PAIRED E17.5.WGBS.rep1 mm10\n",
    "        # bash ${shPATH}/WGBS_processing_publicData.sh ${dirPATH}/CHMsInOtherContexts/CellStateTransition/RetinalDevelopment/GSE87064_Mouse GEO GSM2320286 PAIRED P0.WGBS.rep1 mm10\n",
    "        # bash ${shPATH}/WGBS_processing_publicData.sh ${dirPATH}/CHMsInOtherContexts/CellStateTransition/RetinalDevelopment/GSE87064_Mouse GEO GSM2320287 PAIRED P3.WGBS.rep1 mm10\n",
    "        # bash ${shPATH}/WGBS_processing_publicData.sh ${dirPATH}/CHMsInOtherContexts/CellStateTransition/RetinalDevelopment/GSE87064_Mouse GEO GSM2320288 PAIRED P7.WGBS.rep1 mm10\n",
    "        # bash ${shPATH}/WGBS_processing_publicData.sh ${dirPATH}/CHMsInOtherContexts/CellStateTransition/RetinalDevelopment/GSE87064_Mouse GEO GSM2320289 PAIRED P10.WGBS.rep1 mm10\n",
    "        # bash ${shPATH}/WGBS_processing_publicData.sh ${dirPATH}/CHMsInOtherContexts/CellStateTransition/RetinalDevelopment/GSE87064_Mouse GEO GSM2320290 PAIRED P14.WGBS.rep1 mm10\n",
    "        bash ${shPATH}/WGBS_processing_publicData.sh ${dirPATH}/CHMsInOtherContexts/CellStateTransition/RetinalDevelopment/GSE87064_Mouse GEO GSM2320291 PAIRED P21.WGBS.rep1 mm10\n",
    "    }\n",
    "    # WGBS\n",
    "    \n",
    "    ChIPseq(){\n",
    "        bash ${shPATH}/ChIPseq_processing_publicData.sh ${dirPATH}/CHMsInOtherContexts/CellStateTransition/RetinalDevelopment/GSE87064_Mouse GEO GSM2319491 SINGLE E14.5.H3K9me3.rep1 mm10\n",
    "        bash ${shPATH}/ChIPseq_processing_publicData.sh ${dirPATH}/CHMsInOtherContexts/CellStateTransition/RetinalDevelopment/GSE87064_Mouse GEO GSM2319492 SINGLE E17.5.H3K9me3.rep1 mm10\n",
    "        bash ${shPATH}/ChIPseq_processing_publicData.sh ${dirPATH}/CHMsInOtherContexts/CellStateTransition/RetinalDevelopment/GSE87064_Mouse GEO GSM2319493 SINGLE P0.H3K9me3.rep1 mm10\n",
    "        bash ${shPATH}/ChIPseq_processing_publicData.sh ${dirPATH}/CHMsInOtherContexts/CellStateTransition/RetinalDevelopment/GSE87064_Mouse GEO GSM2319494 SINGLE P3.H3K9me3.rep1 mm10\n",
    "        bash ${shPATH}/ChIPseq_processing_publicData.sh ${dirPATH}/CHMsInOtherContexts/CellStateTransition/RetinalDevelopment/GSE87064_Mouse GEO GSM2319495 SINGLE P7.H3K9me3.rep1 mm10\n",
    "        bash ${shPATH}/ChIPseq_processing_publicData.sh ${dirPATH}/CHMsInOtherContexts/CellStateTransition/RetinalDevelopment/GSE87064_Mouse GEO GSM2319496 SINGLE P10.H3K9me3.rep1 mm10\n",
    "        bash ${shPATH}/ChIPseq_processing_publicData.sh ${dirPATH}/CHMsInOtherContexts/CellStateTransition/RetinalDevelopment/GSE87064_Mouse GEO GSM2319497 SINGLE P14.H3K9me3.rep1 mm10\n",
    "        bash ${shPATH}/ChIPseq_processing_publicData.sh ${dirPATH}/CHMsInOtherContexts/CellStateTransition/RetinalDevelopment/GSE87064_Mouse GEO GSM2319498 SINGLE P21.H3K9me3.rep1 mm10        \n",
    "    }\n",
    "    # ChIPseq\n",
    "    \n",
    "    rmDup(){\n",
    "        # D: Oct-09-2021 09:29 Sat\n",
    "        for sample in E14.5 E17.5 P0 P3 P7 P10 P14 P21\n",
    "        do\n",
    "            cd ${dirPATH}/CHMsInOtherContexts/CellStateTransition/RetinalDevelopment/GSE87064_Mouse/${sample}.H3K9me3.rep1 && \\\n",
    "            samtools rmdup ${sample}.H3K9me3.rep1.bam ${sample}.H3K9me3.rep1.rmDup.bam &\n",
    "        done # for sample end\n",
    "        wait\n",
    "    }\n",
    "    # rmDup\n",
    "    \n",
    "    callCHM(){\n",
    "        # D: Oct-09-2021 09:58 Sat\n",
    "        mkdir -p ${dirPATH}/CHMsInOtherContexts/CellStateTransition/RetinalDevelopment/GSE87064_Mouse/CHMs;cd ${dirPATH}/CHMsInOtherContexts/CellStateTransition/RetinalDevelopment/GSE87064_Mouse/CHMs\n",
    "        for sample in E14.5 E17.5 P0 P3 P7 P10 P14 P21\n",
    "        do\n",
    "            pcar -m callchm \\\n",
    "            -H ${dirPATH}/CHMsInOtherContexts/CellStateTransition/RetinalDevelopment/GSE87064_Mouse/${sample}.H3K9me3.rep1/${sample}.H3K9me3.rep1.rmDup.bam \\\n",
    "            -M ${dirPATH}/CHMsInOtherContexts/CellStateTransition/RetinalDevelopment/GSE87064_Mouse/${sample}.WGBS.rep1/${sample}.WGBS.rep1.sam.G.bed \\\n",
    "            -Z ${anPATH}/mm10/mm10_euch.chrom.sizes \\\n",
    "            -Q ${anPATH}/mm10/mm10.2bit \\\n",
    "            -B 200 \\\n",
    "            -N ${sample}_200bp \\\n",
    "            -T 8\n",
    "        done # for sample end\n",
    "    }\n",
    "    # callCHM\n",
    "    \n",
    "    prepareSignal(){\n",
    "        # D: Feb-17-2022 15:16 Thu\n",
    "        mkdir -p ${dirPATH}/CHMsInOtherContexts/CellStateTransition/RetinalDevelopment/GSE87064_Mouse/Merged;cd ${dirPATH}/CHMsInOtherContexts/CellStateTransition/RetinalDevelopment/GSE87064_Mouse/Merged\n",
    "        K9(){\n",
    "            for sample in E14.5 E17.5 P0 P3 P7 P10 P14 P21\n",
    "            do\n",
    "                # ln -s ${dirPATH}/CHMsInOtherContexts/CellStateTransition/RetinalDevelopment/GSE87064_Mouse/${sample}.H3K9me3.rep1/${sample}.H3K9me3.rep1.rmDup.bw ${sample}.H3K9me3.rmDup.bw\n",
    "                # D: Apr-15-2022 12:21 Fri\n",
    "                # macs2 callpeak -g mm -n ${sample}.H3K9me3.rmDup -B -q 0.05 --nomodel --shift=73 --SPMR --broad -t ${dirPATH}/CHMsInOtherContexts/CellStateTransition/RetinalDevelopment/GSE87064_Mouse/${sample}.H3K9me3.rep1/${sample}.H3K9me3.rep1.rmDup.bam && \\\n",
    "                intersectBed -a ${sample}.H3K9me3.rmDup_treat_pileup.bdg -b ${anPATH}/mm10/mm10_main.chrom.limits -wa -f 1.00 | sort -k1,1 -k2,2n > ${sample}.H3K9me3.rmDup_treat_pileup.bdg.tmp && \\\n",
    "                bedGraphToBigWig ${sample}.H3K9me3.rmDup_treat_pileup.bdg.tmp ${anPATH}/mm10/mm10_main.chrom.sizes ${sample}.H3K9me3.rmDup.bw && \\\n",
    "                rm ${sample}.H3K9me3.rmDup_treat_pileup.bdg.tmp &\n",
    "            done # for sample end\n",
    "        }\n",
    "        K9\n",
    "        \n",
    "        methyl(){\n",
    "            for sample in E14.5 E17.5 P0 P3 P7 P10 P14 P21\n",
    "            do\n",
    "                # ln -s ${dirPATH}/CHMsInOtherContexts/CellStateTransition/RetinalDevelopment/GSE87064_Mouse/${sample}.WGBS.rep1/${sample}.WGBS.rep1.sam.G.bed ${sample}.methyl.sam.G.bed && \\\n",
    "                # grep -v \"#\" ${sample}.methyl.sam.G.bed | cut -f 1-4 | sort -k1,1 -k2,2n > ${sample}.methyl.sam.G.bed.tmp && \\\n",
    "                \n",
    "                bedGraphToBigWig ${sample}.methyl.sam.G.bed.tmp ${anPATH}/mm10/mm10.chrom.sizes ${sample}.methyl.all.bw && \\\n",
    "                rm ${sample}.methyl.sam.G.bed.tmp &\n",
    "            done # for sample end\n",
    "        }\n",
    "        # methyl\n",
    "    }\n",
    "    prepareSignal\n",
    "}\n",
    "# retinal_development\n",
    "\n",
    "dc_development_need_to_be_modified(){\n",
    "    # D: Nov-09-2021 14:46 Tue\n",
    "    # D: Oct-13-2022 18:28 Thu (Need to be modified as the TWGBS data of CDP, cDC, pDC are in DNMT1 deficiency situation, not health condition)\n",
    "    DNAMethylation(){\n",
    "        bash ${shPATH}/WGBS_processing_publicData.sh ${dirPATH}/CHMsInOtherContexts/CellStateTransition/DendriticCellDevelopment GEO GSM2522282 PAIRED HSC.WGBS.rep1 mm10 # GSE95697\n",
    "        bash ${shPATH}/WGBS_processing_publicData.sh ${dirPATH}/CHMsInOtherContexts/CellStateTransition/DendriticCellDevelopment GEO GSM2373810 SINGLE MPP.WGBS.rep1 mm10 # GSE89492\n",
    "        bash ${shPATH}/WGBS_processing_publicData.sh ${dirPATH}/CHMsInOtherContexts/CellStateTransition/DendriticCellDevelopment GEO GSM2373811 SINGLE MPP.WGBS.rep2 mm10 # GSE89492\n",
    "        bash ${shPATH}/WGBS_processing_publicData.sh ${dirPATH}/CHMsInOtherContexts/CellStateTransition/DendriticCellDevelopment GEO GSM4997070 PAIRED CDP.TWGBS.rep1 mm10 # GSE164124\n",
    "        bash ${shPATH}/WGBS_processing_publicData.sh ${dirPATH}/CHMsInOtherContexts/CellStateTransition/DendriticCellDevelopment GEO GSM4997071 PAIRED CDP.TWGBS.rep2 mm10 # GSE164124\n",
    "        bash ${shPATH}/WGBS_processing_publicData.sh ${dirPATH}/CHMsInOtherContexts/CellStateTransition/DendriticCellDevelopment GEO GSM4997072 PAIRED CDP.TWGBS.rep3 mm10 # GSE164124\n",
    "        bash ${shPATH}/WGBS_processing_publicData.sh ${dirPATH}/CHMsInOtherContexts/CellStateTransition/DendriticCellDevelopment GEO GSM4997073 PAIRED CDP.TWGBS.rep4 mm10 # GSE164124\n",
    "        bash ${shPATH}/WGBS_processing_publicData.sh ${dirPATH}/CHMsInOtherContexts/CellStateTransition/DendriticCellDevelopment GEO GSM4997074 PAIRED pDC.TWGBS.rep1 mm10 # GSE164124\n",
    "        bash ${shPATH}/WGBS_processing_publicData.sh ${dirPATH}/CHMsInOtherContexts/CellStateTransition/DendriticCellDevelopment GEO GSM4997075 PAIRED pDC.TWGBS.rep2 mm10 # GSE164124\n",
    "        bash ${shPATH}/WGBS_processing_publicData.sh ${dirPATH}/CHMsInOtherContexts/CellStateTransition/DendriticCellDevelopment GEO GSM4997076 PAIRED pDC.TWGBS.rep3 mm10 # GSE164124\n",
    "        bash ${shPATH}/WGBS_processing_publicData.sh ${dirPATH}/CHMsInOtherContexts/CellStateTransition/DendriticCellDevelopment GEO GSM4997077 PAIRED cDC_CD11b.TWGBS.rep1 mm10 # GSE164124\n",
    "        bash ${shPATH}/WGBS_processing_publicData.sh ${dirPATH}/CHMsInOtherContexts/CellStateTransition/DendriticCellDevelopment GEO GSM4997078 PAIRED cDC_CD11b.TWGBS.rep2 mm10 # GSE164124\n",
    "        bash ${shPATH}/WGBS_processing_publicData.sh ${dirPATH}/CHMsInOtherContexts/CellStateTransition/DendriticCellDevelopment GEO GSM4997079 PAIRED cDC_CD11b.TWGBS.rep3 mm10 # GSE164124\n",
    "        bash ${shPATH}/WGBS_processing_publicData.sh ${dirPATH}/CHMsInOtherContexts/CellStateTransition/DendriticCellDevelopment GEO GSM4997080 PAIRED cDC_CD8a.TWGBS.rep1 mm10 # GSE164124\n",
    "        bash ${shPATH}/WGBS_processing_publicData.sh ${dirPATH}/CHMsInOtherContexts/CellStateTransition/DendriticCellDevelopment GEO GSM4997081 PAIRED cDC_CD8a.TWGBS.rep2 mm10 # GSE164124\n",
    "        bash ${shPATH}/WGBS_processing_publicData.sh ${dirPATH}/CHMsInOtherContexts/CellStateTransition/DendriticCellDevelopment GEO GSM4997082 PAIRED cDC_CD8a.TWGBS.rep3 mm10 # GSE164124\n",
    "    }\n",
    "    # DNAMethylation\n",
    "    \n",
    "    ChIPseq(){\n",
    "        bash ${shPATH}/ChIPseq_processing_publicData.sh ${dirPATH}/CHMsInOtherContexts/CellStateTransition/DendriticCellDevelopment GEO GSM3701318 SINGLE HSC.H3K9me3.rep1 mm10 # GSE129157\n",
    "        bash ${shPATH}/ChIPseq_processing_publicData.sh ${dirPATH}/CHMsInOtherContexts/CellStateTransition/DendriticCellDevelopment GEO GSM1384943 SINGLE MPP.H3K9me3.rep1 mm10 # GSE64767\n",
    "        bash ${shPATH}/ChIPseq_processing_publicData.sh ${dirPATH}/CHMsInOtherContexts/CellStateTransition/DendriticCellDevelopment GEO GSM1384944 SINGLE CDP.H3K9me3.rep1 mm10 # D: Jan-19-2022 18:48 Wed\n",
    "        bash ${shPATH}/ChIPseq_processing_publicData.sh ${dirPATH}/CHMsInOtherContexts/CellStateTransition/DendriticCellDevelopment GEO GSM1384945 SINGLE cDC.H3K9me3.rep1 mm10 # GSE64767\n",
    "        bash ${shPATH}/ChIPseq_processing_publicData.sh ${dirPATH}/CHMsInOtherContexts/CellStateTransition/DendriticCellDevelopment GEO GSM1384946 SINGLE pDC.H3K9me3.rep1 mm10 # GSE64767\n",
    "    }\n",
    "    # ChIPseq\n",
    "    \n",
    "    sampleCor(){\n",
    "        # D: Jan-12-2021 10:12abs Wed\n",
    "        # R: Relative good correlation\n",
    "        mkdir -p ${dirPATH}/CHMsInOtherContexts/CellStateTransition/DendriticCellDevelopment/SampleCor;cd ${dirPATH}/CHMsInOtherContexts/CellStateTransition/DendriticCellDevelopment/SampleCor\n",
    "        # python ${pyPATH}/methyl_processing.py \\\n",
    "            \"unit\" \\\n",
    "            \"${dirPATH}/CHMsInOtherContexts/CellStateTransition/DendriticCellDevelopment/HSC.WGBS.rep1/HSC.WGBS.rep1.sam.G.bed\" \\\n",
    "            \"${dirPATH}/CHMsInOtherContexts/CellStateTransition/DendriticCellDevelopment/MPP.WGBS.rep1/MPP.WGBS.rep1.sam.G.bed\" \\\n",
    "            \"${dirPATH}/CHMsInOtherContexts/CellStateTransition/DendriticCellDevelopment/MPP.WGBS.rep2/MPP.WGBS.rep2.sam.G.bed\" \\\n",
    "            \"${dirPATH}/CHMsInOtherContexts/CellStateTransition/DendriticCellDevelopment/CDP.TWGBS.rep1/CDP.TWGBS.rep1.sam.G.bed\" \\\n",
    "            \"${dirPATH}/CHMsInOtherContexts/CellStateTransition/DendriticCellDevelopment/CDP.TWGBS.rep2/CDP.TWGBS.rep2.sam.G.bed\" \\\n",
    "            \"${dirPATH}/CHMsInOtherContexts/CellStateTransition/DendriticCellDevelopment/CDP.TWGBS.rep3/CDP.TWGBS.rep3.sam.G.bed\" \\\n",
    "            \"${dirPATH}/CHMsInOtherContexts/CellStateTransition/DendriticCellDevelopment/CDP.TWGBS.rep4/CDP.TWGBS.rep4.sam.G.bed\" \\\n",
    "            \"${dirPATH}/CHMsInOtherContexts/CellStateTransition/DendriticCellDevelopment/pDC.TWGBS.rep1/pDC.TWGBS.rep1.sam.G.bed\" \\\n",
    "            \"${dirPATH}/CHMsInOtherContexts/CellStateTransition/DendriticCellDevelopment/pDC.TWGBS.rep2/pDC.TWGBS.rep2.sam.G.bed\" \\\n",
    "            \"${dirPATH}/CHMsInOtherContexts/CellStateTransition/DendriticCellDevelopment/pDC.TWGBS.rep3/pDC.TWGBS.rep3.sam.G.bed\" \\\n",
    "            \"${dirPATH}/CHMsInOtherContexts/CellStateTransition/DendriticCellDevelopment/cDC_CD8a.TWGBS.rep1/cDC_CD8a.TWGBS.rep1.sam.G.bed\" \\\n",
    "            \"${dirPATH}/CHMsInOtherContexts/CellStateTransition/DendriticCellDevelopment/cDC_CD8a.TWGBS.rep2/cDC_CD8a.TWGBS.rep2.sam.G.bed\" \\\n",
    "            \"${dirPATH}/CHMsInOtherContexts/CellStateTransition/DendriticCellDevelopment/cDC_CD8a.TWGBS.rep3/cDC_CD8a.TWGBS.rep3.sam.G.bed\" \\\n",
    "            \"${dirPATH}/CHMsInOtherContexts/CellStateTransition/DendriticCellDevelopment/cDC_CD11b.TWGBS.rep1/cDC_CD11b.TWGBS.rep1.sam.G.bed\" \\\n",
    "            \"${dirPATH}/CHMsInOtherContexts/CellStateTransition/DendriticCellDevelopment/cDC_CD11b.TWGBS.rep2/cDC_CD11b.TWGBS.rep2.sam.G.bed\" \\\n",
    "            \"${dirPATH}/CHMsInOtherContexts/CellStateTransition/DendriticCellDevelopment/cDC_CD11b.TWGBS.rep3/cDC_CD11b.TWGBS.rep3.sam.G.bed\"\n",
    "            \n",
    "        # grep -v \"NA\" all.merged.values.txt > rmNA.mergedged.values.txt\n",
    "        (head -1 rmNA.merged.values.txt && tail -n +2 rmNA.merged.values.txt | grep -w \"^chr[0-9]\\{1,2\\}\") > rmNA_euch.mergedged.values.txt\n",
    "            \n",
    "        # Delete the line with exception in cDC_CD8a.TWGBS.rep2.sam.G.bed : hr4\t70396622\t70396624\t0.839\t31chr4\t70396328\t70396330\t1\t37\t37\tB\tG\t+\t18\t18\t-\t19\t19\tACCGA\n",
    "    }\n",
    "    # sampleCor\n",
    "    \n",
    "    prepareBeforeCallCHM(){\n",
    "        # D: Jan-19-2022 14:48 Wed\n",
    "        mkdir -p ${dirPATH}/CHMsInOtherContexts/CellStateTransition/DendriticCellDevelopment/PreparedBeforeCallCHM;cd ${dirPATH}/CHMsInOtherContexts/CellStateTransition/DendriticCellDevelopment/PreparedBeforeCallCHM\n",
    "        mergeSamGbed(){\n",
    "            ln -s ${dirPATH}/CHMsInOtherContexts/CellStateTransition/DendriticCellDevelopment/HSC.WGBS.rep1/HSC.WGBS.rep1.sam.G.bed HSC.WGBS.sam.G.bed\n",
    "            cat ${dirPATH}/CHMsInOtherContexts/CellStateTransition/DendriticCellDevelopment/*/MPP*sam.G.bed | grep -v \"#chrom\" > MPP.WGBS.sam.G.bed\n",
    "            cat ${dirPATH}/CHMsInOtherContexts/CellStateTransition/DendriticCellDevelopment/*/CDP*sam.G.bed | grep -v \"#chrom\" > CDP.WGBS.sam.G.bed\n",
    "            cat ${dirPATH}/CHMsInOtherContexts/CellStateTransition/DendriticCellDevelopment/*/cDC*sam.G.bed | grep -v \"#chrom\" > cDC.WGBS.sam.G.bed\n",
    "            cat ${dirPATH}/CHMsInOtherContexts/CellStateTransition/DendriticCellDevelopment/*/pDC*sam.G.bed | grep -v \"#chrom\" > pDC.WGBS.sam.G.bed\n",
    "        }\n",
    "        # mergeSamGbed\n",
    "        \n",
    "        rmDup(){\n",
    "            # for sample in HSC MPP CDP cDC pDC\n",
    "            for sample in CDP # D: Jan-19-2022 18:07 Wed\n",
    "            do\n",
    "                samtools rmdup ${dirPATH}/CHMsInOtherContexts/CellStateTransition/DendriticCellDevelopment/${sample}.H3K9me3.rep1/${sample}.H3K9me3.rep1.bam ${sample}.H3K9me3.rep1.rmDup.bam &\n",
    "            done # for sample end\n",
    "            wait\n",
    "        }\n",
    "        # rmDup\n",
    "        \n",
    "        generateBW(){\n",
    "            # D: Jan-20-2022 19:31 Thu\n",
    "            for sample in HSC MPP CDP cDC pDC\n",
    "            do\n",
    "                macs2 callpeak -g mm -n ${sample}.H3K9me3.rep1.rmDup -B -q 0.05 --nomodel --shift=73 --SPMR -t ${sample}.H3K9me3.rep1.rmDup.bam && \\\n",
    "                intersectBed -a ${sample}.H3K9me3.rep1.rmDup_treat_pileup.bdg -b ${anPATH}/mm10/mm10_main.chrom.limits -wa -f 1.00 | sort -k1,1 -k2,2n > ${sample}.H3K9me3.rep1.rmDup_treat_pileup.bdg.tmp && \\\n",
    "                bedGraphToBigWig ${sample}.H3K9me3.rep1.rmDup_treat_pileup.bdg.tmp ${anPATH}/mm10/mm10_main.chrom.sizes ${sample}.H3K9me3.rep1.rmDup.bw &\n",
    "            done # for sample end\n",
    "        }\n",
    "        generateBW\n",
    "    }\n",
    "    # prepareBeforeCallCHM\n",
    "    \n",
    "    callCHM(){\n",
    "        mkdir -p ${dirPATH}/CHMsInOtherContexts/CellStateTransition/DendriticCellDevelopment/CHMs;cd ${dirPATH}/CHMsInOtherContexts/CellStateTransition/DendriticCellDevelopment/CHMs\n",
    "        # for sample in HSC MPP CDP cDC pDC\n",
    "        for sample in CDP cDC pDC\n",
    "        do\n",
    "            pcar -m callchm \\\n",
    "                -H ${dirPATH}/CHMsInOtherContexts/CellStateTransition/DendriticCellDevelopment/PreparedBeforeCallCHM/${sample}.H3K9me3.rep1.rmDup.bam \\\n",
    "                -M ${dirPATH}/CHMsInOtherContexts/CellStateTransition/DendriticCellDevelopment/PreparedBeforeCallCHM/${sample}.WGBS.sam.G.bed \\\n",
    "                -Z ${anPATH}/mm10/mm10_euch.chrom.sizes \\\n",
    "                -Q ${anPATH}/mm10/mm10.2bit \\\n",
    "                -B 200 \\\n",
    "                -N ${sample}_200bp \\\n",
    "                -T 6\n",
    "        done # for sample end\n",
    "    }\n",
    "    # callCHM\n",
    "    \n",
    "    prepareSignal(){\n",
    "        # D: Feb-17-2022 15:12 Thu\n",
    "        mkdir -p ${dirPATH}/CHMsInOtherContexts/CellStateTransition/DendriticCellDevelopment/Merged;cd ${dirPATH}/CHMsInOtherContexts/CellStateTransition/DendriticCellDevelopment/Merged\n",
    "        K9(){\n",
    "            for sample in HSC MPP CDP cDC pDC\n",
    "            do\n",
    "                ln -s ${dirPATH}/CHMsInOtherContexts/CellStateTransition/DendriticCellDevelopment/PreparedBeforeCallCHM/${sample}.H3K9me3.rep1.rmDup.bw ${sample}.H3K9me3.rmDup.bw\n",
    "            done # for sample end\n",
    "        }\n",
    "        # K9\n",
    "        \n",
    "        methyl(){\n",
    "            for sample in HSC MPP CDP cDC pDC\n",
    "            do\n",
    "                # R: Error as multiple lines as replicates => separately now\n",
    "                # ln -s ${dirPATH}/CHMsInOtherContexts/CellStateTransition/DendriticCellDevelopment/PreparedBeforeCallCHM/${sample}.WGBS.sam.G.bed ${sample}.methyl.sam.G.bed\n",
    "                # grep -v \"#\" ${sample}.methyl.sam.G.bed | cut -f 1-4 | sort -k1,1 -k2,2n > ${sample}.methyl.sam.G.bed.tmp && \\\n",
    "                bedGraphToBigWig ${sample}.methyl.sam.G.bed.tmp ${anPATH}/mm10/mm10.chrom.sizes ${sample}.methyl.all.bw && \\\n",
    "                rm ${sample}.methyl.sam.G.bed.tmp & \n",
    "            done # for sample end\n",
    "        }\n",
    "        # methyl\n",
    "    }\n",
    "    # prepareSignal\n",
    "    \n",
    "    RNASeq(){\n",
    "        # D: Apr-21-2022 00:19 Thu\n",
    "        # Data resource: GSE73143\n",
    "        # D: May-29-2022 09:42 Sun\n",
    "        bash ${shPATH}/RNAseq_processing_publicData.sh ${dirPATH}/CHMsInOtherContexts/CellStateTransition/DendriticCellDevelopment GEO GSM1887728 PAIRED MPP.RNASeq.rep1 mm10\n",
    "        # bash ${shPATH}/RNAseq_processing_publicData.sh ${dirPATH}/CHMsInOtherContexts/CellStateTransition/DendriticCellDevelopment GEO GSM1887727 PAIRED CDP.RNASeq.rep1 mm10\n",
    "        # bash ${shPATH}/RNAseq_processing_publicData.sh ${dirPATH}/CHMsInOtherContexts/CellStateTransition/DendriticCellDevelopment GEO GSM1887729 PAIRED cDC.RNASeq.rep1 mm10\n",
    "        # bash ${shPATH}/RNAseq_processing_publicData.sh ${dirPATH}/CHMsInOtherContexts/CellStateTransition/DendriticCellDevelopment GEO GSM1887730 PAIRED pDC.RNASeq.rep1 mm10\n",
    "        \n",
    "        re_download(){\n",
    "            # D: May-27-2022 10:06 Fri\n",
    "            cd ${dirPATH}/CHMsInOtherContexts/CellStateTransition/DendriticCellDevelopment/MPP.RNASeq.rep1\n",
    "            wget -c \"ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR243/008/SRR2435588/SRR2435588_1.fastq.gz\" -O MPP.RNASeq.rep1_R1.fq.gz\n",
    "        }\n",
    "        # re_download\n",
    "    }\n",
    "    RNASeq\n",
    "}\n",
    "# dc_development\n",
    "\n",
    "PGC_development(){\n",
    "    # D: Jan-12-2022 09:29 Wed\n",
    "    # D: Jan-16-2022 10:31 Sun\n",
    "    WGBS(){\n",
    "        bash ${shPATH}/WGBS_processing_publicData.sh ${dirPATH}/CHMsInOtherContexts/CellStateTransition/PGCsDevelopment GEO GSM4578139 SINGLE E9.5.WGBS.rep1 mm10 # GSE151446\n",
    "        # bash ${shPATH}/WGBS_processing_publicData.sh ${dirPATH}/CHMsInOtherContexts/CellStateTransition/PGCsDevelopment GEO GSM2041731 PAIRED E10.5.WGBS.rep1 mm10 # GSE76973\n",
    "        # bash ${shPATH}/WGBS_processing_publicData.sh ${dirPATH}/CHMsInOtherContexts/CellStateTransition/PGCsDevelopment GEO GSM1386026 PAIRED E13.5_female.WGBS.rep1 mm10 # GSE56697\n",
    "        # bash ${shPATH}/WGBS_processing_publicData.sh ${dirPATH}/CHMsInOtherContexts/CellStateTransition/PGCsDevelopment GEO GSM1386027 PAIRED E13.5_male.WGBS.rep1 mm10 # GSE56697\n",
    "    }\n",
    "    # WGBS\n",
    "    \n",
    "    ChIPseq(){\n",
    "        bash ${shPATH}/ChIPseq_processing_publicData.sh ${dirPATH}/CHMsInOtherContexts/CellStateTransition/PGCsDevelopment GEO GSM5230780 PAIRED E9.5.H3K9me3.rep1 mm10 # GSE171695\n",
    "        bash ${shPATH}/ChIPseq_processing_publicData.sh ${dirPATH}/CHMsInOtherContexts/CellStateTransition/PGCsDevelopment GEO GSM4196612 PAIRED E10.5.H3K9me3.rep1 mm10 # GSE141182\n",
    "        bash ${shPATH}/ChIPseq_processing_publicData.sh ${dirPATH}/CHMsInOtherContexts/CellStateTransition/PGCsDevelopment GEO GSM4196616 PAIRED E10.5.H3K9me3.rep2 mm10 # GSE141182\n",
    "        bash ${shPATH}/ChIPseq_processing_publicData.sh ${dirPATH}/CHMsInOtherContexts/CellStateTransition/PGCsDevelopment GEO GSM4977595 PAIRED E10.5.H3K9me3.rep3 mm10 # GSE141182\n",
    "        bash ${shPATH}/ChIPseq_processing_publicData.sh ${dirPATH}/CHMsInOtherContexts/CellStateTransition/PGCsDevelopment GEO GSM4196620 PAIRED E13.5_female.H3K9me3.rep1 mm10 # GSE141182\n",
    "        bash ${shPATH}/ChIPseq_processing_publicData.sh ${dirPATH}/CHMsInOtherContexts/CellStateTransition/PGCsDevelopment GEO GSM4196626 PAIRED E13.5_female.H3K9me3.rep2 mm10 # GSE141182\n",
    "        bash ${shPATH}/ChIPseq_processing_publicData.sh ${dirPATH}/CHMsInOtherContexts/CellStateTransition/PGCsDevelopment GEO GSM4977597 PAIRED E13.5_female.H3K9me3.rep3 mm10 # GSE141182\n",
    "        bash ${shPATH}/ChIPseq_processing_publicData.sh ${dirPATH}/CHMsInOtherContexts/CellStateTransition/PGCsDevelopment GEO GSM4196623 PAIRED E13.5_male.H3K9me3.rep1 mm10 # GSE141182\n",
    "        bash ${shPATH}/ChIPseq_processing_publicData.sh ${dirPATH}/CHMsInOtherContexts/CellStateTransition/PGCsDevelopment GEO GSM4196629 PAIRED E13.5_male.H3K9me3.rep2 mm10 # GSE141182\n",
    "        bash ${shPATH}/ChIPseq_processing_publicData.sh ${dirPATH}/CHMsInOtherContexts/CellStateTransition/PGCsDevelopment GEO GSM4977599 PAIRED E13.5_male.H3K9me3.rep3 mm10 # GSE141182\n",
    "    }\n",
    "    # ChIPseq\n",
    "\n",
    "    redownload(){\n",
    "        # D: Jan-15-2021 15:27 Sat\n",
    "        cd ${dirPATH}/CHMsInOtherContexts/CellStateTransition/PGCsDevelopment/E13.5_male.WGBS.rep1\n",
    "        wget -c ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR128/006/SRR1286816/SRR1286816_1.fastq.gz -O E13.5_male.WGBS.rep1_part1_R1.fq.gz\n",
    "        wget -c ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR128/006/SRR1286816/SRR1286816_2.fastq.gz -O E13.5_male.WGBS.rep1_part1_R2.fq.gz\n",
    "        wget -c ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR128/007/SRR1286817/SRR1286817_1.fastq.gz -O E13.5_male.WGBS.rep1_part2_R1.fq.gz\n",
    "        wget -c ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR128/007/SRR1286817/SRR1286817_2.fastq.gz -O E13.5_male.WGBS.rep1_part2_R2.fq.gz\n",
    "        wget -c ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR128/008/SRR1286818/SRR1286818_1.fastq.gz -O E13.5_male.WGBS.rep1_part3_R1.fq.gz\n",
    "        wget -c ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR128/008/SRR1286818/SRR1286818_2.fastq.gz -O E13.5_male.WGBS.rep1_part3_R2.fq.gz\n",
    "        wget -c ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR128/009/SRR1286819/SRR1286819_1.fastq.gz -O E13.5_male.WGBS.rep1_part4_R1.fq.gz\n",
    "        wget -c ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR128/009/SRR1286819/SRR1286819_2.fastq.gz -O E13.5_male.WGBS.rep1_part4_R2.fq.gz\n",
    "        wget -c ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR128/000/SRR1286820/SRR1286820_1.fastq.gz -O E13.5_male.WGBS.rep1_part5_R1.fq.gz\n",
    "        wget -c ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR128/000/SRR1286820/SRR1286820_2.fastq.gz -O E13.5_male.WGBS.rep1_part5_R2.fq.gz\n",
    "        wget -c ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR128/001/SRR1286821/SRR1286821_1.fastq.gz -O E13.5_male.WGBS.rep1_part6_R1.fq.gz\n",
    "        wget -c ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR128/001/SRR1286821/SRR1286821_2.fastq.gz -O E13.5_male.WGBS.rep1_part6_R2.fq.gz\n",
    "        \n",
    "        cd ${dirPATH}/CHMsInOtherContexts/CellStateTransition/PGCsDevelopment/E9.5.H3K9me3.rep1\n",
    "        wget -c ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR141/040/SRR14182440/SRR14182440_1.fastq.gz -O E9.5.H3K9me3.rep1_R1.fq.gz\n",
    "        wget -c ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR141/040/SRR14182440/SRR14182440_2.fastq.gz -O E9.5.H3K9me3.rep1_R2.fq.gz\n",
    "\n",
    "        cd ${dirPATH}/CHMsInOtherContexts/CellStateTransition/PGCsDevelopment/E10.5.H3K9me3.rep1\n",
    "        wget -c ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR105/000/SRR10560100/SRR10560100_1.fastq.gz -O E10.5.H3K9me3.rep1_R1.fq.gz\n",
    "        wget -c ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR105/000/SRR10560100/SRR10560100_2.fastq.gz -O E10.5.H3K9me3.rep1_R2.fq.gz\n",
    "\n",
    "        cd ${dirPATH}/CHMsInOtherContexts/CellStateTransition/PGCsDevelopment/E10.5.H3K9me3.rep2\n",
    "        wget -c ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR105/004/SRR10560104/SRR10560104_1.fastq.gz -O E10.5.H3K9me3.rep2_R1.fq.gz\n",
    "        wget -c ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR105/004/SRR10560104/SRR10560104_2.fastq.gz -O E10.5.H3K9me3.rep2_R2.fq.gz\n",
    "\n",
    "        cd ${dirPATH}/CHMsInOtherContexts/CellStateTransition/PGCsDevelopment/E10.5.H3K9me3.rep3\n",
    "        wget -c ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR132/072/SRR13296472/SRR13296472_1.fastq.gz -O E10.5.H3K9me3.rep3_R1.fq.gz\n",
    "        wget -c ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR132/072/SRR13296472/SRR13296472_2.fastq.gz -O E10.5.H3K9me3.rep3_R2.fq.gz\n",
    "\n",
    "        cd ${dirPATH}/CHMsInOtherContexts/CellStateTransition/PGCsDevelopment/E13.5_female.H3K9me3.rep1\n",
    "        wget -c ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR105/008/SRR10560108/SRR10560108_1.fastq.gz -O E13.5_female.H3K9me3.rep1_R1.fq.gz\n",
    "        wget -c ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR105/008/SRR10560108/SRR10560108_2.fastq.gz -O E13.5_female.H3K9me3.rep1_R2.fq.gz\n",
    "\n",
    "        cd ${dirPATH}/CHMsInOtherContexts/CellStateTransition/PGCsDevelopment/E13.5_female.H3K9me3.rep2\n",
    "        wget -c ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR105/014/SRR10560114/SRR10560114_1.fastq.gz -O E13.5_female.H3K9me3.rep2_R1.fq.gz\n",
    "        wget -c ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR105/014/SRR10560114/SRR10560114_2.fastq.gz -O E13.5_female.H3K9me3.rep2_R2.fq.gz\n",
    "\n",
    "        cd ${dirPATH}/CHMsInOtherContexts/CellStateTransition/PGCsDevelopment/E13.5_female.H3K9me3.rep3\n",
    "        wget -c ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR132/074/SRR13296474/SRR13296474_1.fastq.gz -O E13.5_female.H3K9me3.rep3_R1.fq.gz\n",
    "        wget -c ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR132/074/SRR13296474/SRR13296474_2.fastq.gz -O E13.5_female.H3K9me3.rep3_R2.fq.gz\n",
    "\n",
    "        cd ${dirPATH}/CHMsInOtherContexts/CellStateTransition/PGCsDevelopment/E13.5_male.H3K9me3.rep1\n",
    "        wget -c ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR105/011/SRR10560111/SRR10560111_1.fastq.gz -O E13.5_male.H3K9me3.rep1_R1.fq.gz\n",
    "        wget -c ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR105/011/SRR10560111/SRR10560111_2.fastq.gz -O E13.5_male.H3K9me3.rep1_R2.fq.gz\n",
    "\n",
    "        cd ${dirPATH}/CHMsInOtherContexts/CellStateTransition/PGCsDevelopment/E13.5_male.H3K9me3.rep2\n",
    "        wget -c ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR105/017/SRR10560117/SRR10560117_1.fastq.gz -O E13.5_male.H3K9me3.rep2_R1.fq.gz\n",
    "        wget -c ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR105/017/SRR10560117/SRR10560117_2.fastq.gz -O E13.5_male.H3K9me3.rep2_R2.fq.gz\n",
    "\n",
    "        cd ${dirPATH}/CHMsInOtherContexts/CellStateTransition/PGCsDevelopment/E13.5_male.H3K9me3.rep3\n",
    "        wget -c ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR132/076/SRR13296476/SRR13296476_1.fastq.gz -O E13.5_male.H3K9me3.rep3_R1.fq.gz\n",
    "        wget -c ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR132/076/SRR13296476/SRR13296476_2.fastq.gz -O E13.5_male.H3K9me3.rep3_R2.fq.gz\n",
    "    }\n",
    "    # redownload\n",
    "    \n",
    "    sampleCor_K9(){\n",
    "        # D: Feb-14-2022 17:07 Mon\n",
    "        # T: executed in 9m 32s, finished 17:30:08 2022-02-14\n",
    "        mkdir -p ${dirPATH}/CHMsInOtherContexts/CellStateTransition/PGCsDevelopment/SampleCor;cd ${dirPATH}/CHMsInOtherContexts/CellStateTransition/PGCsDevelopment/SampleCor\n",
    "        macs2_peak_bw(){\n",
    "            for sample in E9.5 E10.5 E13.5_female E13.5_male\n",
    "            do\n",
    "                for rep in rep1 rep2 rep3\n",
    "                do\n",
    "                    if [ -d ${dirPATH}/CHMsInOtherContexts/CellStateTransition/PGCsDevelopment/${sample}.H3K9me3.${rep} ]\n",
    "                    then\n",
    "                        cd ${dirPATH}/CHMsInOtherContexts/CellStateTransition/PGCsDevelopment/${sample}.H3K9me3.${rep} && \\\n",
    "                        macs2 callpeak -g mm -n ${sample}.H3K9me3.${rep} -B -q 0.05 --nomodel --shift=73 --broad --SPMR -t ${sample}.H3K9me3.${rep}.bam && \\\n",
    "                        intersectBed -a ${sample}.H3K9me3.${rep}_treat_pileup.bdg -b ${anPATH}/mm10/mm10_main.chrom.limits -wa -f 1.00 | sort -k1,1 -k2,2n > ${sample}.H3K9me3.${rep}_treat_pileup.bdg.tmp && \\\n",
    "                        bedGraphToBigWig ${sample}.H3K9me3.${rep}_treat_pileup.bdg.tmp ${anPATH}/mm10/mm10_main.chrom.sizes ${sample}.H3K9me3.${rep}.bw && \\\n",
    "                        cut -f 1-3 ${sample}.H3K9me3.${rep}_peaks.broadPeak > ${sample}.H3K9me3.${rep}.macs2_peaks.bed &\n",
    "                    fi\n",
    "                done # for rep end\n",
    "            done # for sample end\n",
    "        }\n",
    "        # macs2_peak_bw\n",
    "        \n",
    "        mergePeak(){\n",
    "            # D: Feb-14-2022 17:32 Mon\n",
    "            # T: executed in 563ms, finished 17:34:24 2022-02-14\n",
    "            cat ../*H3K9me3*/*.macs2_peaks.bed | sort -k1,1 -k2,2n | mergeBed -i - > mergedPeaks.bed # 34,704\n",
    "        }\n",
    "        # mergePeak\n",
    "        \n",
    "        signal(){\n",
    "            # D: Feb-14-2022 17:35 Mon\n",
    "            # T: executed in 49.8s, finished 17:42:15 2022-02-14\n",
    "            echo -ne \"#Chrom\\tStart\\tEnd\" > title.txt\n",
    "            cat mergedPeaks.bed > mergedPeaks.H3K9me3.txt\n",
    "            for sample in E9.5 E10.5 E13.5_female E13.5_male\n",
    "            do\n",
    "                for rep in rep1 rep2 rep3\n",
    "                do\n",
    "                    if [ -d ${dirPATH}/CHMsInOtherContexts/CellStateTransition/PGCsDevelopment/${sample}.H3K9me3.${rep} ]\n",
    "                    then\n",
    "                        echo -ne \"\\t${sample}.${rep}\" >> title.txt\n",
    "                        bash ${shPATH}/averageSignalInRegionPrimaryOrder.sh \\\n",
    "                            mergedPeaks.bed \\\n",
    "                            ${dirPATH}/CHMsInOtherContexts/CellStateTransition/PGCsDevelopment/${sample}.H3K9me3.${rep}/${sample}.H3K9me3.${rep}.bw \\\n",
    "                            1 \\\n",
    "                            mergedPeaks.H3K9me3_${sample}.${rep}.txt\n",
    "                            \n",
    "                        paste mergedPeaks.H3K9me3.txt mergedPeaks.H3K9me3_${sample}.${rep}.txt > tmp && mv tmp mergedPeaks.H3K9me3.txt\n",
    "                    fi\n",
    "                done # for rep end\n",
    "            done # for sample end\n",
    "            echo -ne \"\\n\" >> title.txt\n",
    "            cat title.txt mergedPeaks.H3K9me3.txt > tmp && mv tmp mergedPeaks.H3K9me3.txt\n",
    "        }\n",
    "        # signal\n",
    "        \n",
    "        sampleCor(){\n",
    "            # D: Feb-14-2022 18:00 Mon\n",
    "            # R: Though relative less correlated as plus rep3, but still use merge one as in paper\n",
    "            Rscript sampleCor.r\n",
    "        }\n",
    "        # sampleCor\n",
    "    }\n",
    "    # sampleCor_K9\n",
    "    \n",
    "    mergeReps_K9(){\n",
    "        # D: Dec-14-2022 20:27 Mon\n",
    "        mkdir -p ${dirPATH}/CHMsInOtherContexts/CellStateTransition/PGCsDevelopment/Merged;cd ${dirPATH}/CHMsInOtherContexts/CellStateTransition/PGCsDevelopment/Merged\n",
    "        samtools view -H ${dirPATH}/CHMsInOtherContexts/CellStateTransition/PGCsDevelopment/E9.5.H3K9me3.rep1/E9.5.H3K9me3.rep1.bam > header.sam\n",
    "        for sample in E9.5 E10.5 E13.5_female E13.5_male\n",
    "        do\n",
    "            samtools merge -@ 30 -h header.sam ${sample}.H3K9me3.bam ${dirPATH}/CHMsInOtherContexts/CellStateTransition/PGCsDevelopment/${sample}.H3K9me3.*/${sample}.H3K9me3.*bam\n",
    "        done # for sample end\n",
    "    }\n",
    "    # mergeReps_K9\n",
    "    \n",
    "    rmDup_K9(){\n",
    "        # D: Feb-14-2022 20:34 Mon\n",
    "        # T: executed in 8m 14s, finished 20:51:14 2022-02-14\n",
    "        mkdir -p ${dirPATH}/CHMsInOtherContexts/CellStateTransition/PGCsDevelopment/Merged;cd ${dirPATH}/CHMsInOtherContexts/CellStateTransition/PGCsDevelopment/Merged\n",
    "        for sample in E9.5 E10.5 E13.5_female E13.5_male\n",
    "        do\n",
    "            samtools rmdup ${sample}.H3K9me3.bam ${sample}.H3K9me3.rmDup.bam &\n",
    "        done # for sample end\n",
    "        wait\n",
    "    }\n",
    "    # rmDup_K9\n",
    "    \n",
    "    callCHM(){\n",
    "        # D: Feb-14-2022 20:43 Mon\n",
    "        # T: executed in 39m 15s, finished 21:47:09 2022-02-14\n",
    "        # R: Number of CHMs varied significantly: \n",
    "        # R: 2,949 E9.5\n",
    "        # R: 11,638 E10.5\n",
    "        # R: 6,168 E13.5_female\n",
    "        # R: 6,603 E13.5_male\n",
    "        \n",
    "        mkdir -p ${dirPATH}/CHMsInOtherContexts/CellStateTransition/PGCsDevelopment/CHMs;cd ${dirPATH}/CHMsInOtherContexts/CellStateTransition/PGCsDevelopment/CHMs\n",
    "        for sample in E9.5 E10.5 E13.5_female E13.5_male\n",
    "        do\n",
    "            pcar -m callchm \\\n",
    "                -H ${dirPATH}/CHMsInOtherContexts/CellStateTransition/PGCsDevelopment/Merged/${sample}.H3K9me3.rmDup.bam \\\n",
    "                -M ${dirPATH}/CHMsInOtherContexts/CellStateTransition/PGCsDevelopment/${sample}.WGBS.rep1/${sample}.WGBS.rep1.sam.G.bed \\\n",
    "                -Z ${anPATH}/mm10/mm10_euch.chrom.sizes \\\n",
    "                -Q ${anPATH}/mm10/mm10.2bit \\\n",
    "                -B 200 \\\n",
    "                -N ${sample}_200bp \\\n",
    "                -T 6\n",
    "        done # for sample end\n",
    "    }\n",
    "    # callCHM\n",
    "    \n",
    "    generateBW(){\n",
    "        # D: Feb-14-2022 21:08 Mon\n",
    "        # T: executed in 16m 21s, finished 22:08:55 2022-02-14\n",
    "        mkdir -p ${dirPATH}/CHMsInOtherContexts/CellStateTransition/PGCsDevelopment/Merged;cd ${dirPATH}/CHMsInOtherContexts/CellStateTransition/PGCsDevelopment/Merged\n",
    "        K9(){\n",
    "            for sample in E9.5 E10.5 E13.5_female E13.5_male\n",
    "            do\n",
    "                macs2 callpeak -g mm -n ${sample}.H3K9me3.rmDup -B -q 0.05 --nomodel --shift=73 --SPMR --broad -t ${sample}.H3K9me3.rmDup.bam && \\\n",
    "                intersectBed -a ${sample}.H3K9me3.rmDup_treat_pileup.bdg -b ${anPATH}/mm10/mm10_main.chrom.limits -wa -f 1.00 | sort -k1,1 -k2,2n > ${sample}.H3K9me3.rmDup_treat_pileup.bdg.tmp && \\\n",
    "                bedGraphToBigWig ${sample}.H3K9me3.rmDup_treat_pileup.bdg.tmp ${anPATH}/mm10/mm10_main.chrom.sizes ${sample}.H3K9me3.rmDup.bw && \\\n",
    "                rm ${sample}.H3K9me3.rmDup_treat_pileup.bdg.tmp &\n",
    "            done # for sample end\n",
    "            wait\n",
    "        }\n",
    "        # K9\n",
    "        \n",
    "        methyl(){\n",
    "            for sample in E9.5 E10.5 E13.5_female E13.5_male\n",
    "            do\n",
    "                # # D: Feb-17-2022 15:09 Thu\n",
    "                # ln -s ${dirPATH}/CHMsInOtherContexts/CellStateTransition/PGCsDevelopment/${sample}.WGBS.rep1/${sample}.WGBS.rep1.sam.G.bed ${sample}.methyl.sam.G.bed && \\\n",
    "                # grep -v '#' ${sample}.methyl.sam.G.bed | cut -f 1-4 | sort -k1,1 -k2,2n > ${sample}.methyl.sam.G.bed.tmp && \\\n",
    "                bedGraphToBigWig ${sample}.methyl.sam.G.bed.tmp ${anPATH}/mm10/mm10.chrom.sizes ${sample}.methyl.all.bw && \\\n",
    "                rm ${sample}.methyl.sam.G.bed.tmp &\n",
    "            done # for sample end\n",
    "            wait\n",
    "        }\n",
    "        methyl\n",
    "    }\n",
    "    # generateBW\n",
    "    \n",
    "    RNASeq(){\n",
    "        # D: Apr-21-2022 00:37 Thu\n",
    "        # Data source: GSE19960\n",
    "        # bash ${shPATH}/RNAseq_processing_publicData.sh ${dirPATH}/CHMsInOtherContexts/CellStateTransition/PGCsDevelopment GEO GSM2041483 PAIRED E10.5.RNASeq.rep1 mm10\n",
    "        # bash ${shPATH}/RNAseq_processing_publicData.sh ${dirPATH}/CHMsInOtherContexts/CellStateTransition/PGCsDevelopment GEO GSM2041484 PAIRED E10.5.RNASeq.rep2 mm10\n",
    "        # bash ${shPATH}/RNAseq_processing_publicData.sh ${dirPATH}/CHMsInOtherContexts/CellStateTransition/PGCsDevelopment GEO GSM4196635 PAIRED E13.5_female.RNASeq.rep1 mm10\n",
    "        # bash ${shPATH}/RNAseq_processing_publicData.sh ${dirPATH}/CHMsInOtherContexts/CellStateTransition/PGCsDevelopment GEO GSM4196636 PAIRED E13.5_female.RNASeq.rep2 mm10\n",
    "        # bash ${shPATH}/RNAseq_processing_publicData.sh ${dirPATH}/CHMsInOtherContexts/CellStateTransition/PGCsDevelopment GEO GSM4196637 PAIRED E13.5_female.RNASeq.rep3 mm10\n",
    "        # bash ${shPATH}/RNAseq_processing_publicData.sh ${dirPATH}/CHMsInOtherContexts/CellStateTransition/PGCsDevelopment GEO GSM4196638 PAIRED E13.5_male.RNASeq.rep1 mm10\n",
    "        bash ${shPATH}/RNAseq_processing_publicData.sh ${dirPATH}/CHMsInOtherContexts/CellStateTransition/PGCsDevelopment GEO GSM4196639 PAIRED E13.5_male.RNASeq.rep2 mm10\n",
    "        bash ${shPATH}/RNAseq_processing_publicData.sh ${dirPATH}/CHMsInOtherContexts/CellStateTransition/PGCsDevelopment GEO GSM4196640 PAIRED E13.5_male.RNASeq.rep3 mm10\n",
    "        bash ${shPATH}/RNAseq_processing_publicData.sh ${dirPATH}/CHMsInOtherContexts/CellStateTransition/PGCsDevelopment GEO GSM4196641 PAIRED E13.5_male.RNASeq.rep4 mm10\n",
    "        \n",
    "        re_download(){\n",
    "            # D: May-27-2022 10:09 Fri\n",
    "            # E13.5_male.RNASeq.rep2\n",
    "            cd ${dirPATH}/CHMsInOtherContexts/CellStateTransition/PGCsDevelopment/E13.5_male.RNASeq.rep2\n",
    "            wget -c \"ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR105/093/SRR10560093/SRR10560093_1.fastq.gz\" -O E13.5_male.RNASeq.rep2_part2_R1.fq.gz\n",
    "            wget -c \"ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR105/093/SRR10560093/SRR10560093_2.fastq.gz\" -O E13.5_male.RNASeq.rep2_part2_R2.fq.gz\n",
    "            \n",
    "            # E13.5_male.RNASeq.rep3\n",
    "            cd ${dirPATH}/CHMsInOtherContexts/CellStateTransition/PGCsDevelopment/E13.5_male.RNASeq.rep3\n",
    "            wget -c \"ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR105/094/SRR10560094/SRR10560094_1.fastq.gz\" -O E13.5_male.RNASeq.rep3_part1_R1.fq.gz\n",
    "            wget -c \"ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR105/094/SRR10560094/SRR10560094_2.fastq.gz\" -O E13.5_male.RNASeq.rep3_part1_R2.fq.gz\n",
    "            wget -c \"ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR105/095/SRR10560095/SRR10560095_1.fastq.gz\" -O E13.5_male.RNASeq.rep3_part2_R1.fq.gz\n",
    "            wget -c \"ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR105/095/SRR10560095/SRR10560095_2.fastq.gz\" -O E13.5_male.RNASeq.rep3_part2_R2.fq.gz\n",
    "            \n",
    "            # E13.5_male.RNASeq.rep4\n",
    "            cd ${dirPATH}/CHMsInOtherContexts/CellStateTransition/PGCsDevelopment/E13.5_male.RNASeq.rep4\n",
    "            wget -c \"ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR105/096/SRR10560096/SRR10560096_2.fastq.gz\" -O E13.5_male.RNASeq.rep4_part1_R2.fq.gz\n",
    "            wget -c \"ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR105/097/SRR10560097/SRR10560097_1.fastq.gz\" -O E13.5_male.RNASeq.rep4_part2_R1.fq.gz\n",
    "            wget -c \"ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR105/097/SRR10560097/SRR10560097_2.fastq.gz\" -O E13.5_male.RNASeq.rep4_part2_R2.fq.gz\n",
    "        }\n",
    "        # re_download\n",
    "    }\n",
    "    RNASeq\n",
    "}\n",
    "# PGC_development\n",
    "\n",
    "spermatogenesis(){\n",
    "    # D: Jan-20-2021 15:39 Thu\n",
    "    DNAMethylation(){\n",
    "        bash ${shPATH}/WGBS_processing_publicData.sh ${dirPATH}/CHMsInOtherContexts/CellStateTransition/Spermatogenesis_GSE137744 GEO GSM4086441 PAIRED US.WGBS.rep1 mm10\n",
    "        bash ${shPATH}/WGBS_processing_publicData.sh ${dirPATH}/CHMsInOtherContexts/CellStateTransition/Spermatogenesis_GSE137744 GEO GSM4086442 PAIRED DS.WGBS.rep1 mm10\n",
    "        bash ${shPATH}/WGBS_processing_publicData.sh ${dirPATH}/CHMsInOtherContexts/CellStateTransition/Spermatogenesis_GSE137744 GEO GSM4086443 PAIRED PS.WGBS.rep1 mm10\n",
    "        bash ${shPATH}/WGBS_processing_publicData.sh ${dirPATH}/CHMsInOtherContexts/CellStateTransition/Spermatogenesis_GSE137744 GEO GSM4086444 PAIRED RS.WGBS.rep1 mm10\n",
    "    }\n",
    "    # DNAMethylation\n",
    "    \n",
    "    ChIPseq(){\n",
    "        bash ${shPATH}/ChIPseq_processing_publicData.sh ${dirPATH}/CHMsInOtherContexts/CellStateTransition/Spermatogenesis_GSE137744 GEO GSM4086425 PAIRED US.H3K9me3.rep1 mm10\n",
    "        bash ${shPATH}/ChIPseq_processing_publicData.sh ${dirPATH}/CHMsInOtherContexts/CellStateTransition/Spermatogenesis_GSE137744 GEO GSM4086426 PAIRED US.H3K9me3.rep2 mm10\n",
    "        bash ${shPATH}/ChIPseq_processing_publicData.sh ${dirPATH}/CHMsInOtherContexts/CellStateTransition/Spermatogenesis_GSE137744 GEO GSM4086427 PAIRED US.H3K9me3.rep3 mm10\n",
    "        bash ${shPATH}/ChIPseq_processing_publicData.sh ${dirPATH}/CHMsInOtherContexts/CellStateTransition/Spermatogenesis_GSE137744 GEO GSM4086429 PAIRED DS.H3K9me3.rep1 mm10\n",
    "        bash ${shPATH}/ChIPseq_processing_publicData.sh ${dirPATH}/CHMsInOtherContexts/CellStateTransition/Spermatogenesis_GSE137744 GEO GSM4086430 PAIRED DS.H3K9me3.rep2 mm10\n",
    "        bash ${shPATH}/ChIPseq_processing_publicData.sh ${dirPATH}/CHMsInOtherContexts/CellStateTransition/Spermatogenesis_GSE137744 GEO GSM4086431 PAIRED DS.H3K9me3.rep3 mm10\n",
    "        bash ${shPATH}/ChIPseq_processing_publicData.sh ${dirPATH}/CHMsInOtherContexts/CellStateTransition/Spermatogenesis_GSE137744 GEO GSM4086433 PAIRED PS.H3K9me3.rep1 mm10\n",
    "        bash ${shPATH}/ChIPseq_processing_publicData.sh ${dirPATH}/CHMsInOtherContexts/CellStateTransition/Spermatogenesis_GSE137744 GEO GSM4086434 PAIRED PS.H3K9me3.rep2 mm10\n",
    "        bash ${shPATH}/ChIPseq_processing_publicData.sh ${dirPATH}/CHMsInOtherContexts/CellStateTransition/Spermatogenesis_GSE137744 GEO GSM4086435 PAIRED PS.H3K9me3.rep3 mm10\n",
    "        bash ${shPATH}/ChIPseq_processing_publicData.sh ${dirPATH}/CHMsInOtherContexts/CellStateTransition/Spermatogenesis_GSE137744 GEO GSM4086437 PAIRED RS.H3K9me3.rep1 mm10\n",
    "        bash ${shPATH}/ChIPseq_processing_publicData.sh ${dirPATH}/CHMsInOtherContexts/CellStateTransition/Spermatogenesis_GSE137744 GEO GSM4086438 PAIRED RS.H3K9me3.rep2 mm10\n",
    "        bash ${shPATH}/ChIPseq_processing_publicData.sh ${dirPATH}/CHMsInOtherContexts/CellStateTransition/Spermatogenesis_GSE137744 GEO GSM4086439 PAIRED RS.H3K9me3.rep3 mm10\n",
    "    }\n",
    "    # ChIPseq\n",
    "    \n",
    "    sampleCor_K9(){\n",
    "        # D: Feb-16-2022 20:09 Wed\n",
    "        mkdir -p ${dirPATH}/CHMsInOtherContexts/CellStateTransition/Spermatogenesis_GSE137744/SampleCor;cd ${dirPATH}/CHMsInOtherContexts/CellStateTransition/Spermatogenesis_GSE137744/SampleCor\n",
    "        macs2_peak_bw(){\n",
    "            # T: executed in 5m 25s, finished 20:24:53 2022-02-16\n",
    "            for sample in US DS PS RS\n",
    "            do\n",
    "                for rep in rep1 rep2 rep3\n",
    "                do\n",
    "                    cd ${dirPATH}/CHMsInOtherContexts/CellStateTransition/Spermatogenesis_GSE137744/${sample}.H3K9me3.${rep} && \\\n",
    "                    macs2 callpeak -g mm -n ${sample}.H3K9me3.${rep} -B -q 0.05 --nomodel --shift=73 --broad --SPMR -t ${sample}.H3K9me3.${rep}.bam && \\\n",
    "                    intersectBed -a ${sample}.H3K9me3.${rep}_treat_pileup.bdg -b ${anPATH}/mm10/mm10_main.chrom.limits -wa -f 1.00 | sort -k1,1 -k2,2n > ${sample}.H3K9me3.${rep}_treat_pileup.bdg.tmp && \\\n",
    "                    bedGraphToBigWig ${sample}.H3K9me3.${rep}_treat_pileup.bdg.tmp ${anPATH}/mm10/mm10_main.chrom.sizes ${sample}.H3K9me3.${rep}.bw && \\\n",
    "                    cut -f 1-3 ${sample}.H3K9me3.${rep}_peaks.broadPeak > ${sample}.H3K9me3.${rep}.macs2_peaks.bed && \\\n",
    "                    rm ${sample}.H3K9me3.${rep}_treat_pileup.bdg.tmp &\n",
    "                done # for rep end\n",
    "            done # for sample end\n",
    "            wait\n",
    "        }\n",
    "        # macs2_peak_bw\n",
    "        \n",
    "        mergePeak(){\n",
    "            # D: Feb-16-2022 20:23 Wed\n",
    "            # T: executed in 869ms, finished 20:30:50 2022-02-16\n",
    "            cat ../*H3K9me3*/*.macs2_peaks.bed | sort -k1,1 -k2,2n | mergeBed -i - > mergedPeaks.bed # 64,987\n",
    "        }\n",
    "        # mergePeak\n",
    "        \n",
    "        signal(){\n",
    "            # D: Feb-16-2022 20:31 Wed\n",
    "            # T: executed in 1m 48.5s, finished 20:36:32 2022-02-16\n",
    "            echo -ne \"#Chrom\\tStart\\tEnd\" > title.txt\n",
    "            cat mergedPeaks.bed > mergedPeaks.H3K9me3.txt\n",
    "            for sample in US DS PS RS\n",
    "            do\n",
    "                for rep in rep1 rep2 rep3\n",
    "                do\n",
    "                    echo -ne \"\\t${sample}.${rep}\" >> title.txt\n",
    "                    bash ${shPATH}/averageSignalInRegionPrimaryOrder.sh \\\n",
    "                        mergedPeaks.bed \\\n",
    "                        ${dirPATH}/CHMsInOtherContexts/CellStateTransition/Spermatogenesis_GSE137744/${sample}.H3K9me3.${rep}/${sample}.H3K9me3.${rep}.bw \\\n",
    "                        1 \\\n",
    "                        mergedPeaks.H3K9me3_${sample}.${rep}.txt\n",
    "                        \n",
    "                    paste mergedPeaks.H3K9me3.txt mergedPeaks.H3K9me3_${sample}.${rep}.txt > tmp && mv tmp mergedPeaks.H3K9me3.txt\n",
    "                done # for rep end\n",
    "            done # for sample end\n",
    "            echo -ne \"\\n\" >> title.txt\n",
    "            cat title.txt mergedPeaks.H3K9me3.txt > tmp && mv tmp mergedPeaks.H3K9me3.txt\n",
    "        }\n",
    "        # signal\n",
    "        \n",
    "        sampleCor(){\n",
    "            # D: Feb-16-2022 20:43 Wed\n",
    "            # R: Highly correlated (R >= 0.92)\n",
    "            Rscript sampleCor.r\n",
    "        }\n",
    "        # sampleCor\n",
    "    }\n",
    "    # sampleCor_K9\n",
    "    \n",
    "    mergeReps_K9(){\n",
    "        # D: Feb-16-2022 20:48 Wed\n",
    "        T: executed in 9m 41s, finished 21:02:23 2022-02-16\n",
    "        mkdir -p ${dirPATH}/CHMsInOtherContexts/CellStateTransition/Spermatogenesis_GSE137744/Merged;cd ${dirPATH}/CHMsInOtherContexts/CellStateTransition/Spermatogenesis_GSE137744/Merged\n",
    "        samtools view -H ${dirPATH}/CHMsInOtherContexts/CellStateTransition/Spermatogenesis_GSE137744/US.H3K9me3.rep1/US.H3K9me3.rep1.bam > header.sam\n",
    "        for sample in US DS PS RS\n",
    "        do\n",
    "            samtools merge -@ 30 -h header.sam ${sample}.H3K9me3.bam ${dirPATH}/CHMsInOtherContexts/CellStateTransition/Spermatogenesis_GSE137744/${sample}.H3K9me3.*/${sample}.H3K9me3.*bam\n",
    "        done # for sample end\n",
    "    }\n",
    "    # mergeReps_K9\n",
    "    \n",
    "    rmDup_K9(){\n",
    "        # D: Feb-16-2022 21:03 Wed\n",
    "        # T: executed in 16m 54s, finished 21:21:11 2022-02-16\n",
    "        mkdir -p ${dirPATH}/CHMsInOtherContexts/CellStateTransition/Spermatogenesis_GSE137744/Merged;cd ${dirPATH}/CHMsInOtherContexts/CellStateTransition/Spermatogenesis_GSE137744/Merged\n",
    "        for sample in US DS PS RS\n",
    "        do\n",
    "            samtools rmdup ${sample}.H3K9me3.bam ${sample}.H3K9me3.rmDup.bam &\n",
    "        done # for sample end\n",
    "        wait\n",
    "    }\n",
    "    # rmDup_K9\n",
    "    \n",
    "    callCHM(){\n",
    "        # D: Feb-16-2022 21:04 Wed\n",
    "        # T: executed in 43m 35s, finished 22:06:18 2022-02-16\n",
    "        mkdir -p ${dirPATH}/CHMsInOtherContexts/CellStateTransition/Spermatogenesis_GSE137744/CHMs;cd ${dirPATH}/CHMsInOtherContexts/CellStateTransition/Spermatogenesis_GSE137744/CHMs\n",
    "        for sample in US DS PS RS\n",
    "        do\n",
    "            pcar -m callchm \\\n",
    "                -H ${dirPATH}/CHMsInOtherContexts/CellStateTransition/Spermatogenesis_GSE137744/Merged/${sample}.H3K9me3.rmDup.bam \\\n",
    "                -M ${dirPATH}/CHMsInOtherContexts/CellStateTransition/Spermatogenesis_GSE137744/${sample}.WGBS.rep1/${sample}.WGBS.rep1.sam.G.bed \\\n",
    "                -Z ${anPATH}/mm10/mm10_euch.chrom.sizes \\\n",
    "                -Q ${anPATH}/mm10/mm10.2bit \\\n",
    "                -B 200 \\\n",
    "                -N ${sample}_200bp \\\n",
    "                -T 6\n",
    "        done # for sample end\n",
    "    }\n",
    "    # callCHM\n",
    "    \n",
    "    prepareSignal(){\n",
    "        # D: Feb-17-2022 15:00 Thu\n",
    "        mkdir -p ${dirPATH}/CHMsInOtherContexts/CellStateTransition/Spermatogenesis_GSE137744/Merged;cd ${dirPATH}/CHMsInOtherContexts/CellStateTransition/Spermatogenesis_GSE137744/Merged\n",
    "        K9(){\n",
    "            for sample in US DS PS RS\n",
    "            do\n",
    "                macs2 callpeak -g mm -n ${sample}.H3K9me3.rmDup -B -q 0.05 --nomodel --shift=73 --SPMR --broad -t ${sample}.H3K9me3.rmDup.bam && \\\n",
    "                intersectBed -a ${sample}.H3K9me3.rmDup_treat_pileup.bdg -b ${anPATH}/mm10/mm10_main.chrom.limits -wa -f 1.00 | sort -k1,1 -k2,2n > ${sample}.H3K9me3.rmDup_treat_pileup.bdg.tmp && \\\n",
    "                bedGraphToBigWig ${sample}.H3K9me3.rmDup_treat_pileup.bdg.tmp ${anPATH}/mm10/mm10_main.chrom.sizes ${sample}.H3K9me3.rmDup.bw && \\\n",
    "                rm ${sample}.H3K9me3.rmDup_treat_pileup.bdg.tmp &\n",
    "            done # for sample end\n",
    "            wait\n",
    "        }\n",
    "        # K9\n",
    "        \n",
    "        methyl(){\n",
    "            for sample in US DS PS RS\n",
    "            do\n",
    "                # ln -s ${dirPATH}/CHMsInOtherContexts/CellStateTransition/Spermatogenesis_GSE137744/${sample}.WGBS.rep1/${sample}.WGBS.rep1.sam.G.bed ${sample}.methyl.sam.G.bed && \\\n",
    "                # grep -v \"#\" ${sample}.methyl.sam.G.bed | cut -f 1-4 | sort -k1,1 -k2,2n > ${sample}.methyl.sam.G.bed.tmp && \\\n",
    "                bedGraphToBigWig ${sample}.methyl.sam.G.bed.tmp ${anPATH}/mm10/mm10.chrom.sizes ${sample}.methyl.all.bw && \\\n",
    "                rm ${sample}.methyl.sam.G.bed.tmp &\n",
    "            done # for sample end\n",
    "            wait\n",
    "        }\n",
    "        methyl\n",
    "    }\n",
    "    # prepareSignal\n",
    "    \n",
    "    RNASeq(){\n",
    "        # D: Apr-21-2022 00:55 Thu\n",
    "        # Data source: GSE55060\n",
    "        bash ${shPATH}/RNAseq_processing_publicData.sh ${dirPATH}/CHMsInOtherContexts/CellStateTransition/Spermatogenesis_GSE137744 GEO GSM1328920 SINGLE GS.RNASeq.rep1 mm10\n",
    "        bash ${shPATH}/RNAseq_processing_publicData.sh ${dirPATH}/CHMsInOtherContexts/CellStateTransition/Spermatogenesis_GSE137744 GEO GSM1328921 SINGLE GS.RNASeq.rep2 mm10\n",
    "        bash ${shPATH}/RNAseq_processing_publicData.sh ${dirPATH}/CHMsInOtherContexts/CellStateTransition/Spermatogenesis_GSE137744 GEO GSM1328922 SINGLE PS.RNASeq.rep1 mm10\n",
    "        bash ${shPATH}/RNAseq_processing_publicData.sh ${dirPATH}/CHMsInOtherContexts/CellStateTransition/Spermatogenesis_GSE137744 GEO GSM1328923 SINGLE PS.RNASeq.rep2 mm10\n",
    "        bash ${shPATH}/RNAseq_processing_publicData.sh ${dirPATH}/CHMsInOtherContexts/CellStateTransition/Spermatogenesis_GSE137744 GEO GSM1328926 SINGLE RS.RNASeq.rep1 mm10\n",
    "        bash ${shPATH}/RNAseq_processing_publicData.sh ${dirPATH}/CHMsInOtherContexts/CellStateTransition/Spermatogenesis_GSE137744 GEO GSM1328927 SINGLE RS.RNASeq.rep2 mm10\n",
    "        \n",
    "        re_download(){\n",
    "            # D: May-27-2022 10:23 Fri\n",
    "            # GS.RNASeq.rep1\n",
    "            cd ${dirPATH}/CHMsInOtherContexts/CellStateTransition/Spermatogenesis_GSE137744/GS.RNASeq.rep1\n",
    "            wget -c \"ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR200/007/SRR2005757/SRR2005757.fastq.gz\" -O GS.RNASeq.rep1.fq.gz\n",
    "            \n",
    "            # GS.RNASeq.rep2\n",
    "            cd ${dirPATH}/CHMsInOtherContexts/CellStateTransition/Spermatogenesis_GSE137744/GS.RNASeq.rep2\n",
    "            wget -c \"ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR200/008/SRR2005758/SRR2005758.fastq.gz\" -O GS.RNASeq.rep2.fq.gz\n",
    "            \n",
    "            # PS.RNASeq.rep1\n",
    "            cd ${dirPATH}/CHMsInOtherContexts/CellStateTransition/Spermatogenesis_GSE137744/PS.RNASeq.rep1\n",
    "            wget -c \"ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR200/009/SRR2005759/SRR2005759.fastq.gz\" -O PS.RNASeq.rep1.fq.gz\n",
    "            \n",
    "            # PS.RNASeq.rep2\n",
    "            cd ${dirPATH}/CHMsInOtherContexts/CellStateTransition/Spermatogenesis_GSE137744/PS.RNASeq.rep2\n",
    "            wget -c \"ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR200/000/SRR2005760/SRR2005760.fastq.gz\" -O PS.RNASeq.rep2.fq.gz\n",
    "            \n",
    "            # RS.RNASeq.rep1\n",
    "            cd ${dirPATH}/CHMsInOtherContexts/CellStateTransition/Spermatogenesis_GSE137744/RS.RNASeq.rep1\n",
    "            wget -c \"ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR200/003/SRR2005763/SRR2005763.fastq.gz\" -O RS.RNASeq.rep1.fq.gz\n",
    "            \n",
    "            # RS.RNASeq.rep2\n",
    "            cd ${dirPATH}/CHMsInOtherContexts/CellStateTransition/Spermatogenesis_GSE137744/RS.RNASeq.rep2\n",
    "            wget -c \"ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR200/004/SRR2005764/SRR2005764.fastq.gz\" -O RS.RNASeq.rep2.fq.gz\n",
    "        }\n",
    "        # re_download\n",
    "    }\n",
    "    RNASeq\n",
    "}\n",
    "# spermatogenesis"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "heading_collapsed": true
   },
   "source": [
    "#### Correlation between DNA methylation and H3K9me3"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 148,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2022-04-18T10:10:24.597443Z",
     "start_time": "2022-04-18T10:10:23.331930Z"
    },
    "code_folding": [
     7,
     17,
     84
    ],
    "hidden": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "null device \n",
      "          1 \n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Warning message:\n",
      "Removed 1 rows containing missing values (geom_hline). \n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "dirPATH=/mnt/Storage/home/yanghui/imprinting/result.2021/Exploring/CHMsInOtherContexts/CellStateTransition\n",
    "shPATH=/mnt/Storage/home/yanghui/scripts/shell\n",
    "rPATH=/mnt/Storage/home/yanghui/scripts/R\n",
    "anPATH=/mnt/Storage/home/yanghui/annotations\n",
    "\n",
    "# R: 在不同生物学过程中DNA甲基化和H3K9me3在富含CpG的区域相关性更高，但在视网膜发育过程中相关性相对较低\n",
    "prepare(){\n",
    "    # D: Apr-14-2022 15:51 Thu\n",
    "    # T: executed in 9.95s, finished 15:55:37 2022-04-14\n",
    "    mkdir -p ${dirPATH}/Prepare;cd ${dirPATH}/Prepare\n",
    "    awk 'BEGIN{FS=OFS=\"\\t\"}{if($4<10){print $1, $2, $3;}}' ${anPATH}/mm10/Bins/mm10.b1kb.euchr.CpGnumber > CpGs_low.bed # 1,722,250\n",
    "    awk 'BEGIN{FS=OFS=\"\\t\"}{if($4>=10 && $4<30){print $1, $2, $3;}}' ${anPATH}/mm10/Bins/mm10.b1kb.euchr.CpGnumber > CpGs_intermediate.bed # 699,059\n",
    "    awk 'BEGIN{FS=OFS=\"\\t\"}{if($4>=30){print $1, $2, $3;}}' ${anPATH}/mm10/Bins/mm10.b1kb.euchr.CpGnumber > CpGs_high.bed # 41,427\n",
    "}\n",
    "# prepare\n",
    "\n",
    "process_cor(){\n",
    "    # D: Apr-14-2022 16:10 Thu\n",
    "    name=${1} # Spermatogenesis_GSE137744\n",
    "    CHM_STAGE=${2} # \"US DS PS RS\"\n",
    "    CHM_STAGE_plot=${3} # \"US;DS;PS;RS\"\n",
    "    \n",
    "    cd ${dirPATH}/${name}/Merged\n",
    "    echo -e \"xValue\\tyValue\\tgroup\" > cor.ggplot_basic.txt\n",
    "    for group in low intermediate high\n",
    "    do\n",
    "        for stage in ${CHM_STAGE}\n",
    "        do\n",
    "#             # DNA methylation level\n",
    "#             bash ${shPATH}/averageMethylInRegionMultipleThreads.sh \\\n",
    "#                 \"${dirPATH}/Prepare/CpGs_${group}.bed\" \\\n",
    "#                 \"${stage}.methyl.sam.G.bed\" \\\n",
    "#                 \"CpGs_${group}.methyl_${stage}.txt\" \\\n",
    "#                 \"2\"\n",
    "            \n",
    "#             # H3K9me3 signal\n",
    "#             bash ${shPATH}/averageSignalInRegionPrimaryOrder.sh \\\n",
    "#                 \"${dirPATH}/Prepare/CpGs_${group}.bed\" \\\n",
    "#                 \"${stage}.H3K9me3.rmDup.bw\" \\\n",
    "#                 \"1\" \\\n",
    "#                 \"CpGs_${group}.H3K9me3_${stage}.txt\"\n",
    "                \n",
    "            Rscript ${rPATH}/calculateCor_for_ggplot.r \\\n",
    "                \"CpGs_${group}.methyl_${stage}.txt\" \\\n",
    "                \"CpGs_${group}.H3K9me3_${stage}.txt\" \\\n",
    "                \"4;1\" \\\n",
    "                \"F\" \\\n",
    "                \"F\" \\\n",
    "                \"${stage}\" \\\n",
    "                \"${group}\" \\\n",
    "                \"cor.ggplot_basic.txt\"\n",
    "        done # for stage end\n",
    "    done # for group end\n",
    "    \n",
    "    Rscript ${rPATH}/ggplot_barplot.r \\\n",
    "        \"cor.ggplot_basic.txt\" \\\n",
    "        \"cor.ggplot_barplot.pdf\" \\\n",
    "        \"${name}\" \\\n",
    "        \"\" \\\n",
    "        \"Pearson's correlation coefficient__n(DNA methylation and H3K9me3)\" \\\n",
    "        \"-0.2;1;0.2\" \\\n",
    "        \"${CHM_STAGE_plot}\" \\\n",
    "        \"low;intermediate;high\" \\\n",
    "        \"dodge\" \\\n",
    "        \"4\" \\\n",
    "        \"3.5\" \\\n",
    "        \"-007\" \\\n",
    "        \"F\"\n",
    "}\n",
    "\n",
    "# # T: executed in 3h 3m 41s, finished 19:21:20 2022-04-14\n",
    "# process_cor PGCsDevelopment \"E9.5 E10.5 E13.5_female E13.5_male\" \"E9.5;E10.5;E13.5_female;E13.5_male\"\n",
    "# process_cor Spermatogenesis_GSE137744 \"US DS PS RS\" \"US;DS;PS;RS\"\n",
    "# process_cor RetinalDevelopment/GSE87064_Mouse \"E14.5 E17.5 P0 P3 P7 P10 P14 P21\" \"E14.5;E17.5;P0;P3;P7;P10;P14;P21\"\n",
    "# process_cor DendriticCellDevelopment \"HSC MPP CDP cDC pDC\" \"HSC;MPP;CDP;cDC;pDC\"\n",
    "\n",
    "# # D: Apr-18-2022 16:11 Mon\n",
    "# process_cor DendriticCellDevelopment \"CDP cDC pDC\" \"CDP;cDC;pDC\"\n",
    "\n",
    "# # D: Apr-18-2022 18:09 Mon\n",
    "# # T: executed in 43.8s, finished 18:09:55 2022-04-18\n",
    "# process_cor DendriticCellDevelopment \"MPP CDP cDC pDC\" \"MPP;CDP;cDC;pDC\"\n",
    "\n",
    "merge_into_one(){\n",
    "    # D: Apr-14-2022 20:57 Thu\n",
    "    # D: Apr-18-2022 18:06 Mon\n",
    "    mkdir -p ${dirPATH}/Prepare;cd ${dirPATH}/Prepare\n",
    "    echo -e \"xValue\\tyValue\\tgroup\" > cor.ggplot_basic.txt\n",
    "    tail -n +2 ${dirPATH}/PGCsDevelopment/Merged/cor.ggplot_basic.txt >> cor.ggplot_basic.txt\n",
    "    tail -n +2 ${dirPATH}/Spermatogenesis_GSE137744/Merged/cor.ggplot_basic.txt >> cor.ggplot_basic.txt\n",
    "    tail -n +2 ${dirPATH}/RetinalDevelopment/GSE87064_Mouse/Merged/cor.ggplot_basic.txt >> cor.ggplot_basic.txt\n",
    "    tail -n +2 ${dirPATH}/DendriticCellDevelopment/Merged/cor.ggplot_basic.txt >> cor.ggplot_basic.txt\n",
    "    \n",
    "    Rscript ${rPATH}/ggplot_barplot.r \\\n",
    "        \"cor.ggplot_basic.txt\" \\\n",
    "        \"cor.ggplot_barplot.pdf\" \\\n",
    "        \"Biological process\" \\\n",
    "        \"\" \\\n",
    "        \"Pearson's correlation coefficient__n(DNA methylation and H3K9me3)\" \\\n",
    "        \"-0.2;1;0.2\" \\\n",
    "        \"E10.5;E13.5_female;E13.5_male;US;DS;PS;RS;E14.5;E17.5;P0;P3;P7;P10;P14;P21;MPP;CDP;cDC;pDC\" \\\n",
    "        \"low;intermediate;high\" \\\n",
    "        \"dodge\" \\\n",
    "        \"8\" \\\n",
    "        \"3.5\" \\\n",
    "        \"-007\" \\\n",
    "        \"F\"\n",
    "}\n",
    "# merge_into_one"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#### Features of stable CHMs in different processes"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "code_folding": [
     0,
     11
    ]
   },
   "outputs": [],
   "source": [
    "%%bash\n",
    "dirPATH=/mnt/Storage/home/yanghui/imprinting/result.2021/Exploring/CHMsInOtherContexts/CellStateTransition\n",
    "shPATH=/mnt/Storage/home/yanghui/scripts/shell\n",
    "rPATH=/mnt/Storage/home/yanghui/scripts/R\n",
    "dataPATH_CHM_earlyEmbryo=/mnt/Storage/home/yanghui/imprinting/result.2021/Filtering/CpGrichedHighMethylH3K9me3/Hybrid/ChromHMM/Manage\n",
    "anPATH=/mnt/Storage/home/yanghui/annotations\n",
    "\n",
    "feature_exploration(){\n",
    "    # D: Mar-14-2022 22:42 Thu\n",
    "    mkdir -p ${dirPATH}/CHMOrganization;cd ${dirPATH}/CHMOrganization\n",
    "    \n",
    "    overlap(){\n",
    "        # D: Apr-15-2022 22:52 Fri\n",
    "        mkdir -p ${dirPATH}/CHMOrganization/Overlap;cd ${dirPATH}/CHMOrganization/Overlap\n",
    "        prepare(){\n",
    "            mkdir -p ${dirPATH}/CHMOrganization/EarlyEmbryogenesis;cd ${dirPATH}/CHMOrganization/EarlyEmbryogenesis\n",
    "            # ln -s ${dataPATH_CHM_earlyEmbryo}/stable.refined.sorted.filteredBySignal.bed stable.filteredBySignal.bed\n",
    "            ln -s ${dataPATH_CHM_earlyEmbryo}/stageSpecific.refined.sorted.filteredBySignal.bed stageSpecific.filteredBySignal.bed # D: Nov-03-2022 11:33 Thu\n",
    "            ln -s /mnt/Storage/home/yanghui/imprinting/result.2021/RevisionNCB/RoleOfH3K9me3AtCHMs/Hybrid_nonCHM_CGI/nonCHM_CGIs_HMR.bed nonCHM_CGIs_HMR_euch.bed # D: Nov-03-2022 12:46 Thu\n",
    "            \n",
    "            cd ${dirPATH}/CHMOrganization/Overlap\n",
    "            # cat ${dirPATH}/CHMOrganization/*/stable.filteredBySignal.bed | sort -k1,1 -k2,2n | mergeBed -i - -d 2000 > Union.bed # 13,416\n",
    "            for process in EarlyEmbryogenesis FemalePGCDevelopment MalePGCDevelopment Spermatogenesis RetinalDevelopment ClassicalDendriticCellDevelopment PlasmacytoidDendriticCellDevelopment\n",
    "            do\n",
    "                intersectBed -u -a Union.bed -b ${dirPATH}/CHMOrganization/${process}/stable.filteredBySignal.bed | \\\n",
    "                    awk 'BEGIN{FS=\"\\t\";OFS=\"_\"}{print $1, $2, $3;}' > ${process}.bed\n",
    "            done # for process end\n",
    "        }\n",
    "        prepare\n",
    "        \n",
    "        Rscript UpSetR.r\n",
    "    }\n",
    "    # overlap\n",
    "    \n",
    "    \n",
    "    kMeansCluster(){\n",
    "        # D: Apr-16-2022 09:41 Sat\n",
    "        mkdir -p ${dirPATH}/CHMOrganization/KMeansCluster;cd ${dirPATH}/CHMOrganization/KMeansCluster\n",
    "        prepare(){\n",
    "            # T: executed in 466ms, finished 09:54:24 2022-04-16\n",
    "            cat ${dirPATH}/CHMOrganization/*/stable.filteredBySignal.bed | \\\n",
    "                cut -f 1-3 | sort -k1,1 -k2,2n | mergeBed -i - -d 2000 -c 2,3 -o median,median | cut -f 1,4-5 > Union_refined.bed # 13,416\n",
    "            echo -ne \"#Chrom\\tStart\\tEnd\" > title.txt\n",
    "            cat Union_refined.bed > Union_refined.existence.txt\n",
    "            for process in EarlyEmbryogenesis FemalePGCDevelopment MalePGCDevelopment Spermatogenesis RetinalDevelopment ClassicalDendriticCellDevelopment PlasmacytoidDendriticCellDevelopment\n",
    "            do\n",
    "                echo -ne \"\\t${process}\" >> title.txt\n",
    "                intersectBed -c -a Union_refined.bed -b ${dirPATH}/CHMOrganization/${process}/stable.filteredBySignal.bed | \\\n",
    "                    cut -f 4 | awk 'BEGIN{FS=OFS=\"\\t\"}{if($1==0){print 0;} else{print 1;}}' | \\\n",
    "                    paste Union_refined.existence.txt - > tmp && mv tmp Union_refined.existence.txt\n",
    "            done # for process end\n",
    "            echo -ne \"\\n\" >> title.txt\n",
    "            cat title.txt Union_refined.existence.txt > tmp && mv tmp Union_refined.existence.txt\n",
    "        }\n",
    "        # prepare\n",
    "    }\n",
    "    # kMeansCluster\n",
    "}\n",
    "# feature_exploration"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#### Universal and process-specific"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 72,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2022-11-07T06:44:09.390680Z",
     "start_time": "2022-11-07T06:44:00.013837Z"
    },
    "code_folding": [
     15,
     47,
     50,
     63,
     94,
     128,
     159,
     177,
     225,
     415,
     431,
     473,
     486,
     499,
     543,
     590,
     615,
     640,
     787,
     925,
     952,
     964,
     981,
     1000,
     1386,
     1390,
     1399,
     1411,
     1413,
     1554,
     1560,
     1561,
     1565,
     1582,
     1599,
     1618,
     1635,
     1654,
     1690,
     1719,
     1732,
     1764,
     1773,
     1802,
     1822,
     1862,
     1902,
     1928,
     2104,
     2108,
     2120,
     2147,
     2173,
     2289,
     2313,
     2317,
     2322,
     2399,
     2402,
     2426,
     2452,
     2492,
     2496,
     2501,
     2522,
     2525,
     2549,
     2575,
     2599,
     2603,
     2608,
     2655,
     2658,
     2684,
     2705,
     2714,
     2727,
     2754,
     2780,
     2804,
     2808,
     2813,
     2851,
     2878,
     2904,
     2932,
     2937,
     2945,
     2948,
     3004,
     3020,
     3042,
     3135,
     3172
    ]
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "null device \n",
      "          1 \n",
      "null device \n",
      "          1 \n",
      "null device \n",
      "          1 \n",
      "null device \n",
      "          1 \n",
      "null device \n",
      "          1 \n",
      "null device \n",
      "          1 \n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "dirPATH=/mnt/Storage/home/yanghui/imprinting/result.2021/Exploring/CHMsInOtherContexts/CellStateTransition\n",
    "pyPATH=/mnt/Storage/home/yanghui/scripts/python\n",
    "shPATH=/mnt/Storage/home/yanghui/scripts/shell\n",
    "binPATH=/mnt/Storage/home/yanghui/scripts/bin\n",
    "rPATH=/mnt/Storage/home/yanghui/scripts/R\n",
    "anPATH=/mnt/Storage/home/yanghui/annotations\n",
    "dataPATH_methyl_hybrid=/mnt/Storage/home/yanghui/data.public/mm10/WGBS/EarlyEmbryos/sam.G.bed\n",
    "dataPATH_H3K9me3_hybrid=/mnt/Storage/home/yanghui/data.public/mm10/ChIP-seq/H3K9me3_GSE97778_SNPTraceablePreImplantation_Gao_2018_NCB/Processed\n",
    "dataPATH_expr_earlyEmbryo=/mnt/Storage/home/yuzhaowei/projects/imprinting/data_mm10/public/RNA-seq/GSE70605_WCF_RNA-seq\n",
    "dataPATH_pad=/mnt/Storage/home/yanghui/data.public/sat4C/GSE65618_deLaat_GR/processed/ESC.PADs_processed.mm10.bed\n",
    "dataPATH_enhancer=/mnt/Storage/home/yanghui/annotations/mm10/RegionEnrichment/mm10_euch.potentialEnhancer.merged.bed\n",
    "dataPATH_promoter=/mnt/Storage/home/yanghui/annotations/mm10/mm10.promoter.refGene.bed\n",
    "\n",
    "mkdir -p ${dirPATH}/CHMOrganization/Universal_specific;cd ${dirPATH}/CHMOrganization/Universal_specific\n",
    "group(){\n",
    "    # D: Apr-17-2022 16:52 Sun\n",
    "    # # T: executed in 441ms, finished 17:02:04 2022-04-17\n",
    "    # awk 'BEGIN{FS=OFS=\"\\t\"}{if($4==1 && $5==1 && $6==1 && $7==1 && $8==1){print $1, $2, $3;}}' ${dirPATH}/CHMOrganization/KMeansCluster/Union_refined.existence.txt > Universal.bed # 2,424\n",
    "    # awk 'BEGIN{FS=OFS=\"\\t\"}{if($4==1 && $5==0 && $6==0 && $7==0 && $8==0){print $1, $2, $3;}}' ${dirPATH}/CHMOrganization/KMeansCluster/Union_refined.existence.txt > EarlyEmbryoSpecific.bed # 599\n",
    "    # awk 'BEGIN{FS=OFS=\"\\t\"}{if($4==0 && $5==1 && $6==0 && $7==0 && $8==0){print $1, $2, $3;}}' ${dirPATH}/CHMOrganization/KMeansCluster/Union_refined.existence.txt > PGCSpecific.bed # 355\n",
    "    # awk 'BEGIN{FS=OFS=\"\\t\"}{if($4==0 && $5==0 && $6==1 && $7==0 && $8==0){print $1, $2, $3;}}' ${dirPATH}/CHMOrganization/KMeansCluster/Union_refined.existence.txt > SpermSpecific.bed # 744\n",
    "    # awk 'BEGIN{FS=OFS=\"\\t\"}{if($4==0 && $5==0 && $6==0 && $7==1 && $8==0){print $1, $2, $3;}}' ${dirPATH}/CHMOrganization/KMeansCluster/Union_refined.existence.txt > RetinalSpecific.bed # 701\n",
    "    # awk 'BEGIN{FS=OFS=\"\\t\"}{if($4==0 && $5==0 && $6==0 && $7==0 && $8==1){print $1, $2, $3;}}' ${dirPATH}/CHMOrganization/KMeansCluster/Union_refined.existence.txt > DCSpecific.bed # 13,287\n",
    "    \n",
    "    # D: Nov-03-2022 11:27 Thu\n",
    "    # T: executed in 759ms, finished 11:42:14 2022-11-03\n",
    "    # nonCHM_CGIs, potential enhancer, promoter\n",
    "    # cat ${dirPATH}/CHMOrganization/*/*.filteredBySignal.bed | intersectBed -v -a ${anPATH}/mm10/mm10.cpgIslandExtUnmasked.bed -b - | \\\n",
    "    #                 grep -w \"^chr[0-9]\\{1,2\\}\" > nonCHM_CGIs.bed # 13,115 \n",
    "    # ln -s ${anPATH}/mm10/RegionEnrichment/mm10_euch.potentialEnhancer.merged.bed enhancer.bed # 1,124,888\n",
    "    # ln -s ${anPATH}/mm10/mm10.promoter.refGene.bed promoter.bed # 47,382\n",
    "    \n",
    "    # D: Nov-06-2022 20:06 Sun\n",
    "    # LTR, LINE, SINE\n",
    "    ln -s ${anPATH}/mm10/Repeats/mm10.LTR.bed LTR.bed\n",
    "    ln -s ${anPATH}/mm10/Repeats/mm10.LINE.bed LINE.bed\n",
    "    ln -s ${anPATH}/mm10/Repeats/mm10.SINE.bed SINE.bed\n",
    "    \n",
    "    \n",
    "    # D: Nov-03-2022 12:47 Thu\n",
    "    # T: executed in 117ms, finished 12:51:39 2022-11-03\n",
    "    # nonCHM_CGIs_HMR\n",
    "    cat ${dirPATH}/CHMOrganization/*/nonCHM_CGIs_HMR_euch.bed | cut -f 1-3 | sort -k1,1 -k2,2n | mergeBed -i - > nonCHM_CGIs_HMR.bed # 3,250\n",
    "}\n",
    "# group\n",
    "\n",
    "signals(){\n",
    "    # D: Apr-17-2022 17:08 Sun\n",
    "    mkdir -p ${dirPATH}/CHMOrganization/Universal_specific/Signals;cd ${dirPATH}/CHMOrganization/Universal_specific/Signals\n",
    "    prepare(){\n",
    "        mkdir -p ${dirPATH}/EarlyEmbryogenesis/Merged;cd ${dirPATH}/EarlyEmbryogenesis/Merged\n",
    "        ln -s ${dataPATH_methyl_hybrid}/2cell.sam.G.bed 2cell.methyl.sam.G.bed\n",
    "        ln -s ${dataPATH_methyl_hybrid}/8cell.sam.G.bed 8cell.methyl.sam.G.bed\n",
    "        ln -s ${dataPATH_methyl_hybrid}/Morula.sam.G.bed Morula.methyl.sam.G.bed\n",
    "        ln -s ${dataPATH_methyl_hybrid}/ICM.sam.G.bed ICM.methyl.sam.G.bed\n",
    "        ln -s ${dataPATH_H3K9me3_hybrid}/bw.files/2cell.H3K9me3.bw 2cell.H3K9me3.rmDup.bw\n",
    "        ln -s ${dataPATH_H3K9me3_hybrid}/bw.files/8cell.H3K9me3.bw 8cell.H3K9me3.rmDup.bw\n",
    "        ln -s ${dataPATH_H3K9me3_hybrid}/bw.files/Morula.H3K9me3.bw Morula.H3K9me3.rmDup.bw\n",
    "        ln -s ${dataPATH_H3K9me3_hybrid}/bw.files/ICM.H3K9me3.bw ICM.H3K9me3.rmDup.bw\n",
    "    }\n",
    "    # prepare\n",
    "    \n",
    "    order(){\n",
    "        # T: executed in 29ms, finished 17:20:24 2022-04-17\n",
    "        mkdir -p ${dirPATH}/CHMOrganization/Universal_specific;cd ${dirPATH}/CHMOrganization/Universal_specific\n",
    "        # cat Universal.bed EarlyEmbryoSpecific.bed PGCSpecific.bed SpermSpecific.bed RetinalSpecific.bed DCSpecific.bed > ordered.bed # 18,110\n",
    "        \n",
    "        # # executed in 147ms, finished 18:32:19 2022-04-17\n",
    "        # intersectBed -wao -a ordered.bed -b ${dirPATH}/CHMOrganization/KMeansCluster/Union_refined.existence.txt -f 1.00 -r | \\\n",
    "        #     cut -f 4-11 > ordered.existence.txt\n",
    "        \n",
    "        # T: executed in 777ms, finished 18:38:32 2022-04-17\n",
    "        Rscript ${rPATH}/heatmap_mat_1.r \\\n",
    "            \"ordered.existence\" \\\n",
    "            \"3\" \\\n",
    "            \"5\" \\\n",
    "            \"5;1\" \\\n",
    "            \"1\" \\\n",
    "            \"0\" \\\n",
    "            \"1\" \\\n",
    "            \"ordered.existence.txt\" \\\n",
    "            \"T\" \\\n",
    "            \"$(seq 4 8)\" \\\n",
    "            \"EarlyEmbryo;PGC;Sperm;Retina;DCs\" \\\n",
    "            \"\" \\\n",
    "            \"13287;13988;14732;15087;15686\" \\\n",
    "            \"-007\" \\\n",
    "            \"pdf\" \\\n",
    "            \"#7570B3\" \\\n",
    "            \"Reverse\"\n",
    "    }\n",
    "    # order\n",
    "    \n",
    "    prepareFiles(){\n",
    "        # T: executed in 84ms, finished 17:28:30 2022-04-17\n",
    "        for stage in 2cell 8cell Morula ICM\n",
    "        do\n",
    "            ln -s ${dirPATH}/EarlyEmbryogenesis/Merged/${stage}.methyl.sam.G.bed ${stage}.methyl.sam.G.bed\n",
    "            ln -s ${dirPATH}/EarlyEmbryogenesis/Merged/${stage}.H3K9me3.rmDup.bw ${stage}.H3K9me3.rmDup.bw\n",
    "        done # for stage end\n",
    "        \n",
    "        for stage in E10.5 E13.5_female E13.5_male\n",
    "        do\n",
    "            ln -s ${dirPATH}/PGCsDevelopment/Merged/${stage}.methyl.sam.G.bed ${stage}.methyl.sam.G.bed\n",
    "            ln -s ${dirPATH}/PGCsDevelopment/Merged/${stage}.H3K9me3.rmDup.bw ${stage}.H3K9me3.rmDup.bw\n",
    "        done # for stage end\n",
    "        \n",
    "        for stage in US DS PS RS\n",
    "        do\n",
    "            ln -s ${dirPATH}/Spermatogenesis_GSE137744/Merged/${stage}.methyl.sam.G.bed ${stage}.methyl.sam.G.bed\n",
    "            ln -s ${dirPATH}/Spermatogenesis_GSE137744/Merged/${stage}.H3K9me3.rmDup.bw ${stage}.H3K9me3.rmDup.bw\n",
    "        done # for stage end\n",
    "        \n",
    "        for stage in E14.5 E17.5 P0 P3 P7 P10 P14 P21\n",
    "        do\n",
    "            ln -s ${dirPATH}/RetinalDevelopment/GSE87064_Mouse/Merged/${stage}.methyl.sam.G.bed ${stage}.methyl.sam.G.bed\n",
    "            ln -s ${dirPATH}/RetinalDevelopment/GSE87064_Mouse/Merged/${stage}.H3K9me3.rmDup.bw ${stage}.H3K9me3.rmDup.bw\n",
    "        done # for stage end\n",
    "        \n",
    "        for stage in MPP CDP cDC pDC\n",
    "        do\n",
    "            ln -s ${dirPATH}/DendriticCellDevelopment/Merged/${stage}.methyl.sam.G.bed ${stage}.methyl.sam.G.bed\n",
    "            ln -s ${dirPATH}/DendriticCellDevelopment/Merged/${stage}.H3K9me3.rmDup.bw ${stage}.H3K9me3.rmDup.bw\n",
    "        done # for stage end\n",
    "    }\n",
    "    # prepareFiles\n",
    "    \n",
    "    methyl(){\n",
    "        echo -ne \"#Chrom\\tStart\\tEnd\" > title.txt\n",
    "        sort -k1,1 -k2,2n ${dirPATH}/CHMOrganization/Universal_specific/ordered.bed > ordered.methyl.txt\n",
    "        for stage in 2cell 8cell Morula ICM E10.5 E13.5_female E13.5_male US DS PS RS E14.5 E17.5 P0 P3 P7 P10 P14 P21 MPP\n",
    "        do\n",
    "            echo -ne \"\\t${stage}\" >> title.txt\n",
    "            bash ${shPATH}/averageMethylInRegionMultipleThreads.sh \\\n",
    "                ${dirPATH}/CHMOrganization/Universal_specific/ordered.bed \\\n",
    "                ${stage}.methyl.sam.G.bed \\\n",
    "                ordered.methyl_${stage}.txt \\\n",
    "                6\n",
    "            sort -k1,1 -k2,2n ordered.methyl_${stage}.txt | cut -f 4 | paste ordered.methyl.txt - > tmp && mv tmp ordered.methyl.txt\n",
    "        done # for stage end\n",
    "        \n",
    "        for stage in CDP cDC pDC\n",
    "        do\n",
    "            echo -ne \"\\t${stage}\" >> title.txt\n",
    "            bash ${shPATH}/averageMethylInRegionMultipleThreads.sh \\\n",
    "                ${dirPATH}/CHMOrganization/Universal_specific/ordered.bed \\\n",
    "                ${stage}.methyl.sam.G.bed \\\n",
    "                ordered.methyl_${stage}.txt \\\n",
    "                2\n",
    "            sort -k1,1 -k2,2n ordered.methyl_${stage}.txt | cut -f 4 | paste ordered.methyl.txt - > tmp && mv tmp ordered.methyl.txt\n",
    "        done # for stage end\n",
    "        echo -ne \"\\n\" >> title.txt\n",
    "        intersectBed -wao -a ${dirPATH}/CHMOrganization/Universal_specific/ordered.bed -b ordered.methyl.txt -f 1.00 -r | \\\n",
    "            cut -f 4-29 >> title.txt\n",
    "        mv title.txt ordered.methyl.final.txt\n",
    "    }\n",
    "    # methyl\n",
    "    \n",
    "    H3K9me3(){\n",
    "        echo -ne \"#Chrom\\tStart\\tEnd\" > title.txt\n",
    "        cat ${dirPATH}/CHMOrganization/Universal_specific/ordered.bed > ordered.H3K9me3.txt\n",
    "        for stage in 2cell 8cell Morula ICM E10.5 E13.5_female E13.5_male US DS PS RS E14.5 E17.5 P0 P3 P7 P10 P14 P21 MPP CDP cDC pDC\n",
    "        do\n",
    "            echo -ne \"\\t${stage}\" >> title.txt\n",
    "            bash ${shPATH}/averageSignalInRegionPrimaryOrder.sh \\\n",
    "                ${dirPATH}/CHMOrganization/Universal_specific/ordered.bed \\\n",
    "                ${stage}.H3K9me3.rmDup.bw \\\n",
    "                1 \\\n",
    "                ${stage}.H3K9me3_${stage}.txt\n",
    "            paste ordered.H3K9me3.txt ${stage}.H3K9me3_${stage}.txt > tmp && mv tmp ordered.H3K9me3.txt\n",
    "        done # for stage end\n",
    "        echo -ne \"\\n\" >> title.txt\n",
    "        cat title.txt ordered.H3K9me3.txt > ordered.H3K9me3.final.txt\n",
    "    }\n",
    "    # H3K9me3\n",
    "    \n",
    "    heatmap(){\n",
    "        # D: Apr-17-2022 18:21 Sun             \n",
    "        # DNA methylation level\n",
    "        Rscript ${rPATH}/heatmap_mat_1.r \\\n",
    "            \"ordered.methyl.final\" \\\n",
    "            \"3\" \\\n",
    "            \"5\" \\\n",
    "            \"5;1\" \\\n",
    "            \"1\" \\\n",
    "            \"0.4\" \\\n",
    "            \"0.6\" \\\n",
    "            \"ordered.methyl.final.txt\" \\\n",
    "            \"T\" \\\n",
    "            \"$(seq 4 26)\" \\\n",
    "            \"\" \\\n",
    "            \"\" \\\n",
    "            \"13287;13988;14732;15087;15686\" \\\n",
    "            \"-007\" \\\n",
    "            \"pdf\" \\\n",
    "            \"#DA404E\" \\\n",
    "            \"Reverse\"\n",
    "                    \n",
    "        # H3K9me3\n",
    "        # T: executed in 4.79s, finished 14:27:51 2022-04-18\n",
    "        Rscript ${rPATH}/heatmap_mat_1.r \\\n",
    "            \"ordered.H3K9me3.final\" \\\n",
    "            \"3\" \\\n",
    "            \"5\" \\\n",
    "            \"5;1\" \\\n",
    "            \"0.3\" \\\n",
    "            \"0\" \\\n",
    "            \"0.3\" \\\n",
    "            \"ordered.H3K9me3.final.txt\" \\\n",
    "            \"T\" \\\n",
    "            \"$(seq 4 26)\" \\\n",
    "            \"\" \\\n",
    "            \"\" \\\n",
    "            \"13287;13988;14732;15087;15686\" \\\n",
    "            \"-007\" \\\n",
    "            \"pdf\" \\\n",
    "            \"#507f9B\" \\\n",
    "            \"Reverse\"\n",
    "    \n",
    "        pdftoppm -png -r 300 ordered.methyl.final.pdf ordered.methyl.final\n",
    "        pdftoppm -png -r 300 ordered.H3K9me3.final.pdf ordered.H3K9me3.final\n",
    "    }\n",
    "    # heatmap\n",
    "    \n",
    "    bin_1kb(){\n",
    "        # D: Apr-18-2022 18:11 Mon\n",
    "        mkdir -p ${dirPATH}/CHMOrganization/1kbBins;cd ${dirPATH}/CHMOrganization/1kbBins\n",
    "        \n",
    "        manage_data(){\n",
    "            echo -e \"xValue\\tyValue\\tgroup\" > methyl.ggplot_basic.txt\n",
    "            echo -e \"xValue\\tyValue\\tgroup\" > H3K9me3.ggplot_basic.txt\n",
    "            # PGC development\n",
    "            for stage in E10.5 E13.5_female E13.5_male\n",
    "            do\n",
    "                cat ${dirPATH}/PGCsDevelopment/Merged/CpGs_*.methyl_${stage}.txt | \\\n",
    "                    awk -v XVALUE=${stage} 'BEGIN{FS=OFS=\"\\t\"}{print XVALUE, $4, \"methyl\";}' >> methyl.ggplot_basic.txt\n",
    "                cat ${dirPATH}/PGCsDevelopment/Merged/CpGs_*.H3K9me3_${stage}.txt | \\\n",
    "                    awk -v XVALUE=${stage} 'BEGIN{FS=OFS=\"\\t\"}{print XVALUE, $1, \"H3K9me3\";}' >> H3K9me3.ggplot_basic.txt\n",
    "            done # for stage end\n",
    "            \n",
    "            # Spermatogenesis_GSE137744\n",
    "            for stage in US DS PS RS\n",
    "            do\n",
    "                cat ${dirPATH}/Spermatogenesis_GSE137744/Merged/CpGs_*.methyl_${stage}.txt | \\\n",
    "                    awk -v XVALUE=${stage} 'BEGIN{FS=OFS=\"\\t\"}{print XVALUE, $4, \"methyl\";}' >> methyl.ggplot_basic.txt\n",
    "                cat ${dirPATH}/Spermatogenesis_GSE137744/Merged/CpGs_*.H3K9me3_${stage}.txt | \\\n",
    "                    awk -v XVALUE=${stage} 'BEGIN{FS=OFS=\"\\t\"}{print XVALUE, $1, \"H3K9me3\";}' >> H3K9me3.ggplot_basic.txt\n",
    "            done # for stage end\n",
    "            \n",
    "            # RetinalDevelopment\n",
    "            for stage in E14.5 E17.5 P0 P3 P7 P10 P14 P21\n",
    "            do\n",
    "                cat ${dirPATH}/RetinalDevelopment/GSE87064_Mouse/Merged/CpGs_*.methyl_${stage}.txt | \\\n",
    "                    awk -v XVALUE=${stage} 'BEGIN{FS=OFS=\"\\t\"}{print XVALUE, $4, \"methyl\";}' >> methyl.ggplot_basic.txt\n",
    "                cat ${dirPATH}/RetinalDevelopment/GSE87064_Mouse/Merged/CpGs_*.H3K9me3_${stage}.txt | \\\n",
    "                    awk -v XVALUE=${stage} 'BEGIN{FS=OFS=\"\\t\"}{print XVALUE, $1, \"H3K9me3\";}' >> H3K9me3.ggplot_basic.txt\n",
    "            done # for stage end\n",
    "            \n",
    "            # DendriticCellDevelopment\n",
    "            for stage in MPP CDP cDC pDC\n",
    "            do\n",
    "                cat ${dirPATH}/DendriticCellDevelopment/Merged/CpGs_*.methyl_${stage}.txt | \\\n",
    "                    awk -v XVALUE=${stage} 'BEGIN{FS=OFS=\"\\t\"}{print XVALUE, $4, \"methyl\";}' >> methyl.ggplot_basic.txt\n",
    "                cat ${dirPATH}/DendriticCellDevelopment/Merged/CpGs_*.H3K9me3_${stage}.txt | \\\n",
    "                    awk -v XVALUE=${stage} 'BEGIN{FS=OFS=\"\\t\"}{print XVALUE, $1, \"H3K9me3\";}' >> H3K9me3.ggplot_basic.txt\n",
    "            done # for stage end\n",
    "        }\n",
    "        # manage_data\n",
    "        \n",
    "        boxplot(){\n",
    "            # # DNA methylation level\n",
    "            # Rscript ${rPATH}/ggplot_boxplot.r \\\n",
    "            #       methyl.ggplot_basic.txt \\\n",
    "            #       methyl.ggplot_boxplot.pdf \\\n",
    "            #       \"DNA methylation level\" \\\n",
    "            #       \"\" \\\n",
    "            #       \"DNA methylation level\" \\\n",
    "            #       \"0\" \\\n",
    "            #       \"1\" \\\n",
    "            #       \"0.2\" \\\n",
    "            #       \"E10.5;E13.5_female;E13.5_male;US;DS;PS;RS;E14.5;E17.5;P0;P3;P7;P10;P14;P21;MPP;CDP;cDC;pDC\" \\\n",
    "            #       \"methyl\" \\\n",
    "            #       \"6\" \\\n",
    "            #       \"4\"\n",
    "                  \n",
    "            # H3K9me3\n",
    "            Rscript ${rPATH}/ggplot_boxplot.r \\\n",
    "                  H3K9me3.ggplot_basic.txt \\\n",
    "                  H3K9me3.ggplot_boxplot.pdf \\\n",
    "                  \"H3K9me3\" \\\n",
    "                  \"\" \\\n",
    "                  \"H3K9me3\" \\\n",
    "                  \"0\" \\\n",
    "                  \"0.2\" \\\n",
    "                  \"0.1\" \\\n",
    "                  \"E10.5;E13.5_female;E13.5_male;US;DS;PS;RS;E14.5;E17.5;P0;P3;P7;P10;P14;P21;MPP;CDP;cDC;pDC\" \\\n",
    "                  \"H3K9me3\" \\\n",
    "                  \"6\" \\\n",
    "                  \"4\"\n",
    "        }\n",
    "        boxplot\n",
    "    }\n",
    "    # bin_1kb\n",
    "    \n",
    "    bin_1kb_CpGrich(){\n",
    "        # D: May-16-2022 17:32 Mon\n",
    "        mkdir -p ${dirPATH}/CHMOrganization/1kbBins;cd ${dirPATH}/CHMOrganization/1kbBins\n",
    "        \n",
    "        manage_data(){\n",
    "            echo -e \"xValue\\tyValue\\tgroup\" > CpGrich_methyl.ggplot_basic.txt\n",
    "            echo -e \"xValue\\tyValue\\tgroup\" > CpGrich_H3K9me3.ggplot_basic.txt\n",
    "            # PGC development\n",
    "            for stage in E10.5 E13.5_female E13.5_male\n",
    "            do\n",
    "                cat ${dirPATH}/PGCsDevelopment/Merged/CpGs_high.methyl_${stage}.txt | \\\n",
    "                    awk -v XVALUE=${stage} 'BEGIN{FS=OFS=\"\\t\"}{print XVALUE, $4, \"methyl\";}' >> CpGrich_methyl.ggplot_basic.txt\n",
    "                cat ${dirPATH}/PGCsDevelopment/Merged/CpGs_high.H3K9me3_${stage}.txt | \\\n",
    "                    awk -v XVALUE=${stage} 'BEGIN{FS=OFS=\"\\t\"}{print XVALUE, $1, \"H3K9me3\";}' >> CpGrich_H3K9me3.ggplot_basic.txt\n",
    "            done # for stage end\n",
    "            \n",
    "            # Spermatogenesis_GSE137744\n",
    "            for stage in US DS PS RS\n",
    "            do\n",
    "                cat ${dirPATH}/Spermatogenesis_GSE137744/Merged/CpGs_high.methyl_${stage}.txt | \\\n",
    "                    awk -v XVALUE=${stage} 'BEGIN{FS=OFS=\"\\t\"}{print XVALUE, $4, \"methyl\";}' >> CpGrich_methyl.ggplot_basic.txt\n",
    "                cat ${dirPATH}/Spermatogenesis_GSE137744/Merged/CpGs_high.H3K9me3_${stage}.txt | \\\n",
    "                    awk -v XVALUE=${stage} 'BEGIN{FS=OFS=\"\\t\"}{print XVALUE, $1, \"H3K9me3\";}' >> CpGrich_H3K9me3.ggplot_basic.txt\n",
    "            done # for stage end\n",
    "            \n",
    "            # RetinalDevelopment\n",
    "            for stage in E14.5 E17.5 P0 P3 P7 P10 P14 P21\n",
    "            do\n",
    "                cat ${dirPATH}/RetinalDevelopment/GSE87064_Mouse/Merged/CpGs_high.methyl_${stage}.txt | \\\n",
    "                    awk -v XVALUE=${stage} 'BEGIN{FS=OFS=\"\\t\"}{print XVALUE, $4, \"methyl\";}' >> CpGrich_methyl.ggplot_basic.txt\n",
    "                cat ${dirPATH}/RetinalDevelopment/GSE87064_Mouse/Merged/CpGs_high.H3K9me3_${stage}.txt | \\\n",
    "                    awk -v XVALUE=${stage} 'BEGIN{FS=OFS=\"\\t\"}{print XVALUE, $1, \"H3K9me3\";}' >> CpGrich_H3K9me3.ggplot_basic.txt\n",
    "            done # for stage end\n",
    "            \n",
    "            # DendriticCellDevelopment\n",
    "            for stage in MPP CDP cDC pDC\n",
    "            do\n",
    "                cat ${dirPATH}/DendriticCellDevelopment/Merged/CpGs_high.methyl_${stage}.txt | \\\n",
    "                    awk -v XVALUE=${stage} 'BEGIN{FS=OFS=\"\\t\"}{print XVALUE, $4, \"methyl\";}' >> CpGrich_methyl.ggplot_basic.txt\n",
    "                cat ${dirPATH}/DendriticCellDevelopment/Merged/CpGs_high.H3K9me3_${stage}.txt | \\\n",
    "                    awk -v XVALUE=${stage} 'BEGIN{FS=OFS=\"\\t\"}{print XVALUE, $1, \"H3K9me3\";}' >> CpGrich_H3K9me3.ggplot_basic.txt\n",
    "            done # for stage end\n",
    "        }\n",
    "        # manage_data\n",
    "        \n",
    "        boxplot(){\n",
    "            # # DNA methylation level\n",
    "            # Rscript ${rPATH}/ggplot_boxplot.r \\\n",
    "            #       CpGrich_methyl.ggplot_basic.txt \\\n",
    "            #       CpGrich_methyl.ggplot_boxplot.pdf \\\n",
    "            #       \"DNA methylation level\" \\\n",
    "            #       \"\" \\\n",
    "            #       \"DNA methylation level\" \\\n",
    "            #       \"0\" \\\n",
    "            #       \"1\" \\\n",
    "            #       \"0.2\" \\\n",
    "            #       \"E10.5;E13.5_female;E13.5_male;US;DS;PS;RS;E14.5;E17.5;P0;P3;P7;P10;P14;P21;MPP;CDP;cDC;pDC\" \\\n",
    "            #       \"methyl\" \\\n",
    "            #       \"6\" \\\n",
    "            #       \"4\"\n",
    "                  \n",
    "            # H3K9me3\n",
    "            Rscript ${rPATH}/ggplot_boxplot.r \\\n",
    "                  CpGrich_H3K9me3.ggplot_basic.txt \\\n",
    "                  CpGrich_H3K9me3.ggplot_boxplot.pdf \\\n",
    "                  \"H3K9me3\" \\\n",
    "                  \"\" \\\n",
    "                  \"H3K9me3\" \\\n",
    "                  \"0\" \\\n",
    "                  \"0.75\" \\\n",
    "                  \"0.25\" \\\n",
    "                  \"E10.5;E13.5_female;E13.5_male;US;DS;PS;RS;E14.5;E17.5;P0;P3;P7;P10;P14;P21;MPP;CDP;cDC;pDC\" \\\n",
    "                  \"H3K9me3\" \\\n",
    "                  \"6\" \\\n",
    "                  \"4\"\n",
    "        }\n",
    "        boxplot\n",
    "    }\n",
    "    bin_1kb_CpGrich\n",
    "    \n",
    "    H3K9me3_profile_DC(){\n",
    "        # D: Apr-24-2022 18:15 Sun\n",
    "        for group in Universal EarlyEmbryoSpecific PGCSpecific SpermSpecific RetinalSpecific DCSpecific\n",
    "        do\n",
    "            for stage in MPP CDP cDC pDC\n",
    "            do\n",
    "                bash ${shPATH}/profileSurroundingPeakCenter.sh \\\n",
    "                    ${dirPATH}/CHMOrganization/Universal_specific/${group}.bed \\\n",
    "                    10000 \\\n",
    "                    10000 \\\n",
    "                    100 \\\n",
    "                    ${stage}.H3K9me3.rmDup.bw \\\n",
    "                    ${stage} \\\n",
    "                    H3K9me3Surrounding${group} \\\n",
    "                    \"-10kb;Center;10kb\" \\\n",
    "                    \"0;50.5;100\" \\\n",
    "                    \"H3K9me3 signal (RPM)\" \\\n",
    "                    \"0\" \\\n",
    "                    \"0.3\" \\\n",
    "                    \"Dark2\"\n",
    "            done # for stage end\n",
    "        done # for group end\n",
    "    }\n",
    "    # H3K9me3_profile_DC\n",
    "}\n",
    "# signals\n",
    "\n",
    "features(){\n",
    "    # D: Apr-17-2022 19:09 Sun\n",
    "    mkdir -p ${dirPATH}/CHMOrganization/Universal_specific/Features;cd ${dirPATH}/CHMOrganization/Universal_specific/Features\n",
    "    regionEnrichment(){\n",
    "        mkdir -p ${dirPATH}/CHMOrganization/Universal_specific/Features/RegionEnrichment;cd ${dirPATH}/CHMOrganization/Universal_specific/Features/RegionEnrichment\n",
    "        each(){\n",
    "            for group in Universal EarlyEmbryoSpecific PGCSpecific SpermSpecific RetinalSpecific DCSpecific\n",
    "            do\n",
    "              bash ${shPATH}/regionEnrichment_userDefined.sh \\\n",
    "               \"${dirPATH}/CHMOrganization/Universal_specific/Features/RegionEnrichment\" \\\n",
    "               \"mm10_euch\" \\\n",
    "               \"${anPATH}/mm10/mm10.chrom.sizes\" \\\n",
    "               \"${dirPATH}/CHMOrganization/Universal_specific/${group}.bed\" \\\n",
    "               \"${anPATH}/mm10/RegionEnrichment/mm10_euch.promoter.merged.bed;${anPATH}/mm10/RegionEnrichment/mm10_euch.geneBody.merged.bed;${anPATH}/mm10/RegionEnrichment/mm10_euch.intergenic.merged.bed;${anPATH}/mm10/RegionEnrichment/mm10_euch.exon.merged.bed;${anPATH}/mm10/RegionEnrichment/mm10_euch.intron.merged.bed;${anPATH}/mm10/RegionEnrichment/mm10_euch.CGI.merged.bed;${anPATH}/mm10/RegionEnrichment/mm10_euch.CGI_Unmasked.merged.bed;${anPATH}/mm10/RegionEnrichment/mm10_euch.SINE.merged.bed;${anPATH}/mm10/RegionEnrichment/mm10_euch.LINE.merged.bed;${anPATH}/mm10/RegionEnrichment/mm10_euch.LTR.merged.bed;${anPATH}/mm10/RegionEnrichment/mm10_euch.PAD.merged.bed;${anPATH}/mm10/RegionEnrichment/mm10_euch.potentialEnhancer.merged.bed;\" \\\n",
    "               \"${group}\"\n",
    "            done # for group end        \n",
    "        }\n",
    "        # each\n",
    "\n",
    "        merge_and_plot(){\n",
    "            # D: Apr-17-2022 19:18 Sun\n",
    "            echo -e \"xValue\\tyValue\\tgroup\" > overlapRatio.ggplot_basic.txt\n",
    "            echo -e \"xValue\\tyValue\\tgroup\" > regionEnrich.ggplot_basic.txt\n",
    "            for group in Universal EarlyEmbryoSpecific PGCSpecific SpermSpecific RetinalSpecific DCSpecific\n",
    "            do\n",
    "                tail -n +2 ${group}.overlaped.txt | awk -v GROUP=${group} 'BEGIN{FS=OFS=\"\\t\"}{print $1, $3*100, GROUP;}' >> overlapRatio.ggplot_basic.txt\n",
    "                tail -n +2 ${group}.EnrichScore.txt | awk -v GROUP=${group} 'BEGIN{FS=OFS=\"\\t\"}{print $1, $2, GROUP;}' >> regionEnrich.ggplot_basic.txt\n",
    "            done # for group end\n",
    "            \n",
    "            Rscript ${rPATH}/ggplot_barplot.r \\\n",
    "                \"overlapRatio.ggplot_basic.txt\" \\\n",
    "                \"overlapRatio.ggplot_barplot.pdf\" \\\n",
    "                \"Overlapped ratio\" \\\n",
    "                \"\" \\\n",
    "                \"Percentage (%)\" \\\n",
    "                \"0;100;20\" \\\n",
    "                \"promoter;geneBody;intergenic;exon;intron;CGI_Unmasked;SINE;LINE;LTR;PAD;potentialEnhancer\" \\\n",
    "                \"Universal;EarlyEmbryoSpecific;PGCSpecific;SpermSpecific;RetinalSpecific;DCSpecific\" \\\n",
    "                \"dodge\" \\\n",
    "                \"10\" \\\n",
    "                \"3.5\" \\\n",
    "                \"-007\" \\\n",
    "                \"F\"\n",
    "                \n",
    "            Rscript ${rPATH}/ggplot_barplot.r \\\n",
    "                \"regionEnrich.ggplot_basic.txt\" \\\n",
    "                \"regionEnrich.ggplot_barplot.pdf\" \\\n",
    "                \"Enrichment score\" \\\n",
    "                \"\" \\\n",
    "                \"Enrichment score\" \\\n",
    "                \"-6;6;3\" \\\n",
    "                \"promoter;geneBody;intergenic;exon;intron;CGI_Unmasked;SINE;LINE;LTR;PAD;potentialEnhancer\" \\\n",
    "                \"Universal;EarlyEmbryoSpecific;PGCSpecific;SpermSpecific;RetinalSpecific;DCSpecific\" \\\n",
    "                \"dodge\" \\\n",
    "                \"10\" \\\n",
    "                \"3.5\" \\\n",
    "                \"-1;1\" \\\n",
    "                \"F\"\n",
    "        }\n",
    "        # merge_and_plot\n",
    "        \n",
    "        LINE_F(){\n",
    "            # D: Apr-17-2022 22:11 Sun\n",
    "            mkdir -p ${dirPATH}/CHMOrganization/Universal_specific/Features/RegionEnrichment/LINE/SubFamily;cd ${dirPATH}/CHMOrganization/Universal_specific/Features/RegionEnrichment/LINE/SubFamily\n",
    "            for group in Universal EarlyEmbryoSpecific PGCSpecific SpermSpecific RetinalSpecific DCSpecific\n",
    "            do\n",
    "                bash ${shPATH}/regionEnrichment_LINE_mm10.sh \\\n",
    "                    ${dirPATH}/CHMOrganization/Universal_specific/Features/RegionEnrichment/LINE/SubFamily \\\n",
    "                    ${dirPATH}/CHMOrganization/Universal_specific/${group}.bed \\\n",
    "                    ${group}\n",
    "            done # for group end\n",
    "        }\n",
    "        # LINE_F\n",
    "        \n",
    "        LTR_F(){\n",
    "            # D: Apr-17-2022 22:14 Sun\n",
    "            mkdir -p ${dirPATH}/CHMOrganization/Universal_specific/Features/RegionEnrichment/LTR/SubFamily;cd ${dirPATH}/CHMOrganization/Universal_specific/Features/RegionEnrichment/LTR/SubFamily\n",
    "            for group in Universal EarlyEmbryoSpecific PGCSpecific SpermSpecific RetinalSpecific DCSpecific\n",
    "            do\n",
    "                bash ${shPATH}/regionEnrichment_LTR_mm10.sh \\\n",
    "                    ${dirPATH}/CHMOrganization/Universal_specific/Features/RegionEnrichment/LTR/SubFamily \\\n",
    "                    ${dirPATH}/CHMOrganization/Universal_specific/${group}.bed \\\n",
    "                    ${group}\n",
    "            done # for group end\n",
    "        }\n",
    "        # LTR_F\n",
    "        \n",
    "        LINE_F_sina(){\n",
    "            # D: Apr-17-2022 22:58 Sun\n",
    "            mkdir -p ${dirPATH}/CHMOrganization/Universal_specific/Features/RegionEnrichment/LINE/SubFamily;cd ${dirPATH}/CHMOrganization/Universal_specific/Features/RegionEnrichment/LINE/SubFamily\n",
    "            # echo -e \"xValue\\tyValue\\tgroup\\tcol\\tlabel\" > enrichmentScore.ggplot_sina.txt\n",
    "            # for group in Universal EarlyEmbryoSpecific PGCSpecific SpermSpecific RetinalSpecific DCSpecific\n",
    "            # do\n",
    "            #     for fml in CR1 Dong-R4 Jockey L1 L1-Tx1 L2 Penelope RTE-BovB RTE-X\n",
    "            #     do\n",
    "            #         tail -n +2 ${group}.EnrichScore_LINE_${fml}.txt | \\\n",
    "            #             awk -v XVALUE=${group} -v COL=${fml} 'BEGIN{FS=OFS=\"\\t\"}{\\\n",
    "            #             if($2==\"-inf\"){{print XVALUE, \"-8\", \"LINE\", COL, \"\";}} else{\\\n",
    "            #             if($2>=3){print XVALUE, $2, \"LINE\", COL, $1;} else{print XVALUE, $2, \"LINE\", COL, \"\";}}}' >> enrichmentScore.ggplot_sina.txt\n",
    "            #     done # for fml end\n",
    "            # done # for group end\n",
    "            # \n",
    "            # Rscript ${rPATH}/ggplot_sina_colorfull.r \\\n",
    "            #     enrichmentScore.ggplot_sina.txt \\\n",
    "            #     enrichmentScore.ggplot_sina.pdf \\\n",
    "            #     \"Enrichment score\" \\\n",
    "            #     \"-8;8;2\" \\\n",
    "            #     \"Universal;EarlyEmbryoSpecific;PGCSpecific;SpermSpecific;RetinalSpecific;DCSpecific\" \\\n",
    "            #     \"LINE\" \\\n",
    "            #     \"CR1;Dong-R4;Jockey;L1;L1-Tx1;L2;Penelope;RTE-BovB;RTE-X\" \\\n",
    "            #     \"#CE0013;#16557A;#C7A609;#87C232;#64C0AB;#A14C94;#8B7E75;#1187CD;#000000\" \\\n",
    "            #     \"0.2\" \\\n",
    "            #     \"0.75\" \\\n",
    "            #     \"1;3\" \\\n",
    "            #     \"5\" \\\n",
    "            #     \"5\"\n",
    "            \n",
    "            echo -e \"xValue\\tyValue\\tgroup\\tcol\\tlabel\" > enrichmentScore.ggplot_sina_details.txt\n",
    "            for group in Universal EarlyEmbryoSpecific PGCSpecific SpermSpecific RetinalSpecific DCSpecific\n",
    "            do\n",
    "                for fml in CR1 Dong-R4 Jockey L1 L1-Tx1 L2 Penelope RTE-BovB RTE-X\n",
    "                do\n",
    "                    tail -n +2 ${group}.EnrichScore_LINE_${fml}.txt | \\\n",
    "                        awk -v XVALUE=${group} -v COL=${fml} 'BEGIN{FS=OFS=\"\\t\"}{\\\n",
    "                        if($2==\"-inf\"){{print XVALUE, \"-8\", \"LINE\", COL, $1;}} else{\\\n",
    "                        print XVALUE, $2, \"LINE\", COL, $1;}}' >> enrichmentScore.ggplot_sina_details.txt\n",
    "                done # for fml end\n",
    "            done # for group end\n",
    "        }\n",
    "        # LINE_F_sina\n",
    "        \n",
    "        LTR_F_sina(){\n",
    "            # D: Apr-17-2022 23:34 Sun\n",
    "            mkdir -p ${dirPATH}/CHMOrganization/Universal_specific/Features/RegionEnrichment/LTR/SubFamily;cd ${dirPATH}/CHMOrganization/Universal_specific/Features/RegionEnrichment/LTR/SubFamily\n",
    "            # echo -e \"xValue\\tyValue\\tgroup\\tcol\\tlabel\" > enrichmentScore.ggplot_sina.txt\n",
    "            # for group in Universal EarlyEmbryoSpecific PGCSpecific SpermSpecific RetinalSpecific DCSpecific\n",
    "            # do\n",
    "            #     for fml in ERV1 \"ERV1?\" ERVK \"ERVK?\" ERVL ERVL-MaLR \"ERVL?\" Gypsy \"Gypsy?\"\n",
    "            #     do\n",
    "            #         tail -n +2 ${group}.EnrichScore_LTR_${fml}.txt | \\\n",
    "            #             awk -v XVALUE=${group} -v COL=${fml} 'BEGIN{FS=OFS=\"\\t\"}{\\\n",
    "            #             if($2==\"-inf\"){{print XVALUE, \"-8\", \"LTR\", COL, \"\";}} else{\\\n",
    "            #             if($2>=-10){print XVALUE, $2, \"LTR\", COL, $1;} else{print XVALUE, $2, \"LINE\", COL, \"\";}}}' >> enrichmentScore.ggplot_sina.txt\n",
    "            #     done # for fml end\n",
    "            # done # for group end\n",
    "            \n",
    "            # Rscript ${rPATH}/ggplot_sina_colorfull.r \\\n",
    "            #     enrichmentScore.ggplot_sina.txt \\\n",
    "            #     enrichmentScore.ggplot_sina.pdf \\\n",
    "            #     \"Enrichment score\" \\\n",
    "            #     \"-8;8;2\" \\\n",
    "            #     \"Universal;EarlyEmbryoSpecific;PGCSpecific;SpermSpecific;RetinalSpecific;DCSpecific\" \\\n",
    "            #     \"LTR\" \\\n",
    "            #     \"ERV1;ERV1?;ERVK;ERVK?;ERVL;ERVL-MaLR;ERVL?;Gypsy;Gypsy?\" \\\n",
    "            #     \"#CE0013;#16557A;#C7A609;#87C232;#64C0AB;#A14C94;#8B7E75;#1187CD;#000000\" \\\n",
    "            #     \"0.2\" \\\n",
    "            #     \"0.75\" \\\n",
    "            #     \"1\" \\\n",
    "            #     \"5\" \\\n",
    "            #     \"5\"\n",
    "            \n",
    "            Rscript ${rPATH}/ggplot_sina_colorfull.r \\\n",
    "                enrichmentScore.ggplot_sina.txt \\\n",
    "                enrichmentScore_filtered.ggplot_sina.pdf \\\n",
    "                \"Enrichment score\" \\\n",
    "                \"-8;8;2\" \\\n",
    "                \"Universal;EarlyEmbryoSpecific;PGCSpecific;SpermSpecific;RetinalSpecific;DCSpecific\" \\\n",
    "                \"LTR\" \\\n",
    "                \"ERV1;ERVK;ERVL;ERVL-MaLR;Gypsy\" \\\n",
    "                \"#CE0013;#C7A609;#64C0AB;#A14C94;#1187CD\" \\\n",
    "                \"0.2\" \\\n",
    "                \"0.75\" \\\n",
    "                \"1\" \\\n",
    "                \"5\" \\\n",
    "                \"5\"\n",
    "        }\n",
    "        # LTR_F_sina\n",
    "        \n",
    "        LINE_F_sina_enrich_pie(){\n",
    "            # D: Apr-18-2022 10:59 Mon\n",
    "            mkdir -p ${dirPATH}/CHMOrganization/Universal_specific/Features/RegionEnrichment/LINE/SubFamily;cd ${dirPATH}/CHMOrganization/Universal_specific/Features/RegionEnrichment/LINE/SubFamily\n",
    "            for group in Universal EarlyEmbryoSpecific PGCSpecific SpermSpecific RetinalSpecific DCSpecific\n",
    "            do\n",
    "                # echo -e \"value\\tgroup\" > ${group}.enriched.basic_pieplot.txt\n",
    "                # for fml in CR1 Dong-R4 Jockey L1 L1-Tx1 L2 Penelope RTE-BovB RTE-X\n",
    "                # do\n",
    "                #     tail -n +2 ${group}.EnrichScore_LINE_${fml}.txt | \\\n",
    "                #         awk -v GROUP=${fml} 'BEGIN{FS=OFS=\"\\t\";TOTAL=0;}{\\\n",
    "                #         if($2>=1){TOTAL+=1;}}END{print TOTAL, GROUP;}' >> ${group}.enriched.basic_pieplot.txt\n",
    "                # done # for fml end\n",
    "                \n",
    "                Rscript ${rPATH}/ggplot_pie.r \\\n",
    "                    ${group}.enriched.basic_pieplot.txt \\\n",
    "                    ${group} \\\n",
    "                    \"CR1;Dong-R4;Jockey;L1;L1-Tx1;L2;Penelope;RTE-BovB;RTE-X\" \\\n",
    "                    \"#CE0013;#16557A;#C7A609;#87C232;#64C0AB;#A14C94;#8B7E75;#1187CD;#000000\" \\\n",
    "                    T \\\n",
    "                    3.5 \\\n",
    "                    3.5\n",
    "            done # for group end\n",
    "        }\n",
    "        # LINE_F_sina_enrich_pie\n",
    "        \n",
    "        LTR_F_sina_enrich_pie(){\n",
    "            # D: Apr-18-2022 11:44 Mon\n",
    "            mkdir -p ${dirPATH}/CHMOrganization/Universal_specific/Features/RegionEnrichment/LTR/SubFamily;cd ${dirPATH}/CHMOrganization/Universal_specific/Features/RegionEnrichment/LTR/SubFamily\n",
    "            for group in Universal EarlyEmbryoSpecific PGCSpecific SpermSpecific RetinalSpecific DCSpecific\n",
    "            do\n",
    "                echo -e \"value\\tgroup\" > ${group}.enriched.basic_pieplot.txt\n",
    "                for fml in ERV1 \"ERV1?\" ERVK \"ERVK?\" ERVL ERVL-MaLR \"ERVL?\" Gypsy \"Gypsy?\"\n",
    "                do\n",
    "                    tail -n +2 ${group}.EnrichScore_LTR_${fml}.txt | \\\n",
    "                        awk -v GROUP=${fml} 'BEGIN{FS=OFS=\"\\t\";TOTAL=0;}{\\\n",
    "                        if($2>=1){TOTAL+=1;}}END{print TOTAL, GROUP;}' >> ${group}.enriched.basic_pieplot.txt\n",
    "                done # for fml end\n",
    "                \n",
    "                Rscript ${rPATH}/ggplot_pie.r \\\n",
    "                    ${group}.enriched.basic_pieplot.txt \\\n",
    "                    ${group} \\\n",
    "                    \"ERV1;ERV1?;ERVK;ERVK?;ERVL;ERVL-MaLR;ERVL?;Gypsy;Gypsy?\" \\\n",
    "                    \"#CE0013;#16557A;#C7A609;#87C232;#64C0AB;#A14C94;#8B7E75;#1187CD;#000000\" \\\n",
    "                    T \\\n",
    "                    3.5 \\\n",
    "                    3.5\n",
    "            done # for group end\n",
    "        }\n",
    "        # LTR_F_sina_enrich_pie\n",
    "        \n",
    "        LINE_F_sina_enrich_group(){\n",
    "            # D: Apr-18-2022 12:22 Mon\n",
    "            # T: executed in 81ms, finished 12:36:47 2022-04-18\n",
    "            mkdir -p ${dirPATH}/CHMOrganization/Universal_specific/Features/RegionEnrichment/LINE/SubFamily;cd ${dirPATH}/CHMOrganization/Universal_specific/Features/RegionEnrichment/LINE/SubFamily\n",
    "            # cat *.EnrichScore_*.txt | grep -v \"EnrichScore\" | \\\n",
    "            #     awk 'BEGIN{FS=OFS=\"\\t\";}{if($2!=\"-inf\" && $2>=1){print $1;}}' | sort -u > subfml_enriched.txt # 40\n",
    "            #     \n",
    "            # cat *.EnrichScore_*.txt | grep -v \"EnrichScore\" | \\\n",
    "            #     awk 'BEGIN{FS=OFS=\"\\t\";}{if($2==\"-inf\" || $2<1){print $1;}}' | sort -u | grep -wvf subfml_enriched.txt - > subfml_nonenriched.txt # 201\n",
    "            #     \n",
    "            # # T: executed in 334ms, finished 12:44:11 2022-04-18\n",
    "            # for subfml in $(cut -f 1 subfml_enriched.txt)\n",
    "            # do\n",
    "            #     cat ${anPATH}/mm10/Repeats/LINE/*/mm10.${subfml}.bed | cut -f 1-3 >> subfml_enriched.bed #\n",
    "            # done # for subfml\n",
    "            \n",
    "            # for subfml in $(cut -f 1 subfml_nonenriched.txt)\n",
    "            # do\n",
    "            #     cat ${anPATH}/mm10/Repeats/LINE/*/mm10.${subfml}.bed | cut -f 1-3 >> subfml_nonenriched.bed #\n",
    "            # done # for subfml\n",
    "            \n",
    "            enriched_manage_for_table(){\n",
    "                # D: Apr-19-2022 21:12 Tue\n",
    "                # T: executed in 525ms, finished 21:18:13 2022-04-19\n",
    "                echo -e \"Process\\tFamily\\tSubfamily\" > enriched.LINE.txt\n",
    "                for group in Universal EarlyEmbryoSpecific PGCSpecific SpermSpecific RetinalSpecific DCSpecific\n",
    "                do\n",
    "                    for fml in CR1 Dong-R4 Jockey L1 L1-Tx1 L2 Penelope RTE-BovB RTE-X\n",
    "                    do\n",
    "                        tail -n +2 ${group}.EnrichScore_LINE_${fml}.txt | \\\n",
    "                            awk -v PROCESS=${group} -v FAMILY=${fml} 'BEGIN{FS=OFS=\"\\t\"}{\\\n",
    "                            if($2!=\"-inf\" && $2>=1){print PROCESS, FAMILY, $1;}}' >> enriched.LINE.txt\n",
    "                    done # for fml end\n",
    "                done # for group end\n",
    "            }\n",
    "            # enriched_manage_for_table\n",
    "            \n",
    "            enriched_overlap(){\n",
    "                # D: Apr-19-2022 21:57 Tue\n",
    "                Rscript UpSetR.r\n",
    "                \n",
    "            }\n",
    "            # enriched_overlap\n",
    "            \n",
    "            phastCons60wayPlacental(){\n",
    "                # D: Apr-18-2022 12:51 Mon\n",
    "                echo -e \"xValue\\tyValue\\tgroup\" > phastCons60wayPlacental.ggplot_basic.txt\n",
    "                for group in enriched nonenriched\n",
    "                do\n",
    "                    bash ${shPATH}/averageSignalInRegion.sh \\\n",
    "                        subfml_${group}.bed \\\n",
    "                        ${anPATH}/mm10/Conservation/mm10.60way.phastCons60wayPlacental.bw \\\n",
    "                        1 \\\n",
    "                        ${group}.phastConsPlacental.txt\n",
    "                    awk -v GROUP=${group} 'BEGIN{FS=OFS=\"\\t\"}{print \"LINE\", $4, GROUP;}' \\\n",
    "                        ${group}.phastConsPlacental.txt >> phastCons60wayPlacental.ggplot_basic.txt\n",
    "                done # for group end\n",
    "                \n",
    "                # Violin plot\n",
    "                Rscript ${rPATH}/ggplot_violin.r \\\n",
    "                    phastCons60wayPlacental.ggplot_basic.txt \\\n",
    "                    phastCons60wayPlacental.ggplot_violin.pdf \\\n",
    "                    LINE \\\n",
    "                    \"\" \\\n",
    "                    \"Conservation score\" \\\n",
    "                    0 \\\n",
    "                    0.6 \\\n",
    "                    0.2 \\\n",
    "                    \"LINE\" \\\n",
    "                    \"enriched;nonenriched\" \\\n",
    "                    4 \\\n",
    "                    3\n",
    "                \n",
    "                # box plot\n",
    "                Rscript ${rPATH}/ggplot_boxplot.r \\\n",
    "                    phastCons60wayPlacental.ggplot_basic.txt \\\n",
    "                    phastCons60wayPlacental.ggplot_boxplot.pdf \\\n",
    "                    LINE \\\n",
    "                    \"\" \\\n",
    "                    \"Conservation score\" \\\n",
    "                    0 \\\n",
    "                    0.4 \\\n",
    "                    0.2 \\\n",
    "                    \"LINE\" \\\n",
    "                    \"enriched;nonenriched\" \\\n",
    "                    4 \\\n",
    "                    3\n",
    "            }\n",
    "            phastCons60wayPlacental\n",
    "            \n",
    "            significance_log(){\n",
    "                echo -e \"\n",
    "                    [1] wilcox.test of LINE\n",
    "                    # A tibble: 1 × 8\n",
    "                      .y.    group1   group2          p p.adj p.format p.signif method  \n",
    "                      <chr>  <chr>    <chr>       <dbl> <dbl> <chr>    <chr>    <chr>   \n",
    "                    1 yValue enriched nonenriched     0     0 <2e-16   ****     Wilcoxon\"\n",
    "            }\n",
    "            \n",
    "            phastCons(){\n",
    "                # D: Apr-18-2022 13:26 Mon\n",
    "                echo -e \"xValue\\tyValue\\tgroup\" > phastCons.ggplot_basic.txt\n",
    "                for group in enriched nonenriched\n",
    "                do\n",
    "                    bash ${shPATH}/averageSignalInRegion.sh \\\n",
    "                        subfml_${group}.bed \\\n",
    "                        ${anPATH}/mm10/Conservation/mm10.60way.phastCons.bw \\\n",
    "                        1 \\\n",
    "                        ${group}.phastConsPlacental.txt\n",
    "                    awk -v GROUP=${group} 'BEGIN{FS=OFS=\"\\t\"}{print \"LINE\", $4, GROUP;}' \\\n",
    "                        ${group}.phastConsPlacental.txt >> phastCons.ggplot_basic.txt\n",
    "                done # for group end\n",
    "                \n",
    "                # Violin plot\n",
    "                Rscript ${rPATH}/ggplot_violin.r \\\n",
    "                    phastCons.ggplot_basic.txt \\\n",
    "                    phastCons.ggplot_violin.pdf \\\n",
    "                    LINE \\\n",
    "                    \"\" \\\n",
    "                    \"Conservation score\" \\\n",
    "                    0 \\\n",
    "                    0.4 \\\n",
    "                    0.2 \\\n",
    "                    \"LINE\" \\\n",
    "                    \"enriched;nonenriched\" \\\n",
    "                    4 \\\n",
    "                    3\n",
    "                \n",
    "                # box plot\n",
    "                Rscript ${rPATH}/ggplot_boxplot.r \\\n",
    "                    phastCons.ggplot_basic.txt \\\n",
    "                    phastCons.ggplot_boxplot.pdf \\\n",
    "                    LINE \\\n",
    "                    \"\" \\\n",
    "                    \"Conservation score\" \\\n",
    "                    0 \\\n",
    "                    0.4 \\\n",
    "                    0.2 \\\n",
    "                    \"LINE\" \\\n",
    "                    \"enriched;nonenriched\" \\\n",
    "                    4 \\\n",
    "                    3\n",
    "            }\n",
    "            # phastCons\n",
    "        }\n",
    "        # LINE_F_sina_enrich_group\n",
    "        \n",
    "        LTR_F_sina_enrich_group(){\n",
    "            # # D: Apr-18-2022 13:15 Mon\n",
    "            # # T: \n",
    "            mkdir -p ${dirPATH}/CHMOrganization/Universal_specific/Features/RegionEnrichment/LTR/SubFamily;cd ${dirPATH}/CHMOrganization/Universal_specific/Features/RegionEnrichment/LTR/SubFamily\n",
    "            # cat *.EnrichScore_*.txt | grep -v \"EnrichScore\" | \\\n",
    "            #     awk 'BEGIN{FS=OFS=\"\\t\";}{if($2!=\"-inf\" && $2>=1){print $1;}}' | sort -u > subfml_enriched.txt # \n",
    "            #     \n",
    "            # cat *.EnrichScore_*.txt | grep -v \"EnrichScore\" | \\\n",
    "            #     awk 'BEGIN{FS=OFS=\"\\t\";}{if($2==\"-inf\" || $2<1){print $1;}}' | sort -u | grep -wvf subfml_enriched.txt - > subfml_nonenriched.txt # \n",
    "            #     \n",
    "            # # T: \n",
    "            # for subfml in $(cut -f 1 subfml_enriched.txt)\n",
    "            # do\n",
    "            #     cat ${anPATH}/mm10/Repeats/LTR/*/mm10.${subfml}.bed | cut -f 1-3 >> subfml_enriched.bed # 238\n",
    "            # done # for subfml\n",
    "            # \n",
    "            # for subfml in $(cut -f 1 subfml_nonenriched.txt)\n",
    "            # do\n",
    "            #     cat ${anPATH}/mm10/Repeats/LTR/*/mm10.${subfml}.bed | cut -f 1-3 >> subfml_nonenriched.bed # 413\n",
    "            # done # for subfml\n",
    "            \n",
    "            enriched_manage_for_table(){\n",
    "                # D: Apr-20-2022 00:23 Wed\n",
    "                # T: executed in 269ms, finished 00:25:00 2022-04-20\n",
    "                echo -e \"Process\\tFamily\\tSubfamily\" > enriched.LTR.txt\n",
    "                for group in Universal EarlyEmbryoSpecific PGCSpecific SpermSpecific RetinalSpecific DCSpecific\n",
    "                do\n",
    "                    for fml in ERV1 \"ERV1?\" ERVK \"ERVK?\" ERVL ERVL-MaLR \"ERVL?\" Gypsy \"Gypsy?\"\n",
    "                    do\n",
    "                        tail -n +2 ${group}.EnrichScore_LTR_${fml}.txt | \\\n",
    "                            awk -v PROCESS=${group} -v FAMILY=${fml} 'BEGIN{FS=OFS=\"\\t\"}{\\\n",
    "                            if($2!=\"-inf\" && $2>=1){print PROCESS, FAMILY, $1;}}' >> enriched.LTR.txt\n",
    "                    done # for fml end\n",
    "                done # for group end\n",
    "            }\n",
    "            # enriched_manage_for_table\n",
    "            \n",
    "            enriched_overlap(){\n",
    "                # D: Apr-20-2022 00:25 Wed\n",
    "                Rscript UpSetR.r\n",
    "                \n",
    "            }\n",
    "            # enriched_overlap\n",
    "            \n",
    "            phastConsPlacental(){\n",
    "                # D: \n",
    "                echo -e \"xValue\\tyValue\\tgroup\" > phastCons60wayPlacental.ggplot_basic.txt\n",
    "                for group in enriched nonenriched\n",
    "                do\n",
    "                    bash ${shPATH}/averageSignalInRegion.sh \\\n",
    "                        subfml_${group}.bed \\\n",
    "                        ${anPATH}/mm10/Conservation/mm10.60way.phastCons60wayPlacental.bw \\\n",
    "                        1 \\\n",
    "                        ${group}.phastConsPlacental.txt\n",
    "                    awk -v GROUP=${group} 'BEGIN{FS=OFS=\"\\t\"}{print \"LTR\", $4, GROUP;}' \\\n",
    "                        ${group}.phastConsPlacental.txt >> phastCons60wayPlacental.ggplot_basic.txt\n",
    "                done # for group end\n",
    "                \n",
    "                # Violin plot\n",
    "                Rscript ${rPATH}/ggplot_violin.r \\\n",
    "                    phastCons60wayPlacental.ggplot_basic.txt \\\n",
    "                    phastCons60wayPlacental.ggplot_violin.pdf \\\n",
    "                    LTR \\\n",
    "                    \"\" \\\n",
    "                    \"Conservation score\" \\\n",
    "                    0 \\\n",
    "                    0.6 \\\n",
    "                    0.2 \\\n",
    "                    \"LTR\" \\\n",
    "                    \"enriched;nonenriched\" \\\n",
    "                    4 \\\n",
    "                    3\n",
    "                \n",
    "                # box plot\n",
    "                Rscript ${rPATH}/ggplot_boxplot.r \\\n",
    "                    phastCons60wayPlacental.ggplot_basic.txt \\\n",
    "                    phastCons60wayPlacental.ggplot_boxplot.pdf \\\n",
    "                    LTR \\\n",
    "                    \"\" \\\n",
    "                    \"Conservation score\" \\\n",
    "                    0 \\\n",
    "                    0.4 \\\n",
    "                    0.2 \\\n",
    "                    \"LTR\" \\\n",
    "                    \"enriched;nonenriched\" \\\n",
    "                    4 \\\n",
    "                    3\n",
    "            }\n",
    "            # phastConsPlacental\n",
    "            \n",
    "            phastCons(){\n",
    "                # D: \n",
    "                echo -e \"xValue\\tyValue\\tgroup\" > phastCons.ggplot_basic.txt\n",
    "                for group in enriched nonenriched\n",
    "                do\n",
    "                    bash ${shPATH}/averageSignalInRegion.sh \\\n",
    "                        subfml_${group}.bed \\\n",
    "                        ${anPATH}/mm10/Conservation/mm10.60way.phastCons.bw \\\n",
    "                        1 \\\n",
    "                        ${group}.phastConsPlacental.txt\n",
    "                    awk -v GROUP=${group} 'BEGIN{FS=OFS=\"\\t\"}{print \"LTR\", $4, GROUP;}' \\\n",
    "                        ${group}.phastConsPlacental.txt >> phastCons.ggplot_basic.txt\n",
    "                done # for group end\n",
    "                \n",
    "                # Violin plot\n",
    "                Rscript ${rPATH}/ggplot_violin.r \\\n",
    "                    phastCons.ggplot_basic.txt \\\n",
    "                    phastCons.ggplot_violin.pdf \\\n",
    "                    LTR \\\n",
    "                    \"\" \\\n",
    "                    \"Conservation score\" \\\n",
    "                    0 \\\n",
    "                    0.6 \\\n",
    "                    0.2 \\\n",
    "                    \"LTR\" \\\n",
    "                    \"enriched;nonenriched\" \\\n",
    "                    4 \\\n",
    "                    3\n",
    "                \n",
    "                # box plot\n",
    "                Rscript ${rPATH}/ggplot_boxplot.r \\\n",
    "                    phastCons.ggplot_basic.txt \\\n",
    "                    phastCons.ggplot_boxplot.pdf \\\n",
    "                    LTR \\\n",
    "                    \"\" \\\n",
    "                    \"Conservation score\" \\\n",
    "                    0 \\\n",
    "                    0.4 \\\n",
    "                    0.2 \\\n",
    "                    \"LTR\" \\\n",
    "                    \"enriched;nonenriched\" \\\n",
    "                    4 \\\n",
    "                    3\n",
    "            }\n",
    "            # phastCons\n",
    "        }\n",
    "        # LTR_F_sina_enrich_group\n",
    "        \n",
    "        LTR_F_sina_enrich_pie_filtered(){\n",
    "            # D: May-02-2022 19:37 Mon\n",
    "            mkdir -p ${dirPATH}/CHMOrganization/Universal_specific/Features/RegionEnrichment/LTR/SubFamily;cd ${dirPATH}/CHMOrganization/Universal_specific/Features/RegionEnrichment/LTR/SubFamily\n",
    "            for group in Universal EarlyEmbryoSpecific PGCSpecific SpermSpecific RetinalSpecific DCSpecific\n",
    "            do\n",
    "                echo -e \"value\\tgroup\" > ${group}.enriched.basic_pieplot_filtered.txt\n",
    "                for fml in ERV1 ERVK ERVL ERVL-MaLR Gypsy\n",
    "                do\n",
    "                    tail -n +2 ${group}.EnrichScore_LTR_${fml}.txt | \\\n",
    "                        awk -v GROUP=${fml} 'BEGIN{FS=OFS=\"\\t\";TOTAL=0;}{\\\n",
    "                        if($2>=1){TOTAL+=1;}}END{print TOTAL, GROUP;}' >> ${group}.enriched.basic_pieplot_filtered.txt\n",
    "                done # for fml end\n",
    "                \n",
    "                Rscript ${rPATH}/ggplot_pie.r \\\n",
    "                    ${group}.enriched.basic_pieplot_filtered.txt \\\n",
    "                    ${group}_filtered \\\n",
    "                    \"ERV1;ERVK;ERVL;ERVL-MaLR;Gypsy\" \\\n",
    "                    \"#CE0013;#C7A609;#64C0AB;#A14C94;#1187CD\" \\\n",
    "                    T \\\n",
    "                    3.5 \\\n",
    "                    3.5\n",
    "            done # for group end\n",
    "        }\n",
    "        # LTR_F_sina_enrich_pie_filtered\n",
    "    }\n",
    "    # regionEnrichment\n",
    "    \n",
    "    length(){\n",
    "        # D: Apr-19-2022 11:58 Tue\n",
    "        mkdir -p ${dirPATH}/CHMOrganization/Universal_specific/Features/Length;cd ${dirPATH}/CHMOrganization/Universal_specific/Features/Length\n",
    "        manage_length(){\n",
    "            echo -e \"xValue\\tyValue\\tgroup\" > length.ggplot_basic.txt\n",
    "            for group in Universal EarlyEmbryoSpecific PGCSpecific SpermSpecific RetinalSpecific DCSpecific\n",
    "            do\n",
    "                awk -v XVALUE=${group} 'BEGIN{FS=OFS=\"\\t\"}{print XVALUE, $3-$2, \"Length\";}' ${dirPATH}/CHMOrganization/Universal_specific/${group}.bed >> length.ggplot_basic.txt\n",
    "            done # for group end\n",
    "        }\n",
    "        # manage_length\n",
    "        \n",
    "        boxplot(){\n",
    "            Rscript ${rPATH}/ggplot_boxplot.r \\\n",
    "                length.ggplot_basic.txt \\\n",
    "                length.ggplot_boxplot.pdf \\\n",
    "                Length \\\n",
    "                \"\" \\\n",
    "                \"Length of stable CHMs\" \\\n",
    "                0 \\\n",
    "                12000 \\\n",
    "                2000 \\\n",
    "                \"Universal;EarlyEmbryoSpecific;PGCSpecific;SpermSpecific;RetinalSpecific;DCSpecific\" \\\n",
    "                Length \\\n",
    "                3.5 \\\n",
    "                3.5\n",
    "        }\n",
    "        # boxplot\n",
    "        \n",
    "        violinplot(){\n",
    "            Rscript ${rPATH}/ggplot_violin.r \\\n",
    "                length.ggplot_basic.txt \\\n",
    "                length.ggplot_violinplot.pdf \\\n",
    "                Length \\\n",
    "                \"\" \\\n",
    "                \"Length of stable CHMs\" \\\n",
    "                0 \\\n",
    "                12000 \\\n",
    "                2000 \\\n",
    "                \"Universal;EarlyEmbryoSpecific;PGCSpecific;SpermSpecific;RetinalSpecific;DCSpecific\" \\\n",
    "                Length \\\n",
    "                4.5 \\\n",
    "                3.5\n",
    "        }\n",
    "        violinplot\n",
    "    }\n",
    "    # length\n",
    "    \n",
    "    targetGenes(){\n",
    "        # D: Apr-20-2022 12:27 Wed\n",
    "        # N: Genes within 2kb of stable CHMs\n",
    "        mkdir -p ${dirPATH}/CHMOrganization/Universal_specific/Features/TargetGenes;cd ${dirPATH}/CHMOrganization/Universal_specific/Features/TargetGenes\n",
    "        genesWithin2kb(){\n",
    "            # T: executed in 1.43s, finished 12:37:11 2022-04-20\n",
    "            mkdir -p ${dirPATH}/CHMOrganization/Universal_specific/Features/TargetGenes/Within_2kb;cd ${dirPATH}/CHMOrganization/Universal_specific/Features/TargetGenes/Within_2kb\n",
    "            manage(){\n",
    "                for group in Universal EarlyEmbryoSpecific PGCSpecific SpermSpecific RetinalSpecific DCSpecific\n",
    "                do\n",
    "                     windowBed -w 2000 -a ${dirPATH}/CHMOrganization/Universal_specific/${group}.bed -b ${anPATH}/mm10/mm10.refGene.withGeneSymbol.bed | \\\n",
    "                         cut -f 8 | sort -u > ${group}.targetGenes.txt # 477, 158, 85, 117, 333, 6694\n",
    "                done # for group end\n",
    "                \n",
    "                sort -R DCSpecific.targetGenes.txt | head -3000 > DCSpecific.targetGenes_rand1.txt \n",
    "                sort -R DCSpecific.targetGenes.txt | head -3000 > DCSpecific.targetGenes_rand2.txt \n",
    "                sort -R DCSpecific.targetGenes.txt | head -3000 > DCSpecific.targetGenes_rand3.txt \n",
    "            }\n",
    "            # manage\n",
    "            \n",
    "            DAVID_2021(){\n",
    "                prepare(){\n",
    "                    # BP_ALL\n",
    "                    wget -c \"https://david.ncifcrf.gov/data/download/t2t_11E66E8CC5961650429689442.txt\" -O Universal.BP_ALL.txt\n",
    "                    wget -c \"https://david.ncifcrf.gov/data/download/t2t_11E66E8CC5961650430110573.txt\" -O EarlyEmbryoSpecific.BP_ALL.txt\n",
    "                    wget -c \"https://david.ncifcrf.gov/data/download/t2t_11E66E8CC5961650430255416.txt\" -O PGCSpecific.BP_ALL.txt\n",
    "                    wget -c \"https://david.ncifcrf.gov/data/download/t2t_11E66E8CC5961650430412492.txt\" -O SpermSpecific.BP_ALL.txt\n",
    "                    wget -c \"https://david.ncifcrf.gov/data/download/t2t_11E66E8CC5961650430733111.txt\" -O RetinalSpecific.BP_ALL.txt\n",
    "                    wget -c \"https://david.ncifcrf.gov/data/download/t2t_11E66E8CC5961650431134297.txt\" -O DCSpecific_rand1.BP_ALL.txt\n",
    "                    wget -c \"https://david.ncifcrf.gov/data/download/t2t_69CB942DB5841650433251835.txt\" -O DCSpecific_rand2.BP_ALL.txt\n",
    "                    wget -c \"https://david.ncifcrf.gov/data/download/t2t_69CB942DB5841650433433267.txt\" -O DCSpecific_rand3.BP_ALL.txt\n",
    "                    \n",
    "                    # CC_ALL\n",
    "                    wget -c \"https://david.ncifcrf.gov/data/download/t2t_11E66E8CC5961650429894357.txt\" -O Universal.CC_ALL.txt\n",
    "                    wget -c \"https://david.ncifcrf.gov/data/download/t2t_11E66E8CC5961650430135432.txt\" -O EarlyEmbryoSpecific.CC_ALL.txt\n",
    "                    wget -c \"https://david.ncifcrf.gov/data/download/t2t_11E66E8CC5961650430284696.txt\" -O PGCSpecific.CC_ALL.txt\n",
    "                    wget -c \"https://david.ncifcrf.gov/data/download/t2t_11E66E8CC5961650430437186.txt\" -O SpermSpecific.CC_ALL.txt\n",
    "                    wget -c \"https://david.ncifcrf.gov/data/download/t2t_11E66E8CC5961650430760375.txt\" -O RetinalSpecific.CC_ALL.txt\n",
    "                    wget -c \"https://david.ncifcrf.gov/data/download/t2t_11E66E8CC5961650431234916.txt\" -O DCSpecific_rand1.CC_ALL.txt\n",
    "                    wget -c \"https://david.ncifcrf.gov/data/download/t2t_69CB942DB5841650433292623.txt\" -O DCSpecific_rand2.CC_ALL.txt\n",
    "                    wget -c \"https://david.ncifcrf.gov/data/download/t2t_69CB942DB5841650433458403.txt\" -O DCSpecific_rand3.CC_ALL.txt\n",
    "                    \n",
    "                    # MF_ALL\n",
    "                    wget -c \"https://david.ncifcrf.gov/data/download/t2t_11E66E8CC5961650429964292.txt\" -O Universal.MF_ALL.txt\n",
    "                    wget -c \"https://david.ncifcrf.gov/data/download/t2t_11E66E8CC5961650430160923.txt\" -O EarlyEmbryoSpecific.MF_ALL.txt\n",
    "                    wget -c \"https://david.ncifcrf.gov/data/download/tr211E66E8CC5961650430319114.txt\" -O PGCSpecific.MF_ALL.txt\n",
    "                    wget -c \"https://david.ncifcrf.gov/data/download/t2t_11E66E8CC5961650430465227.txt\" -O SpermSpecific.MF_ALL.txt\n",
    "                    wget -c \"https://david.ncifcrf.gov/data/download/t2t_11E66E8CC5961650430784349.txt\" -O RetinalSpecific.MF_ALL.txt\n",
    "                    wget -c \"https://david.ncifcrf.gov/data/download/t2t_11E66E8CC5961650431268625.txt\" -O DCSpecific_rand1.MF_ALL.txt\n",
    "                    wget -c \"https://david.ncifcrf.gov/data/download/t2t_69CB942DB5841650433318768.txt\" -O DCSpecific_rand2.MF_ALL.txt\n",
    "                    wget -c \"https://david.ncifcrf.gov/data/download/t2t_69CB942DB5841650433478106.txt\" -O DCSpecific_rand3.MF_ALL.txt\n",
    "\n",
    "                    # KEGG\n",
    "                    wget -c \"https://david.ncifcrf.gov/data/download/t2t_11E66E8CC5961650429988439.txt\" -O Universal.KEGG.txt\n",
    "                    wget -c \"https://david.ncifcrf.gov/data/download/t2t_11E66E8CC5961650430183836.txt\" -O EarlyEmbryoSpecific.KEGG.txt\n",
    "                    wget -c \"https://david.ncifcrf.gov/data/download/tr211E66E8CC5961650430498930.txt\" -O SpermSpecific.KEGG.txt\n",
    "                    wget -c \"https://david.ncifcrf.gov/data/download/t2t_11E66E8CC5961650430806571.txt\" -O RetinalSpecific.KEGG.txt\n",
    "                    wget -c \"https://david.ncifcrf.gov/data/download/t2t_69CB942DB5841650433010184.txt\" -O DCSpecific_rand1.KEGG.txt\n",
    "                    wget -c \"https://david.ncifcrf.gov/data/download/t2t_69CB942DB5841650433341879.txt\" -O DCSpecific_rand2.KEGG.txt\n",
    "                    wget -c \"https://david.ncifcrf.gov/data/download/t2t_69CB942DB5841650433497196.txt\" -O DCSpecific_rand3.KEGG.txt\n",
    "\n",
    "                    # BIOGRID_INTERACTION\n",
    "                    wget -c \"https://david.ncifcrf.gov/data/download/t2t_11E66E8CC5961650430019286.txt\" -O Universal.BIOGRID_INTERACTION.txt\n",
    "                    wget -c \"https://david.ncifcrf.gov/data/download/t2t_11E66E8CC5961650430213512.txt\" -O EarlyEmbryoSpecific.BIOGRID_INTERACTION.txt\n",
    "                    wget -c \"https://david.ncifcrf.gov/data/download/t2t_11E66E8CC5961650430833732.txt\" -O RetinalSpecific.BIOGRID_INTERACTION.txt\n",
    "                    wget -c \"https://david.ncifcrf.gov/data/download/t2t_69CB942DB5841650433038448.txt\" -O DCSpecific_rand1.BIOGRID_INTERACTION.txt\n",
    "                    wget -c \"https://david.ncifcrf.gov/data/download/t2t_69CB942DB5841650433374195.txt\" -O DCSpecific_rand2.BIOGRID_INTERACTION.txt\n",
    "                    wget -c \"https://david.ncifcrf.gov/data/download/t2t_69CB942DB5841650433519683.txt\" -O DCSpecific_rand3.BIOGRID_INTERACTION.txt\n",
    "                }\n",
    "                # prepare\n",
    "                \n",
    "                plot(){\n",
    "                    # D: Apr-20-2022 13:48 Wed\n",
    "                    # for group in Universal EarlyEmbryoSpecific PGCSpecific SpermSpecific RetinalSpecific DCSpecific DCSpecific_rand1 DCSpecific_rand2 DCSpecific_rand3\n",
    "                    for group in DCSpecific_rand1 DCSpecific_rand2 DCSpecific_rand3\n",
    "                    do\n",
    "                        for type in BP_ALL CC_ALL MF_ALL KEGG BIOGRID_INTERACTION\n",
    "                        do\n",
    "                            if [ -f ${group}.${type}.txt ]\n",
    "                            then\n",
    "                                python ${binPATH}/GO.py ${group}.${type}.txt\n",
    "                                Rscript ${rPATH}/GO.r ${group}.${type}.plot.txt ${group}.${type}.plot.pdf ${group} 6 3\n",
    "                            fi\n",
    "                        done # for type end\n",
    "                    done # for group end\n",
    "                }\n",
    "                # plot\n",
    "                \n",
    "                special_KEGG(){\n",
    "                    wget -c \"https://david.ncifcrf.gov/data/download/tr2C2BD5B1F15B71650452360159.txt\" -O RetinalSpecific.KEGG.txt\n",
    "                }\n",
    "                # special_KEGG\n",
    "                \n",
    "                enriched_GO_Term_genome_enrichment(){\n",
    "                    # D: May-16-2022 22:55 Mon\n",
    "                    GO_0007155_cell_adhesion(){\n",
    "                        # cut -f 2 GO_0007155_cell_adhesion.txt | sort -u > GO_0007155_cell_adhesion.genes.txt # 1,488\n",
    "                        # python ${pyPATH}/ExtractSubset.py ExtractSubset GO_0007155_cell_adhesion.genes.txt ${anPATH}/mm10/mm10.refGene.withGeneSymbol.bed 4 GO_0007155_cell_adhesion.genes.bed # 3,028\n",
    "                        # cut -f 1-3 GO_0007155_cell_adhesion.genes.bed | sort -u | sort -k1,1 -k2,2n | mergeBed -i - > GO_0007155_cell_adhesion.genes.merged.bed # 1,453\n",
    "                          \n",
    "                        bash ${shPATH}/regionEnrichment_userDefined.sh \\\n",
    "                           \"${dirPATH}/CHMOrganization/Universal_specific/Features/TargetGenes/Within_2kb\" \\\n",
    "                           \"mm10_euch\" \\\n",
    "                           \"${anPATH}/mm10/mm10.chrom.sizes\" \\\n",
    "                           \"${dirPATH}/CHMOrganization/Universal_specific/Features/TargetGenes/Within_2kb/GO_0007155_cell_adhesion.genes.merged.bed\" \\\n",
    "                           \"${anPATH}/mm10/RegionEnrichment/mm10_euch.promoter.merged.bed;${anPATH}/mm10/RegionEnrichment/mm10_euch.geneBody.merged.bed;${anPATH}/mm10/RegionEnrichment/mm10_euch.intergenic.merged.bed;${anPATH}/mm10/RegionEnrichment/mm10_euch.exon.merged.bed;${anPATH}/mm10/RegionEnrichment/mm10_euch.intron.merged.bed;${anPATH}/mm10/RegionEnrichment/mm10_euch.CGI.merged.bed;${anPATH}/mm10/RegionEnrichment/mm10_euch.CGI_Unmasked.merged.bed;${anPATH}/mm10/RegionEnrichment/mm10_euch.SINE.merged.bed;${anPATH}/mm10/RegionEnrichment/mm10_euch.LINE.merged.bed;${anPATH}/mm10/RegionEnrichment/mm10_euch.LTR.merged.bed;${anPATH}/mm10/RegionEnrichment/mm10_euch.PAD.merged.bed;${anPATH}/mm10/RegionEnrichment/mm10_euch.potentialEnhancer.merged.bed;\" \\\n",
    "                           \"GO_0007155_cell_adhesion.genes\"\n",
    "                    }\n",
    "                    GO_0007155_cell_adhesion\n",
    "                    \n",
    "                    GO_0098742_cell-cell_adhesion_via_plasma-membrane_adhesion_molecules(){\n",
    "                        # D: May-16-2022 23:39 Mon\n",
    "                        cut -f 2 GO_0098742_cell-cell_adhesion_via_plasma-membrane_adhesion_molecules.txt | sort -u > GO_0098742_cell-cell_adhesion_via_plasma-membrane_adhesion_molecules.genes.txt # 216\n",
    "                        python ${pyPATH}/ExtractSubset.py ExtractSubset GO_0098742_cell-cell_adhesion_via_plasma-membrane_adhesion_molecules.genes.txt ${anPATH}/mm10/mm10.refGene.withGeneSymbol.bed 4 GO_0098742_cell-cell_adhesion_via_plasma-membrane_adhesion_molecules.genes.bed # 467\n",
    "                        cut -f 1-3 GO_0098742_cell-cell_adhesion_via_plasma-membrane_adhesion_molecules.genes.bed | sort -u | sort -k1,1 -k2,2n | mergeBed -i - > GO_0098742_cell-cell_adhesion_via_plasma-membrane_adhesion_molecules.genes.merged.bed # 209\n",
    "                          \n",
    "                        bash ${shPATH}/regionEnrichment_userDefined.sh \\\n",
    "                           \"${dirPATH}/CHMOrganization/Universal_specific/Features/TargetGenes/Within_2kb\" \\\n",
    "                           \"mm10_euch\" \\\n",
    "                           \"${anPATH}/mm10/mm10.chrom.sizes\" \\\n",
    "                           \"${dirPATH}/CHMOrganization/Universal_specific/Features/TargetGenes/Within_2kb/GO_0098742_cell-cell_adhesion_via_plasma-membrane_adhesion_molecules.genes.merged.bed\" \\\n",
    "                           \"${anPATH}/mm10/RegionEnrichment/mm10_euch.promoter.merged.bed;${anPATH}/mm10/RegionEnrichment/mm10_euch.geneBody.merged.bed;${anPATH}/mm10/RegionEnrichment/mm10_euch.intergenic.merged.bed;${anPATH}/mm10/RegionEnrichment/mm10_euch.exon.merged.bed;${anPATH}/mm10/RegionEnrichment/mm10_euch.intron.merged.bed;${anPATH}/mm10/RegionEnrichment/mm10_euch.CGI.merged.bed;${anPATH}/mm10/RegionEnrichment/mm10_euch.CGI_Unmasked.merged.bed;${anPATH}/mm10/RegionEnrichment/mm10_euch.SINE.merged.bed;${anPATH}/mm10/RegionEnrichment/mm10_euch.LINE.merged.bed;${anPATH}/mm10/RegionEnrichment/mm10_euch.LTR.merged.bed;${anPATH}/mm10/RegionEnrichment/mm10_euch.PAD.merged.bed;${anPATH}/mm10/RegionEnrichment/mm10_euch.potentialEnhancer.merged.bed;\" \\\n",
    "                           \"GO_0098742_cell-cell_adhesion_via_plasma-membrane_adhesion_molecules.genes\"\n",
    "                    }\n",
    "                    # GO_0098742_cell-cell_adhesion_via_plasma-membrane_adhesion_molecules\n",
    "                    \n",
    "                    GO_0007156_homophilic_cell_adhesion_via_plasma_membrane_adhesion_molecules(){\n",
    "                        # D: May-16-2022 23:46 Mon\n",
    "                        cut -f 2 GO_0007156_homophilic_cell_adhesion_via_plasma_membrane_adhesion_molecules.txt | sort -u > GO_0007156_homophilic_cell_adhesion_via_plasma_membrane_adhesion_molecules.genes.txt # 117\n",
    "                        python ${pyPATH}/ExtractSubset.py ExtractSubset GO_0007156_homophilic_cell_adhesion_via_plasma_membrane_adhesion_molecules.genes.txt ${anPATH}/mm10/mm10.refGene.withGeneSymbol.bed 4 GO_0007156_homophilic_cell_adhesion_via_plasma_membrane_adhesion_molecules.genes.bed # 212\n",
    "                        cut -f 1-3 GO_0007156_homophilic_cell_adhesion_via_plasma_membrane_adhesion_molecules.genes.bed | sort -u | sort -k1,1 -k2,2n | mergeBed -i - > GO_0007156_homophilic_cell_adhesion_via_plasma_membrane_adhesion_molecules.genes.merged.bed # 111\n",
    "                          \n",
    "                        bash ${shPATH}/regionEnrichment_userDefined.sh \\\n",
    "                           \"${dirPATH}/CHMOrganization/Universal_specific/Features/TargetGenes/Within_2kb\" \\\n",
    "                           \"mm10_euch\" \\\n",
    "                           \"${anPATH}/mm10/mm10.chrom.sizes\" \\\n",
    "                           \"${dirPATH}/CHMOrganization/Universal_specific/Features/TargetGenes/Within_2kb/GO_0007156_homophilic_cell_adhesion_via_plasma_membrane_adhesion_molecules.genes.merged.bed\" \\\n",
    "                           \"${anPATH}/mm10/RegionEnrichment/mm10_euch.promoter.merged.bed;${anPATH}/mm10/RegionEnrichment/mm10_euch.geneBody.merged.bed;${anPATH}/mm10/RegionEnrichment/mm10_euch.intergenic.merged.bed;${anPATH}/mm10/RegionEnrichment/mm10_euch.exon.merged.bed;${anPATH}/mm10/RegionEnrichment/mm10_euch.intron.merged.bed;${anPATH}/mm10/RegionEnrichment/mm10_euch.CGI.merged.bed;${anPATH}/mm10/RegionEnrichment/mm10_euch.CGI_Unmasked.merged.bed;${anPATH}/mm10/RegionEnrichment/mm10_euch.SINE.merged.bed;${anPATH}/mm10/RegionEnrichment/mm10_euch.LINE.merged.bed;${anPATH}/mm10/RegionEnrichment/mm10_euch.LTR.merged.bed;${anPATH}/mm10/RegionEnrichment/mm10_euch.PAD.merged.bed;${anPATH}/mm10/RegionEnrichment/mm10_euch.potentialEnhancer.merged.bed;\" \\\n",
    "                           \"GO_0007156_homophilic_cell_adhesion_via_plasma_membrane_adhesion_molecules.genes\"\n",
    "                    }\n",
    "                    # GO_0007156_homophilic_cell_adhesion_via_plasma_membrane_adhesion_molecules\n",
    "                    \n",
    "                    all_enriched_GO_terms(){\n",
    "                        # D: May-17-2022 09:56 Tue\n",
    "                        cat *.BP_ALL.plot.txt *.MF_ALL.plot.txt *.CC_ALL.plot.txt | \\\n",
    "                            awk -F '[~\\t]' 'BEGIN{OFS=\"\\t\"}{if($3>=1.30103){print $1;}}' | \\\n",
    "                            sort -u > all_enriched_GO_terms.txt # 711 => too much => only showed\n",
    "                    }\n",
    "                    # all_enriched_GO_terms\n",
    "                    \n",
    "                    showed_enriched_GO_terms(){\n",
    "                        # # D: May-17-2022 10:27 Tue\n",
    "                        # for group in Universal EarlyEmbryoSpecific PGCSpecific SpermSpecific RetinalSpecific\n",
    "                        # do\n",
    "                        #     for category in BP MF CC\n",
    "                        #     do\n",
    "                        #         sort -u ${group}.${category}_ALL.plot.txt | \\\n",
    "                        #             awk -F '[~\\t]' 'BEGIN{OFS=\"\\t\"}{if($3>=1.30103){print $1;}}' | \\\n",
    "                        #             sort -t$'\\t' -k3,3nr | head -10 | cut -f 1 >> showed_enriched_GO_terms.txt\n",
    "                        #     done # for category end\n",
    "                        # done # for group end\n",
    "                        # \n",
    "                        # # DCSpecific\n",
    "                        # for category in BP MF CC\n",
    "                        # do\n",
    "                        #     sort -u DCSpecific_rand1.${category}_ALL.plot.txt | \\\n",
    "                        #         awk -F '[~\\t]' 'BEGIN{OFS=\"\\t\"}{if($3>=1.30103){print $1;}}' | \\\n",
    "                        #         sort -t$'\\t' -k3,3nr | head -10 | cut -f 1 >> showed_enriched_GO_terms.txt\n",
    "                        # done # for category end\n",
    "                        # \n",
    "                        # sort -u showed_enriched_GO_terms.txt > showed_enriched_GO_terms.txt.tmp && mv showed_enriched_GO_terms.txt.tmp showed_enriched_GO_terms.txt # 103\n",
    "                        \n",
    "                        # # GO term enrichment\n",
    "                        # echo -e \"xValue\\tyValue\\tgroup\\tcol\\tlabel\" > showed_enriched_GO_terms_used.ggplot_sina.txt\n",
    "                        # for GOTerm in $(cat showed_enriched_GO_terms_used.txt)\n",
    "                        # do\n",
    "                        #     cut -f 2 ${GOTerm}.txt | sort -u > ${GOTerm}.genes.txt\n",
    "                        #     python ${pyPATH}/ExtractSubset.py ExtractSubset ${GOTerm}.genes.txt ${anPATH}/mm10/mm10.refGene.withGeneSymbol.bed 4 ${GOTerm}.genes.bed\n",
    "                        #     cut -f 1-3 ${GOTerm}.genes.bed | sort -u | sort -k1,1 -k2,2n | mergeBed -i - > ${GOTerm}.genes.merged.bed\n",
    "                        #     bash ${shPATH}/regionEnrichment_userDefined.sh \\\n",
    "                        #        \"${dirPATH}/CHMOrganization/Universal_specific/Features/TargetGenes/Within_2kb\" \\\n",
    "                        #        \"mm10_euch\" \\\n",
    "                        #        \"${anPATH}/mm10/mm10.chrom.sizes\" \\\n",
    "                        #        \"${dirPATH}/CHMOrganization/Universal_specific/Features/TargetGenes/Within_2kb/${GOTerm}.genes.merged.bed\" \\\n",
    "                        #        \"${anPATH}/mm10/RegionEnrichment/mm10_euch.promoter.merged.bed;${anPATH}/mm10/RegionEnrichment/mm10_euch.geneBody.merged.bed;${anPATH}/mm10/RegionEnrichment/mm10_euch.intergenic.merged.bed;${anPATH}/mm10/RegionEnrichment/mm10_euch.exon.merged.bed;${anPATH}/mm10/RegionEnrichment/mm10_euch.intron.merged.bed;${anPATH}/mm10/RegionEnrichment/mm10_euch.CGI.merged.bed;${anPATH}/mm10/RegionEnrichment/mm10_euch.CGI_Unmasked.merged.bed;${anPATH}/mm10/RegionEnrichment/mm10_euch.SINE.merged.bed;${anPATH}/mm10/RegionEnrichment/mm10_euch.LINE.merged.bed;${anPATH}/mm10/RegionEnrichment/mm10_euch.LTR.merged.bed;${anPATH}/mm10/RegionEnrichment/mm10_euch.PAD.merged.bed;${anPATH}/mm10/RegionEnrichment/mm10_euch.potentialEnhancer.merged.bed;\" \\\n",
    "                        #        \"${GOTerm}.genes\"\n",
    "                        # \n",
    "                        #     # Prepare for Sina plot\n",
    "                        #     \n",
    "                        #     tail -n +2 ${GOTerm}.genes.EnrichScore.txt | \\\n",
    "                        #         awk 'BEGIN{FS=OFS=\"\\t\"}{if($2!=\"-inf\"){print $1, $2, \"All\", \"All\", \"All\";}}' >> showed_enriched_GO_terms_used.ggplot_sina.txt\n",
    "                        # done # for GOTerm end\n",
    "                        \n",
    "                        Sina_plot(){\n",
    "                            Rscript ${rPATH}/ggplot_sina_colorfull.r \\\n",
    "                                \"showed_enriched_GO_terms_used.ggplot_sina.txt\" \\\n",
    "                                \"showed_enriched_GO_terms_used.ggplot_sina.pdf\" \\\n",
    "                                \"Enrichment score\" \\\n",
    "                                \"-8;8;2\" \\\n",
    "                                \"promoter;geneBody;intergenic;exon;intron;CGI_Unmasked;SINE;LINE;LTR\" \\\n",
    "                                \"All\" \\\n",
    "                                \"All\" \\\n",
    "                                \"Black\" \\\n",
    "                                \"0.2\" \\\n",
    "                                \"0.75\" \\\n",
    "                                \"1;3\" \\\n",
    "                                \"5\" \\\n",
    "                                \"5\"\n",
    "                        }\n",
    "                        # Sina_plot\n",
    "                        \n",
    "                        violin_plot(){\n",
    "                            # D: May-17-2022 12:04 Tue\n",
    "                            echo -e \"xValue\\tyValue\\tgroup\" > showed_enriched_GO_terms_used.ggplot_violin.txt\n",
    "                            for GOTerm in $(cat showed_enriched_GO_terms_used.txt)\n",
    "                            do\n",
    "                                tail -n +2 ${GOTerm}.genes.EnrichScore.txt | \\\n",
    "                                    awk 'BEGIN{FS=OFS=\"\\t\"}{if($2!=\"-inf\"){print $1, $2, \"All\";}}' >> showed_enriched_GO_terms_used.ggplot_violin.txt\n",
    "                            done # for GOTerm\n",
    "                            \n",
    "                            Rscript ${rPATH}/ggplot_violin.r \\\n",
    "                                \"showed_enriched_GO_terms_used.ggplot_violin.txt\" \\\n",
    "                                \"showed_enriched_GO_terms_used.ggplot_violin.pdf\" \\\n",
    "                                \"\" \\\n",
    "                                \"\" \\\n",
    "                                \"Enrichment score\" \\\n",
    "                                \"-3\" \\\n",
    "                                \"3\" \\\n",
    "                                \"1\" \\\n",
    "                                \"promoter;geneBody;intergenic;exon;intron;CGI_Unmasked;SINE;LINE;LTR\" \\\n",
    "                                \"All\" \\\n",
    "                                \"5\" \\\n",
    "                                \"5\"\n",
    "                        }\n",
    "                        # violin_plot\n",
    "                        \n",
    "                        jitter_plot(){\n",
    "                            # D: May-17-2022 12:11 Tue\n",
    "                            Rscript ${rPATH}/ggplot_jitter.r \\\n",
    "                                \"showed_enriched_GO_terms_used.ggplot_violin.txt\" \\\n",
    "                                \"showed_enriched_GO_terms_used.ggplot_jitter.pdf\" \\\n",
    "                                \"\" \\\n",
    "                                \"\" \\\n",
    "                                \"Enrichment score\" \\\n",
    "                                \"-2\" \\\n",
    "                                \"3\" \\\n",
    "                                \"1\" \\\n",
    "                                \"promoter;geneBody;intergenic;exon;intron;CGI_Unmasked;SINE;LINE;LTR\" \\\n",
    "                                \"All\" \\\n",
    "                                \"6\" \\\n",
    "                                \"4\" \\\n",
    "                                \"1\"\n",
    "                        }\n",
    "                        jitter_plot\n",
    "                        \n",
    "                        boxplot_plot(){\n",
    "                            # D: May-17-2022 13:26 Tue                        \n",
    "                            Rscript ${rPATH}/ggplot_boxplot.r \\\n",
    "                                \"showed_enriched_GO_terms_used.ggplot_violin.txt\" \\\n",
    "                                \"showed_enriched_GO_terms_used.ggplot_boxplot.pdf\" \\\n",
    "                                \"\" \\\n",
    "                                \"\" \\\n",
    "                                \"Enrichment score\" \\\n",
    "                                \"-2\" \\\n",
    "                                \"3\" \\\n",
    "                                \"1\" \\\n",
    "                                \"promoter;geneBody;intergenic;exon;intron;CGI_Unmasked;SINE;LINE;LTR\" \\\n",
    "                                \"All\" \\\n",
    "                                \"6\" \\\n",
    "                                \"4\" \\\n",
    "                                \"1\"\n",
    "                        }\n",
    "                        # boxplot_plot\n",
    "                    }\n",
    "                    # showed_enriched_GO_terms\n",
    "                }\n",
    "                # enriched_GO_Term_genome_enrichment\n",
    "            }\n",
    "            # DAVID_2021\n",
    "            \n",
    "            expr(){\n",
    "                # D: May-31-2022 14:41 Tue\n",
    "                mkdir -p ${dirPATH}/CHMOrganization/Universal_specific/Features/TargetGenes/Within_2kb/Expression;cd ${dirPATH}/CHMOrganization/Universal_specific/Features/TargetGenes/Within_2kb/Expression\n",
    "                prepare_expr(){\n",
    "                    echo -e \"xValue\\tyValue\\tgroup\" > targetGenes.expr.ggplot_basic.txt\n",
    "                    for group in Universal EarlyEmbryoSpecific PGCSpecific SpermSpecific RetinalSpecific DCSpecific\n",
    "                    do\n",
    "                        windowBed -w 2000 -a ${dirPATH}/CHMOrganization/Universal_specific/${group}.bed -b ${anPATH}/mm10/mm10.refGene.withGeneSymbol.bed | \\\n",
    "                            cut -f 7 | sort -u > ${group}.targetGenes_RefSeq.txt # \n",
    "                            \n",
    "                        # Early embryos\n",
    "                        for stage in 2-cell 8-cell Morula ICM\n",
    "                        do\n",
    "                            for rep in 1 2 3 4 5 6\n",
    "                            do\n",
    "                                if [ -d ${dataPATH_expr_earlyEmbryo}/${stage}_${rep} ]\n",
    "                                then\n",
    "                                    python ${pyPATH}/ExtractSubset.py ExtractSubset ${group}.targetGenes_RefSeq.txt ${dataPATH_expr_earlyEmbryo}/${stage}_${rep}/t_data.ctab 5 ${group}.targetGenes_RefSeq.expr_${stage}.${rep}.txt\n",
    "                                    awk -v XVALUE=${stage} -v GROUP=${group} 'BEGIN{FS=OFS=\"\\t\"}{print XVALUE, log($12+1)/log(2), GROUP;}' ${group}.targetGenes_RefSeq.expr_${stage}.${rep}.txt >> targetGenes.expr.ggplot_basic.txt\n",
    "                                fi\n",
    "                            done # for rep end\n",
    "                        done # for stage end\n",
    "                        \n",
    "                        # PGC development\n",
    "                        for stage in E10.5 E13.5_female E13.5_male\n",
    "                        do\n",
    "                            for rep in 1 2 3 4\n",
    "                            do\n",
    "                                if [ -d ${dirPATH}/PGCsDevelopment/${stage}.RNASeq.rep${rep} ]\n",
    "                                then\n",
    "                                    python ${pyPATH}/ExtractSubset.py ExtractSubset ${group}.targetGenes_RefSeq.txt ${dirPATH}/PGCsDevelopment/${stage}.RNASeq.rep${rep}/${stage}.RNASeq.rep${rep}.t_data.ctab 5 ${group}.targetGenes_RefSeq.expr_${stage}.${rep}.txt\n",
    "                                    awk -v XVALUE=${stage} -v GROUP=${group} 'BEGIN{FS=OFS=\"\\t\"}{print XVALUE, log($12+1)/log(2), GROUP;}' ${group}.targetGenes_RefSeq.expr_${stage}.${rep}.txt >> targetGenes.expr.ggplot_basic.txt\n",
    "                                fi\n",
    "                            done # for rep end\n",
    "                        done # for stage end\n",
    "                        \n",
    "                        # Spermatogenesis\n",
    "                        for stage in GS PS RS\n",
    "                        do\n",
    "                            for rep in 1 2\n",
    "                            do\n",
    "                                if [ -d ${dirPATH}/Spermatogenesis_GSE137744/${stage}.RNASeq.rep${rep} ]\n",
    "                                then\n",
    "                                    python ${pyPATH}/ExtractSubset.py ExtractSubset ${group}.targetGenes_RefSeq.txt ${dirPATH}/Spermatogenesis_GSE137744/${stage}.RNASeq.rep${rep}/${stage}.RNASeq.rep${rep}.t_data.ctab 5 ${group}.targetGenes_RefSeq.expr_${stage}.${rep}.txt\n",
    "                                    awk -v XVALUE=${stage} -v GROUP=${group} 'BEGIN{FS=OFS=\"\\t\"}{print XVALUE, log($12+1)/log(2), GROUP;}' ${group}.targetGenes_RefSeq.expr_${stage}.${rep}.txt >> targetGenes.expr.ggplot_basic.txt\n",
    "                                fi\n",
    "                            done # for rep end\n",
    "                        done # for stage end\n",
    "                        \n",
    "                        # Retinal development\n",
    "                        for stage in E14.5 E17.5 P0 P3 P7 P10 P14 P21\n",
    "                        do\n",
    "                            for rep in 1 2\n",
    "                            do\n",
    "                                if [ -d ${dirPATH}/RetinalDevelopment/GSE87064_Mouse/${stage}.RNASeq.rep${rep} ]\n",
    "                                then\n",
    "                                    python ${pyPATH}/ExtractSubset.py ExtractSubset ${group}.targetGenes_RefSeq.txt ${dirPATH}/RetinalDevelopment/GSE87064_Mouse/${stage}.RNASeq.rep${rep}/${stage}.RNASeq.rep${rep}.t_data.ctab 5 ${group}.targetGenes_RefSeq.expr_${stage}.${rep}.txt\n",
    "                                    awk -v XVALUE=${stage} -v GROUP=${group} 'BEGIN{FS=OFS=\"\\t\"}{print XVALUE, log($12+1)/log(2), GROUP;}' ${group}.targetGenes_RefSeq.expr_${stage}.${rep}.txt >> targetGenes.expr.ggplot_basic.txt\n",
    "                                fi\n",
    "                            done # for rep end\n",
    "                        done # for stage end\n",
    "                        \n",
    "                        # DC differentiation\n",
    "                        for stage in MPP CDP cDC pDC\n",
    "                        do\n",
    "                            for rep in 1\n",
    "                            do\n",
    "                                if [ -d ${dirPATH}/DendriticCellDevelopment/${stage}.RNASeq.rep${rep} ]\n",
    "                                then\n",
    "                                    python ${pyPATH}/ExtractSubset.py ExtractSubset ${group}.targetGenes_RefSeq.txt ${dirPATH}/DendriticCellDevelopment/${stage}.RNASeq.rep${rep}/${stage}.RNASeq.rep${rep}.t_data.ctab 5 ${group}.targetGenes_RefSeq.expr_${stage}.${rep}.txt\n",
    "                                    awk -v XVALUE=${stage} -v GROUP=${group} 'BEGIN{FS=OFS=\"\\t\"}{print XVALUE, log($12+1)/log(2), GROUP;}' ${group}.targetGenes_RefSeq.expr_${stage}.${rep}.txt >> targetGenes.expr.ggplot_basic.txt\n",
    "                                fi\n",
    "                            done # for rep end\n",
    "                        done # for stage end\n",
    "                    done # for group end\n",
    "                }\n",
    "                # prepare_expr\n",
    "                \n",
    "                ggplot_boxplot(){\n",
    "                    # D: May-31-2022 15:41 Tue\n",
    "                    for group in Universal EarlyEmbryoSpecific PGCSpecific SpermSpecific RetinalSpecific DCSpecific\n",
    "                    do\n",
    "                        Rscript ${rPATH}/ggplot_boxplot.r \\\n",
    "                            targetGenes.expr.ggplot_basic.txt \\\n",
    "                            targetGenes_${group}.expr.ggplot_boxplot.pdf \\\n",
    "                            ${group} \\\n",
    "                            \"\" \\\n",
    "                            \"log2(FPKM+1)\" \\\n",
    "                            \"0\" \\\n",
    "                            \"6\" \\\n",
    "                            \"2\" \\\n",
    "                            \"2-cell;8-cell;Morula;ICM;E10.5;E13.5_female;E13.5_male;GS;PS;RS;E14.5;E17.5;P0;P3;P7;P10;P14;P21;MPP;CDP;cDC;pDC\" \\\n",
    "                            ${group} \\\n",
    "                            \"6\" \\\n",
    "                            \"4\" \\\n",
    "                            \"2\"\n",
    "                    done # for group end     \n",
    "                }\n",
    "                ggplot_boxplot\n",
    "            }\n",
    "            expr\n",
    "        }\n",
    "        genesWithin2kb\n",
    "    }\n",
    "    # targetGenes\n",
    "    \n",
    "    motifFinding(){\n",
    "        # D: Apr-19-2022 00:23 Tue\n",
    "        # Run in py2 environment\n",
    "        mkdir -p ${dirPATH}/CHMOrganization/Universal_specific/Features/MotifFinding;cd ${dirPATH}/CHMOrganization/Universal_specific/Features/MotifFinding\n",
    "        MDSeqPosF(){\n",
    "            for group in Universal EarlyEmbryoSpecific PGCSpecific SpermSpecific RetinalSpecific DCSpecific\n",
    "            do\n",
    "                MDSeqPos.py -d -m cistrome.xml -O MDSeqPos_cistrome_${group} ${dirPATH}/CHMOrganization/Universal_specific/${group}.bed mm10 && \\\n",
    "                MDSeqPos.py -d -m transfac.xml -O MDSeqPos_transfac_${group} ${dirPATH}/CHMOrganization/Universal_specific/${group}.bed mm10 &\n",
    "            done # for group end        \n",
    "        }\n",
    "        # MDSeqPosF\n",
    "        \n",
    "        Manage_cistrome(){\n",
    "            # D: Apr-20-2022 22:02 Wed\n",
    "            mkdir -p ${dirPATH}/CHMOrganization/Universal_specific/Features/MotifFinding/Manage_cistrome;cd ${dirPATH}/CHMOrganization/Universal_specific/Features/MotifFinding/Manage_cistrome\n",
    "            format(){\n",
    "                # T: executed in 207ms, finished 22:13:09 2022-04-20\n",
    "                for group in Universal EarlyEmbryoSpecific PGCSpecific SpermSpecific RetinalSpecific DCSpecific\n",
    "                do\n",
    "                    awk 'BEGIN{FS=OFS=\"\\t\"}{if(NF==10){print $2, $3, $4, $5, $6, $7, $8, $9, $10} else{print $0;}}' ${group}.txt > ${group}.formated.txt # \n",
    "                done # for group end\n",
    "            }\n",
    "            # format\n",
    "            \n",
    "            DNABindingDomain(){\n",
    "                # D: Apr-20-2022 22:14 Wed\n",
    "                prepare(){\n",
    "                    # T: executed in 134ms, finished 22:24:12 2022-04-20\n",
    "                    echo -e \"xValue\\tyValue\\tgroup\" > DNABindingDomain.ggplot_basic.txt\n",
    "                    for group in Universal EarlyEmbryoSpecific PGCSpecific SpermSpecific RetinalSpecific DCSpecific\n",
    "                    do\n",
    "                        tail -n +2 ${group}.formated.txt | cut -f 3 | sort | \\\n",
    "                            uniq -c | sed -e 's/^ *//;s/ /\\t/' | \\\n",
    "                            awk -v XVALUE=${group} 'BEGIN{FS=OFS=\"\\t\"}{print XVALUE, $1, $2;}' >> DNABindingDomain.ggplot_basic.txt\n",
    "                    done # for group end\n",
    "                }\n",
    "                # prepare\n",
    "                \n",
    "                barplot(){\n",
    "                    Rscript ${rPATH}/ggplot_barplot_percentage.r \\\n",
    "                        \"DNABindingDomain.ggplot_basic.txt\" \\\n",
    "                        \"DNABindingDomain\" \\\n",
    "                        \"\" \\\n",
    "                        \"Percentage\" \\\n",
    "                        \"Arid Domain Family;AT-hook Domain Family;BED-type-zinc finger Family;BetaBetaAlpha-zinc finger Family;BetaBetaAlpha-zinc finger Family,AT-hook Domain Family;BetaBetaAlpha-zinc finger Family (zf-C2H2);CENP-B box binding Family;CP2 Transcription Factor Domain Family;DMRT Domain Family;DNA binding homeobox and Different Transcription Factors (DDT) Domain Family;DNA-binding LAG-1-like Domain Family;Ets Domain Family;Forkhead Domain Family;GATA Domain Family;Glial Cells Missing (GCM) Domain Family;Grainyhead;GTF2I Domain Family;Helix-Loop-Helix Family;High Mobility Group (Box) Family;Homeobox and leucine zippe Family;Homeodomain Family;Hormone-nuclear Receptor Family;Interferon Regulatory Factor;Leucine zipper Family;Leucine Zipper Family;Loop-Sheet-Helix Family;MADS Box Family;MH1 Domain Family;Myb Domain Family;NF-Y CCAAT-Binding Protein Family;None;Nuclear Factor I-CCAAT-binding Transcription Factor (NFI-CTF) Family;PROX;Rel Homology Region Family;RFX Domain Family;Ring finger domain family;Runt Domain Family;Sand Domain Family;Stat Protein Family;TATA box-binding Family;Transcription Factor Family;Transcription Factor T-Domain\" \\\n",
    "                        \"Universal;EarlyEmbryoSpecific;PGCSpecific;SpermSpecific;RetinalSpecific;DCSpecific\" \\\n",
    "                        \"F\" \\\n",
    "                        \"16\" \\\n",
    "                        \"8\" \\\n",
    "                        0 \\\n",
    "                        100 \\\n",
    "                        20\n",
    "                }\n",
    "                # barplot\n",
    "                \n",
    "                pieplot(){\n",
    "                    # D: Apr-21-2022 15:18 Thu\n",
    "                    for group in Universal EarlyEmbryoSpecific PGCSpecific SpermSpecific RetinalSpecific DCSpecific\n",
    "                    do\n",
    "                        # echo -e \"value\\tgroup\" > DNABindingDomain_${group}.ggplot_twoColumns.txt\n",
    "                        # tail -n +2 ${group}.formated.txt | awk 'BEGIN{FS=OFS=\"\\t\"}{print $3;}' | \\\n",
    "                        #     awk 'BEGIN{FS=\"\\t\"}{print \"\\\"\"$1\"\\\"\"}' | sort | uniq -c | \\\n",
    "                        #     sed -e 's/^ *//;s/ /\\t/' | sort -k1,1n >> DNABindingDomain_${group}.ggplot_twoColumns.txt\n",
    "                        \n",
    "                        Rscript ${rPATH}/ggplot_pie_sortByPercent.r \\\n",
    "                            DNABindingDomain_${group}.ggplot_twoColumns.txt \\\n",
    "                            DNABindingDomain_${group} \\\n",
    "                            T \\\n",
    "                            12 \\\n",
    "                            12\n",
    "                    done # for group end\n",
    "                }\n",
    "                # pieplot\n",
    "            }\n",
    "            # DNABindingDomain\n",
    "            \n",
    "            enrichedFactors(){\n",
    "                # D: Apr-21-2022 14:03 Thu\n",
    "                prepare(){\n",
    "                    # T: executed in 108ms, finished 14:41:20 2022-04-21\n",
    "                    cat *.formated.txt | grep -v \"collapsed_id\" | awk 'BEGIN{FS=OFS=\"\\t\"}{print $2}' | \\\n",
    "                        sed -e 's/::/\\n/g;s/\\b\\([[:alpha:][:alnum:]]\\)\\([[:alpha:][:alnum:]]*\\)\\b/\\u\\1\\L\\2/g' | sort -u > enriched.factors.txt # 484\n",
    "                }\n",
    "                # prepare\n",
    "                \n",
    "                existence(){\n",
    "                    # # D: Apr-21-2022 14:46 Thu\n",
    "                    # for factor in $(cut -f 1 enriched.factors.txt)\n",
    "                    # do\n",
    "                    #     echo -e \"${factor}\" >> enriched.factors.ordered.txt\n",
    "                    #     for group in Universal EarlyEmbryoSpecific PGCSpecific SpermSpecific RetinalSpecific DCSpecific\n",
    "                    #     do\n",
    "                    #         grep -wi \"${factor}\" ${group}.formated.txt | awk 'END{if(NR>0){print 1;} else{print 0;}}' >> enriched.factors.existence_${group}.txt\n",
    "                    #     done # for group end\n",
    "                    # done # for factor end\n",
    "                    \n",
    "                    # T: executed in 76ms, finished 16:43:16 2022-04-21\n",
    "                    paste enriched.factors.ordered.txt enriched.factors.existence_Universal.txt enriched.factors.existence_EarlyEmbryoSpecific.txt enriched.factors.existence_PGCSpecific.txt enriched.factors.existence_SpermSpecific.txt enriched.factors.existence_RetinalSpecific.txt enriched.factors.existence_DCSpecific.txt > enriched.factors.existence.txt\n",
    "                }\n",
    "                # existence\n",
    "                \n",
    "                overlap(){\n",
    "                    # T: executed in 129ms, finished 16:25:19 2022-04-21\n",
    "                    for group in Universal EarlyEmbryoSpecific PGCSpecific SpermSpecific RetinalSpecific DCSpecific\n",
    "                    do\n",
    "                        cat ${group}.formated.txt | grep -v \"collapsed_id\" | awk 'BEGIN{FS=OFS=\"\\t\"}{print $2}' | \\\n",
    "                            sed -e 's/::/\\n/g;s/\\b\\([[:alpha:][:alnum:]]\\)\\([[:alpha:][:alnum:]]*\\)\\b/\\u\\1\\L\\2/g' | sort -u > enriched_${group}.factors.txt #\n",
    "                    done # for group end\n",
    "                    \n",
    "                    Rscript UpSetR.r\n",
    "                }\n",
    "                # overlap\n",
    "        \n",
    "                existence_zs_n_15(){\n",
    "                    # D: Apr-21-2022 16:50 Thu\n",
    "                    # T: executed in 119ms, finished 16:53:16 2022-04-21\n",
    "                    # cat *.formated.txt | grep -v \"collapsed_id\" | awk 'BEGIN{FS=OFS=\"\\t\"}{if($6<-15){print $2}}' | \\\n",
    "                    #     sed -e 's/::/\\n/g;s/\\b\\([[:alpha:][:alnum:]]\\)\\([[:alpha:][:alnum:]]*\\)\\b/\\u\\1\\L\\2/g' | sort -u > enriched.factors_zs_n_15.txt # 82\n",
    "                    \n",
    "                    # for factor in $(cut -f 1 enriched.factors_zs_n_15.txt)\n",
    "                    # do\n",
    "                    #     echo -e \"${factor}\" >> enriched.factors_zs_n_15.ordered.txt\n",
    "                    #     for group in Universal EarlyEmbryoSpecific PGCSpecific SpermSpecific RetinalSpecific DCSpecific\n",
    "                    #     do\n",
    "                    #         grep -wi \"${factor}\" ${group}.formated.txt | awk 'END{if(NR>0){print 1;} else{print 0;}}' >> enriched.factors_zs_n_15.existence_${group}.txt\n",
    "                    #     done # for group end\n",
    "                    # done # for factor end\n",
    "                    \n",
    "                    # paste enriched.factors_zs_n_15.ordered.txt enriched.factors_zs_n_15.existence_Universal.txt enriched.factors_zs_n_15.existence_EarlyEmbryoSpecific.txt enriched.factors_zs_n_15.existence_PGCSpecific.txt enriched.factors_zs_n_15.existence_SpermSpecific.txt enriched.factors_zs_n_15.existence_RetinalSpecific.txt enriched.factors_zs_n_15.existence_DCSpecific.txt > enriched.factors_zs_n_15.existence.txt\n",
    "                    \n",
    "                    # Rscript UpSetR_zs_n_15.r\n",
    "                    \n",
    "                    group(){\n",
    "                        # D: Apr-21-2022 16:59 Thu\n",
    "                        # T: executed in 122ms, finished 17:02:36 2022-04-21\n",
    "                        awk 'BEGIN{FS=OFS=\"\\t\"}{if($2==0 && $3==1 && $4==1 && $5==1 && $6==1 && $7==1){print $1;}}' enriched.factors_zs_n_15.existence.txt > enriched.factors_zs_n_15.processSpecificCommon.txt # 25\n",
    "                        awk 'BEGIN{FS=OFS=\"\\t\"}{if($2==1 && $3==1 && $4==1 && $5==1 && $6==1 && $7==1){print $1;}}' enriched.factors_zs_n_15.existence.txt > enriched.factors_zs_n_15.All.txt # 15\n",
    "                        awk 'BEGIN{FS=OFS=\"\\t\"}{if($2==1 && $3==0 && $4==0 && $5==0 && $6==0 && $7==0){print $1;}}' enriched.factors_zs_n_15.existence.txt > enriched.factors_zs_n_15.UniversalSpecific.txt # 11\n",
    "                    }\n",
    "                    # group\n",
    "                    \n",
    "                    pieplot(){\n",
    "                        for group in processSpecificCommon All UniversalSpecific\n",
    "                        do\n",
    "                            # echo -e \"value\\tgroup\" > DNABindingDomain_${group}.ggplot_twoColumns.txt\n",
    "                            # awk 'BEGIN{FS=OFS=\"\\t\"}{print $2}' enriched.factors_zs_n_15.${group}.txt | \\\n",
    "                            #     awk 'BEGIN{FS=\"\\t\"}{print \"\\\"\"$1\"\\\"\"}' | sort | uniq -c | \\\n",
    "                            #     sed -e 's/^ *//;s/ /\\t/' | sort -k1,1n >> DNABindingDomain_${group}.ggplot_twoColumns.txt\n",
    "                                \n",
    "                            Rscript ${rPATH}/ggplot_pie_sortByPercent.r \\\n",
    "                                DNABindingDomain_${group}.ggplot_twoColumns.txt \\\n",
    "                                DNABindingDomain_${group} \\\n",
    "                                T \\\n",
    "                                7.5 \\\n",
    "                                7.5\n",
    "                        done # for group end\n",
    "                    }\n",
    "                    pieplot\n",
    "                }\n",
    "                existence_zs_n_15\n",
    "            }\n",
    "            enrichedFactors\n",
    "        }\n",
    "        Manage_cistrome\n",
    "    }\n",
    "    # motifFinding\n",
    "    \n",
    "    functions(){\n",
    "        # D: Oct-30-2022 20:28 Sun\n",
    "        # F: Potential function exploration\n",
    "        mkdir -p ${dirPATH}/CHMOrganization/Universal_specific/Functions;cd ${dirPATH}/CHMOrganization/Universal_specific/Functions\n",
    "        DSB(){\n",
    "            mkdir -p ${dirPATH}/CHMOrganization/Universal_specific/Functions/DSB;cd ${dirPATH}/CHMOrganization/Universal_specific/Functions/DSB\n",
    "            END_seq(){\n",
    "                Testis(){\n",
    "                    # D: Oct-30-2022 20:34 Sun\n",
    "                    # I: Mahgoub_Elife_2020_GSE139289:https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE139289\n",
    "                    mkdir -p ${dirPATH}/CHMOrganization/Universal_specific/Functions/DSB/Testis;cd ${dirPATH}/CHMOrganization/Universal_specific/Functions/DSB/Testis\n",
    "                    download(){\n",
    "                        # D: Oct-30-2022 20:35 Sun\n",
    "                        # T: executed in 1m 40.8s, finished 20:43:24 2022-10-30\n",
    "                        wget -c \"https://ftp.ncbi.nlm.nih.gov/geo/samples/GSM4136nnn/GSM4136235/suppl/GSM4136235_WT_END_seq_Juvenile.bed.gz\" -O Testis_Juvenile.peaks.bed.gz\n",
    "                        gunzip Testis_Juvenile.peaks.bed.gz\n",
    "                        wget -c \"https://ftp.ncbi.nlm.nih.gov/geo/samples/GSM4136nnn/GSM4136235/suppl/GSM4136235_WT_END_seq_Juvenile.bigwig\" -O Testis_Juvenile.bw\n",
    "                        wget -c \"https://ftp.ncbi.nlm.nih.gov/geo/samples/GSM4136nnn/GSM4136237/suppl/GSM4136237_WT_END_seq_Adult.bed.gz\" -O Testis_Adult.peaks.bed.gz\n",
    "                        gunzip Testis_Adult.peaks.bed.gz\n",
    "                        wget -c \"https://ftp.ncbi.nlm.nih.gov/geo/samples/GSM4136nnn/GSM4136237/suppl/GSM4136237_WT_END_seq_Adult.bigwig\" -O Testis_Adult.bw\n",
    "                        \n",
    "                        # D: Oct-30-2022 21:07 Sun\n",
    "                        # T: executed in 235ms, finished 21:10:43 2022-10-30\n",
    "                        sort -k1,1 -k2,2n Testis_Juvenile.peaks.bed > Testis_Juvenile.peaks.bed.tmp && mv Testis_Juvenile.peaks.bed.tmp Testis_Juvenile.peaks.bed\n",
    "                        sort -k1,1 -k2,2n Testis_Adult.peaks.bed > Testis_Adult.peaks.bed.tmp && mv Testis_Adult.peaks.bed.tmp Testis_Adult.peaks.bed\n",
    "                    }\n",
    "                    # download\n",
    "                    \n",
    "                    peakOverlap(){\n",
    "                        # D: Oct-30-2022 20:46 Sun\n",
    "                        # R: Few overlap\n",
    "                        mkdir -p ${dirPATH}/CHMOrganization/Universal_specific/Functions/DSB/Testis/PeakOverlap;cd ${dirPATH}/CHMOrganization/Universal_specific/Functions/DSB/Testis/PeakOverlap\n",
    "                        echo -e \"xValue\\ttotal\\toverlaped\\tyValue\\tgroup\" > peakOverlap.ggplot_barplot.txt\n",
    "                        for group in Universal EarlyEmbryoSpecific PGCSpecific SpermSpecific RetinalSpecific DCSpecific\n",
    "                        do\n",
    "                            Total=$(cat ${dirPATH}/CHMOrganization/Universal_specific/${group}.bed | wc -l)\n",
    "                            for sample in Juvenile Adult\n",
    "                            do\n",
    "                                Overlaped=$(intersectBed -u -a ${dirPATH}/CHMOrganization/Universal_specific/${group}.bed -b ../Testis_${sample}.peaks.bed | wc -l)\n",
    "                                echo -e \"${group}\\t${Total}\\t${Overlaped}\\t${sample}\" | awk 'BEGIN{FS=OFS=\"\\t\"}{print $1, $2, $3, $3/$2,$4;}' >> peakOverlap.ggplot_barplot.txt\n",
    "                            done # for sample end\n",
    "                        done # for group end\n",
    "                    }\n",
    "                    # peakOverlap\n",
    "                    \n",
    "                    peakWithin2kb(){\n",
    "                        # D: Oct-30-2022 21:02 Sun\n",
    "                        mkdir -p ${dirPATH}/CHMOrganization/Universal_specific/Functions/DSB/Testis/PeakWithin2kb;cd ${dirPATH}/CHMOrganization/Universal_specific/Functions/DSB/Testis/PeakWithin2kb\n",
    "                        echo -e \"xValue\\ttotal\\twithin2kb\\tyValue\\tgroup\" > peakWithin2kb.ggplot_barplot.txt\n",
    "                        for group in Universal EarlyEmbryoSpecific PGCSpecific SpermSpecific RetinalSpecific DCSpecific\n",
    "                        do\n",
    "                            Total=$(cat ${dirPATH}/CHMOrganization/Universal_specific/${group}.bed | wc -l)\n",
    "                            for sample in Juvenile Adult\n",
    "                            do\n",
    "                                Within=$(cut -f 1-3 ../Testis_${sample}.peaks.bed | \\\n",
    "                                    closestBed -d -a ${dirPATH}/CHMOrganization/Universal_specific/${group}.bed -b - | \\\n",
    "                                    awk 'BEGIN{FS=OFS=\"\\t\"}{if($7<=2000){print $1, $2, $3;}}' | \\\n",
    "                                    sort -u | wc -l)\n",
    "                                echo -e \"${group}\\t${Total}\\t${Within}\\t${sample}\" | awk 'BEGIN{FS=OFS=\"\\t\"}{print $1, $2, $3, $3/$2,$4;}' >> peakWithin2kb.ggplot_barplot.txt\n",
    "                            done # for sample end\n",
    "                        done # for group end\n",
    "                    }\n",
    "                    # peakWithin2kb\n",
    "                    \n",
    "                    distance(){\n",
    "                        # D: Oct-30-2022 21:22 Sun\n",
    "                        # F: Distance to END-seq peaks\n",
    "\n",
    "                        mkdir -p ${dirPATH}/CHMOrganization/Universal_specific/Functions/DSB/Testis/Distance;cd ${dirPATH}/CHMOrganization/Universal_specific/Functions/DSB/Testis/Distance\n",
    "                        # echo -e \"xValue\\tyValue\\tgroup\" > distance.ggplot_basic.txt\n",
    "                        # for group in Universal EarlyEmbryoSpecific PGCSpecific SpermSpecific RetinalSpecific DCSpecific\n",
    "                        # do\n",
    "                        #     for sample in Juvenile Adult\n",
    "                        #     do\n",
    "                        #         cut -f 1-3 ../Testis_${sample}.peaks.bed | \\\n",
    "                        #             closestBed -d -a ${dirPATH}/CHMOrganization/Universal_specific/${group}.bed -b - | \\\n",
    "                        #             cut -f 1,2,3,7 | sort -u | \\\n",
    "                        #             awk -v XVALUE=${group} -v GROUP=${sample} 'BEGIN{FS=OFS=\"\\t\"}{print XVALUE, $4, GROUP;}' >> distance.ggplot_basic.txt\n",
    "                        #     done # for sample end\n",
    "                        # done # for group end\n",
    "                        \n",
    "                        boxplot(){\n",
    "                            Rscript ${rPATH}/ggplot_boxplot.r \\\n",
    "                                distance.ggplot_basic.txt \\\n",
    "                                distance.ggplot_boxplot.pdf \\\n",
    "                                distance \\\n",
    "                                \"\" \\\n",
    "                                \"distance of stable CHMs to END-seq peaks\" \\\n",
    "                                0 \\\n",
    "                                2500000 \\\n",
    "                                500000 \\\n",
    "                                \"Universal;EarlyEmbryoSpecific;PGCSpecific;SpermSpecific;RetinalSpecific;DCSpecific\" \\\n",
    "                                \"Juvenile;Adult\" \\\n",
    "                                6 \\\n",
    "                                4\n",
    "                        }\n",
    "                        # boxplot\n",
    "                    }\n",
    "                    # distance\n",
    "                    \n",
    "                    signalsSurrounding(){\n",
    "                        # D: Oct-30-2022 22:14 Sun\n",
    "                        mkdir -p ${dirPATH}/CHMOrganization/Universal_specific/Functions/DSB/Testis/SignalSurrounding;cd ${dirPATH}/CHMOrganization/Universal_specific/Functions/DSB/Testis/SignalSurrounding\n",
    "                        for sample in Juvenile Adult\n",
    "                        do\n",
    "                            # bash ${shPATH}/eachProfileSurroundingPeakCenter.sh \\\n",
    "                            #     ${dirPATH}/CHMOrganization/Universal_specific/ordered.bed \\\n",
    "                            #     10000 \\\n",
    "                            #     100 \\\n",
    "                            #     ${dirPATH}/CHMOrganization/Universal_specific/Functions/DSB/Testis/Testis_${sample}.bw \\\n",
    "                            #     ordered.Testis_${sample}\n",
    "                                \n",
    "                            Rscript ${rPATH}/heatmap_mat_1.r \\\n",
    "                                \"ordered.Testis_${sample}\" \\\n",
    "                                \"2.5\" \\\n",
    "                                \"10\" \\\n",
    "                                \"9;1\" \\\n",
    "                                \"0.8\" \\\n",
    "                                \"0\" \\\n",
    "                                \"0.8\" \\\n",
    "                                \"ordered.Testis_${sample}\" \\\n",
    "                                \"F\" \\\n",
    "                                \"$(seq 1 100)\" \\\n",
    "                                \"\" \\\n",
    "                                \"StableCHMs\" \\\n",
    "                                \"13287;13988;14732;15087;15686\" \\\n",
    "                                \"-007\" \\\n",
    "                                \"pdf\" \\\n",
    "                                \"#984EA3\" \\\n",
    "                                \"Reverse\"\n",
    "                                \n",
    "                            pdftoppm -png -r 720 ordered.Testis_${sample}.pdf ordered.Testis_${sample}\n",
    "                        done # for sample end\n",
    "                    }\n",
    "                    # signalsSurrounding\n",
    "                    \n",
    "                    avgSignalSurrounding(){\n",
    "                        # D: Oct-30-2022 22:59 Sun\n",
    "                        mkdir -p ${dirPATH}/CHMOrganization/Universal_specific/Functions/DSB/Testis/AvgSignalSurrounding;cd ${dirPATH}/CHMOrganization/Universal_specific/Functions/DSB/Testis/AvgSignalSurrounding\n",
    "                        for sample in Juvenile Adult\n",
    "                        do\n",
    "                            # for group in Universal EarlyEmbryoSpecific PGCSpecific SpermSpecific RetinalSpecific DCSpecific nonCHM_CGIs promoter enhancer nonCHM_CGIs_HMR\n",
    "                            for group in LTR LINE SINE\n",
    "                            do\n",
    "                                bash ${shPATH}/profileSurroundingPeakCenter.sh \\\n",
    "                                    ${dirPATH}/CHMOrganization/Universal_specific/${group}.bed \\\n",
    "                                    50000 \\\n",
    "                                    50000 \\\n",
    "                                    40 \\\n",
    "                                    ${dirPATH}/CHMOrganization/Universal_specific/Functions/DSB/Testis/Testis_${sample}.bw \\\n",
    "                                    ${group} \\\n",
    "                                    \"Testis_${sample}\" \\\n",
    "                                    \"-50kb;;50kb\" \\\n",
    "                                    \"0;20.5;40\" \\\n",
    "                                    \"DSB signal\" \\\n",
    "                                    \"0.4\" \\\n",
    "                                    \"0.9\" \\\n",
    "                                    \"Dark2\"\n",
    "                            done # for group end\n",
    "                        done # for sample end\n",
    "                    }\n",
    "                    avgSignalSurrounding\n",
    "                }\n",
    "                # Testis\n",
    "                \n",
    "                ESC(){\n",
    "                    # D: Oct-31-2022 00:02 Mon\n",
    "                    # T: Same to Tillo_NatComm_2021_ESC_GSE165161\n",
    "                    mkdir -p ${dirPATH}/CHMOrganization/Universal_specific/Functions/DSB/ESC;cd ${dirPATH}/CHMOrganization/Universal_specific/Functions/DSB/ESC\n",
    "                    download(){\n",
    "                        # T: \n",
    "                        wget -c \"https://ftp.ncbi.nlm.nih.gov/geo/samples/GSM5027nnn/GSM5027367/suppl/GSM5027367_ESC_DUX_Clone1_ENDseq_pos.mm10.RPKM.bw\" -O ESC.rep1.pos.bw\n",
    "                        wget -c \"https://ftp.ncbi.nlm.nih.gov/geo/samples/GSM5027nnn/GSM5027367/suppl/GSM5027367_ESC_DUX_Clone1_ENDseq_neg.mm10.RPKM.bw\" -O ESC.rep1.neg.bw\n",
    "                        wget -c \"https://ftp.ncbi.nlm.nih.gov/geo/samples/GSM5027nnn/GSM5027369/suppl/GSM5027369_ESC_DUX_Clone2_ENDseq_pos.mm10.RPKM.bw\" -O ESC.rep2.pos.bw\n",
    "                        wget -c \"https://ftp.ncbi.nlm.nih.gov/geo/samples/GSM5027nnn/GSM5027369/suppl/GSM5027369_ESC_DUX_Clone2_ENDseq_neg.mm10.RPKM.bw\" -O ESC.rep2.neg.bw\n",
    "                    }\n",
    "                    # download\n",
    "                    \n",
    "                    avgSignalSurrounding(){\n",
    "                        # D: Oct-31-2022 00:13 Mon\n",
    "                        mkdir -p ${dirPATH}/CHMOrganization/Universal_specific/Functions/DSB/ESC/AvgSignalSurrounding;cd ${dirPATH}/CHMOrganization/Universal_specific/Functions/DSB/ESC/AvgSignalSurrounding\n",
    "                        for rep in rep1 rep2\n",
    "                        do\n",
    "                            for type in pos neg\n",
    "                            do\n",
    "                                # for group in Universal EarlyEmbryoSpecific PGCSpecific SpermSpecific RetinalSpecific DCSpecific nonCHM_CGIs promoter enhancer nonCHM_CGIs_HMR\n",
    "                                for group in nonCHM_CGIs_HMR\n",
    "                                do\n",
    "                                    bash ${shPATH}/profileSurroundingPeakCenter.sh \\\n",
    "                                        ${dirPATH}/CHMOrganization/Universal_specific/${group}.bed \\\n",
    "                                        50000 \\\n",
    "                                        50000 \\\n",
    "                                        40 \\\n",
    "                                        ${dirPATH}/CHMOrganization/Universal_specific/Functions/DSB/ESC/ESC.${rep}.${type}.bw \\\n",
    "                                        ${group}.${type} \\\n",
    "                                        \"ESC.${rep}\" \\\n",
    "                                        \"-50kb;;50kb\" \\\n",
    "                                        \"0;20.5;40\" \\\n",
    "                                        \"DSB signal\" \\\n",
    "                                        \"0.05\" \\\n",
    "                                        \"0.06\" \\\n",
    "                                        \"Dark2\"\n",
    "                                done # for group end\n",
    "                            done # for type end\n",
    "                        done # for rep end\n",
    "                    }\n",
    "                    avgSignalSurrounding\n",
    "                }\n",
    "                # ESC\n",
    "                \n",
    "                BCell(){\n",
    "                    # D: Oct-31-2022 00:36 Mon\n",
    "                    mkdir -p ${dirPATH}/CHMOrganization/Universal_specific/Functions/DSB/BCell;cd ${dirPATH}/CHMOrganization/Universal_specific/Functions/DSB/BCell\n",
    "                    download(){\n",
    "                        wget -c \"https://ftp.ncbi.nlm.nih.gov/geo/samples/GSM4160nnn/GSM4160241/suppl/GSM4160241_190124_WT-BCell-NT.si.pos.bw\" -O BCell.pos.bw\n",
    "                        wget -c \"https://ftp.ncbi.nlm.nih.gov/geo/samples/GSM4160nnn/GSM4160241/suppl/GSM4160241_190124_WT-BCell-NT.si.neg.bw\" -O BCell.neg.bw\n",
    "                    }\n",
    "                    # download\n",
    "                    \n",
    "                    avgSignalSurrounding(){\n",
    "                        # D: Oct-31-2022 00:13 Mon\n",
    "                        mkdir -p ${dirPATH}/CHMOrganization/Universal_specific/Functions/DSB/BCell/AvgSignalSurrounding;cd ${dirPATH}/CHMOrganization/Universal_specific/Functions/DSB/BCell/AvgSignalSurrounding\n",
    "                            for type in pos neg\n",
    "                            do\n",
    "                                # for group in Universal EarlyEmbryoSpecific PGCSpecific SpermSpecific RetinalSpecific DCSpecific nonCHM_CGIs promoter enhancer nonCHM_CGIs_HMR\n",
    "                                for group in nonCHM_CGIs_HMR\n",
    "                                do\n",
    "                                    bash ${shPATH}/profileSurroundingPeakCenter.sh \\\n",
    "                                        ${dirPATH}/CHMOrganization/Universal_specific/${group}.bed \\\n",
    "                                        20000 \\\n",
    "                                        20000 \\\n",
    "                                        40 \\\n",
    "                                        ${dirPATH}/CHMOrganization/Universal_specific/Functions/DSB/BCell/BCell.${type}.bw \\\n",
    "                                        ${group}.${type} \\\n",
    "                                        \"BCell\" \\\n",
    "                                        \"-20kb;;20kb\" \\\n",
    "                                        \"0;20.5;40\" \\\n",
    "                                        \"DSB signal\" \\\n",
    "                                        \"-0.05\" \\\n",
    "                                        \"0.015\" \\\n",
    "                                        \"Dark2\"\n",
    "                                done # for group end\n",
    "                            done # for type end\n",
    "                    }\n",
    "                    # avgSignalSurrounding\n",
    "                }\n",
    "                # BCell\n",
    "                \n",
    "                Tubbs_Cell_2018_B_T_cell(){\n",
    "                    # D: Nov-06-2022 11:50 Sun\n",
    "                    mkdir -p ${dirPATH}/CHMOrganization/Universal_specific/Functions/DSB/Tubbs_Cell_2018;cd ${dirPATH}/CHMOrganization/Universal_specific/Functions/DSB/Tubbs_Cell_2018\n",
    "                    download(){\n",
    "                        # T: executed in 14m 14s, finished 12:14:20 2022-11-06\n",
    "                        wget -c \"https://ftp.ncbi.nlm.nih.gov/geo/samples/GSM3227nnn/GSM3227944/suppl/GSM3227944_END_seq_aB_28h_HU_28h_10mM_rep1.bw\" -O aB_28h_HU_28h_10mM_rep1.bw\n",
    "                        wget -c \"ftp://ftp.ncbi.nlm.nih.gov/geo/samples/GSM3227nnn/GSM3227945/suppl/GSM3227945_END_seq_aB_28h_HU_28h_10mM_rep2.bw\" -O aB_28h_HU_28h_10mM_rep2.bw\n",
    "                        wget -c \"https://ftp.ncbi.nlm.nih.gov/geo/samples/GSM3227nnn/GSM3227946/suppl/GSM3227946_END_seq_aB_28h_HU_6h_10mM_rep1.bw\" -O aB_28h_HU_6h_10mM_rep1.bw\n",
    "                        wget -c \"https://ftp.ncbi.nlm.nih.gov/geo/samples/GSM3227nnn/GSM3227947/suppl/GSM3227947_END_seq_aB_28h_NT_rep1.bw\" -O aB_28h_NT_rep1.bw\n",
    "                        wget -c \"https://ftp.ncbi.nlm.nih.gov/geo/samples/GSM3227nnn/GSM3227948/suppl/GSM3227948_END_seq_aB_28h_NT_rep2.bw\" -O aB_28h_NT_rep2.bw\n",
    "                        wget -c \"https://ftp.ncbi.nlm.nih.gov/geo/samples/GSM3227nnn/GSM3227949/suppl/GSM3227949_END_seq_aB_28h_HU_28h_0.5mM_rep1.bw\" -O aB_28h_HU_28h_0.5mM_rep1.bw\n",
    "                        wget -c \"https://ftp.ncbi.nlm.nih.gov/geo/samples/GSM3227nnn/GSM3227950/suppl/GSM3227950_END_seq_aT_28h_HU_28h_2mM_rep1.bw\" -O aT_28h_HU_28h_2mM_rep1.bw\n",
    "                        wget -c \"https://ftp.ncbi.nlm.nih.gov/geo/samples/GSM3227nnn/GSM3227953/suppl/GSM3227953_END_seq_aB_28h_CDC2i_CDC7i_0.5h_HU_6h_10mM_rep1.bw\" -O aB_28h_CDC2i_CDC7i_0.5h_HU_6h_10mM_rep1.bw\n",
    "                        wget -c \"https://ftp.ncbi.nlm.nih.gov/geo/samples/GSM3227nnn/GSM3227954/suppl/GSM3227954_END_seq_aB_28h_CDC2i_CDC7i_0.5h_HU_6h_0.5mM_rep1.bw\" -O aB_28h_CDC2i_CDC7i_0.5h_HU_6h_0.5mM_rep1.bw\n",
    "                        wget -c \"https://ftp.ncbi.nlm.nih.gov/geo/samples/GSM3227nnn/GSM3227955/suppl/GSM3227955_END_seq_aB_28h_ATRi_28h_rep1.bw\" -O aB_28h_ATRi_28h_rep1.bw\n",
    "                        wget -c \"https://ftp.ncbi.nlm.nih.gov/geo/samples/GSM3227nnn/GSM3227956/suppl/GSM3227956_END_seq_aB_28h_HU_28h_0.5mM_BL6_eij_rep1.bw\" -O aB_28h_HU_28h_0.5mM_BL6_eij_rep1.bw\n",
    "                        wget -c \"https://ftp.ncbi.nlm.nih.gov/geo/samples/GSM3227nnn/GSM3227957/suppl/GSM3227957_END_seq_aB_28h_HU_28h_0.5mM_CAST_eij_rep1.bw\" -O aB_28h_HU_28h_0.5mM_CAST_eij_rep1.bw\n",
    "                    }\n",
    "                    # download\n",
    "                    \n",
    "                    avgSignalSurrounding(){\n",
    "                        # D: Nov-06-2022 12:01 Sun\n",
    "                        # T: executed in 49m 57s, finished 13:07:59 2022-11-06\n",
    "                        mkdir -p ${dirPATH}/CHMOrganization/Universal_specific/Functions/DSB/Tubbs_Cell_2018/AvgSignalSurrounding;cd ${dirPATH}/CHMOrganization/Universal_specific/Functions/DSB/Tubbs_Cell_2018/AvgSignalSurrounding\n",
    "                        for sample in aB_28h_HU_28h_10mM_rep1 aB_28h_HU_28h_10mM_rep2 aB_28h_HU_6h_10mM_rep1 aB_28h_NT_rep1 aB_28h_NT_rep2 aB_28h_HU_28h_0.5mM_rep1 aT_28h_HU_28h_2mM_rep1 aB_28h_CDC2i_CDC7i_0.5h_HU_6h_10mM_rep1 aB_28h_CDC2i_CDC7i_0.5h_HU_6h_0.5mM_rep1 aB_28h_ATRi_28h_rep1 aB_28h_HU_28h_0.5mM_BL6_eij_rep1 aB_28h_HU_28h_0.5mM_CAST_eij_rep1\n",
    "                        do\n",
    "                            for group in Universal EarlyEmbryoSpecific PGCSpecific SpermSpecific RetinalSpecific DCSpecific nonCHM_CGIs promoter enhancer nonCHM_CGIs_HMR\n",
    "                            do\n",
    "                                bash ${shPATH}/profileSurroundingPeakCenter.sh \\\n",
    "                                    ${dirPATH}/CHMOrganization/Universal_specific/${group}.bed \\\n",
    "                                    50000 \\\n",
    "                                    50000 \\\n",
    "                                    40 \\\n",
    "                                    ${dirPATH}/CHMOrganization/Universal_specific/Functions/DSB/Tubbs_Cell_2018/${sample}.bw \\\n",
    "                                    ${group} \\\n",
    "                                    \"${sample}\" \\\n",
    "                                    \"-50kb;;50kb\" \\\n",
    "                                    \"0;20.5;40\" \\\n",
    "                                    \"DSB signal\" \\\n",
    "                                    \"0.4\" \\\n",
    "                                    \"0.9\" \\\n",
    "                                    \"Dark2\"\n",
    "                            done # for group end\n",
    "                            \n",
    "                            Rscript ${rPATH}/line_profileSurroundingPeak.r \\\n",
    "                                \"${sample}\" \\\n",
    "                                \"${sample}\" \\\n",
    "                                4 \\\n",
    "                                3.5 \\\n",
    "                                \"-50kb;;50kb\" \\\n",
    "                                \"0;20.5;40\" \\\n",
    "                                \"DSB signal\" \\\n",
    "                                \"0\" \\\n",
    "                                \"0.5\" \\\n",
    "                                \"${sample}\" \\\n",
    "                                \"Dark2\"\n",
    "                        done # for sample end\n",
    "                    }\n",
    "                    # avgSignalSurrounding\n",
    "                    \n",
    "                    avgSignalSurrounding_20kb_50(){\n",
    "                        # D: Nov-06-2022 14:04 Sun\n",
    "                        # T: \n",
    "                        mkdir -p ${dirPATH}/CHMOrganization/Universal_specific/Functions/DSB/Tubbs_Cell_2018/AvgSignalSurrounding_20kb_50;cd ${dirPATH}/CHMOrganization/Universal_specific/Functions/DSB/Tubbs_Cell_2018/AvgSignalSurrounding_20kb_50\n",
    "                        for sample in aB_28h_HU_28h_10mM_rep1 aB_28h_HU_28h_10mM_rep2 aB_28h_HU_6h_10mM_rep1 aB_28h_NT_rep1 aB_28h_NT_rep2 aB_28h_HU_28h_0.5mM_rep1 aT_28h_HU_28h_2mM_rep1 aB_28h_CDC2i_CDC7i_0.5h_HU_6h_10mM_rep1 aB_28h_CDC2i_CDC7i_0.5h_HU_6h_0.5mM_rep1 aB_28h_ATRi_28h_rep1 aB_28h_HU_28h_0.5mM_BL6_eij_rep1 aB_28h_HU_28h_0.5mM_CAST_eij_rep1\n",
    "                        do\n",
    "                            for group in Universal EarlyEmbryoSpecific PGCSpecific SpermSpecific RetinalSpecific DCSpecific nonCHM_CGIs promoter enhancer nonCHM_CGIs_HMR\n",
    "                            do\n",
    "                                bash ${shPATH}/profileSurroundingPeakCenter.sh \\\n",
    "                                    ${dirPATH}/CHMOrganization/Universal_specific/${group}.bed \\\n",
    "                                    20000 \\\n",
    "                                    20000 \\\n",
    "                                    50 \\\n",
    "                                    ${dirPATH}/CHMOrganization/Universal_specific/Functions/DSB/Tubbs_Cell_2018/${sample}.bw \\\n",
    "                                    ${group} \\\n",
    "                                    \"${sample}\" \\\n",
    "                                    \"-20kb;;20kb\" \\\n",
    "                                    \"0;25.5;50\" \\\n",
    "                                    \"DSB signal\" \\\n",
    "                                    \"0.02\" \\\n",
    "                                    \"0.5\" \\\n",
    "                                    \"Dark2\"\n",
    "                            done # for group end\n",
    "                            \n",
    "                            Rscript ${rPATH}/line_profileSurroundingPeak.r \\\n",
    "                                \"${sample}\" \\\n",
    "                                \"${sample}\" \\\n",
    "                                4 \\\n",
    "                                3.5 \\\n",
    "                                \"-20kb;;20kb\" \\\n",
    "                                \"0;25.5;50\" \\\n",
    "                                \"DSB signal\" \\\n",
    "                                \"0\" \\\n",
    "                                \"0.5\" \\\n",
    "                                \"${sample}\" \\\n",
    "                                \"Dark2\"\n",
    "                        done # for sample end\n",
    "                    }\n",
    "                    # avgSignalSurrounding_20kb_50\n",
    "                    \n",
    "                    avgSignalSurrounding_10kb_50(){\n",
    "                        # D: Nov-06-2022 19:23 Sun\n",
    "                        # T: executed in 5m 23s, finished 19:29:44 2022-11-06\n",
    "                        mkdir -p ${dirPATH}/CHMOrganization/Universal_specific/Functions/DSB/Tubbs_Cell_2018/AvgSignalSurrounding_10kb_50;cd ${dirPATH}/CHMOrganization/Universal_specific/Functions/DSB/Tubbs_Cell_2018/AvgSignalSurrounding_10kb_50\n",
    "                        # for sample in aB_28h_HU_28h_10mM_rep1 aB_28h_HU_28h_10mM_rep2 aB_28h_HU_6h_10mM_rep1 aB_28h_NT_rep1 aB_28h_NT_rep2 aB_28h_HU_28h_0.5mM_rep1 aT_28h_HU_28h_2mM_rep1 aB_28h_CDC2i_CDC7i_0.5h_HU_6h_10mM_rep1 aB_28h_CDC2i_CDC7i_0.5h_HU_6h_0.5mM_rep1 aB_28h_ATRi_28h_rep1 aB_28h_HU_28h_0.5mM_BL6_eij_rep1 aB_28h_HU_28h_0.5mM_CAST_eij_rep1\n",
    "                        # do\n",
    "                        #     # for group in Universal EarlyEmbryoSpecific PGCSpecific SpermSpecific RetinalSpecific DCSpecific nonCHM_CGIs promoter nonCHM_CGIs_HMR LTR LINE SINE\n",
    "                        #     for group in LTR LINE SINE\n",
    "                        #     do\n",
    "                        #         bash ${shPATH}/profileSurroundingPeakCenter.sh \\\n",
    "                        #             ${dirPATH}/CHMOrganization/Universal_specific/${group}.bed \\\n",
    "                        #             10000 \\\n",
    "                        #             10000 \\\n",
    "                        #             50 \\\n",
    "                        #             ${dirPATH}/CHMOrganization/Universal_specific/Functions/DSB/Tubbs_Cell_2018/${sample}.bw \\\n",
    "                        #             ${group} \\\n",
    "                        #             \"${sample}\" \\\n",
    "                        #             \"-10kb;;10kb\" \\\n",
    "                        #             \"0;25.5;50\" \\\n",
    "                        #             \"DSB signal\" \\\n",
    "                        #             \"0.02\" \\\n",
    "                        #             \"0.5\" \\\n",
    "                        #             \"Dark2\"\n",
    "                        #     done # for group end\n",
    "                        # done # for sample end\n",
    "                        \n",
    "                        each_plot(){\n",
    "                            # D: Nov-06-2022 19:35 Sun\n",
    "                            # aB_28h_HU_28h_10mM_rep1\n",
    "                            Rscript ${rPATH}/line_profileSurroundingPeak.r \\\n",
    "                                \"aB_28h_HU_28h_10mM_rep1\" \\\n",
    "                                \"aB_28h_HU_28h_10mM_rep1\" \\\n",
    "                                4 \\\n",
    "                                3.5 \\\n",
    "                                \"-10kb;;10kb\" \\\n",
    "                                \"0;25.5;50\" \\\n",
    "                                \"DSB signal\" \\\n",
    "                                \"0.04\" \\\n",
    "                                \"0.12\" \\\n",
    "                                \"aB_28h_HU_28h_10mM_rep1\" \\\n",
    "                                \"Dark2\"\n",
    "                                \n",
    "                            # aB_28h_HU_28h_10mM_rep2\n",
    "                            Rscript ${rPATH}/line_profileSurroundingPeak.r \\\n",
    "                                \"aB_28h_HU_28h_10mM_rep2\" \\\n",
    "                                \"aB_28h_HU_28h_10mM_rep2\" \\\n",
    "                                4 \\\n",
    "                                3.5 \\\n",
    "                                \"-10kb;;10kb\" \\\n",
    "                                \"0;25.5;50\" \\\n",
    "                                \"DSB signal\" \\\n",
    "                                \"0.04\" \\\n",
    "                                \"0.12\" \\\n",
    "                                \"aB_28h_HU_28h_10mM_rep2\" \\\n",
    "                                \"Dark2\"\n",
    "                                \n",
    "                            # aB_28h_HU_6h_10mM_rep1\n",
    "                            Rscript ${rPATH}/line_profileSurroundingPeak.r \\\n",
    "                                \"aB_28h_HU_6h_10mM_rep1\" \\\n",
    "                                \"aB_28h_HU_6h_10mM_rep1\" \\\n",
    "                                4 \\\n",
    "                                3.5 \\\n",
    "                                \"-10kb;;10kb\" \\\n",
    "                                \"0;25.5;50\" \\\n",
    "                                \"DSB signal\" \\\n",
    "                                \"0.027\" \\\n",
    "                                \"0.056\" \\\n",
    "                                \"aB_28h_HU_6h_10mM_rep1\" \\\n",
    "                                \"Dark2\"\n",
    "                        \n",
    "                            # aB_28h_NT_rep1\n",
    "                            Rscript ${rPATH}/line_profileSurroundingPeak.r \\\n",
    "                                \"aB_28h_NT_rep1\" \\\n",
    "                                \"aB_28h_NT_rep1\" \\\n",
    "                                4 \\\n",
    "                                3.5 \\\n",
    "                                \"-10kb;;10kb\" \\\n",
    "                                \"0;25.5;50\" \\\n",
    "                                \"DSB signal\" \\\n",
    "                                \"0.03\" \\\n",
    "                                \"0.09\" \\\n",
    "                                \"aB_28h_NT_rep1\" \\\n",
    "                                \"Dark2\"\n",
    "                        \n",
    "                            # aB_28h_NT_rep2\n",
    "                            Rscript ${rPATH}/line_profileSurroundingPeak.r \\\n",
    "                                \"aB_28h_NT_rep2\" \\\n",
    "                                \"aB_28h_NT_rep2\" \\\n",
    "                                4 \\\n",
    "                                3.5 \\\n",
    "                                \"-10kb;;10kb\" \\\n",
    "                                \"0;25.5;50\" \\\n",
    "                                \"DSB signal\" \\\n",
    "                                \"0.04\" \\\n",
    "                                \"0.08\" \\\n",
    "                                \"aB_28h_NT_rep2\" \\\n",
    "                                \"Dark2\"\n",
    "\n",
    "                            # aB_28h_HU_28h_0.5mM_rep1\n",
    "                            Rscript ${rPATH}/line_profileSurroundingPeak.r \\\n",
    "                                \"aB_28h_HU_28h_0.5mM_rep1\" \\\n",
    "                                \"aB_28h_HU_28h_0.5mM_rep1\" \\\n",
    "                                4 \\\n",
    "                                3.5 \\\n",
    "                                \"-10kb;;10kb\" \\\n",
    "                                \"0;25.5;50\" \\\n",
    "                                \"DSB signal\" \\\n",
    "                                \"0.03\" \\\n",
    "                                \"0.13\" \\\n",
    "                                \"aB_28h_HU_28h_0.5mM_rep1\" \\\n",
    "                                \"Dark2\"\n",
    "                        \n",
    "                            # aT_28h_HU_28h_2mM_rep1\n",
    "                            Rscript ${rPATH}/line_profileSurroundingPeak.r \\\n",
    "                                \"aT_28h_HU_28h_2mM_rep1\" \\\n",
    "                                \"aT_28h_HU_28h_2mM_rep1\" \\\n",
    "                                4 \\\n",
    "                                3.5 \\\n",
    "                                \"-10kb;;10kb\" \\\n",
    "                                \"0;25.5;50\" \\\n",
    "                                \"DSB signal\" \\\n",
    "                                \"0.045\" \\\n",
    "                                \"0.065\" \\\n",
    "                                \"aT_28h_HU_28h_2mM_rep1\" \\\n",
    "                                \"Dark2\"\n",
    "                                \n",
    "                            # aB_28h_CDC2i_CDC7i_0.5h_HU_6h_10mM_rep1\n",
    "                            Rscript ${rPATH}/line_profileSurroundingPeak.r \\\n",
    "                                \"aB_28h_CDC2i_CDC7i_0.5h_HU_6h_10mM_rep1\" \\\n",
    "                                \"aB_28h_CDC2i_CDC7i_0.5h_HU_6h_10mM_rep1\" \\\n",
    "                                4 \\\n",
    "                                3.5 \\\n",
    "                                \"-10kb;;10kb\" \\\n",
    "                                \"0;25.5;50\" \\\n",
    "                                \"DSB signal\" \\\n",
    "                                \"0.06\" \\\n",
    "                                \"0.16\" \\\n",
    "                                \"aB_28h_CDC2i_CDC7i_0.5h_HU_6h_10mM_rep1\" \\\n",
    "                                \"Dark2\"\n",
    "                                \n",
    "                            # aB_28h_CDC2i_CDC7i_0.5h_HU_6h_0.5mM_rep1\n",
    "                            Rscript ${rPATH}/line_profileSurroundingPeak.r \\\n",
    "                                \"aB_28h_CDC2i_CDC7i_0.5h_HU_6h_0.5mM_rep1\" \\\n",
    "                                \"aB_28h_CDC2i_CDC7i_0.5h_HU_6h_0.5mM_rep1\" \\\n",
    "                                4 \\\n",
    "                                3.5 \\\n",
    "                                \"-10kb;;10kb\" \\\n",
    "                                \"0;25.5;50\" \\\n",
    "                                \"DSB signal\" \\\n",
    "                                \"0.05\" \\\n",
    "                                \"0.15\" \\\n",
    "                                \"aB_28h_CDC2i_CDC7i_0.5h_HU_6h_0.5mM_rep1\" \\\n",
    "                                \"Dark2\"\n",
    "                                \n",
    "                            # aB_28h_ATRi_28h_rep1\n",
    "                            Rscript ${rPATH}/line_profileSurroundingPeak.r \\\n",
    "                                \"aB_28h_ATRi_28h_rep1\" \\\n",
    "                                \"aB_28h_ATRi_28h_rep1\" \\\n",
    "                                4 \\\n",
    "                                3.5 \\\n",
    "                                \"-10kb;;10kb\" \\\n",
    "                                \"0;25.5;50\" \\\n",
    "                                \"DSB signal\" \\\n",
    "                                \"0.038\" \\\n",
    "                                \"0.088\" \\\n",
    "                                \"aB_28h_ATRi_28h_rep1\" \\\n",
    "                                \"Dark2\"\n",
    "                                \n",
    "                            # aB_28h_HU_28h_0.5mM_BL6_eij_rep1\n",
    "                            Rscript ${rPATH}/line_profileSurroundingPeak.r \\\n",
    "                                \"aB_28h_HU_28h_0.5mM_BL6_eij_rep1\" \\\n",
    "                                \"aB_28h_HU_28h_0.5mM_BL6_eij_rep1\" \\\n",
    "                                4 \\\n",
    "                                3.5 \\\n",
    "                                \"-10kb;;10kb\" \\\n",
    "                                \"0;25.5;50\" \\\n",
    "                                \"DSB signal\" \\\n",
    "                                \"0.09\" \\\n",
    "                                \"0.16\" \\\n",
    "                                \"aB_28h_HU_28h_0.5mM_BL6_eij_rep1\" \\\n",
    "                                \"Dark2\"\n",
    "                                \n",
    "                            # aB_28h_HU_28h_0.5mM_CAST_eij_rep1\n",
    "                            Rscript ${rPATH}/line_profileSurroundingPeak.r \\\n",
    "                                \"aB_28h_HU_28h_0.5mM_CAST_eij_rep1\" \\\n",
    "                                \"aB_28h_HU_28h_0.5mM_CAST_eij_rep1\" \\\n",
    "                                4 \\\n",
    "                                3.5 \\\n",
    "                                \"-10kb;;10kb\" \\\n",
    "                                \"0;25.5;50\" \\\n",
    "                                \"DSB signal\" \\\n",
    "                                \"0.15\" \\\n",
    "                                \"0.25\" \\\n",
    "                                \"aB_28h_HU_28h_0.5mM_CAST_eij_rep1\" \\\n",
    "                                \"Dark2\"\n",
    "                        }\n",
    "                        each_plot\n",
    "                    }\n",
    "                    # avgSignalSurrounding_10kb_50\n",
    "                }\n",
    "                # Tubbs_Cell_2018_B_T_cell\n",
    "                \n",
    "                Nacson_unpublished_MEF_GSE147468(){\n",
    "                    # D: Nov-06-2022 19:28 Sun\n",
    "                    # I: https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSM4431523\n",
    "                    mkdir -p ${dirPATH}/CHMOrganization/Universal_specific/Functions/DSB/Nacson_unpublished_MEF_GSE147468;cd ${dirPATH}/CHMOrganization/Universal_specific/Functions/DSB/Nacson_unpublished_MEF_GSE147468\n",
    "                    download(){\n",
    "                        wget -c \"https://ftp.ncbi.nlm.nih.gov/geo/samples/GSM4431nnn/GSM4431520/suppl/GSM4431520_WT_END-seq.pos.bw\" -O MEF.WT.pos.bw\n",
    "                        wget -c \"https://ftp.ncbi.nlm.nih.gov/geo/samples/GSM4431nnn/GSM4431520/suppl/GSM4431520_WT_END-seq.neg.bw\" -O MEF.WT.neg.bw\n",
    "                        wget -c \"https://ftp.ncbi.nlm.nih.gov/geo/samples/GSM4431nnn/GSM4431521/suppl/GSM4431521_CC_END-seq.pos.bw\" -O MEF.CC.pos.bw\n",
    "                        wget -c \"https://ftp.ncbi.nlm.nih.gov/geo/samples/GSM4431nnn/GSM4431521/suppl/GSM4431521_CC_END-seq.neg.bw\" -O MEF.CC.neg.bw\n",
    "                        wget -c \"https://ftp.ncbi.nlm.nih.gov/geo/samples/GSM4431nnn/GSM4431522/suppl/GSM4431522_Del11_END-seq.pos.bw\" -O MEF.Del11.pos.bw\n",
    "                        wget -c \"https://ftp.ncbi.nlm.nih.gov/geo/samples/GSM4431nnn/GSM4431522/suppl/GSM4431522_Del11_END-seq.neg.bw\" -O MEF.Del11.neg.bw\n",
    "                        wget -c \"https://ftp.ncbi.nlm.nih.gov/geo/samples/GSM4431nnn/GSM4431523/suppl/GSM4431523_Del11_CC_END-seq.pos.bw\" -O MEF.Del11_CC.pos.bw\n",
    "                        wget -c \"https://ftp.ncbi.nlm.nih.gov/geo/samples/GSM4431nnn/GSM4431523/suppl/GSM4431523_Del11_CC_END-seq.neg.bw\" -O MEF.Del11_CC.neg.bw\n",
    "                    }\n",
    "                    # download\n",
    "                    \n",
    "                    avgSignalSurrounding_10kb_50(){\n",
    "                        # D: Nov-06-2022 21:26 Sun\n",
    "                        mkdir -p ${dirPATH}/CHMOrganization/Universal_specific/Functions/DSB/Nacson_unpublished_MEF_GSE147468/AvgSignalSurrounding_10kb_50;cd ${dirPATH}/CHMOrganization/Universal_specific/Functions/DSB/Nacson_unpublished_MEF_GSE147468/AvgSignalSurrounding_10kb_50\n",
    "                        pos_part(){\n",
    "                            for sample in WT CC Del11 Del11_CC\n",
    "                            do\n",
    "                                for group in Universal EarlyEmbryoSpecific PGCSpecific SpermSpecific RetinalSpecific DCSpecific nonCHM_CGIs promoter nonCHM_CGIs_HMR\n",
    "                                do\n",
    "                                    bash ${shPATH}/profileSurroundingPeakCenter.sh \\\n",
    "                                        ${dirPATH}/CHMOrganization/Universal_specific/${group}.bed \\\n",
    "                                        10000 \\\n",
    "                                        10000 \\\n",
    "                                        50 \\\n",
    "                                        ${dirPATH}/CHMOrganization/Universal_specific/Functions/DSB/Nacson_unpublished_MEF_GSE147468/MEF.${sample}.pos.bw \\\n",
    "                                        ${group} \\\n",
    "                                        \"MEF.${sample}.pos\" \\\n",
    "                                        \"-10kb;;10kb\" \\\n",
    "                                        \"0;25.5;50\" \\\n",
    "                                        \"DSB signal\" \\\n",
    "                                        \"0.02\" \\\n",
    "                                        \"0.5\" \\\n",
    "                                        \"Dark2\"\n",
    "                                done # for group end\n",
    "                            done # for sample end                       \n",
    "                        }\n",
    "                        pos_part\n",
    "                        \n",
    "                        neg_part(){\n",
    "                            for sample in WT CC Del11 Del11_CC\n",
    "                            do\n",
    "                                for group in Universal EarlyEmbryoSpecific PGCSpecific SpermSpecific RetinalSpecific DCSpecific nonCHM_CGIs promoter nonCHM_CGIs_HMR\n",
    "                                do\n",
    "                                    bash ${shPATH}/profileSurroundingPeakCenter.sh \\\n",
    "                                        ${dirPATH}/CHMOrganization/Universal_specific/${group}.bed \\\n",
    "                                        10000 \\\n",
    "                                        10000 \\\n",
    "                                        50 \\\n",
    "                                        ${dirPATH}/CHMOrganization/Universal_specific/Functions/DSB/Nacson_unpublished_MEF_GSE147468/MEF.${sample}.neg.bw \\\n",
    "                                        ${group} \\\n",
    "                                        \"MEF.${sample}.neg\" \\\n",
    "                                        \"-10kb;;10kb\" \\\n",
    "                                        \"0;25.5;50\" \\\n",
    "                                        \"DSB signal\" \\\n",
    "                                        \"-0.5\" \\\n",
    "                                        \"-0.02\" \\\n",
    "                                        \"Dark2\"\n",
    "                                done # for group end\n",
    "                            done # for sample end                       \n",
    "                        }\n",
    "                        neg_part\n",
    "                    }\n",
    "                    # avgSignalSurrounding_10kb_50\n",
    "                    \n",
    "                    each_plot(){\n",
    "                        # D: Nov-07-2022 10:59 Mon\n",
    "                        # MEF.WT.pos\n",
    "                        Rscript ${rPATH}/line_profileSurroundingPeak.r \\\n",
    "                            \"MEF.WT.pos\" \\\n",
    "                            \"MEF.WT.pos\" \\\n",
    "                            4 \\\n",
    "                            3.5 \\\n",
    "                            \"-10kb;;10kb\" \\\n",
    "                            \"0;25.5;50\" \\\n",
    "                            \"DSB signal\" \\\n",
    "                            \"0.04\" \\\n",
    "                            \"0.2\" \\\n",
    "                            \"MEF.WT.pos\" \\\n",
    "                            \"Dark2\"\n",
    "                            \n",
    "                        # MEF.CC.pos\n",
    "                        Rscript ${rPATH}/line_profileSurroundingPeak.r \\\n",
    "                            \"MEF.CC.pos\" \\\n",
    "                            \"MEF.CC.pos\" \\\n",
    "                            4 \\\n",
    "                            3.5 \\\n",
    "                            \"-10kb;;10kb\" \\\n",
    "                            \"0;25.5;50\" \\\n",
    "                            \"DSB signal\" \\\n",
    "                            \"0.05\" \\\n",
    "                            \"0.2\" \\\n",
    "                            \"MEF.CC.pos\" \\\n",
    "                            \"Dark2\"\n",
    "                            \n",
    "                        # MEF.Del11.pos\n",
    "                        Rscript ${rPATH}/line_profileSurroundingPeak.r \\\n",
    "                            \"MEF.Del11.pos\" \\\n",
    "                            \"MEF.Del11.pos\" \\\n",
    "                            4 \\\n",
    "                            3.5 \\\n",
    "                            \"-10kb;;10kb\" \\\n",
    "                            \"0;25.5;50\" \\\n",
    "                            \"DSB signal\" \\\n",
    "                            \"0.06\" \\\n",
    "                            \"0.2\" \\\n",
    "                            \"MEF.Del11.pos\" \\\n",
    "                            \"Dark2\"\n",
    "                            \n",
    "                        # MEF.Del11_CC.pos\n",
    "                        Rscript ${rPATH}/line_profileSurroundingPeak.r \\\n",
    "                            \"MEF.Del11_CC.pos\" \\\n",
    "                            \"MEF.Del11_CC.pos\" \\\n",
    "                            4 \\\n",
    "                            3.5 \\\n",
    "                            \"-10kb;;10kb\" \\\n",
    "                            \"0;25.5;50\" \\\n",
    "                            \"DSB signal\" \\\n",
    "                            \"0.04\" \\\n",
    "                            \"0.2\" \\\n",
    "                            \"MEF.Del11_CC.pos\" \\\n",
    "                            \"Dark2\"\n",
    "                            \n",
    "                        # MEF.WT.neg\n",
    "                        Rscript ${rPATH}/line_profileSurroundingPeak.r \\\n",
    "                            \"MEF.WT.neg\" \\\n",
    "                            \"MEF.WT.neg\" \\\n",
    "                            4 \\\n",
    "                            3.5 \\\n",
    "                            \"-10kb;;10kb\" \\\n",
    "                            \"0;25.5;50\" \\\n",
    "                            \"DSB signal\" \\\n",
    "                            \"-0.04\" \\\n",
    "                            \"-0.2\" \\\n",
    "                            \"MEF.WT.neg\" \\\n",
    "                            \"Dark2\"\n",
    "                            \n",
    "                        # MEF.CC.neg\n",
    "                        Rscript ${rPATH}/line_profileSurroundingPeak.r \\\n",
    "                            \"MEF.CC.neg\" \\\n",
    "                            \"MEF.CC.neg\" \\\n",
    "                            4 \\\n",
    "                            3.5 \\\n",
    "                            \"-10kb;;10kb\" \\\n",
    "                            \"0;25.5;50\" \\\n",
    "                            \"DSB signal\" \\\n",
    "                            \"-0.05\" \\\n",
    "                            \"-0.2\" \\\n",
    "                            \"MEF.CC.neg\" \\\n",
    "                            \"Dark2\"\n",
    "                            \n",
    "                        # MEF.Del11.neg\n",
    "                        Rscript ${rPATH}/line_profileSurroundingPeak.r \\\n",
    "                            \"MEF.Del11.neg\" \\\n",
    "                            \"MEF.Del11.neg\" \\\n",
    "                            4 \\\n",
    "                            3.5 \\\n",
    "                            \"-10kb;;10kb\" \\\n",
    "                            \"0;25.5;50\" \\\n",
    "                            \"DSB signal\" \\\n",
    "                            \"-0.05\" \\\n",
    "                            \"-0.2\" \\\n",
    "                            \"MEF.Del11.neg\" \\\n",
    "                            \"Dark2\"\n",
    "                            \n",
    "                        # MEF.Del11_CC.neg\n",
    "                        Rscript ${rPATH}/line_profileSurroundingPeak.r \\\n",
    "                            \"MEF.Del11_CC.neg\" \\\n",
    "                            \"MEF.Del11_CC.neg\" \\\n",
    "                            4 \\\n",
    "                            3.5 \\\n",
    "                            \"-10kb;;10kb\" \\\n",
    "                            \"0;25.5;50\" \\\n",
    "                            \"DSB signal\" \\\n",
    "                            \"-0.04\" \\\n",
    "                            \"-0.2\" \\\n",
    "                            \"MEF.Del11_CC.neg\" \\\n",
    "                            \"Dark2\"\n",
    "                    }\n",
    "                    # each_plot\n",
    "                    \n",
    "                    each_plot_auto(){\n",
    "                        # D: Nov-07-2022 11:22 Mon\n",
    "                        mkdir -p ${dirPATH}/CHMOrganization/Universal_specific/Functions/DSB/Nacson_unpublished_MEF_GSE147468/AvgSignalSurrounding_10kb_50;cd ${dirPATH}/CHMOrganization/Universal_specific/Functions/DSB/Nacson_unpublished_MEF_GSE147468/AvgSignalSurrounding_10kb_50\n",
    "                        for sample in WT CC Del11 Del11_CC\n",
    "                        do\n",
    "                            for type in pos neg\n",
    "                            do\n",
    "                                Rscript ${rPATH}/line_profileSurroundingPeak_auto.r \\\n",
    "                                    \"MEF.${sample}.${type}\" \\\n",
    "                                    \"MEF.${sample}.${type}\" \\\n",
    "                                    4 \\\n",
    "                                    3.5 \\\n",
    "                                    \"-10kb;;10kb\" \\\n",
    "                                    \"0;25.5;50\" \\\n",
    "                                    \"DSB signal\" \\\n",
    "                                    \"MEF.${sample}.${type}\" \\\n",
    "                                    \"Dark2\"\n",
    "                            done # for type end\n",
    "                        done # for sample end\n",
    "                    }\n",
    "                    each_plot_auto\n",
    "                }\n",
    "                # Nacson_unpublished_MEF_GSE147468\n",
    "                \n",
    "                Sciascia_Elife_2020_BCell_GSE140372(){\n",
    "                    # D: Nov-06-2022 21:34 Sun\n",
    "                    # T: https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE140372\n",
    "                    mkdir -p ${dirPATH}/CHMOrganization/Universal_specific/Functions/DSB/Sciascia_Elife_2020_BCell_GSE140372;cd ${dirPATH}/CHMOrganization/Universal_specific/Functions/DSB/Sciascia_Elife_2020_BCell_GSE140372\n",
    "                    download(){\n",
    "                        wget -c \"https://www.ncbi.nlm.nih.gov/geo/download/?acc=GSE140372&format=file\"\n",
    "                    }\n",
    "                    # download\n",
    "                    \n",
    "                    manage(){\n",
    "                        # D: Nov-07-2022 08:58 Mon\n",
    "                        # T: executed in 332ms, finished 09:12:27 2022-11-07\n",
    "                        # tar -xvf \"index.html?acc=GSE140372&format=file\" && mv \"index.html?acc=GSE140372&format=file\" GSE140372_RAW.tar\n",
    "                        mv GSM4160241_190124_WT-BCell-NT.si.neg.bw GSM4160241_WT_BCell_NT.neg.bw\n",
    "                        mv GSM4160241_190124_WT-BCell-NT.si.pos.bw GSM4160241_WT_BCell_NT.pos.bw\n",
    "                        mv GSM4160242_190124_WT-BCell-ETO.si.neg.bw GSM4160242_WT_BCell_ETO.neg.bw\n",
    "                        mv GSM4160242_190124_WT-BCell-ETO.si.pos.bw GSM4160242_WT_BCell_ETO.pos.bw\n",
    "                        mv GSM4160243_190124_WT-BCell-ETO-WO.si.neg.bw GSM4160243_WT_BCell_ETO_WO.neg.bw\n",
    "                        mv GSM4160243_190124_WT-BCell-ETO-WO.si.pos.bw GSM4160243_WT_BCell_ETO_WO.pos.bw\n",
    "                        mv GSM4160244_190124_WT-BCell-ETO-WO-2h.si.neg.bw GSM4160244_WT_BCell_ETO_WO_2h.neg.bw\n",
    "                        mv GSM4160244_190124_WT-BCell-ETO-WO-2h.si.pos.bw GSM4160244_WT_BCell_ETO_WO_2h.pos.bw\n",
    "                        mv GSM4160245_190124_WT-BCell-BTZ.si.neg.bw GSM4160245_WT_BCell_BTZ.neg.bw\n",
    "                        mv GSM4160245_190124_WT-BCell-BTZ.si.pos.bw GSM4160245_WT_BCell_BTZ.pos.bw\n",
    "                        mv GSM4160246_190124_WT-BCell-BTZ-ETO.si.neg.bw GSM4160246_WT_BCell_BTZ_ETO.neg.bw\n",
    "                        mv GSM4160246_190124_WT-BCell-BTZ-ETO.si.pos.bw GSM4160246_WT_BCell_BTZ_ETO.pos.bw\n",
    "                        mv GSM4160247_190124_WT-BCell-BTZ-ETO-WO.si.neg.bw GSM4160247_WT_BCell_BTZ_ETO_WO.neg.bw\n",
    "                        mv GSM4160247_190124_WT-BCell-BTZ-ETO-WO.si.pos.bw GSM4160247_WT_BCell_BTZ_ETO_WO.pos.bw\n",
    "                        mv GSM4160248_190124_WT-BCell-BTZ-ETO-WO-2h.si.neg.bw GSM4160248_WT_BCell_BTZ_ETO_WO_2h.neg.bw\n",
    "                        mv GSM4160248_190124_WT-BCell-BTZ-ETO-WO-2h.si.pos.bw GSM4160248_WT_BCell_BTZ_ETO_WO_2h.pos.bw\n",
    "                        mv GSM4160249_190810_WT-BCell-ETO-E7.si.neg.bw GSM4160249_WT_BCell_ETO_E7.neg.bw\n",
    "                        mv GSM4160249_190810_WT-BCell-ETO-E7.si.pos.bw GSM4160249_WT_BCell_ETO_E7.pos.bw\n",
    "                        mv GSM4160250_190810_WT-BCell-ETO-ExoT.si.neg.bw GSM4160250_WT_BCell_ETO_ExoT.neg.bw\n",
    "                        mv GSM4160250_190810_WT-BCell-ETO-ExoT.si.pos.bw GSM4160250_WT_BCell_ETO_ExoT.pos.bw\n",
    "                        mv GSM4160251_190810_WT-BCell-ETO-WO-0h-E7.si.neg.bw GSM4160251_WT_BCell_ETO_WO_0h_E7.neg.bw\n",
    "                        mv GSM4160251_190810_WT-BCell-ETO-WO-0h-E7.si.pos.bw GSM4160251_WT_BCell_ETO_WO_0h_E7.pos.bw\n",
    "                        mv GSM4160252_190810_WT-BCell-ETO-WO-0h-ExoT.si.neg.bw GSM4160252_WT_BCell_ETO_WO_0h_ExoT.neg.bw\n",
    "                        mv GSM4160252_190810_WT-BCell-ETO-WO-0h-ExoT.si.pos.bw GSM4160252_WT_BCell_ETO_WO_0h_ExoT.pos.bw\n",
    "                        mv GSM4160253_190810_WT-BCell-BTZ-ETO-E7.si.neg.bw GSM4160253_WT_BCell_BTZ_ETO_E7.neg.bw\n",
    "                        mv GSM4160253_190810_WT-BCell-BTZ-ETO-E7.si.pos.bw GSM4160253_WT_BCell_BTZ_ETO_E7.pos.bw\n",
    "                        mv GSM4160254_190810_WT-BCell-BTZ-ETO-ExoT.si.neg.bw GSM4160254_WT_BCell_BTZ_ETO_ExoT.neg.bw\n",
    "                        mv GSM4160254_190810_WT-BCell-BTZ-ETO-ExoT.si.pos.bw GSM4160254_WT_BCell_BTZ_ETO_ExoT.pos.bw\n",
    "                        mv GSM4160255_190810_WT-BCell-BTZ-ETO-WO-0h-E7.si.neg.bw GSM4160255_WT_BCell_BTZ_ETO_WO_0h_E7.neg.bw\n",
    "                        mv GSM4160255_190810_WT-BCell-BTZ-ETO-WO-0h-E7.si.pos.bw GSM4160255_WT_BCell_BTZ_ETO_WO_0h_E7.pos.bw\n",
    "                        mv GSM4160256_190810_WT-BCell-BTZ-ETO-WO-0h-ExoT.si.neg.bw GSM4160256_WT_BCell_BTZ_ETO_WO_0h_ExoT.neg.bw\n",
    "                        mv GSM4160256_190810_WT-BCell-BTZ-ETO-WO-0h-ExoT.si.pos.bw GSM4160256_WT_BCell_BTZ_ETO_WO_0h_ExoT.pos.bw\n",
    "                        mv GSM4160257_190518_WT-BCell-1h-BTZ-ETO.si.neg.bw GSM4160257_WT_BCell_1h_BTZ_ETO.neg.bw\n",
    "                        mv GSM4160257_190518_WT-BCell-1h-BTZ-ETO.si.pos.bw GSM4160257_WT_BCell_1h_BTZ_ETO.pos.bw\n",
    "                        mv GSM4160258_190518_WT-BCell-1h-BTZ-ETO-WO-2h.si.neg.bw GSM4160258_WT_BCell_1h_BTZ_ETO_WO_2h.neg.bw\n",
    "                        mv GSM4160258_190518_WT-BCell-1h-BTZ-ETO-WO-2h.si.pos.bw GSM4160258_WT_BCell_1h_BTZ_ETO_WO_2h.pos.bw\n",
    "                        mv GSM4160259_190518_WT-BCell-BTZ-ETO.si.neg.bw GSM4160259_WT_BCell_BTZ_ETO.neg.bw\n",
    "                        mv GSM4160259_190518_WT-BCell-BTZ-ETO.si.pos.bw GSM4160259_WT_BCell_BTZ_ETO.pos.bw\n",
    "                        mv GSM4160260_190518_WT-BCell-BTZ-ETO-WO-2h.si.neg.bw GSM4160260_WT_BCell_BTZ_ETO_WO_2h.neg.bw\n",
    "                        mv GSM4160260_190518_WT-BCell-BTZ-ETO-WO-2h.si.pos.bw GSM4160260_WT_BCell_BTZ_ETO_WO_2h.pos.bw\n",
    "                        mv GSM4160261_190518_WT-BCell-ETO-2h-BTZ-WO.si.neg.bw GSM4160261_WT_BCell_ETO_2h_BTZ_WO.neg.bw\n",
    "                        mv GSM4160261_190518_WT-BCell-ETO-2h-BTZ-WO.si.pos.bw GSM4160261_WT_BCell_ETO_2h_BTZ_WO.pos.bw\n",
    "                        mv GSM4160262_190518_WT-BCell-ETO.si.neg.bw GSM4160262_WT_BCell_ETO.neg.bw\n",
    "                        mv GSM4160262_190518_WT-BCell-ETO.si.pos.bw GSM4160262_WT_BCell_ETO.pos.bw\n",
    "                        mv GSM4160263_190518_WT-BCell-ETO-WO-2h.si.neg.bw GSM4160263_WT_BCell_ETO_WO_2h.neg.bw\n",
    "                        mv GSM4160263_190518_WT-BCell-ETO-WO-2h.si.pos.bw GSM4160263_WT_BCell_ETO_WO_2h.pos.bw\n",
    "                        mv GSM4160264_190816_WT-BCell-ETO-E7.si.neg.bw GSM4160264_WT_BCell_ETO_E7.neg.bw\n",
    "                        mv GSM4160264_190816_WT-BCell-ETO-E7.si.pos.bw GSM4160264_WT_BCell_ETO_E7.pos.bw\n",
    "                        mv GSM4160265_190816_RNF4-BCell-ETO-E7.si.neg.bw GSM4160265_RNF4_BCell_ETO_E7.neg.bw\n",
    "                        mv GSM4160265_190816_RNF4-BCell-ETO-E7.si.pos.bw GSM4160265_RNF4_BCell_ETO_E7.pos.bw\n",
    "                        mv GSM4160266_190816_WT-BCell-ETO-ExoT.si.neg.bw GSM4160266_WT_BCell_ETO_ExoT.neg.bw\n",
    "                        mv GSM4160266_190816_WT-BCell-ETO-ExoT.si.pos.bw GSM4160266_WT_BCell_ETO_ExoT.pos.bw\n",
    "                        mv GSM4160267_190816_RNF4-BCell-ETO-ExoT.si.neg.bw GSM4160267_RNF4_BCell_ETO_ExoT.neg.bw\n",
    "                        mv GSM4160267_190816_RNF4-BCell-ETO-ExoT.si.pos.bw GSM4160267_RNF4_BCell_ETO_ExoT.pos.bw\n",
    "                        mv GSM4207022_190503_WT-BCell-NT.si.neg.bw GSM4207022_WT_BCell_NT.neg.bw\n",
    "                        mv GSM4207022_190503_WT-BCell-NT.si.pos.bw GSM4207022_WT_BCell_NT.pos.bw\n",
    "                        mv GSM4207023_190503_WT-BCell-ETO.si.neg.bw GSM4207023_WT_BCell_ETO.neg.bw\n",
    "                        mv GSM4207023_190503_WT-BCell-ETO.si.pos.bw GSM4207023_WT_BCell_ETO.pos.bw\n",
    "                        mv GSM4207024_190503_WT-BCell-ETO-WO.si.neg.bw GSM4207024_WT_BCell_ETO_WO.neg.bw\n",
    "                        mv GSM4207024_190503_WT-BCell-ETO-WO.si.pos.bw GSM4207024_WT_BCell_ETO_WO.pos.bw\n",
    "                        mv GSM4207025_190503_WT-BCell-ETO-WO-2h.si.neg.bw GSM4207025_WT_BCell_ETO_WO_2h.neg.bw\n",
    "                        mv GSM4207025_190503_WT-BCell-ETO-WO-2h.si.pos.bw GSM4207025_WT_BCell_ETO_WO_2h.pos.bw\n",
    "                        mv GSM4207026_190503_WT-BCell-BTZ.si.neg.bw GSM4207026_WT_BCell_BTZ.neg.bw\n",
    "                        mv GSM4207026_190503_WT-BCell-BTZ.si.pos.bw GSM4207026_WT_BCell_BTZ.pos.bw\n",
    "                        mv GSM4207027_190503_WT-BCell-BTZ-ETO.si.neg.bw GSM4207027_WT_BCell_BTZ_ETO.neg.bw\n",
    "                        mv GSM4207027_190503_WT-BCell-BTZ-ETO.si.pos.bw GSM4207027_WT_BCell_BTZ_ETO.pos.bw\n",
    "                        mv GSM4207028_190503_WT-BCell-BTZ-ETO-WO.si.neg.bw GSM4207028_WT_BCell_BTZ_ETO_WO.neg.bw\n",
    "                        mv GSM4207028_190503_WT-BCell-BTZ-ETO-WO.si.pos.bw GSM4207028_WT_BCell_BTZ_ETO_WO.pos.bw\n",
    "                        mv GSM4207029_190503_WT-BCell-BTZ-ETO-WO-2h.si.neg.bw GSM4207029_WT_BCell_BTZ_ETO_WO_2h.neg.bw\n",
    "                        mv GSM4207029_190503_WT-BCell-BTZ-ETO-WO-2h.si.pos.bw GSM4207029_WT_BCell_BTZ_ETO_WO_2h.pos.bw\n",
    "                    }\n",
    "                    # manage\n",
    "                    \n",
    "                    avgSignalSurrounding_10kb_50(){\n",
    "                        # D: Nov-07-2022 09:13 Mon\n",
    "                        mkdir -p ${dirPATH}/CHMOrganization/Universal_specific/Functions/DSB/Sciascia_Elife_2020_BCell_GSE140372/AvgSignalSurrounding_10kb_50;cd ${dirPATH}/CHMOrganization/Universal_specific/Functions/DSB/Sciascia_Elife_2020_BCell_GSE140372/AvgSignalSurrounding_10kb_50\n",
    "                        pos_part(){\n",
    "                            for sample in GSM4160241_WT_BCell_NT GSM4160242_WT_BCell_ETO GSM4160243_WT_BCell_ETO_WO GSM4160244_WT_BCell_ETO_WO_2h GSM4160245_WT_BCell_BTZ GSM4160246_WT_BCell_BTZ_ETO GSM4160247_WT_BCell_BTZ_ETO_WO GSM4160248_WT_BCell_BTZ_ETO_WO_2h GSM4160249_WT_BCell_ETO_E7 GSM4160250_WT_BCell_ETO_ExoT GSM4160251_WT_BCell_ETO_WO_0h_E7 GSM4160252_WT_BCell_ETO_WO_0h_ExoT GSM4160253_WT_BCell_BTZ_ETO_E7 GSM4160254_WT_BCell_BTZ_ETO_ExoT GSM4160255_WT_BCell_BTZ_ETO_WO_0h_E7 GSM4160256_WT_BCell_BTZ_ETO_WO_0h_ExoT GSM4160257_WT_BCell_1h_BTZ_ETO GSM4160258_WT_BCell_1h_BTZ_ETO_WO_2h GSM4160259_WT_BCell_BTZ_ETO GSM4160260_WT_BCell_BTZ_ETO_WO_2h GSM4160261_WT_BCell_ETO_2h_BTZ_WO GSM4160262_WT_BCell_ETO GSM4160263_WT_BCell_ETO_WO_2h GSM4160264_WT_BCell_ETO_E7 GSM4160265_RNF4_BCell_ETO_E7 GSM4160266_WT_BCell_ETO_ExoT GSM4160267_RNF4_BCell_ETO_ExoT GSM4207022_WT_BCell_NT GSM4207023_WT_BCell_ETO GSM4207024_WT_BCell_ETO_WO GSM4207025_WT_BCell_ETO_WO_2h GSM4207026_WT_BCell_BTZ GSM4207027_WT_BCell_BTZ_ETO GSM4207028_WT_BCell_BTZ_ETO_WO GSM4207029_WT_BCell_BTZ_ETO_WO_2h\n",
    "                            do\n",
    "                                for group in Universal EarlyEmbryoSpecific PGCSpecific SpermSpecific RetinalSpecific DCSpecific nonCHM_CGIs promoter nonCHM_CGIs_HMR\n",
    "                                do\n",
    "                                    bash ${shPATH}/profileSurroundingPeakCenter.sh \\\n",
    "                                        ${dirPATH}/CHMOrganization/Universal_specific/${group}.bed \\\n",
    "                                        10000 \\\n",
    "                                        10000 \\\n",
    "                                        50 \\\n",
    "                                        ${dirPATH}/CHMOrganization/Universal_specific/Functions/DSB/Sciascia_Elife_2020_BCell_GSE140372/${sample}.pos.bw \\\n",
    "                                        ${group} \\\n",
    "                                        \"${sample}.pos\" \\\n",
    "                                        \"-10kb;;10kb\" \\\n",
    "                                        \"0;25.5;50\" \\\n",
    "                                        \"DSB signal\" \\\n",
    "                                        \"0.02\" \\\n",
    "                                        \"0.1\" \\\n",
    "                                        \"Dark2\"\n",
    "                                done # for group end\n",
    "                            done # for sample end                       \n",
    "                        }\n",
    "                        pos_part\n",
    "                        \n",
    "                        neg_part(){\n",
    "                            for sample in GSM4160241_WT_BCell_NT GSM4160242_WT_BCell_ETO GSM4160243_WT_BCell_ETO_WO GSM4160244_WT_BCell_ETO_WO_2h GSM4160245_WT_BCell_BTZ GSM4160246_WT_BCell_BTZ_ETO GSM4160247_WT_BCell_BTZ_ETO_WO GSM4160248_WT_BCell_BTZ_ETO_WO_2h GSM4160249_WT_BCell_ETO_E7 GSM4160250_WT_BCell_ETO_ExoT GSM4160251_WT_BCell_ETO_WO_0h_E7 GSM4160252_WT_BCell_ETO_WO_0h_ExoT GSM4160253_WT_BCell_BTZ_ETO_E7 GSM4160254_WT_BCell_BTZ_ETO_ExoT GSM4160255_WT_BCell_BTZ_ETO_WO_0h_E7 GSM4160256_WT_BCell_BTZ_ETO_WO_0h_ExoT GSM4160257_WT_BCell_1h_BTZ_ETO GSM4160258_WT_BCell_1h_BTZ_ETO_WO_2h GSM4160259_WT_BCell_BTZ_ETO GSM4160260_WT_BCell_BTZ_ETO_WO_2h GSM4160261_WT_BCell_ETO_2h_BTZ_WO GSM4160262_WT_BCell_ETO GSM4160263_WT_BCell_ETO_WO_2h GSM4160264_WT_BCell_ETO_E7 GSM4160265_RNF4_BCell_ETO_E7 GSM4160266_WT_BCell_ETO_ExoT GSM4160267_RNF4_BCell_ETO_ExoT GSM4207022_WT_BCell_NT GSM4207023_WT_BCell_ETO GSM4207024_WT_BCell_ETO_WO GSM4207025_WT_BCell_ETO_WO_2h GSM4207026_WT_BCell_BTZ GSM4207027_WT_BCell_BTZ_ETO GSM4207028_WT_BCell_BTZ_ETO_WO GSM4207029_WT_BCell_BTZ_ETO_WO_2h\n",
    "                            do\n",
    "                                for group in Universal EarlyEmbryoSpecific PGCSpecific SpermSpecific RetinalSpecific DCSpecific nonCHM_CGIs promoter nonCHM_CGIs_HMR\n",
    "                                do\n",
    "                                    bash ${shPATH}/profileSurroundingPeakCenter.sh \\\n",
    "                                        ${dirPATH}/CHMOrganization/Universal_specific/${group}.bed \\\n",
    "                                        10000 \\\n",
    "                                        10000 \\\n",
    "                                        50 \\\n",
    "                                        ${dirPATH}/CHMOrganization/Universal_specific/Functions/DSB/Sciascia_Elife_2020_BCell_GSE140372/${sample}.neg.bw \\\n",
    "                                        ${group} \\\n",
    "                                        \"${sample}.neg\" \\\n",
    "                                        \"-10kb;;10kb\" \\\n",
    "                                        \"0;25.5;50\" \\\n",
    "                                        \"DSB signal\" \\\n",
    "                                        \"-0.1\" \\\n",
    "                                        \"-0.02\" \\\n",
    "                                        \"Dark2\"\n",
    "                                done # for group end\n",
    "                            done # for sample end                       \n",
    "                        }\n",
    "                        neg_part\n",
    "                    }\n",
    "                    # avgSignalSurrounding_10kb_50\n",
    "                    \n",
    "                    each_plot(){\n",
    "                        # D: Nov-07-2022 11:09 Mon\n",
    "                        # # GSM4160241_WT_BCell_NT.pos\n",
    "                        # Rscript ${rPATH}/line_profileSurroundingPeak.r \\\n",
    "                        #     \"GSM4160241_WT_BCell_NT.pos\" \\\n",
    "                        #     \"GSM4160241_WT_BCell_NT.pos\" \\\n",
    "                        #     4 \\\n",
    "                        #     3.5 \\\n",
    "                        #     \"-10kb;;10kb\" \\\n",
    "                        #     \"0;25.5;50\" \\\n",
    "                        #     \"DSB signal\" \\\n",
    "                        #     \"0.0009\" \\\n",
    "                        #     \"0.0038\" \\\n",
    "                        #     \"GSM4160241_WT_BCell_NT.pos\" \\\n",
    "                        #     \"Dark2\"\n",
    "\n",
    "                        mkdir -p ${dirPATH}/CHMOrganization/Universal_specific/Functions/DSB/Sciascia_Elife_2020_BCell_GSE140372/AvgSignalSurrounding_10kb_50;cd ${dirPATH}/CHMOrganization/Universal_specific/Functions/DSB/Sciascia_Elife_2020_BCell_GSE140372/AvgSignalSurrounding_10kb_50\n",
    "                        for sample in GSM4160241_WT_BCell_NT GSM4160242_WT_BCell_ETO GSM4160243_WT_BCell_ETO_WO GSM4160244_WT_BCell_ETO_WO_2h GSM4160245_WT_BCell_BTZ GSM4160246_WT_BCell_BTZ_ETO GSM4160247_WT_BCell_BTZ_ETO_WO GSM4160248_WT_BCell_BTZ_ETO_WO_2h GSM4160249_WT_BCell_ETO_E7 GSM4160250_WT_BCell_ETO_ExoT GSM4160251_WT_BCell_ETO_WO_0h_E7 GSM4160252_WT_BCell_ETO_WO_0h_ExoT GSM4160253_WT_BCell_BTZ_ETO_E7 GSM4160254_WT_BCell_BTZ_ETO_ExoT GSM4160255_WT_BCell_BTZ_ETO_WO_0h_E7 GSM4160256_WT_BCell_BTZ_ETO_WO_0h_ExoT GSM4160257_WT_BCell_1h_BTZ_ETO GSM4160258_WT_BCell_1h_BTZ_ETO_WO_2h GSM4160259_WT_BCell_BTZ_ETO GSM4160260_WT_BCell_BTZ_ETO_WO_2h GSM4160261_WT_BCell_ETO_2h_BTZ_WO GSM4160262_WT_BCell_ETO GSM4160263_WT_BCell_ETO_WO_2h GSM4160264_WT_BCell_ETO_E7 GSM4160265_RNF4_BCell_ETO_E7 GSM4160266_WT_BCell_ETO_ExoT GSM4160267_RNF4_BCell_ETO_ExoT GSM4207022_WT_BCell_NT GSM4207023_WT_BCell_ETO GSM4207024_WT_BCell_ETO_WO GSM4207025_WT_BCell_ETO_WO_2h GSM4207026_WT_BCell_BTZ GSM4207027_WT_BCell_BTZ_ETO GSM4207028_WT_BCell_BTZ_ETO_WO GSM4207029_WT_BCell_BTZ_ETO_WO_2h\n",
    "                        do\n",
    "                            # for type in pos neg\n",
    "                            for type in neg\n",
    "                            do\n",
    "                                Rscript ${rPATH}/line_profileSurroundingPeak_auto.r \\\n",
    "                                    \"${sample}.${type}\" \\\n",
    "                                    \"${sample}.${type}\" \\\n",
    "                                    4 \\\n",
    "                                    3.5 \\\n",
    "                                    \"-10kb;;10kb\" \\\n",
    "                                    \"0;25.5;50\" \\\n",
    "                                    \"DSB signal\" \\\n",
    "                                    \"${sample}.${type}\" \\\n",
    "                                    \"Dark2\"\n",
    "                            done # for type end\n",
    "                        done # for sample end\n",
    "                            \n",
    "                    }\n",
    "                    each_plot\n",
    "                }\n",
    "                # Sciascia_Elife_2020_BCell_GSE140372\n",
    "                \n",
    "                Callen_MolCell_2020_MEF_GSE133806(){\n",
    "                    # D: Nov-06-2022 22:15 Sun\n",
    "                    # I: https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE133806\n",
    "                    mkdir -p ${dirPATH}/CHMOrganization/Universal_specific/Functions/DSB/Callen_MolCell_2020_MEF_GSE133806;cd ${dirPATH}/CHMOrganization/Universal_specific/Functions/DSB/Callen_MolCell_2020_MEF_GSE133806\n",
    "                    download(){\n",
    "                        wget -c \"https://www.ncbi.nlm.nih.gov/geo/download/?acc=GSE133806&format=file\"\n",
    "                    }\n",
    "                    # download\n",
    "                    \n",
    "                    manage(){\n",
    "                        # D: Nov-07-2022 09:20 Mon\n",
    "                        mv GSM3927555_2018-07-06-ENDseq-AsiSI-MEFS-WT_S5_R1_001.neg.bw GSM3927555_AsiSI_MEFS_WT_S5_R1_001.neg.bw\n",
    "                        mv GSM3927555_2018-07-06-ENDseq-AsiSI-MEFS-WT_S5_R1_001.pos.bw GSM3927555_AsiSI_MEFS_WT_S5_R1_001.pos.bw\n",
    "                        mv GSM3927556_180302-53BP1-MEF-AsisI.neg.bw GSM3927556_53BP1_MEF_AsisI.neg.bw\n",
    "                        mv GSM3927556_180302-53BP1-MEF-AsisI.pos.bw GSM3927556_53BP1_MEF_AsisI.pos.bw\n",
    "                        mv GSM3927557_180302-S25A-MEF-AsisI.neg.bw GSM3927557_S25A_MEF_AsisI.neg.bw\n",
    "                        mv GSM3927557_180302-S25A-MEF-AsisI.pos.bw GSM3927557_S25A_MEF_AsisI.pos.bw\n",
    "                        mv GSM3927558_2018-07-06-ENDseq-AsiSI-MEFS-BRCA1ko_S6_R1_001.neg.bw GSM3927558_AsiSI_MEFS_BRCA1ko_S6_R1_001.neg.bw\n",
    "                        mv GSM3927558_2018-07-06-ENDseq-AsiSI-MEFS-BRCA1ko_S6_R1_001.pos.bw GSM3927558_AsiSI_MEFS_BRCA1ko_S6_R1_001.pos.bw\n",
    "                        mv GSM3927559_2018-07-06-ENDseq-AsiSI-MEFS-BRCA1-53BP1-dko_S7_R1_001.neg.bw GSM3927559_AsiSI_MEFS_BRCA1_53BP1_dko_S7_R1_001.neg.bw\n",
    "                        mv GSM3927559_2018-07-06-ENDseq-AsiSI-MEFS-BRCA1-53BP1-dko_S7_R1_001.pos.bw GSM3927559_AsiSI_MEFS_BRCA1_53BP1_dko_S7_R1_001.pos.bw\n",
    "                        mv GSM3927560_2018-07-06-ENDseq-AsiSI-MEFS-BRCA1-S25A-dko_S8_R1_001.neg.bw GSM3927560_AsiSI_MEFS_BRCA1_S25A_dko_S8_R1_001.neg.bw\n",
    "                        mv GSM3927560_2018-07-06-ENDseq-AsiSI-MEFS-BRCA1-S25A-dko_S8_R1_001.pos.bw GSM3927560_AsiSI_MEFS_BRCA1_S25A_dko_S8_R1_001.pos.bw\n",
    "                        mv GSM3927561_181211-14223-Asis1-8hTAM.neg.bw GSM3927561_Asis1_8hTAM.neg.bw\n",
    "                        mv GSM3927561_181211-14223-Asis1-8hTAM.pos.bw GSM3927561_Asis1_8hTAM.pos.bw\n",
    "                        mv GSM3927562_181211-WTAsis1-8hTAM.neg.bw GSM3927562_WTAsis1_8hTAM.neg.bw\n",
    "                        mv GSM3927562_181211-WTAsis1-8hTAM.pos.bw GSM3927562_WTAsis1_8hTAM.pos.bw\n",
    "                    }\n",
    "                    # manage\n",
    "                    \n",
    "                    avgSignalSurrounding_10kb_50(){\n",
    "                        # D: Nov-07-2022 09:29 Mon\n",
    "                        mkdir -p ${dirPATH}/CHMOrganization/Universal_specific/Functions/DSB/Callen_MolCell_2020_MEF_GSE133806/AvgSignalSurrounding_10kb_50;cd ${dirPATH}/CHMOrganization/Universal_specific/Functions/DSB/Callen_MolCell_2020_MEF_GSE133806/AvgSignalSurrounding_10kb_50\n",
    "                        pos_part(){\n",
    "                            for sample in GSM3927555_AsiSI_MEFS_WT_S5_R1_001 GSM3927556_53BP1_MEF_AsisI GSM3927557_S25A_MEF_AsisI GSM3927558_AsiSI_MEFS_BRCA1ko_S6_R1_001 GSM3927559_AsiSI_MEFS_BRCA1_53BP1_dko_S7_R1_001 GSM3927560_AsiSI_MEFS_BRCA1_S25A_dko_S8_R1_001 GSM3927561_Asis1_8hTAM GSM3927562_WTAsis1_8hTAM\n",
    "                            do\n",
    "                                for group in Universal EarlyEmbryoSpecific PGCSpecific SpermSpecific RetinalSpecific DCSpecific nonCHM_CGIs promoter nonCHM_CGIs_HMR\n",
    "                                do\n",
    "                                    bash ${shPATH}/profileSurroundingPeakCenter.sh \\\n",
    "                                        ${dirPATH}/CHMOrganization/Universal_specific/${group}.bed \\\n",
    "                                        10000 \\\n",
    "                                        10000 \\\n",
    "                                        50 \\\n",
    "                                        ${dirPATH}/CHMOrganization/Universal_specific/Functions/DSB/Callen_MolCell_2020_MEF_GSE133806/${sample}.pos.bw \\\n",
    "                                        ${group} \\\n",
    "                                        \"${sample}.pos\" \\\n",
    "                                        \"-10kb;;10kb\" \\\n",
    "                                        \"0;25.5;50\" \\\n",
    "                                        \"DSB signal\" \\\n",
    "                                        \"0\" \\\n",
    "                                        \"0.1\" \\\n",
    "                                        \"Dark2\"\n",
    "                                done # for group end\n",
    "                            done # for sample end                       \n",
    "                        }\n",
    "                        pos_part\n",
    "                        \n",
    "                        neg_part(){\n",
    "                            for sample in GSM3927555_AsiSI_MEFS_WT_S5_R1_001 GSM3927556_53BP1_MEF_AsisI GSM3927557_S25A_MEF_AsisI GSM3927558_AsiSI_MEFS_BRCA1ko_S6_R1_001 GSM3927559_AsiSI_MEFS_BRCA1_53BP1_dko_S7_R1_001 GSM3927560_AsiSI_MEFS_BRCA1_S25A_dko_S8_R1_001 GSM3927561_Asis1_8hTAM GSM3927562_WTAsis1_8hTAM\n",
    "                            do\n",
    "                                for group in Universal EarlyEmbryoSpecific PGCSpecific SpermSpecific RetinalSpecific DCSpecific nonCHM_CGIs promoter nonCHM_CGIs_HMR\n",
    "                                do\n",
    "                                    bash ${shPATH}/profileSurroundingPeakCenter.sh \\\n",
    "                                        ${dirPATH}/CHMOrganization/Universal_specific/${group}.bed \\\n",
    "                                        10000 \\\n",
    "                                        10000 \\\n",
    "                                        50 \\\n",
    "                                        ${dirPATH}/CHMOrganization/Universal_specific/Functions/DSB/Callen_MolCell_2020_MEF_GSE133806/${sample}.neg.bw \\\n",
    "                                        ${group} \\\n",
    "                                        \"${sample}.neg\" \\\n",
    "                                        \"-10kb;;10kb\" \\\n",
    "                                        \"0;25.5;50\" \\\n",
    "                                        \"DSB signal\" \\\n",
    "                                        \"-0.1\" \\\n",
    "                                        \"0\" \\\n",
    "                                        \"Dark2\"\n",
    "                                done # for group end\n",
    "                            done # for sample end                       \n",
    "                        }\n",
    "                        neg_part\n",
    "                    }\n",
    "                    # avgSignalSurrounding_10kb_50\n",
    "                    \n",
    "                    each_plot(){\n",
    "                        # D: Nov-07-2022 11:41 Mon\n",
    "                        mkdir -p ${dirPATH}/CHMOrganization/Universal_specific/Functions/DSB/Callen_MolCell_2020_MEF_GSE133806/AvgSignalSurrounding_10kb_50;cd ${dirPATH}/CHMOrganization/Universal_specific/Functions/DSB/Callen_MolCell_2020_MEF_GSE133806/AvgSignalSurrounding_10kb_50\n",
    "                        for sample in GSM3927555_AsiSI_MEFS_WT_S5_R1_001 GSM3927556_53BP1_MEF_AsisI GSM3927557_S25A_MEF_AsisI GSM3927558_AsiSI_MEFS_BRCA1ko_S6_R1_001 GSM3927559_AsiSI_MEFS_BRCA1_53BP1_dko_S7_R1_001 GSM3927560_AsiSI_MEFS_BRCA1_S25A_dko_S8_R1_001 GSM3927561_Asis1_8hTAM GSM3927562_WTAsis1_8hTAM\n",
    "                        do\n",
    "                            for type in pos neg\n",
    "                            do\n",
    "                                Rscript ${rPATH}/line_profileSurroundingPeak_auto.r \\\n",
    "                                    \"${sample}.${type}\" \\\n",
    "                                    \"${sample}.${type}\" \\\n",
    "                                    4 \\\n",
    "                                    3.5 \\\n",
    "                                    \"-10kb;;10kb\" \\\n",
    "                                    \"0;25.5;50\" \\\n",
    "                                    \"DSB signal\" \\\n",
    "                                    \"${sample}.${type}\" \\\n",
    "                                    \"Dark2\"\n",
    "                            done # for type end\n",
    "                        done # for sample end\n",
    "                    }\n",
    "                    each_plot\n",
    "                }\n",
    "                # Callen_MolCell_2020_MEF_GSE133806\n",
    "                \n",
    "                Canela_MolCell_2019_B_MEF_preB_T_Neuron_GSE129524(){\n",
    "                    # D: Nov-06-2022 22:18 Sun\n",
    "                    # I: https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE129524\n",
    "                    mkdir -p ${dirPATH}/CHMOrganization/Universal_specific/Functions/DSB/Canela_MolCell_2019_B_MEF_preB_T_Neuron_GSE129524;cd ${dirPATH}/CHMOrganization/Universal_specific/Functions/DSB/Canela_MolCell_2019_B_MEF_preB_T_Neuron_GSE129524\n",
    "                    download(){\n",
    "                        wget -c \"https://www.ncbi.nlm.nih.gov/geo/download/?acc=GSE129524&format=file\"\n",
    "                    }\n",
    "                    # download\n",
    "                    \n",
    "                    manage(){\n",
    "                        # D: Nov-07-2022 09:31 Mon\n",
    "                        # tar -xvf \"index.html?acc=GSE129524&format=file\" && mv \"index.html?acc=GSE129524&format=file\" GSE129524_RAW.tar\n",
    "                        mv GSM3714864_END_seq_PB_LIG4_2_5h_TRANSLOCATION.bw GSM3714864_preB_LIG4_2_5h_TRANSLOCATION.bw\n",
    "                        mv GSM3714865_END_seq_PB_LIG4_AsiSI_TRANSLOCATION.bw GSM3714865_preB_LIG4_AsiSI_TRANSLOCATION.bw\n",
    "                        mv GSM3714866_END_seq_PB_LIG4_2_5h_DRB_TRX.bw GSM3714866_preB_LIG4_2_5h_DRB_TRX.bw\n",
    "                        mv GSM3714867_END_seq_ExoT_PB_LIG4_2_5h_DRB_TRX.bw GSM3714867_ExoT_preB_LIG4_2_5h_DRB_TRX.bw\n",
    "                        mv GSM3714868_END_seq_PB_LIG4_AsiSI_DRB_TRX.bw GSM3714868_preB_LIG4_AsiSI_DRB_TRX.bw\n",
    "                        mv GSM3714869_END_seq_PB_LIG4_2_5h_IFNB_TRX.bw GSM3714869_preB_LIG4_2_5h_IFNB_TRX.bw\n",
    "                        mv GSM3714870_END_seq_ExoT_PB_LIG4_2_5h_IFNB_TRX.bw GSM3714870_ExoT_preB_LIG4_2_5h_IFNB_TRX.bw\n",
    "                        mv GSM3714871_END_seq_PB_LIG4_0_5h_TIME_COURSE.bw GSM3714871_preB_LIG4_0_5h_TIME_COURSE.bw\n",
    "                        mv GSM3714872_END_seq_PB_LIG4_1h_TIME_COURSE.bw GSM3714872_preB_LIG4_1h_TIME_COURSE.bw\n",
    "                        mv GSM3714873_END_seq_PB_LIG4_2_5h_TIME_COURSE.bw GSM3714873_preB_LIG4_2_5h_TIME_COURSE.bw\n",
    "                        mv GSM3714874_END_seq_PB_LIG4_5h_TIME_COURSE.bw GSM3714874_preB_LIG4_5h_TIME_COURSE.bw\n",
    "                        mv GSM3714875_END_seq_PB_LIG4_7_5h_TIME_COURSE.bw GSM3714875_preB_LIG4_7_5h_TIME_COURSE.bw\n",
    "                        mv GSM3714876_END_seq_PB_LIG4_12_5h_TIME_COURSE.bw GSM3714876_preB_LIG4_12_5h_TIME_COURSE.bw\n",
    "                        mv GSM3714877_END_seq_ExoT_PB_LIG4_0_5h_TIME_COURSE.bw GSM3714877_ExoT_preB_LIG4_0_5h_TIME_COURSE.bw\n",
    "                        mv GSM3714878_END_seq_ExoT_PB_LIG4_1h_TIME_COURSE.bw GSM3714878_ExoT_preB_LIG4_1h_TIME_COURSE.bw\n",
    "                        mv GSM3714879_END_seq_ExoT_PB_LIG4_2_5h_TIME_COURSE.bw GSM3714879_ExoT_preB_LIG4_2_5h_TIME_COURSE.bw\n",
    "                        mv GSM3714880_END_seq_ExoT_PB_LIG4_5h_TIME_COURSE.bw GSM3714880_ExoT_preB_LIG4_5h_TIME_COURSE.bw\n",
    "                        mv GSM3714881_END_seq_ExoT_PB_LIG4_7_5h_TIME_COURSE.bw GSM3714881_ExoT_preB_LIG4_7_5h_TIME_COURSE.bw\n",
    "                        mv GSM3714882_END_seq_ExoT_PB_LIG4_12_5h_TIME_COURSE.bw GSM3714882_ExoT_preB_LIG4_12_5h_TIME_COURSE.bw\n",
    "                        mv GSM3714883_END_seq_PB_LIG4_NT.bw GSM3714883_preB_LIG4_NT.bw\n",
    "                        mv GSM3714884_END_seq_ExoT_PB_LIG4_NT.bw GSM3714884_ExoT_preB_LIG4_NT.bw\n",
    "                        mv GSM3714885_END_seq_PB_LIG4_2_5h_REVERSIBILITY.bw GSM3714885_preB_LIG4_2_5h_REVERSIBILITY.bw\n",
    "                        mv GSM3714886_END_seq_ExoT_PB_LIG4_2_5h_REVERSIBILITY.bw GSM3714886_ExoT_preB_LIG4_2_5h_REVERSIBILITY.bw\n",
    "                        mv GSM3714887_END_seq_PB_LIG4_WASH_2_5h_REVERSIBILITY.bw GSM3714887_preB_LIG4_WASH_2_5h_REVERSIBILITY.bw\n",
    "                        mv GSM3714888_END_seq_ExoT_PB_LIG4_WASH_2_5h_REVERSIBILITY.bw GSM3714888_ExoT_preB_LIG4_WASH_2_5h_REVERSIBILITY.bw\n",
    "                        mv GSM3714889_END_seq_PB_LIG4_TDP2_2_5h_REVERSIBILITY.bw GSM3714889_preB_LIG4_TDP2_2_5h_REVERSIBILITY.bw\n",
    "                        mv GSM3714890_END_seq_ExoT_PB_LIG4_TDP2_2_5h_REVERSIBILITY.bw GSM3714890_ExoT_preB_LIG4_TDP2_2_5h_REVERSIBILITY.bw\n",
    "                        mv GSM3714891_END_seq_PB_LIG4_TDP2_WASH_2_5h_REVERSIBILITY.bw GSM3714891_preB_LIG4_TDP2_WASH_2_5h_REVERSIBILITY.bw\n",
    "                        mv GSM3714892_END_seq_ExoT_PB_LIG4_TDP2_WASH_2_5h_REVERSIBILITY.bw GSM3714892_ExoT_preB_LIG4_TDP2_WASH_2_5h_REVERSIBILITY.bw\n",
    "                        mv GSM3714893_END_seq_PB_LIG4_EPOX_2_5h_REVERSIBILITY.bw GSM3714893_preB_LIG4_EPOX_2_5h_REVERSIBILITY.bw\n",
    "                        mv GSM3714894_END_seq_ExoT_PB_LIG4_EPOX_2_5h_REVERSIBILITY.bw GSM3714894_ExoT_preB_LIG4_EPOX_2_5h_REVERSIBILITY.bw\n",
    "                        mv GSM3714895_END_seq_PB_LIG4_EPOX_WASH_2_5h_REVERSIBILITY.bw GSM3714895_preB_LIG4_EPOX_WASH_2_5h_REVERSIBILITY.bw\n",
    "                        mv GSM3714896_END_seq_ExoT_PB_LIG4_EPOX_WASH_2_5h_REVERSIBILITY.bw GSM3714896_ExoT_preB_LIG4_EPOX_WASH_2_5h_REVERSIBILITY.bw\n",
    "                        mv GSM3714897_END_seq_PB_LIG4_WASH_0_5h_REVERSIBILITY.bw GSM3714897_preB_LIG4_WASH_0_5h_REVERSIBILITY.bw\n",
    "                        mv GSM3714898_END_seq_ExoT_PB_LIG4_WASH_0_5h_REVERSIBILITY.bw GSM3714898_ExoT_preB_LIG4_WASH_0_5h_REVERSIBILITY.bw\n",
    "                        mv GSM3714899_END_seq_MEF_Smc3_PP_0_5h.bw GSM3714899_MEF_Smc3_PP_0_5h.bw\n",
    "                        mv GSM3714900_END_seq_MEF_Smc3_FF_0_5h.bw GSM3714900_MEF_Smc3_FF_0_5h.bw\n",
    "                        mv GSM3714901_END_seq_MEF_Wapl_FF_0_5h.bw GSM3714901_MEF_Wapl_FF_0_5h.bw\n",
    "                        mv GSM3714902_END_seq_MEF_Wapl_FF_CRE_0_5h.bw GSM3714902_MEF_Wapl_FF_CRE_0_5h.bw\n",
    "                        mv GSM3714906_END_seq_aB_TOP2Bko_0_5h_rep1_2_pileup.bw GSM3714906_aB_TOP2Bko_0_5h_rep1_2_pileup.bw\n",
    "                        mv GSM3714908_END_seq_aB_WT_0_5h_rep1_2_pileup.bw GSM3714908_aB_WT_0_5h_rep1_2_pileup.bw\n",
    "                    }\n",
    "                    # manage\n",
    "                    \n",
    "                    avgSignalSurrounding_10kb_50(){\n",
    "                        # D: Nov-07-2022 09:41 Mon\n",
    "                        mkdir -p ${dirPATH}/CHMOrganization/Universal_specific/Functions/DSB/Canela_MolCell_2019_B_MEF_preB_T_Neuron_GSE129524/AvgSignalSurrounding_10kb_50;cd ${dirPATH}/CHMOrganization/Universal_specific/Functions/DSB/Canela_MolCell_2019_B_MEF_preB_T_Neuron_GSE129524/AvgSignalSurrounding_10kb_50\n",
    "                        pos_part(){\n",
    "                            for sample in GSM3714864_preB_LIG4_2_5h_TRANSLOCATION GSM3714865_preB_LIG4_AsiSI_TRANSLOCATION GSM3714866_preB_LIG4_2_5h_DRB_TRX GSM3714867_ExoT_preB_LIG4_2_5h_DRB_TRX GSM3714868_preB_LIG4_AsiSI_DRB_TRX GSM3714869_preB_LIG4_2_5h_IFNB_TRX GSM3714870_ExoT_preB_LIG4_2_5h_IFNB_TRX GSM3714871_preB_LIG4_0_5h_TIME_COURSE GSM3714872_preB_LIG4_1h_TIME_COURSE GSM3714873_preB_LIG4_2_5h_TIME_COURSE GSM3714874_preB_LIG4_5h_TIME_COURSE GSM3714875_preB_LIG4_7_5h_TIME_COURSE GSM3714876_preB_LIG4_12_5h_TIME_COURSE GSM3714877_ExoT_preB_LIG4_0_5h_TIME_COURSE GSM3714878_ExoT_preB_LIG4_1h_TIME_COURSE GSM3714879_ExoT_preB_LIG4_2_5h_TIME_COURSE GSM3714880_ExoT_preB_LIG4_5h_TIME_COURSE GSM3714881_ExoT_preB_LIG4_7_5h_TIME_COURSE GSM3714882_ExoT_preB_LIG4_12_5h_TIME_COURSE GSM3714883_preB_LIG4_NT GSM3714884_ExoT_preB_LIG4_NT GSM3714885_preB_LIG4_2_5h_REVERSIBILITY GSM3714886_ExoT_preB_LIG4_2_5h_REVERSIBILITY GSM3714887_preB_LIG4_WASH_2_5h_REVERSIBILITY GSM3714888_ExoT_preB_LIG4_WASH_2_5h_REVERSIBILITY GSM3714889_preB_LIG4_TDP2_2_5h_REVERSIBILITY GSM3714890_ExoT_preB_LIG4_TDP2_2_5h_REVERSIBILITY GSM3714891_preB_LIG4_TDP2_WASH_2_5h_REVERSIBILITY GSM3714892_ExoT_preB_LIG4_TDP2_WASH_2_5h_REVERSIBILITY GSM3714893_preB_LIG4_EPOX_2_5h_REVERSIBILITY GSM3714894_ExoT_preB_LIG4_EPOX_2_5h_REVERSIBILITY GSM3714895_preB_LIG4_EPOX_WASH_2_5h_REVERSIBILITY GSM3714896_ExoT_preB_LIG4_EPOX_WASH_2_5h_REVERSIBILITY GSM3714897_preB_LIG4_WASH_0_5h_REVERSIBILITY GSM3714898_ExoT_preB_LIG4_WASH_0_5h_REVERSIBILITY GSM3714899_MEF_Smc3_PP_0_5h GSM3714900_MEF_Smc3_FF_0_5h GSM3714901_MEF_Wapl_FF_0_5h GSM3714902_MEF_Wapl_FF_CRE_0_5h GSM3714906_aB_TOP2Bko_0_5h_rep1_2_pileup GSM3714908_aB_WT_0_5h_rep1_2_pileup\n",
    "                            do\n",
    "                                for group in Universal EarlyEmbryoSpecific PGCSpecific SpermSpecific RetinalSpecific DCSpecific nonCHM_CGIs promoter nonCHM_CGIs_HMR\n",
    "                                do\n",
    "                                    bash ${shPATH}/profileSurroundingPeakCenter.sh \\\n",
    "                                        ${dirPATH}/CHMOrganization/Universal_specific/${group}.bed \\\n",
    "                                        10000 \\\n",
    "                                        10000 \\\n",
    "                                        50 \\\n",
    "                                        ${dirPATH}/CHMOrganization/Universal_specific/Functions/DSB/Canela_MolCell_2019_B_MEF_preB_T_Neuron_GSE129524/${sample}.bw \\\n",
    "                                        ${group} \\\n",
    "                                        \"${sample}\" \\\n",
    "                                        \"-10kb;;10kb\" \\\n",
    "                                        \"0;25.5;50\" \\\n",
    "                                        \"DSB signal\" \\\n",
    "                                        \"0\" \\\n",
    "                                        \"0.1\" \\\n",
    "                                        \"Dark2\"\n",
    "                                done # for group end\n",
    "                            done # for sample end\n",
    "                        }\n",
    "                        pos_part\n",
    "                    }\n",
    "                    # avgSignalSurrounding_10kb_50\n",
    "                    \n",
    "                    each_plot(){\n",
    "                        # D: Nov-07-2022 11:43 Mon\n",
    "                        mkdir -p ${dirPATH}/CHMOrganization/Universal_specific/Functions/DSB/Canela_MolCell_2019_B_MEF_preB_T_Neuron_GSE129524/AvgSignalSurrounding_10kb_50;cd ${dirPATH}/CHMOrganization/Universal_specific/Functions/DSB/Canela_MolCell_2019_B_MEF_preB_T_Neuron_GSE129524/AvgSignalSurrounding_10kb_50\n",
    "                        for sample in GSM3714864_preB_LIG4_2_5h_TRANSLOCATION GSM3714865_preB_LIG4_AsiSI_TRANSLOCATION GSM3714866_preB_LIG4_2_5h_DRB_TRX GSM3714867_ExoT_preB_LIG4_2_5h_DRB_TRX GSM3714868_preB_LIG4_AsiSI_DRB_TRX GSM3714869_preB_LIG4_2_5h_IFNB_TRX GSM3714870_ExoT_preB_LIG4_2_5h_IFNB_TRX GSM3714871_preB_LIG4_0_5h_TIME_COURSE GSM3714872_preB_LIG4_1h_TIME_COURSE GSM3714873_preB_LIG4_2_5h_TIME_COURSE GSM3714874_preB_LIG4_5h_TIME_COURSE GSM3714875_preB_LIG4_7_5h_TIME_COURSE GSM3714876_preB_LIG4_12_5h_TIME_COURSE GSM3714877_ExoT_preB_LIG4_0_5h_TIME_COURSE GSM3714878_ExoT_preB_LIG4_1h_TIME_COURSE GSM3714879_ExoT_preB_LIG4_2_5h_TIME_COURSE GSM3714880_ExoT_preB_LIG4_5h_TIME_COURSE GSM3714881_ExoT_preB_LIG4_7_5h_TIME_COURSE GSM3714882_ExoT_preB_LIG4_12_5h_TIME_COURSE GSM3714883_preB_LIG4_NT GSM3714884_ExoT_preB_LIG4_NT GSM3714885_preB_LIG4_2_5h_REVERSIBILITY GSM3714886_ExoT_preB_LIG4_2_5h_REVERSIBILITY GSM3714887_preB_LIG4_WASH_2_5h_REVERSIBILITY GSM3714888_ExoT_preB_LIG4_WASH_2_5h_REVERSIBILITY GSM3714889_preB_LIG4_TDP2_2_5h_REVERSIBILITY GSM3714890_ExoT_preB_LIG4_TDP2_2_5h_REVERSIBILITY GSM3714891_preB_LIG4_TDP2_WASH_2_5h_REVERSIBILITY GSM3714892_ExoT_preB_LIG4_TDP2_WASH_2_5h_REVERSIBILITY GSM3714893_preB_LIG4_EPOX_2_5h_REVERSIBILITY GSM3714894_ExoT_preB_LIG4_EPOX_2_5h_REVERSIBILITY GSM3714895_preB_LIG4_EPOX_WASH_2_5h_REVERSIBILITY GSM3714896_ExoT_preB_LIG4_EPOX_WASH_2_5h_REVERSIBILITY GSM3714897_preB_LIG4_WASH_0_5h_REVERSIBILITY GSM3714898_ExoT_preB_LIG4_WASH_0_5h_REVERSIBILITY GSM3714899_MEF_Smc3_PP_0_5h GSM3714900_MEF_Smc3_FF_0_5h GSM3714901_MEF_Wapl_FF_0_5h GSM3714902_MEF_Wapl_FF_CRE_0_5h GSM3714906_aB_TOP2Bko_0_5h_rep1_2_pileup GSM3714908_aB_WT_0_5h_rep1_2_pileup\n",
    "                        do\n",
    "                            Rscript ${rPATH}/line_profileSurroundingPeak_auto.r \\\n",
    "                                \"${sample}\" \\\n",
    "                                \"${sample}\" \\\n",
    "                                4 \\\n",
    "                                3.5 \\\n",
    "                                \"-10kb;;10kb\" \\\n",
    "                                \"0;25.5;50\" \\\n",
    "                                \"DSB signal\" \\\n",
    "                                \"${sample}\" \\\n",
    "                                \"Dark2\"\n",
    "                        done # for sample end\n",
    "                    }\n",
    "                    each_plot\n",
    "                }\n",
    "                # Canela_MolCell_2019_B_MEF_preB_T_Neuron_GSE129524\n",
    "                \n",
    "                Tillo_NatComm_2021_ESC_GSE165161(){\n",
    "                    # D: Nov-06-2022 22:21 Sun\n",
    "                    # I: https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE165161\n",
    "                    mkdir -p ${dirPATH}/CHMOrganization/Universal_specific/Functions/DSB/Tillo_NatComm_2021_ESC_GSE165161;cd ${dirPATH}/CHMOrganization/Universal_specific/Functions/DSB/Tillo_NatComm_2021_ESC_GSE165161\n",
    "                    download(){\n",
    "                        wget -c \"https://www.ncbi.nlm.nih.gov/geo/download/?acc=GSE165161&format=file\"\n",
    "                    }\n",
    "                    # download\n",
    "                    \n",
    "                    manage(){\n",
    "                        # D: Nov-07-2022 09:45 Mon\n",
    "                        mv GSM5027366_ESC_DUX_Clone1_DOX_16h_ENDseq_neg.mm10.RPKM.bw GSM5027366_ESC_DUX_DOX_16h.neg.bw\n",
    "                        mv GSM5027366_ESC_DUX_Clone1_DOX_16h_ENDseq_pos.mm10.RPKM.bw GSM5027366_ESC_DUX_DOX_16h.pos.bw\n",
    "                        mv GSM5027367_ESC_DUX_Clone1_ENDseq_neg.mm10.RPKM.bw GSM5027367_ESC_DUX.neg.bw\n",
    "                        mv GSM5027367_ESC_DUX_Clone1_ENDseq_pos.mm10.RPKM.bw GSM5027367_ESC_DUX.pos.bw\n",
    "                        mv GSM5027368_ESC_DUX_Clone2_DOX_16h_ENDseq_neg.mm10.RPKM.bw GSM5027368_ESC_DUX_DOX_16h.neg.bw\n",
    "                        mv GSM5027368_ESC_DUX_Clone2_DOX_16h_ENDseq_pos.mm10.RPKM.bw GSM5027368_ESC_DUX_DOX_16h.pos.bw\n",
    "                        mv GSM5027369_ESC_DUX_Clone2_ENDseq_neg.mm10.RPKM.bw GSM5027369_ESC_DUX.neg.bw\n",
    "                        mv GSM5027369_ESC_DUX_Clone2_ENDseq_pos.mm10.RPKM.bw GSM5027369_ESC_DUX.pos.bw\n",
    "                    }\n",
    "                    # manage\n",
    "                    \n",
    "                    avgSignalSurrounding_10kb_50(){\n",
    "                        # D: Nov-07-2022 09:46 Mon\n",
    "                        mkdir -p ${dirPATH}/CHMOrganization/Universal_specific/Functions/DSB/Tillo_NatComm_2021_ESC_GSE165161/AvgSignalSurrounding_10kb_50;cd ${dirPATH}/CHMOrganization/Universal_specific/Functions/DSB/Tillo_NatComm_2021_ESC_GSE165161/AvgSignalSurrounding_10kb_50\n",
    "                        pos_part(){\n",
    "                            for sample in GSM5027366_ESC_DUX_DOX_16h GSM5027367_ESC_DUX GSM5027368_ESC_DUX_DOX_16h GSM5027369_ESC_DUX\n",
    "                            do\n",
    "                                for group in Universal EarlyEmbryoSpecific PGCSpecific SpermSpecific RetinalSpecific DCSpecific nonCHM_CGIs promoter nonCHM_CGIs_HMR\n",
    "                                do\n",
    "                                    bash ${shPATH}/profileSurroundingPeakCenter.sh \\\n",
    "                                        ${dirPATH}/CHMOrganization/Universal_specific/${group}.bed \\\n",
    "                                        10000 \\\n",
    "                                        10000 \\\n",
    "                                        50 \\\n",
    "                                        ${dirPATH}/CHMOrganization/Universal_specific/Functions/DSB/Tillo_NatComm_2021_ESC_GSE165161/${sample}.pos.bw \\\n",
    "                                        ${group} \\\n",
    "                                        \"${sample}.pos\" \\\n",
    "                                        \"-10kb;;10kb\" \\\n",
    "                                        \"0;25.5;50\" \\\n",
    "                                        \"DSB signal\" \\\n",
    "                                        \"0\" \\\n",
    "                                        \"0.1\" \\\n",
    "                                        \"Dark2\"\n",
    "                                done # for group end\n",
    "                            done # for sample end                       \n",
    "                        }\n",
    "                        pos_part\n",
    "                        \n",
    "                        neg_part(){\n",
    "                            for sample in GSM5027366_ESC_DUX_DOX_16h GSM5027367_ESC_DUX GSM5027368_ESC_DUX_DOX_16h GSM5027369_ESC_DUX\n",
    "                            do\n",
    "                                for group in Universal EarlyEmbryoSpecific PGCSpecific SpermSpecific RetinalSpecific DCSpecific nonCHM_CGIs promoter nonCHM_CGIs_HMR\n",
    "                                do\n",
    "                                    bash ${shPATH}/profileSurroundingPeakCenter.sh \\\n",
    "                                        ${dirPATH}/CHMOrganization/Universal_specific/${group}.bed \\\n",
    "                                        10000 \\\n",
    "                                        10000 \\\n",
    "                                        50 \\\n",
    "                                        ${dirPATH}/CHMOrganization/Universal_specific/Functions/DSB/Tillo_NatComm_2021_ESC_GSE165161/${sample}.neg.bw \\\n",
    "                                        ${group} \\\n",
    "                                        \"${sample}.neg\" \\\n",
    "                                        \"-10kb;;10kb\" \\\n",
    "                                        \"0;25.5;50\" \\\n",
    "                                        \"DSB signal\" \\\n",
    "                                        \"-0.1\" \\\n",
    "                                        \"0\" \\\n",
    "                                        \"Dark2\"\n",
    "                                done # for group end\n",
    "                            done # for sample end                       \n",
    "                        }\n",
    "                        neg_part\n",
    "                    }\n",
    "                    # avgSignalSurrounding_10kb_50\n",
    "                    \n",
    "                    each_plot(){\n",
    "                        # D: Nov-07-2022 11:45 Mon\n",
    "                        mkdir -p ${dirPATH}/CHMOrganization/Universal_specific/Functions/DSB/Tillo_NatComm_2021_ESC_GSE165161/AvgSignalSurrounding_10kb_50;cd ${dirPATH}/CHMOrganization/Universal_specific/Functions/DSB/Tillo_NatComm_2021_ESC_GSE165161/AvgSignalSurrounding_10kb_50\n",
    "                        for sample in GSM5027366_ESC_DUX_DOX_16h GSM5027367_ESC_DUX GSM5027368_ESC_DUX_DOX_16h GSM5027369_ESC_DUX\n",
    "                        do\n",
    "                            for type in pos neg\n",
    "                            do\n",
    "                                Rscript ${rPATH}/line_profileSurroundingPeak_auto.r \\\n",
    "                                    \"${sample}.${type}\" \\\n",
    "                                    \"${sample}.${type}\" \\\n",
    "                                    4 \\\n",
    "                                    3.5 \\\n",
    "                                    \"-10kb;;10kb\" \\\n",
    "                                    \"0;25.5;50\" \\\n",
    "                                    \"DSB signal\" \\\n",
    "                                    \"${sample}.${type}\" \\\n",
    "                                    \"Dark2\"\n",
    "                            done # for type end\n",
    "                        done # for sample end\n",
    "                    }\n",
    "                    each_plot\n",
    "                }\n",
    "                # Tillo_NatComm_2021_ESC_GSE165161\n",
    "                \n",
    "                Fowler_Elife_2022_preB_GSE186085(){\n",
    "                    # D: Nov-06-2022 22:23 Sun\n",
    "                    # I: https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE186085\n",
    "                    mkdir -p ${dirPATH}/CHMOrganization/Universal_specific/Functions/DSB/Fowler_Elife_2022_preB_GSE186085;cd ${dirPATH}/CHMOrganization/Universal_specific/Functions/DSB/Fowler_Elife_2022_preB_GSE186085\n",
    "                    download(){\n",
    "                        wget -c \"https://www.ncbi.nlm.nih.gov/geo/download/?acc=GSE186085&format=file\"\n",
    "                    }\n",
    "                    # download\n",
    "                    \n",
    "                    manage(){\n",
    "                        # D: Nov-07-2022 09:49 Mon\n",
    "                        # tar -xvf \"index.html?acc=GSE186085&format=file\" && mv \"index.html?acc=GSE186085&format=file\" GSE186085_RAW.tar\n",
    "                        mv GSM5631734_Abl-Lig4-G2-AsisI-DNAPKi-4h-ENDseq_rep1.neg.bw GSM5631734_Abl_Lig4_G2_AsisI_DNAPKi_4h_rep1.neg.bw\n",
    "                        mv GSM5631734_Abl-Lig4-G2-AsisI-DNAPKi-4h-ENDseq_rep1.pos.bw GSM5631734_Abl_Lig4_G2_AsisI_DNAPKi_4h_rep1.pos.bw\n",
    "                        mv GSM5631735_Abl-Lig4-G2-AsisI-DNAPKi-4h-ENDseq_rep2.neg.bw GSM5631735_Abl_Lig4_G2_AsisI_DNAPKi_4h_rep2.neg.bw\n",
    "                        mv GSM5631735_Abl-Lig4-G2-AsisI-DNAPKi-4h-ENDseq_rep2.pos.bw GSM5631735_Abl_Lig4_G2_AsisI_DNAPKi_4h_rep2.pos.bw\n",
    "                        mv GSM5631736_Abl-Lig4-G2-AsisI-DNAPKi-8h-ENDseq_rep1.neg.bw GSM5631736_Abl_Lig4_G2_AsisI_DNAPKi_8h_rep1.neg.bw\n",
    "                        mv GSM5631736_Abl-Lig4-G2-AsisI-DNAPKi-8h-ENDseq_rep1.pos.bw GSM5631736_Abl_Lig4_G2_AsisI_DNAPKi_8h_rep1.pos.bw\n",
    "                        mv GSM5631737_Abl-Lig4-G2-AsisI-DNAPKi-8h-ENDseq_rep2.neg.bw GSM5631737_Abl_Lig4_G2_AsisI_DNAPKi_8h_rep2.neg.bw\n",
    "                        mv GSM5631737_Abl-Lig4-G2-AsisI-DNAPKi-8h-ENDseq_rep2.pos.bw GSM5631737_Abl_Lig4_G2_AsisI_DNAPKi_8h_rep2.pos.bw\n",
    "                        mv GSM5631738_Abl-Lig4-G2-AsisI-NT-4h-ENDseq_rep1.neg.bw GSM5631738_Abl_Lig4_G2_AsisI_NT_4h_rep1.neg.bw\n",
    "                        mv GSM5631738_Abl-Lig4-G2-AsisI-NT-4h-ENDseq_rep1.pos.bw GSM5631738_Abl_Lig4_G2_AsisI_NT_4h_rep1.pos.bw\n",
    "                        mv GSM5631739_Abl-Lig4-G2-AsisI-NT-4h-ENDseq_rep2.neg.bw GSM5631739_Abl_Lig4_G2_AsisI_NT_4h_rep2.neg.bw\n",
    "                        mv GSM5631739_Abl-Lig4-G2-AsisI-NT-4h-ENDseq_rep2.pos.bw GSM5631739_Abl_Lig4_G2_AsisI_NT_4h_rep2.pos.bw\n",
    "                        mv GSM5631740_Abl-Lig4-G2-AsisI-NT-8h-ENDseq_rep1.neg.bw GSM5631740_Abl_Lig4_G2_AsisI_NT_8h_rep1.neg.bw\n",
    "                        mv GSM5631740_Abl-Lig4-G2-AsisI-NT-8h-ENDseq_rep1.pos.bw GSM5631740_Abl_Lig4_G2_AsisI_NT_8h_rep1.pos.bw\n",
    "                        mv GSM5631741_Abl-Lig4-G2-AsisI-NT-8h-ENDseq_rep2.neg.bw GSM5631741_Abl_Lig4_G2_AsisI_NT_8h_rep2.neg.bw\n",
    "                        mv GSM5631741_Abl-Lig4-G2-AsisI-NT-8h-ENDseq_rep2.pos.bw GSM5631741_Abl_Lig4_G2_AsisI_NT_8h_rep2.pos.bw\n",
    "                        mv GSM5631742_Abl-Lig4-STI-AsisI-DNAPKi-4h-ENDseq_rep1.neg.bw GSM5631742_Abl_Lig4_STI_AsisI_DNAPKi_4h_rep1.neg.bw\n",
    "                        mv GSM5631742_Abl-Lig4-STI-AsisI-DNAPKi-4h-ENDseq_rep1.pos.bw GSM5631742_Abl_Lig4_STI_AsisI_DNAPKi_4h_rep1.pos.bw\n",
    "                        mv GSM5631743_Abl-Lig4-STI-AsisI-DNAPKi-4h-ENDseq_rep2.neg.bw GSM5631743_Abl_Lig4_STI_AsisI_DNAPKi_4h_rep2.neg.bw\n",
    "                        mv GSM5631743_Abl-Lig4-STI-AsisI-DNAPKi-4h-ENDseq_rep2.pos.bw GSM5631743_Abl_Lig4_STI_AsisI_DNAPKi_4h_rep2.pos.bw\n",
    "                        mv GSM5631744_Abl-Lig4-STI-AsisI-DNAPKi-8h-ENDseq_rep1.neg.bw GSM5631744_Abl_Lig4_STI_AsisI_DNAPKi_8h_rep1.neg.bw\n",
    "                        mv GSM5631744_Abl-Lig4-STI-AsisI-DNAPKi-8h-ENDseq_rep1.pos.bw GSM5631744_Abl_Lig4_STI_AsisI_DNAPKi_8h_rep1.pos.bw\n",
    "                        mv GSM5631745_Abl-Lig4-STI-AsisI-DNAPKi-8h-ENDseq_rep2.neg.bw GSM5631745_Abl_Lig4_STI_AsisI_DNAPKi_8h_rep2.neg.bw\n",
    "                        mv GSM5631745_Abl-Lig4-STI-AsisI-DNAPKi-8h-ENDseq_rep2.pos.bw GSM5631745_Abl_Lig4_STI_AsisI_DNAPKi_8h_rep2.pos.bw\n",
    "                        mv GSM5631746_Abl-Lig4-STI-AsisI-NT-4h-ENDseq_rep1.neg.bw GSM5631746_Abl_Lig4_STI_AsisI_NT_4h_rep1.neg.bw\n",
    "                        mv GSM5631746_Abl-Lig4-STI-AsisI-NT-4h-ENDseq_rep1.pos.bw GSM5631746_Abl_Lig4_STI_AsisI_NT_4h_rep1.pos.bw\n",
    "                        mv GSM5631747_Abl-Lig4-STI-AsisI-NT-4h-ENDseq_rep2.neg.bw GSM5631747_Abl_Lig4_STI_AsisI_NT_4h_rep2.neg.bw\n",
    "                        mv GSM5631747_Abl-Lig4-STI-AsisI-NT-4h-ENDseq_rep2.pos.bw GSM5631747_Abl_Lig4_STI_AsisI_NT_4h_rep2.pos.bw\n",
    "                        mv GSM5631748_Abl-Lig4-STI-AsisI-NT-8h-ENDseq_rep1.neg.bw GSM5631748_Abl_Lig4_STI_AsisI_NT_8h_rep1.neg.bw\n",
    "                        mv GSM5631748_Abl-Lig4-STI-AsisI-NT-8h-ENDseq_rep1.pos.bw GSM5631748_Abl_Lig4_STI_AsisI_NT_8h_rep1.pos.bw\n",
    "                        mv GSM5631749_Abl-Lig4-STI-AsisI-NT-8h-ENDseq_rep2.neg.bw GSM5631749_Abl_Lig4_STI_AsisI_NT_8h_rep2.neg.bw\n",
    "                        mv GSM5631749_Abl-Lig4-STI-AsisI-NT-8h-ENDseq_rep2.pos.bw GSM5631749_Abl_Lig4_STI_AsisI_NT_8h_rep2.pos.bw\n",
    "                    }\n",
    "                    # manage\n",
    "                    \n",
    "                    avgSignalSurrounding_10kb_50(){\n",
    "                        # D: Nov-07-2022 09:54 Mon\n",
    "                        mkdir -p ${dirPATH}/CHMOrganization/Universal_specific/Functions/DSB/Fowler_Elife_2022_preB_GSE186085/AvgSignalSurrounding_10kb_50;cd ${dirPATH}/CHMOrganization/Universal_specific/Functions/DSB/Fowler_Elife_2022_preB_GSE186085/AvgSignalSurrounding_10kb_50\n",
    "                        pos_part(){\n",
    "                            for sample in GSM5631734_Abl_Lig4_G2_AsisI_DNAPKi_4h_rep1 GSM5631735_Abl_Lig4_G2_AsisI_DNAPKi_4h_rep2 GSM5631736_Abl_Lig4_G2_AsisI_DNAPKi_8h_rep1 GSM5631737_Abl_Lig4_G2_AsisI_DNAPKi_8h_rep2 GSM5631738_Abl_Lig4_G2_AsisI_NT_4h_rep1 GSM5631739_Abl_Lig4_G2_AsisI_NT_4h_rep2 GSM5631740_Abl_Lig4_G2_AsisI_NT_8h_rep1 GSM5631741_Abl_Lig4_G2_AsisI_NT_8h_rep2 GSM5631742_Abl_Lig4_STI_AsisI_DNAPKi_4h_rep1 GSM5631743_Abl_Lig4_STI_AsisI_DNAPKi_4h_rep2 GSM5631744_Abl_Lig4_STI_AsisI_DNAPKi_8h_rep1 GSM5631745_Abl_Lig4_STI_AsisI_DNAPKi_8h_rep2 GSM5631746_Abl_Lig4_STI_AsisI_NT_4h_rep1 GSM5631747_Abl_Lig4_STI_AsisI_NT_4h_rep2 GSM5631748_Abl_Lig4_STI_AsisI_NT_8h_rep1 GSM5631749_Abl_Lig4_STI_AsisI_NT_8h_rep2\n",
    "                            do\n",
    "                                for group in Universal EarlyEmbryoSpecific PGCSpecific SpermSpecific RetinalSpecific DCSpecific nonCHM_CGIs promoter nonCHM_CGIs_HMR\n",
    "                                do\n",
    "                                    bash ${shPATH}/profileSurroundingPeakCenter.sh \\\n",
    "                                        ${dirPATH}/CHMOrganization/Universal_specific/${group}.bed \\\n",
    "                                        10000 \\\n",
    "                                        10000 \\\n",
    "                                        50 \\\n",
    "                                        ${dirPATH}/CHMOrganization/Universal_specific/Functions/DSB/Fowler_Elife_2022_preB_GSE186085/${sample}.pos.bw \\\n",
    "                                        ${group} \\\n",
    "                                        \"${sample}.pos\" \\\n",
    "                                        \"-10kb;;10kb\" \\\n",
    "                                        \"0;25.5;50\" \\\n",
    "                                        \"DSB signal\" \\\n",
    "                                        \"0\" \\\n",
    "                                        \"0.1\" \\\n",
    "                                        \"Dark2\"\n",
    "                                done # for group end\n",
    "                            done # for sample end                       \n",
    "                        }\n",
    "                        pos_part\n",
    "                        \n",
    "                        neg_part(){\n",
    "                            for sample in GSM5631734_Abl_Lig4_G2_AsisI_DNAPKi_4h_rep1 GSM5631735_Abl_Lig4_G2_AsisI_DNAPKi_4h_rep2 GSM5631736_Abl_Lig4_G2_AsisI_DNAPKi_8h_rep1 GSM5631737_Abl_Lig4_G2_AsisI_DNAPKi_8h_rep2 GSM5631738_Abl_Lig4_G2_AsisI_NT_4h_rep1 GSM5631739_Abl_Lig4_G2_AsisI_NT_4h_rep2 GSM5631740_Abl_Lig4_G2_AsisI_NT_8h_rep1 GSM5631741_Abl_Lig4_G2_AsisI_NT_8h_rep2 GSM5631742_Abl_Lig4_STI_AsisI_DNAPKi_4h_rep1 GSM5631743_Abl_Lig4_STI_AsisI_DNAPKi_4h_rep2 GSM5631744_Abl_Lig4_STI_AsisI_DNAPKi_8h_rep1 GSM5631745_Abl_Lig4_STI_AsisI_DNAPKi_8h_rep2 GSM5631746_Abl_Lig4_STI_AsisI_NT_4h_rep1 GSM5631747_Abl_Lig4_STI_AsisI_NT_4h_rep2 GSM5631748_Abl_Lig4_STI_AsisI_NT_8h_rep1 GSM5631749_Abl_Lig4_STI_AsisI_NT_8h_rep2\n",
    "                            do\n",
    "                                for group in Universal EarlyEmbryoSpecific PGCSpecific SpermSpecific RetinalSpecific DCSpecific nonCHM_CGIs promoter nonCHM_CGIs_HMR\n",
    "                                do\n",
    "                                    bash ${shPATH}/profileSurroundingPeakCenter.sh \\\n",
    "                                        ${dirPATH}/CHMOrganization/Universal_specific/${group}.bed \\\n",
    "                                        10000 \\\n",
    "                                        10000 \\\n",
    "                                        50 \\\n",
    "                                        ${dirPATH}/CHMOrganization/Universal_specific/Functions/DSB/Fowler_Elife_2022_preB_GSE186085/${sample}.neg.bw \\\n",
    "                                        ${group} \\\n",
    "                                        \"${sample}.neg\" \\\n",
    "                                        \"-10kb;;10kb\" \\\n",
    "                                        \"0;25.5;50\" \\\n",
    "                                        \"DSB signal\" \\\n",
    "                                        \"-0.1\" \\\n",
    "                                        \"0\" \\\n",
    "                                        \"Dark2\"\n",
    "                                done # for group end\n",
    "                            done # for sample end                       \n",
    "                        }\n",
    "                        neg_part\n",
    "                    }\n",
    "                    # avgSignalSurrounding_10kb_50\n",
    "                    \n",
    "                    each_plot(){\n",
    "                        # D: Nov-07-2022 11:46 Mon\n",
    "                        mkdir -p ${dirPATH}/CHMOrganization/Universal_specific/Functions/DSB/Fowler_Elife_2022_preB_GSE186085/AvgSignalSurrounding_10kb_50;cd ${dirPATH}/CHMOrganization/Universal_specific/Functions/DSB/Fowler_Elife_2022_preB_GSE186085/AvgSignalSurrounding_10kb_50\n",
    "                        for sample in GSM5631734_Abl_Lig4_G2_AsisI_DNAPKi_4h_rep1 GSM5631735_Abl_Lig4_G2_AsisI_DNAPKi_4h_rep2 GSM5631736_Abl_Lig4_G2_AsisI_DNAPKi_8h_rep1 GSM5631737_Abl_Lig4_G2_AsisI_DNAPKi_8h_rep2 GSM5631738_Abl_Lig4_G2_AsisI_NT_4h_rep1 GSM5631739_Abl_Lig4_G2_AsisI_NT_4h_rep2 GSM5631740_Abl_Lig4_G2_AsisI_NT_8h_rep1 GSM5631741_Abl_Lig4_G2_AsisI_NT_8h_rep2 GSM5631742_Abl_Lig4_STI_AsisI_DNAPKi_4h_rep1 GSM5631743_Abl_Lig4_STI_AsisI_DNAPKi_4h_rep2 GSM5631744_Abl_Lig4_STI_AsisI_DNAPKi_8h_rep1 GSM5631745_Abl_Lig4_STI_AsisI_DNAPKi_8h_rep2 GSM5631746_Abl_Lig4_STI_AsisI_NT_4h_rep1 GSM5631747_Abl_Lig4_STI_AsisI_NT_4h_rep2 GSM5631748_Abl_Lig4_STI_AsisI_NT_8h_rep1 GSM5631749_Abl_Lig4_STI_AsisI_NT_8h_rep2\n",
    "                        do\n",
    "                            for type in pos neg\n",
    "                            do\n",
    "                                Rscript ${rPATH}/line_profileSurroundingPeak_auto.r \\\n",
    "                                    \"${sample}.${type}\" \\\n",
    "                                    \"${sample}.${type}\" \\\n",
    "                                    4 \\\n",
    "                                    3.5 \\\n",
    "                                    \"-10kb;;10kb\" \\\n",
    "                                    \"0;25.5;50\" \\\n",
    "                                    \"DSB signal\" \\\n",
    "                                    \"${sample}.${type}\" \\\n",
    "                                    \"Dark2\"\n",
    "                            done # for type end\n",
    "                        done # for sample end\n",
    "                    }\n",
    "                    each_plot\n",
    "                }\n",
    "                # Fowler_Elife_2022_preB_GSE186085\n",
    "                \n",
    "                Wu_Nature_2021_Neuron_GSE167257(){\n",
    "                    # D: Nov-06-2022 22L28 Sun\n",
    "                    # I: https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE167257\n",
    "                    mkdir -p ${dirPATH}/CHMOrganization/Universal_specific/Functions/DSB/Wu_Nature_2021_Neuron_GSE167257;cd ${dirPATH}/CHMOrganization/Universal_specific/Functions/DSB/Wu_Nature_2021_Neuron_GSE167257\n",
    "                    download(){\n",
    "                        wget -c \"https://www.ncbi.nlm.nih.gov/geo/download/?acc=GSE167257&format=file\"\n",
    "                    }\n",
    "                    # download\n",
    "                    \n",
    "                    manage(){\n",
    "                        # D: Nov-07-2022 10:11 Mon\n",
    "                        tar -xvf \"index.html?acc=GSE167257&format=file\" && mv \"index.html?acc=GSE167257&format=file\" GSE167257_RAW.tar\n",
    "                        mv GSM5100463_pre-B_WT_AsiSI-Dox_SARseq.bw GSM5100463_preB_WT_AsiSI_Dox_SARseq.bw\n",
    "                        mv GSM5100464_pre-B_WT_AsiSI-noDox_SARseq.bw GSM5100464_preB_WT_AsiSI_noDox_SARseq.bw\n",
    "                    }\n",
    "                    # manage\n",
    "                    \n",
    "                    avgSignalSurrounding_10kb_50(){\n",
    "                        # D: Nov-07-2022 10:54 Mon\n",
    "                        mkdir -p ${dirPATH}/CHMOrganization/Universal_specific/Functions/DSB/Wu_Nature_2021_Neuron_GSE167257/AvgSignalSurrounding_10kb_50;cd ${dirPATH}/CHMOrganization/Universal_specific/Functions/DSB/Wu_Nature_2021_Neuron_GSE167257/AvgSignalSurrounding_10kb_50\n",
    "                        pos_part(){\n",
    "                            for sample in GSM5100463_preB_WT_AsiSI_Dox_SARseq GSM5100464_preB_WT_AsiSI_noDox_SARseq\n",
    "                            do\n",
    "                                for group in Universal EarlyEmbryoSpecific PGCSpecific SpermSpecific RetinalSpecific DCSpecific nonCHM_CGIs promoter nonCHM_CGIs_HMR\n",
    "                                do\n",
    "                                    bash ${shPATH}/profileSurroundingPeakCenter.sh \\\n",
    "                                        ${dirPATH}/CHMOrganization/Universal_specific/${group}.bed \\\n",
    "                                        10000 \\\n",
    "                                        10000 \\\n",
    "                                        50 \\\n",
    "                                        ${dirPATH}/CHMOrganization/Universal_specific/Functions/DSB/Wu_Nature_2021_Neuron_GSE167257/${sample}.bw \\\n",
    "                                        ${group} \\\n",
    "                                        \"${sample}\" \\\n",
    "                                        \"-10kb;;10kb\" \\\n",
    "                                        \"0;25.5;50\" \\\n",
    "                                        \"DSB signal\" \\\n",
    "                                        \"0\" \\\n",
    "                                        \"0.1\" \\\n",
    "                                        \"Dark2\"\n",
    "                                done # for group end\n",
    "                            done # for sample end                       \n",
    "                        }\n",
    "                        pos_part\n",
    "                    }\n",
    "                    # avgSignalSurrounding_10kb_50\n",
    "                    \n",
    "                    each_plot(){\n",
    "                        # D: Nov-07-2022 11:47 Mon\n",
    "                        mkdir -p ${dirPATH}/CHMOrganization/Universal_specific/Functions/DSB/Wu_Nature_2021_Neuron_GSE167257/AvgSignalSurrounding_10kb_50;cd ${dirPATH}/CHMOrganization/Universal_specific/Functions/DSB/Wu_Nature_2021_Neuron_GSE167257/AvgSignalSurrounding_10kb_50\n",
    "                        for sample in GSM5100463_preB_WT_AsiSI_Dox_SARseq GSM5100464_preB_WT_AsiSI_noDox_SARseq\n",
    "                        do\n",
    "                            Rscript ${rPATH}/line_profileSurroundingPeak_auto.r \\\n",
    "                                \"${sample}\" \\\n",
    "                                \"${sample}\" \\\n",
    "                                4 \\\n",
    "                                3.5 \\\n",
    "                                \"-10kb;;10kb\" \\\n",
    "                                \"0;25.5;50\" \\\n",
    "                                \"SSB signal\" \\\n",
    "                                \"${sample}\" \\\n",
    "                                \"Dark2\"\n",
    "                        done # for sample end\n",
    "                    }\n",
    "                    each_plot\n",
    "                }\n",
    "                Wu_Nature_2021_Neuron_GSE167257\n",
    "            }\n",
    "            END_seq\n",
    "        }\n",
    "        DSB\n",
    "    }\n",
    "    # functions\n",
    "    \n",
    "    genomeDistribution(){\n",
    "        # D: Nov-06-2022 14:18 Sun\n",
    "        mkdir -p ${dirPATH}/CHMOrganization/Universal_specific/Features/GenomeDistribution;cd ${dirPATH}/CHMOrganization/Universal_specific/Features/GenomeDistribution\n",
    "        PAD_regulatory(){\n",
    "            # D: Nov-06-2022 14:39 Sun\n",
    "            mkdir -p ${dirPATH}/CHMOrganization/Universal_specific/Features/GenomeDistribution/PAD_regulatory;cd ${dirPATH}/CHMOrganization/Universal_specific/Features/GenomeDistribution/PAD_regulatory\n",
    "            # echo -e \"Group\\tTotal\\tPAD\\tEnhancer\\tPromoter\\tnonPAD_enhancer\\tnonPAD_promoter\\tPAD(%)\\tEnhancer(%)\\tPromoter(%)\\tnonPAD_enhancer(%)\\tnonPAD_promoter(%)\" > PAD_regulatory.statistics.txt\n",
    "            # for group in Universal EarlyEmbryoSpecific PGCSpecific SpermSpecific RetinalSpecific DCSpecific\n",
    "            # do\n",
    "            #     N_TOTAL=$(cat ${dirPATH}/CHMOrganization/Universal_specific/${group}.bed | wc -l)\n",
    "            #     N_PAD=$(intersectBed -u -a ${dirPATH}/CHMOrganization/Universal_specific/${group}.bed -b ${dataPATH_pad} | wc -l)\n",
    "            #     N_ENHANCER=$(intersectBed -u -a ${dirPATH}/CHMOrganization/Universal_specific/${group}.bed -b ${dataPATH_enhancer} | wc -l)\n",
    "            #     N_PROMOTER=$(intersectBed -u -a ${dirPATH}/CHMOrganization/Universal_specific/${group}.bed -b ${dataPATH_promoter} | wc -l)\n",
    "            #     N_NON_PAD_ENHANCER=$(intersectBed -v -a ${dirPATH}/CHMOrganization/Universal_specific/${group}.bed -b ${dataPATH_pad} | intersectBed -u -a - -b ${dataPATH_enhancer} | wc -l)\n",
    "            #     N_NON_PAD_PROMOTER=$(intersectBed -v -a ${dirPATH}/CHMOrganization/Universal_specific/${group}.bed -b ${dataPATH_pad} | intersectBed -u -a - -b ${dataPATH_promoter} | wc -l)\n",
    "            #     echo -e \"${group}\\t${N_TOTAL}\\t${N_PAD}\\t${N_ENHANCER}\\t${N_PROMOTER}\\t${N_NON_PAD_ENHANCER}\\t${N_NON_PAD_PROMOTER}\" | \\\n",
    "            #         awk 'BEGIN{FS=OFS=\"\\t\"}{print $1, $2, $3, $4, $5, $6, $7, $3/$2, $4/$2, $5/$2, $6/$2, $7/$2;}' >> PAD_regulatory.statistics.txt\n",
    "            # done # for group end\n",
    "            \n",
    "            PAD_plot(){\n",
    "                # D: Nov-06-2022 15:02 Sun\n",
    "                echo -e \"xValue\\tyValue\\tgroup\" > PAD.ggplot_basic.txt\n",
    "                tail -n +2 PAD_regulatory.statistics.txt | \\\n",
    "                    awk 'BEGIN{FS=OFS=\"\\t\"}{print $1, $8*100, \"PAD\"}' >> PAD.ggplot_basic.txt\n",
    "                Rscript ${rPATH}/ggplot_barplot.r \\\n",
    "                    \"PAD.ggplot_basic.txt\" \\\n",
    "                    \"PAD.ggplot_barplot.pdf\" \\\n",
    "                    \"PAD\" \\\n",
    "                    \"\" \\\n",
    "                    \"Percentage of CHMs located in PAD\" \\\n",
    "                    \"0;100;20\" \\\n",
    "                    \"Universal;EarlyEmbryoSpecific;PGCSpecific;SpermSpecific;RetinalSpecific;DCSpecific\" \\\n",
    "                    \"PAD\" \\\n",
    "                    \"dodge\" \\\n",
    "                    \"3\" \\\n",
    "                    \"5\" \\\n",
    "                    \"-007\" \\\n",
    "                    \"F\"\n",
    "            }\n",
    "            # PAD_plot\n",
    "            \n",
    "            Enhancer_plot(){\n",
    "                # D: Nov-06-2022 15:02 Sun\n",
    "                echo -e \"xValue\\tyValue\\tgroup\" > Enhancer.ggplot_basic.txt\n",
    "                tail -n +2 PAD_regulatory.statistics.txt | \\\n",
    "                    awk 'BEGIN{FS=OFS=\"\\t\"}{print $1, $9*100, \"Enhancer\"}' >> Enhancer.ggplot_basic.txt\n",
    "                Rscript ${rPATH}/ggplot_barplot.r \\\n",
    "                    \"Enhancer.ggplot_basic.txt\" \\\n",
    "                    \"Enhancer.ggplot_barplot.pdf\" \\\n",
    "                    \"Enhancer\" \\\n",
    "                    \"\" \\\n",
    "                    \"Percentage of CHMs located in Enhancer\" \\\n",
    "                    \"0;100;20\" \\\n",
    "                    \"Universal;EarlyEmbryoSpecific;PGCSpecific;SpermSpecific;RetinalSpecific;DCSpecific\" \\\n",
    "                    \"Enhancer\" \\\n",
    "                    \"dodge\" \\\n",
    "                    \"3\" \\\n",
    "                    \"5\" \\\n",
    "                    \"-007\" \\\n",
    "                    \"F\"\n",
    "            }\n",
    "            Enhancer_plot\n",
    "            \n",
    "            Promoter_plot(){\n",
    "                # D: Nov-06-2022 15:02 Sun\n",
    "                echo -e \"xValue\\tyValue\\tgroup\" > Promoter.ggplot_basic.txt\n",
    "                tail -n +2 PAD_regulatory.statistics.txt | \\\n",
    "                    awk 'BEGIN{FS=OFS=\"\\t\"}{print $1, $10*100, \"Promoter\"}' >> Promoter.ggplot_basic.txt\n",
    "                Rscript ${rPATH}/ggplot_barplot.r \\\n",
    "                    \"Promoter.ggplot_basic.txt\" \\\n",
    "                    \"Promoter.ggplot_barplot.pdf\" \\\n",
    "                    \"Promoter\" \\\n",
    "                    \"\" \\\n",
    "                    \"Percentage of CHMs located in Promoter\" \\\n",
    "                    \"0;100;20\" \\\n",
    "                    \"Universal;EarlyEmbryoSpecific;PGCSpecific;SpermSpecific;RetinalSpecific;DCSpecific\" \\\n",
    "                    \"Promoter\" \\\n",
    "                    \"dodge\" \\\n",
    "                    \"3\" \\\n",
    "                    \"5\" \\\n",
    "                    \"-007\" \\\n",
    "                    \"F\"\n",
    "            }\n",
    "            Promoter_plot\n",
    "            \n",
    "            nonPAD_Enhancer_plot(){\n",
    "                # D: Nov-06-2022 15:02 Sun\n",
    "                echo -e \"xValue\\tyValue\\tgroup\" > nonPAD_Enhancer.ggplot_basic.txt\n",
    "                tail -n +2 PAD_regulatory.statistics.txt | \\\n",
    "                    awk 'BEGIN{FS=OFS=\"\\t\"}{print $1, $11*100, \"nonPAD_Enhancer\"}' >> nonPAD_Enhancer.ggplot_basic.txt\n",
    "                Rscript ${rPATH}/ggplot_barplot.r \\\n",
    "                    \"nonPAD_Enhancer.ggplot_basic.txt\" \\\n",
    "                    \"nonPAD_Enhancer.ggplot_barplot.pdf\" \\\n",
    "                    \"nonPAD_Enhancer\" \\\n",
    "                    \"\" \\\n",
    "                    \"Percentage of CHMs located in nonPAD_Enhancer\" \\\n",
    "                    \"0;100;20\" \\\n",
    "                    \"Universal;EarlyEmbryoSpecific;PGCSpecific;SpermSpecific;RetinalSpecific;DCSpecific\" \\\n",
    "                    \"nonPAD_Enhancer\" \\\n",
    "                    \"dodge\" \\\n",
    "                    \"3\" \\\n",
    "                    \"5\" \\\n",
    "                    \"-007\" \\\n",
    "                    \"F\"\n",
    "            }\n",
    "            nonPAD_Enhancer_plot\n",
    "            \n",
    "            nonPAD_Promoter_plot(){\n",
    "                # D: Nov-06-2022 15:02 Sun\n",
    "                echo -e \"xValue\\tyValue\\tgroup\" > nonPAD_Promoter.ggplot_basic.txt\n",
    "                tail -n +2 PAD_regulatory.statistics.txt | \\\n",
    "                    awk 'BEGIN{FS=OFS=\"\\t\"}{print $1, $12*100, \"nonPAD_Promoter\"}' >> nonPAD_Promoter.ggplot_basic.txt\n",
    "                Rscript ${rPATH}/ggplot_barplot.r \\\n",
    "                    \"nonPAD_Promoter.ggplot_basic.txt\" \\\n",
    "                    \"nonPAD_Promoter.ggplot_barplot.pdf\" \\\n",
    "                    \"nonPAD_Promoter\" \\\n",
    "                    \"\" \\\n",
    "                    \"Percentage of CHMs located in nonPAD_Promoter\" \\\n",
    "                    \"0;100;20\" \\\n",
    "                    \"Universal;EarlyEmbryoSpecific;PGCSpecific;SpermSpecific;RetinalSpecific;DCSpecific\" \\\n",
    "                    \"nonPAD_Promoter\" \\\n",
    "                    \"dodge\" \\\n",
    "                    \"3\" \\\n",
    "                    \"5\" \\\n",
    "                    \"-007\" \\\n",
    "                    \"F\"\n",
    "            }\n",
    "            nonPAD_Promoter_plot\n",
    "        }\n",
    "        # PAD_regulatory\n",
    "        \n",
    "        Repeats(){\n",
    "            # D: Nov-06-2022 15:26 Sun\n",
    "            mkdir -p ${dirPATH}/CHMOrganization/Universal_specific/Features/GenomeDistribution/Repeats;cd ${dirPATH}/CHMOrganization/Universal_specific/Features/GenomeDistribution/Repeats\n",
    "            simpleOverlap(){\n",
    "                # D: Nov-06-2022 16:15 Sun\n",
    "                # R: Almost CHMs located in repeats, especially in LTR\n",
    "                echo -e \"xValue\\tyValue\\tgroup\" > repeats.ggplot_basic.txt\n",
    "                echo -e \"xValue\\tyValue\\tgroup\" > LINE.ggplot_basic.txt\n",
    "                echo -e \"xValue\\tyValue\\tgroup\" > SINE.ggplot_basic.txt\n",
    "                echo -e \"xValue\\tyValue\\tgroup\" > LTR.ggplot_basic.txt\n",
    "                for group in Universal EarlyEmbryoSpecific PGCSpecific SpermSpecific RetinalSpecific DCSpecific\n",
    "                do\n",
    "                    N_TOTAL=$(cat ${dirPATH}/CHMOrganization/Universal_specific/${group}.bed | wc -l)\n",
    "                    for type in repeats LINE SINE LTR\n",
    "                    do\n",
    "                        N_TYPE=$(intersectBed -u -a ${dirPATH}/CHMOrganization/Universal_specific/${group}.bed -b ${anPATH}/mm10/Repeats/mm10.${type}.bed | wc -l)\n",
    "                        echo -e \"${group}\\t${N_TOTAL}\\t${N_TYPE}\\t${type}\" | awk 'BEGIN{FS=OFS=\"\\t\"}{print $1, $3/$2*100, $4;}' >> ${type}.ggplot_basic.txt\n",
    "                    done # for type end\n",
    "                done # for group end\n",
    "                \n",
    "                for type in repeats LINE SINE LTR\n",
    "                do\n",
    "                    Rscript ${rPATH}/ggplot_barplot.r \\\n",
    "                        \"${type}.ggplot_basic.txt\" \\\n",
    "                        \"${type}.ggplot_barplot.pdf\" \\\n",
    "                        \"${type}\" \\\n",
    "                        \"\" \\\n",
    "                        \"Percentage of CHMs located in ${type}\" \\\n",
    "                        \"0;100;20\" \\\n",
    "                        \"Universal;EarlyEmbryoSpecific;PGCSpecific;SpermSpecific;RetinalSpecific;DCSpecific\" \\\n",
    "                        \"${type}\" \\\n",
    "                        \"dodge\" \\\n",
    "                        \"3\" \\\n",
    "                        \"5\" \\\n",
    "                        \"-007\" \\\n",
    "                        \"F\"\n",
    "                done # for type end\n",
    "            }\n",
    "            # simpleOverlap\n",
    "            \n",
    "            overlapDetails(){\n",
    "                # D: Nov-06-2022 16:43 Sun\n",
    "                mkdir -p ${dirPATH}/CHMOrganization/Universal_specific/Features/GenomeDistribution/Repeats/OverlapDetails;cd ${dirPATH}/CHMOrganization/Universal_specific/Features/GenomeDistribution/Repeats/OverlapDetails\n",
    "                # chr start end repeats LTR LINE SINE\n",
    "                for group in Universal EarlyEmbryoSpecific PGCSpecific SpermSpecific RetinalSpecific DCSpecific\n",
    "                do\n",
    "                    # cut -f 1-3 ${dirPATH}/CHMOrganization/Universal_specific/${group}.bed > ${group}.overlapDetails.txt\n",
    "                    # intersectBed -c -a ${group}.overlapDetails.txt -b ${anPATH}/mm10/Repeats/mm10.repeats.bed | \\\n",
    "                    #     awk 'BEGIN{FS=OFS=\"\\t\"}{if($4>0){print $1, $2, $3, 1;} else{print $1, $2, $3, 0;}}' > tmp && mv tmp ${group}.overlapDetails.txt\n",
    "                    # intersectBed -c -a ${group}.overlapDetails.txt -b ${anPATH}/mm10/Repeats/mm10.LTR.bed | \\\n",
    "                    #     awk 'BEGIN{FS=OFS=\"\\t\"}{if($5>0){print $1, $2, $3, $4, 1;} else{print $1, $2, $3, $4, 0;}}' > tmp && mv tmp ${group}.overlapDetails.txt\n",
    "                    # intersectBed -c -a ${group}.overlapDetails.txt -b ${anPATH}/mm10/Repeats/mm10.LINE.bed | \\\n",
    "                    #     awk 'BEGIN{FS=OFS=\"\\t\"}{if($6>0){print $1, $2, $3, $4, $5, 1;} else{print $1, $2, $3, $4, $5, 0;}}' > tmp && mv tmp ${group}.overlapDetails.txt\n",
    "                    # intersectBed -c -a ${group}.overlapDetails.txt -b ${anPATH}/mm10/Repeats/mm10.SINE.bed | \\\n",
    "                    #     awk 'BEGIN{FS=OFS=\"\\t\"}{if($7>0){print $1, $2, $3, $4, $5, $6, 1;} else{print $1, $2, $3, $4, $5, $6, 0;}}' > tmp && mv tmp ${group}.overlapDetails.txt\n",
    "                    intersectBed -c -a ${group}.overlapDetails.txt -b ${anPATH}/mm10/Repeats/mm10.others.bed | \\\n",
    "                        awk 'BEGIN{FS=OFS=\"\\t\"}{if($8>0){print $1, $2, $3, $4, $5, $6, $7, 1;} else{print $1, $2, $3, $4, $5, $6, $7, 0;}}' > tmp && mv tmp ${group}.overlapDetails.txt\n",
    "                done # for group end\n",
    "            }\n",
    "            # overlapDetails\n",
    "            \n",
    "            overlapGroups(){\n",
    "                # D: Nov-07-2022 13:55 Mon\n",
    "                mkdir -p ${dirPATH}/CHMOrganization/Universal_specific/Features/GenomeDistribution/Repeats/OverlapDetails;cd ${dirPATH}/CHMOrganization/Universal_specific/Features/GenomeDistribution/Repeats/OverlapDetails\n",
    "                piechart(){\n",
    "                    for group in Universal EarlyEmbryoSpecific PGCSpecific SpermSpecific RetinalSpecific DCSpecific\n",
    "                    do\n",
    "                        # echo -e \"value\\tgroup\" > ${group}.overlapDetails.ggplot_pieplot.txt\n",
    "                        # cut -f 4- ${group}.overlapDetails.txt | sort | uniq -c | awk 'BEGIN{FS=\" \";OFS=\"\\t\"}{print $1,\"g_\"$2$3$4$5$6}' >> ${group}.overlapDetails.ggplot_pieplot.txt\n",
    "                        Rscript ${rPATH}/ggplot_pie.r \\\n",
    "                            ${group}.overlapDetails.ggplot_pieplot.txt \\\n",
    "                            ${group}.overlapDetails \\\n",
    "                            \"g_00000;g_10001;g_10010;g_10011;g_10100;g_10101;g_10110;g_10111;g_11000;g_11001;g_11010;g_11011;g_11100;g_11101;g_11110;g_11111\" \\\n",
    "                            \"#ffffff;#7fa718;#00007E;#0000FF;#007E00;#00FF00;#007E7E;#00FFFF;#7E0000;#FF0000;#7E007E;#FF00FF;#7E7E00;#FFFF00;#7E7E7E;#000000\" \\\n",
    "                            T \\\n",
    "                            6 \\\n",
    "                            6\n",
    "                    done # for group end\n",
    "                }\n",
    "                piechart\n",
    "            }\n",
    "            overlapGroups\n",
    "        }\n",
    "        Repeats\n",
    "    }\n",
    "    genomeDistribution\n",
    "}\n",
    "features"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3 (ipykernel)",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.8.10"
  },
  "toc": {
   "base_numbering": 1,
   "nav_menu": {},
   "number_sections": true,
   "sideBar": true,
   "skip_h1_title": false,
   "title_cell": "Table of Contents",
   "title_sidebar": "Contents",
   "toc_cell": false,
   "toc_position": {
    "height": "calc(100% - 180px)",
    "left": "10px",
    "top": "150px",
    "width": "245px"
   },
   "toc_section_display": true,
   "toc_window_display": true
  }
 },
 "nbformat": 4,
 "nbformat_minor": 4
}
