{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [],
   "source": [
    "import pandas as pd\n",
    "import numpy as np\n",
    "import os"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [],
   "source": [
    "os.chdir('/mnt/Storage/home/wangyiman/CHMsInOtherContexts/CellStateTransition/DataCollection')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Stage</th>\n",
       "      <th>Data</th>\n",
       "      <th>GSMID</th>\n",
       "      <th>GSEID</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>E11.5 Liver</td>\n",
       "      <td>H3K9me3 ChIP-seq</td>\n",
       "      <td>GSM2192347</td>\n",
       "      <td>GSE82974</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>E11.5 Liver</td>\n",
       "      <td>H3K9me3 ChIP-seq</td>\n",
       "      <td>GSM2192348</td>\n",
       "      <td>GSE82974</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>E12.5 Liver</td>\n",
       "      <td>H3K9me3 ChIP-seq</td>\n",
       "      <td>GSM2191248</td>\n",
       "      <td>GSE82470</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>E12.5 Liver</td>\n",
       "      <td>H3K9me3 ChIP-seq</td>\n",
       "      <td>GSM2191249</td>\n",
       "      <td>GSE82470</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>E13.5 Liver</td>\n",
       "      <td>H3K9me3 ChIP-seq</td>\n",
       "      <td>GSM2191298</td>\n",
       "      <td>GSE82494</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "         Stage              Data       GSMID     GSEID\n",
       "0  E11.5 Liver  H3K9me3 ChIP-seq  GSM2192347  GSE82974\n",
       "1  E11.5 Liver  H3K9me3 ChIP-seq  GSM2192348  GSE82974\n",
       "2  E12.5 Liver  H3K9me3 ChIP-seq  GSM2191248  GSE82470\n",
       "3  E12.5 Liver  H3K9me3 ChIP-seq  GSM2191249  GSE82470\n",
       "4  E13.5 Liver  H3K9me3 ChIP-seq  GSM2191298  GSE82494"
      ]
     },
     "execution_count": 4,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "data_geo_dict = {}\n",
    "for tissue in ['heart', 'liver'] :\n",
    "    data_geo_dict[tissue] = pd.read_csv(f'DataCollection_add_{tissue}.csv', header = 0, sep = \",\", usecols = ['Stage', 'Data', 'GSMID', 'GSEID'])\n",
    "\n",
    "data_geo_dict[tissue].head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Run</th>\n",
       "      <th>Assay Type</th>\n",
       "      <th>LibraryLayout</th>\n",
       "      <th>SRA Study</th>\n",
       "      <th>GEO_Accession (exp)</th>\n",
       "      <th>Tissue</th>\n",
       "      <th>dev_stage</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>SRR10049450</td>\n",
       "      <td>ChIP-Seq</td>\n",
       "      <td>SINGLE</td>\n",
       "      <td>SRP013703</td>\n",
       "      <td>GSM4048835</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>SRR10049470</td>\n",
       "      <td>ChIP-Seq</td>\n",
       "      <td>SINGLE</td>\n",
       "      <td>SRP013703</td>\n",
       "      <td>GSM4050822</td>\n",
       "      <td>NaN</td>\n",
       "      <td>embryonic</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>SRR10049471</td>\n",
       "      <td>ChIP-Seq</td>\n",
       "      <td>SINGLE</td>\n",
       "      <td>SRP013703</td>\n",
       "      <td>GSM4050823</td>\n",
       "      <td>NaN</td>\n",
       "      <td>embryonic</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>SRR10049474</td>\n",
       "      <td>ChIP-Seq</td>\n",
       "      <td>SINGLE</td>\n",
       "      <td>SRP013703</td>\n",
       "      <td>GSM4050830</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>SRR10049475</td>\n",
       "      <td>ChIP-Seq</td>\n",
       "      <td>SINGLE</td>\n",
       "      <td>SRP013703</td>\n",
       "      <td>GSM4050831</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4347</th>\n",
       "      <td>SRR16812120</td>\n",
       "      <td>DNase-Hypersensitivity</td>\n",
       "      <td>PAIRED</td>\n",
       "      <td>SRP013703</td>\n",
       "      <td>GSM5669081</td>\n",
       "      <td>right cerebral cortex</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4348</th>\n",
       "      <td>SRR16808739</td>\n",
       "      <td>Bisulfite-Seq</td>\n",
       "      <td>SINGLE</td>\n",
       "      <td>SRP013703</td>\n",
       "      <td>GSM5667532</td>\n",
       "      <td>olfactory bulb</td>\n",
       "      <td>adult</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4349</th>\n",
       "      <td>SRR16814931</td>\n",
       "      <td>Bisulfite-Seq</td>\n",
       "      <td>SINGLE</td>\n",
       "      <td>SRP013703</td>\n",
       "      <td>GSM5669986</td>\n",
       "      <td>cerebellum</td>\n",
       "      <td>adult</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4350</th>\n",
       "      <td>SRR16808574</td>\n",
       "      <td>DNase-Hypersensitivity</td>\n",
       "      <td>PAIRED</td>\n",
       "      <td>SRP013703</td>\n",
       "      <td>GSM5667489</td>\n",
       "      <td>heart</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4351</th>\n",
       "      <td>SRR16809038</td>\n",
       "      <td>DNase-Hypersensitivity</td>\n",
       "      <td>PAIRED</td>\n",
       "      <td>SRP013703</td>\n",
       "      <td>GSM5667605</td>\n",
       "      <td>lung</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>4352 rows × 7 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "              Run              Assay Type LibraryLayout  SRA Study  \\\n",
       "0     SRR10049450                ChIP-Seq        SINGLE  SRP013703   \n",
       "1     SRR10049470                ChIP-Seq        SINGLE  SRP013703   \n",
       "2     SRR10049471                ChIP-Seq        SINGLE  SRP013703   \n",
       "3     SRR10049474                ChIP-Seq        SINGLE  SRP013703   \n",
       "4     SRR10049475                ChIP-Seq        SINGLE  SRP013703   \n",
       "...           ...                     ...           ...        ...   \n",
       "4347  SRR16812120  DNase-Hypersensitivity        PAIRED  SRP013703   \n",
       "4348  SRR16808739           Bisulfite-Seq        SINGLE  SRP013703   \n",
       "4349  SRR16814931           Bisulfite-Seq        SINGLE  SRP013703   \n",
       "4350  SRR16808574  DNase-Hypersensitivity        PAIRED  SRP013703   \n",
       "4351  SRR16809038  DNase-Hypersensitivity        PAIRED  SRP013703   \n",
       "\n",
       "     GEO_Accession (exp)                 Tissue  dev_stage  \n",
       "0             GSM4048835                    NaN        NaN  \n",
       "1             GSM4050822                    NaN  embryonic  \n",
       "2             GSM4050823                    NaN  embryonic  \n",
       "3             GSM4050830                    NaN        NaN  \n",
       "4             GSM4050831                    NaN        NaN  \n",
       "...                  ...                    ...        ...  \n",
       "4347          GSM5669081  right cerebral cortex        NaN  \n",
       "4348          GSM5667532         olfactory bulb      adult  \n",
       "4349          GSM5669986             cerebellum      adult  \n",
       "4350          GSM5667489                  heart        NaN  \n",
       "4351          GSM5667605                   lung        NaN  \n",
       "\n",
       "[4352 rows x 7 columns]"
      ]
     },
     "execution_count": 18,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "sra_meta = pd.read_csv(\"SraRunTable_ENCODE_mouse_embryo.csv\", sep = \",\", header = 0, \n",
    "                       usecols = ['Run', 'Assay Type', \"SRA Study\", \"GEO_Accession (exp)\", 'Tissue', 'dev_stage', 'LibraryLayout'])\n",
    "sra_meta"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 81,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Stage</th>\n",
       "      <th>Data</th>\n",
       "      <th>GSMID</th>\n",
       "      <th>GSEID</th>\n",
       "      <th>Run</th>\n",
       "      <th>Assay Type</th>\n",
       "      <th>LibraryLayout</th>\n",
       "      <th>SRA Study</th>\n",
       "      <th>GEO_Accession (exp)</th>\n",
       "      <th>Tissue</th>\n",
       "      <th>dev_stage</th>\n",
       "      <th>sample_name</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>E11.5 Liver</td>\n",
       "      <td>H3K9me3 ChIP-seq</td>\n",
       "      <td>GSM2192347</td>\n",
       "      <td>GSE82974</td>\n",
       "      <td>SRR3653117</td>\n",
       "      <td>ChIP-Seq</td>\n",
       "      <td>SINGLE</td>\n",
       "      <td>SRP013703</td>\n",
       "      <td>GSM2192347</td>\n",
       "      <td>liver</td>\n",
       "      <td>embryonic</td>\n",
       "      <td>E11.5.H3K9me3.rep1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>E11.5 Liver</td>\n",
       "      <td>H3K9me3 ChIP-seq</td>\n",
       "      <td>GSM2192347</td>\n",
       "      <td>GSE82974</td>\n",
       "      <td>SRR3653118</td>\n",
       "      <td>ChIP-Seq</td>\n",
       "      <td>SINGLE</td>\n",
       "      <td>SRP013703</td>\n",
       "      <td>GSM2192347</td>\n",
       "      <td>liver</td>\n",
       "      <td>embryonic</td>\n",
       "      <td>E11.5.H3K9me3.rep1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>E11.5 Liver</td>\n",
       "      <td>H3K9me3 ChIP-seq</td>\n",
       "      <td>GSM2192348</td>\n",
       "      <td>GSE82974</td>\n",
       "      <td>SRR3653119</td>\n",
       "      <td>ChIP-Seq</td>\n",
       "      <td>SINGLE</td>\n",
       "      <td>SRP013703</td>\n",
       "      <td>GSM2192348</td>\n",
       "      <td>liver</td>\n",
       "      <td>embryonic</td>\n",
       "      <td>E11.5.H3K9me3.rep2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>E11.5 Liver</td>\n",
       "      <td>H3K9me3 ChIP-seq</td>\n",
       "      <td>GSM2192348</td>\n",
       "      <td>GSE82974</td>\n",
       "      <td>SRR3653120</td>\n",
       "      <td>ChIP-Seq</td>\n",
       "      <td>SINGLE</td>\n",
       "      <td>SRP013703</td>\n",
       "      <td>GSM2192348</td>\n",
       "      <td>liver</td>\n",
       "      <td>embryonic</td>\n",
       "      <td>E11.5.H3K9me3.rep2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>E12.5 Liver</td>\n",
       "      <td>H3K9me3 ChIP-seq</td>\n",
       "      <td>GSM2191248</td>\n",
       "      <td>GSE82470</td>\n",
       "      <td>SRR3651489</td>\n",
       "      <td>ChIP-Seq</td>\n",
       "      <td>SINGLE</td>\n",
       "      <td>SRP013703</td>\n",
       "      <td>GSM2191248</td>\n",
       "      <td>liver</td>\n",
       "      <td>embryonic</td>\n",
       "      <td>E12.5.H3K9me3.rep1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>69</th>\n",
       "      <td>E16.5 Liver</td>\n",
       "      <td>WGBS</td>\n",
       "      <td>GSM2191215</td>\n",
       "      <td>GSE82456</td>\n",
       "      <td>SRR3651455</td>\n",
       "      <td>Bisulfite-Seq</td>\n",
       "      <td>SINGLE</td>\n",
       "      <td>SRP013703</td>\n",
       "      <td>GSM2191215</td>\n",
       "      <td>liver</td>\n",
       "      <td>embryonic</td>\n",
       "      <td>E16.5.WGBS.rep4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>70</th>\n",
       "      <td>P0 Liver</td>\n",
       "      <td>WGBS</td>\n",
       "      <td>GSM2191921</td>\n",
       "      <td>GSE82780</td>\n",
       "      <td>SRR3652380</td>\n",
       "      <td>Bisulfite-Seq</td>\n",
       "      <td>SINGLE</td>\n",
       "      <td>SRP013703</td>\n",
       "      <td>GSM2191921</td>\n",
       "      <td>liver</td>\n",
       "      <td>postnatal</td>\n",
       "      <td>P0.WGBS.rep1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>71</th>\n",
       "      <td>P0 Liver</td>\n",
       "      <td>WGBS</td>\n",
       "      <td>GSM2191922</td>\n",
       "      <td>GSE82780</td>\n",
       "      <td>SRR3652381</td>\n",
       "      <td>Bisulfite-Seq</td>\n",
       "      <td>SINGLE</td>\n",
       "      <td>SRP013703</td>\n",
       "      <td>GSM2191922</td>\n",
       "      <td>liver</td>\n",
       "      <td>postnatal</td>\n",
       "      <td>P0.WGBS.rep2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>72</th>\n",
       "      <td>P0 Liver</td>\n",
       "      <td>WGBS</td>\n",
       "      <td>GSM2191923</td>\n",
       "      <td>GSE82780</td>\n",
       "      <td>SRR3652382</td>\n",
       "      <td>Bisulfite-Seq</td>\n",
       "      <td>SINGLE</td>\n",
       "      <td>SRP013703</td>\n",
       "      <td>GSM2191923</td>\n",
       "      <td>liver</td>\n",
       "      <td>postnatal</td>\n",
       "      <td>P0.WGBS.rep3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>73</th>\n",
       "      <td>P0 Liver</td>\n",
       "      <td>WGBS</td>\n",
       "      <td>GSM2191924</td>\n",
       "      <td>GSE82780</td>\n",
       "      <td>SRR3652383</td>\n",
       "      <td>Bisulfite-Seq</td>\n",
       "      <td>SINGLE</td>\n",
       "      <td>SRP013703</td>\n",
       "      <td>GSM2191924</td>\n",
       "      <td>liver</td>\n",
       "      <td>postnatal</td>\n",
       "      <td>P0.WGBS.rep4</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>74 rows × 12 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "          Stage              Data       GSMID     GSEID         Run  \\\n",
       "0   E11.5 Liver  H3K9me3 ChIP-seq  GSM2192347  GSE82974  SRR3653117   \n",
       "1   E11.5 Liver  H3K9me3 ChIP-seq  GSM2192347  GSE82974  SRR3653118   \n",
       "2   E11.5 Liver  H3K9me3 ChIP-seq  GSM2192348  GSE82974  SRR3653119   \n",
       "3   E11.5 Liver  H3K9me3 ChIP-seq  GSM2192348  GSE82974  SRR3653120   \n",
       "4   E12.5 Liver  H3K9me3 ChIP-seq  GSM2191248  GSE82470  SRR3651489   \n",
       "..          ...               ...         ...       ...         ...   \n",
       "69  E16.5 Liver              WGBS  GSM2191215  GSE82456  SRR3651455   \n",
       "70     P0 Liver              WGBS  GSM2191921  GSE82780  SRR3652380   \n",
       "71     P0 Liver              WGBS  GSM2191922  GSE82780  SRR3652381   \n",
       "72     P0 Liver              WGBS  GSM2191923  GSE82780  SRR3652382   \n",
       "73     P0 Liver              WGBS  GSM2191924  GSE82780  SRR3652383   \n",
       "\n",
       "       Assay Type LibraryLayout  SRA Study GEO_Accession (exp) Tissue  \\\n",
       "0        ChIP-Seq        SINGLE  SRP013703          GSM2192347  liver   \n",
       "1        ChIP-Seq        SINGLE  SRP013703          GSM2192347  liver   \n",
       "2        ChIP-Seq        SINGLE  SRP013703          GSM2192348  liver   \n",
       "3        ChIP-Seq        SINGLE  SRP013703          GSM2192348  liver   \n",
       "4        ChIP-Seq        SINGLE  SRP013703          GSM2191248  liver   \n",
       "..            ...           ...        ...                 ...    ...   \n",
       "69  Bisulfite-Seq        SINGLE  SRP013703          GSM2191215  liver   \n",
       "70  Bisulfite-Seq        SINGLE  SRP013703          GSM2191921  liver   \n",
       "71  Bisulfite-Seq        SINGLE  SRP013703          GSM2191922  liver   \n",
       "72  Bisulfite-Seq        SINGLE  SRP013703          GSM2191923  liver   \n",
       "73  Bisulfite-Seq        SINGLE  SRP013703          GSM2191924  liver   \n",
       "\n",
       "    dev_stage         sample_name  \n",
       "0   embryonic  E11.5.H3K9me3.rep1  \n",
       "1   embryonic  E11.5.H3K9me3.rep1  \n",
       "2   embryonic  E11.5.H3K9me3.rep2  \n",
       "3   embryonic  E11.5.H3K9me3.rep2  \n",
       "4   embryonic  E12.5.H3K9me3.rep1  \n",
       "..        ...                 ...  \n",
       "69  embryonic     E16.5.WGBS.rep4  \n",
       "70  postnatal        P0.WGBS.rep1  \n",
       "71  postnatal        P0.WGBS.rep2  \n",
       "72  postnatal        P0.WGBS.rep3  \n",
       "73  postnatal        P0.WGBS.rep4  \n",
       "\n",
       "[74 rows x 12 columns]"
      ]
     },
     "execution_count": 81,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "## merge geo id, sra id together\n",
    "## generate sample name\n",
    "\n",
    "geo_sra_merge_dict = {}\n",
    "sample_name_dict = {}\n",
    "sample_name_dict['heart'] = {}\n",
    "sample_name_dict['liver'] = {}\n",
    "for tissue in ['heart', 'liver'] :\n",
    "    geo_sra_merge_dict[tissue] = pd.merge(data_geo_dict[tissue], sra_meta, left_on = 'GSMID', right_on = 'GEO_Accession (exp)')\n",
    "    for stage in geo_sra_merge_dict[tissue]['Stage'].unique() :\n",
    "        df = geo_sra_merge_dict[tissue].loc[geo_sra_merge_dict[tissue]['Stage'] == stage].reset_index()\n",
    "        for k,data in enumerate(df['Data'].unique()) :\n",
    "            sub_df = df.loc[df['Data'] == data].reset_index()            \n",
    "            for j,gsm in enumerate(sub_df['GSMID'].unique()) :\n",
    "                sub_sub_df = sub_df.loc[sub_df['GSMID'] == gsm].drop('level_0', axis = 1).reset_index()\n",
    "                data_type = data if data == 'WGBS' else data.split(' ')[0] if data[-8:] == \"ChIP-seq\" else \"RNASeq\"\n",
    "                sample_name_dict[tissue][gsm] = f\"{stage.split()[0]}.{data_type}.rep{j+1}\"\n",
    "        \n",
    "    sample_name_df = pd.DataFrame.from_dict(sample_name_dict[tissue], orient='index').reset_index()\n",
    "    sample_name_df.columns = ['GSMID', 'sample_name']\n",
    "    geo_sra_merge_dict[tissue] = pd.merge(geo_sra_merge_dict[tissue], sample_name_df, on = 'GSMID')\n",
    "    geo_sra_merge_dict[tissue].to_csv(f'geo_sra_id_{tissue}.csv', sep = \",\", index = False, header = True)\n",
    "\n",
    "geo_sra_merge_dict[tissue] "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "metadata": {},
   "outputs": [],
   "source": [
    "## sra list for raw data downloading\n",
    "\n",
    "for tissue in ['heart', 'liver'] :\n",
    "    pd.DataFrame(geo_sra_merge_dict[tissue]['Run'].unique()).to_csv(f'download_sra_list_{tissue}.txt', header = False, index = False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 114,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "\n",
      "cd ~/CHMsInOtherContexts/CellStateTransition/HeartDevelopment\n",
      "mkdir E10.5.WGBS.rep1 && cd E10.5.WGBS.rep1\n",
      "ln -sf ../0_raw_data/SRR5833022.fastq.gz E10.5.WGBS.rep1.fq.gz && cd ../\n",
      "mkdir E10.5.WGBS.rep2 && cd E10.5.WGBS.rep2\n",
      "ln -sf ../0_raw_data/SRR5833023.fastq.gz E10.5.WGBS.rep2.fq.gz && cd ../\n",
      "mkdir E10.5.WGBS.rep3 && cd E10.5.WGBS.rep3\n",
      "ln -sf ../0_raw_data/SRR5833024.fastq.gz E10.5.WGBS.rep3.fq.gz && cd ../\n",
      "mkdir E10.5.WGBS.rep4 && cd E10.5.WGBS.rep4\n",
      "ln -sf ../0_raw_data/SRR5833025.fastq.gz E10.5.WGBS.rep4.fq.gz && cd ../\n",
      "mkdir E11.5.WGBS.rep1 && cd E11.5.WGBS.rep1\n",
      "ln -sf ../0_raw_data/SRR3652576.fastq.gz E11.5.WGBS.rep1.fq.gz && cd ../\n",
      "mkdir E11.5.WGBS.rep2 && cd E11.5.WGBS.rep2\n",
      "ln -sf ../0_raw_data/SRR3652577.fastq.gz E11.5.WGBS.rep2.fq.gz && cd ../\n",
      "mkdir E11.5.WGBS.rep3 && cd E11.5.WGBS.rep3\n",
      "ln -sf ../0_raw_data/SRR3652578.fastq.gz E11.5.WGBS.rep3.fq.gz && cd ../\n",
      "mkdir E11.5.WGBS.rep4 && cd E11.5.WGBS.rep4\n",
      "ln -sf ../0_raw_data/SRR3652579.fastq.gz E11.5.WGBS.rep4.fq.gz && cd ../\n",
      "mkdir E12.5.WGBS.rep1 && cd E12.5.WGBS.rep1\n",
      "ln -sf ../0_raw_data/SRR3651703.fastq.gz E12.5.WGBS.rep1.fq.gz && cd ../\n",
      "mkdir E12.5.WGBS.rep2 && cd E12.5.WGBS.rep2\n",
      "ln -sf ../0_raw_data/SRR3651704.fastq.gz E12.5.WGBS.rep2.fq.gz && cd ../\n",
      "mkdir E12.5.WGBS.rep3 && cd E12.5.WGBS.rep3\n",
      "ln -sf ../0_raw_data/SRR3651705.fastq.gz E12.5.WGBS.rep3.fq.gz && cd ../\n",
      "mkdir E12.5.WGBS.rep4 && cd E12.5.WGBS.rep4\n",
      "ln -sf ../0_raw_data/SRR3651706.fastq.gz E12.5.WGBS.rep4.fq.gz && cd ../\n",
      "mkdir E13.5.WGBS.rep1 && cd E13.5.WGBS.rep1\n",
      "ln -sf ../0_raw_data/SRR3651479.fastq.gz E13.5.WGBS.rep1.fq.gz && cd ../\n",
      "mkdir E13.5.WGBS.rep2 && cd E13.5.WGBS.rep2\n",
      "ln -sf ../0_raw_data/SRR3651480.fastq.gz E13.5.WGBS.rep2.fq.gz && cd ../\n",
      "mkdir E13.5.WGBS.rep3 && cd E13.5.WGBS.rep3\n",
      "ln -sf ../0_raw_data/SRR3651481.fastq.gz E13.5.WGBS.rep3.fq.gz && cd ../\n",
      "mkdir E13.5.WGBS.rep4 && cd E13.5.WGBS.rep4\n",
      "ln -sf ../0_raw_data/SRR3651482.fastq.gz E13.5.WGBS.rep4.fq.gz && cd ../\n",
      "mkdir E14.5.WGBS.rep1 && cd E14.5.WGBS.rep1\n",
      "ln -sf ../0_raw_data/SRR3652612.fastq.gz E14.5.WGBS.rep1.fq.gz && cd ../\n",
      "mkdir E14.5.WGBS.rep2 && cd E14.5.WGBS.rep2\n",
      "ln -sf ../0_raw_data/SRR3652613.fastq.gz E14.5.WGBS.rep2.fq.gz && cd ../\n",
      "mkdir E14.5.WGBS.rep3 && cd E14.5.WGBS.rep3\n",
      "ln -sf ../0_raw_data/SRR3652614.fastq.gz E14.5.WGBS.rep3.fq.gz && cd ../\n",
      "mkdir E14.5.WGBS.rep4 && cd E14.5.WGBS.rep4\n",
      "ln -sf ../0_raw_data/SRR3652615.fastq.gz E14.5.WGBS.rep4.fq.gz && cd ../\n",
      "mkdir E15.5.WGBS.rep1 && cd E15.5.WGBS.rep1\n",
      "ln -sf ../0_raw_data/SRR3651737.fastq.gz E15.5.WGBS.rep1.fq.gz && cd ../\n",
      "mkdir E15.5.WGBS.rep2 && cd E15.5.WGBS.rep2\n",
      "ln -sf ../0_raw_data/SRR3651738.fastq.gz E15.5.WGBS.rep2.fq.gz && cd ../\n",
      "mkdir E15.5.WGBS.rep3 && cd E15.5.WGBS.rep3\n",
      "ln -sf ../0_raw_data/SRR3651739.fastq.gz E15.5.WGBS.rep3.fq.gz && cd ../\n",
      "mkdir E15.5.WGBS.rep4 && cd E15.5.WGBS.rep4\n",
      "ln -sf ../0_raw_data/SRR3651740.fastq.gz E15.5.WGBS.rep4.fq.gz && cd ../\n",
      "mkdir E16.5.WGBS.rep1 && cd E16.5.WGBS.rep1\n",
      "ln -sf ../0_raw_data/SRR3651253.fastq.gz E16.5.WGBS.rep1.fq.gz && cd ../\n",
      "mkdir E16.5.WGBS.rep2 && cd E16.5.WGBS.rep2\n",
      "ln -sf ../0_raw_data/SRR3651254.fastq.gz E16.5.WGBS.rep2.fq.gz && cd ../\n",
      "mkdir E16.5.WGBS.rep3 && cd E16.5.WGBS.rep3\n",
      "ln -sf ../0_raw_data/SRR3651255.fastq.gz E16.5.WGBS.rep3.fq.gz && cd ../\n",
      "mkdir E16.5.WGBS.rep4 && cd E16.5.WGBS.rep4\n",
      "ln -sf ../0_raw_data/SRR3651256.fastq.gz E16.5.WGBS.rep4.fq.gz && cd ../\n",
      "mkdir P0.WGBS.rep1 && cd P0.WGBS.rep1\n",
      "ln -sf ../0_raw_data/SRR3652006.fastq.gz P0.WGBS.rep1.fq.gz && cd ../\n",
      "mkdir P0.WGBS.rep2 && cd P0.WGBS.rep2\n",
      "ln -sf ../0_raw_data/SRR3652007.fastq.gz P0.WGBS.rep2.fq.gz && cd ../\n",
      "mkdir P0.WGBS.rep3 && cd P0.WGBS.rep3\n",
      "ln -sf ../0_raw_data/SRR3652008.fastq.gz P0.WGBS.rep3.fq.gz && cd ../\n",
      "mkdir P0.WGBS.rep4 && cd P0.WGBS.rep4\n",
      "ln -sf ../0_raw_data/SRR3652009.fastq.gz P0.WGBS.rep4.fq.gz && cd ../\n",
      "mkdir E10.5.H3K9me3.rep1 && cd E10.5.H3K9me3.rep1\n",
      "ln -sf ../0_raw_data/SRR4235654.fastq.gz E10.5.H3K9me3.rep1.fq.gz && cd ../\n",
      "mkdir E10.5.H3K9me3.rep2 && cd E10.5.H3K9me3.rep2\n",
      "ln -sf ../0_raw_data/SRR4235655.fastq.gz E10.5.H3K9me3.rep2.fq.gz && cd ../\n",
      "mkdir E11.5.H3K9me3.rep1 && cd E11.5.H3K9me3.rep1\n",
      "ln -sf ../0_raw_data/SRR3652172.fastq.gz E11.5.H3K9me3.rep1_part1.fq.gz\n",
      "ln -sf ../0_raw_data/SRR3652173.fastq.gz E11.5.H3K9me3.rep1_part2.fq.gz && cd ../\n",
      "cat E11.5.H3K9me3.rep1/E11.5.H3K9me3.rep1_part*.fq.gz > E11.5.H3K9me3.rep1/E11.5.H3K9me3.rep1.fq.gz && rm E11.5.H3K9me3.rep1/E11.5.H3K9me3.rep1_part*.fq.gz\n",
      "mkdir E11.5.H3K9me3.rep2 && cd E11.5.H3K9me3.rep2\n",
      "ln -sf ../0_raw_data/SRR3652174.fastq.gz E11.5.H3K9me3.rep2.fq.gz && cd ../\n",
      "mkdir E12.5.H3K9me3.rep1 && cd E12.5.H3K9me3.rep1\n",
      "ln -sf ../0_raw_data/SRR3652407.fastq.gz E12.5.H3K9me3.rep1.fq.gz && cd ../\n",
      "mkdir E12.5.H3K9me3.rep2 && cd E12.5.H3K9me3.rep2\n",
      "ln -sf ../0_raw_data/SRR3652408.fastq.gz E12.5.H3K9me3.rep2.fq.gz && cd ../\n",
      "mkdir E13.5.H3K9me3.rep1 && cd E13.5.H3K9me3.rep1\n",
      "ln -sf ../0_raw_data/SRR3651385.fastq.gz E13.5.H3K9me3.rep1.fq.gz && cd ../\n",
      "mkdir E13.5.H3K9me3.rep2 && cd E13.5.H3K9me3.rep2\n",
      "ln -sf ../0_raw_data/SRR3651386.fastq.gz E13.5.H3K9me3.rep2.fq.gz && cd ../\n",
      "mkdir E14.5.H3K9me3.rep1 && cd E14.5.H3K9me3.rep1\n",
      "ln -sf ../0_raw_data/SRR3651163.fastq.gz E14.5.H3K9me3.rep1_part1.fq.gz\n",
      "ln -sf ../0_raw_data/SRR3651164.fastq.gz E14.5.H3K9me3.rep1_part2.fq.gz && cd ../\n",
      "cat E14.5.H3K9me3.rep1/E14.5.H3K9me3.rep1_part*.fq.gz > E14.5.H3K9me3.rep1/E14.5.H3K9me3.rep1.fq.gz && rm E14.5.H3K9me3.rep1/E14.5.H3K9me3.rep1_part*.fq.gz\n",
      "mkdir E14.5.H3K9me3.rep2 && cd E14.5.H3K9me3.rep2\n",
      "ln -sf ../0_raw_data/SRR3651165.fastq.gz E14.5.H3K9me3.rep2_part1.fq.gz\n",
      "ln -sf ../0_raw_data/SRR3651166.fastq.gz E14.5.H3K9me3.rep2_part2.fq.gz\n",
      "ln -sf ../0_raw_data/SRR3651167.fastq.gz E14.5.H3K9me3.rep2_part3.fq.gz && cd ../\n",
      "cat E14.5.H3K9me3.rep2/E14.5.H3K9me3.rep2_part*.fq.gz > E14.5.H3K9me3.rep2/E14.5.H3K9me3.rep2.fq.gz && rm E14.5.H3K9me3.rep2/E14.5.H3K9me3.rep2_part*.fq.gz\n",
      "mkdir E15.5.H3K9me3.rep1 && cd E15.5.H3K9me3.rep1\n",
      "ln -sf ../0_raw_data/SRR3651354.fastq.gz E15.5.H3K9me3.rep1.fq.gz && cd ../\n",
      "mkdir E15.5.H3K9me3.rep2 && cd E15.5.H3K9me3.rep2\n",
      "ln -sf ../0_raw_data/SRR3651355.fastq.gz E15.5.H3K9me3.rep2.fq.gz && cd ../\n",
      "mkdir E16.5.H3K9me3.rep1 && cd E16.5.H3K9me3.rep1\n",
      "ln -sf ../0_raw_data/SRR3651436.fastq.gz E16.5.H3K9me3.rep1.fq.gz && cd ../\n",
      "mkdir E16.5.H3K9me3.rep2 && cd E16.5.H3K9me3.rep2\n",
      "ln -sf ../0_raw_data/SRR3651437.fastq.gz E16.5.H3K9me3.rep2.fq.gz && cd ../\n",
      "mkdir P0.H3K9me3.rep1 && cd P0.H3K9me3.rep1\n",
      "ln -sf ../0_raw_data/SRR3653355.fastq.gz P0.H3K9me3.rep1_part1.fq.gz\n",
      "ln -sf ../0_raw_data/SRR3653356.fastq.gz P0.H3K9me3.rep1_part2.fq.gz && cd ../\n",
      "cat P0.H3K9me3.rep1/P0.H3K9me3.rep1_part*.fq.gz > P0.H3K9me3.rep1/P0.H3K9me3.rep1.fq.gz && rm P0.H3K9me3.rep1/P0.H3K9me3.rep1_part*.fq.gz\n",
      "mkdir P0.H3K9me3.rep2 && cd P0.H3K9me3.rep2\n",
      "ln -sf ../0_raw_data/SRR3653357.fastq.gz P0.H3K9me3.rep2_part1.fq.gz\n",
      "ln -sf ../0_raw_data/SRR3653358.fastq.gz P0.H3K9me3.rep2_part2.fq.gz && cd ../\n",
      "cat P0.H3K9me3.rep2/P0.H3K9me3.rep2_part*.fq.gz > P0.H3K9me3.rep2/P0.H3K9me3.rep2.fq.gz && rm P0.H3K9me3.rep2/P0.H3K9me3.rep2_part*.fq.gz\n",
      "mkdir E10.5.input.rep1 && cd E10.5.input.rep1\n",
      "ln -sf ../0_raw_data/SRR4235703.fastq.gz E10.5.input.rep1.fq.gz && cd ../\n",
      "mkdir E10.5.input.rep2 && cd E10.5.input.rep2\n",
      "ln -sf ../0_raw_data/SRR4235704.fastq.gz E10.5.input.rep2.fq.gz && cd ../\n",
      "mkdir E11.5.input.rep1 && cd E11.5.input.rep1\n",
      "ln -sf ../0_raw_data/SRR3651841.fastq.gz E11.5.input.rep1.fq.gz && cd ../\n",
      "mkdir E11.5.input.rep2 && cd E11.5.input.rep2\n",
      "ln -sf ../0_raw_data/SRR3651842.fastq.gz E11.5.input.rep2.fq.gz && cd ../\n",
      "mkdir E12.5.input.rep1 && cd E12.5.input.rep1\n",
      "ln -sf ../0_raw_data/SRR3652651.fastq.gz E12.5.input.rep1.fq.gz && cd ../\n",
      "mkdir E12.5.input.rep2 && cd E12.5.input.rep2\n",
      "ln -sf ../0_raw_data/SRR3652652.fastq.gz E12.5.input.rep2_part1.fq.gz\n",
      "ln -sf ../0_raw_data/SRR3652653.fastq.gz E12.5.input.rep2_part2.fq.gz && cd ../\n",
      "cat E12.5.input.rep2/E12.5.input.rep2_part*.fq.gz > E12.5.input.rep2/E12.5.input.rep2.fq.gz && rm E12.5.input.rep2/E12.5.input.rep2_part*.fq.gz\n",
      "mkdir E13.5.input.rep1 && cd E13.5.input.rep1\n",
      "ln -sf ../0_raw_data/SRR3651324.fastq.gz E13.5.input.rep1_part1.fq.gz\n",
      "ln -sf ../0_raw_data/SRR3651325.fastq.gz E13.5.input.rep1_part2.fq.gz && cd ../\n",
      "cat E13.5.input.rep1/E13.5.input.rep1_part*.fq.gz > E13.5.input.rep1/E13.5.input.rep1.fq.gz && rm E13.5.input.rep1/E13.5.input.rep1_part*.fq.gz\n",
      "mkdir E13.5.input.rep2 && cd E13.5.input.rep2\n",
      "ln -sf ../0_raw_data/SRR3651326.fastq.gz E13.5.input.rep2_part1.fq.gz\n",
      "ln -sf ../0_raw_data/SRR3651327.fastq.gz E13.5.input.rep2_part2.fq.gz\n",
      "ln -sf ../0_raw_data/SRR3651328.fastq.gz E13.5.input.rep2_part3.fq.gz && cd ../\n",
      "cat E13.5.input.rep2/E13.5.input.rep2_part*.fq.gz > E13.5.input.rep2/E13.5.input.rep2.fq.gz && rm E13.5.input.rep2/E13.5.input.rep2_part*.fq.gz\n",
      "mkdir E14.5.input.rep1 && cd E14.5.input.rep1\n",
      "ln -sf ../0_raw_data/SRR3653110.fastq.gz E14.5.input.rep1.fq.gz && cd ../\n",
      "mkdir E14.5.input.rep2 && cd E14.5.input.rep2\n",
      "ln -sf ../0_raw_data/SRR3653111.fastq.gz E14.5.input.rep2.fq.gz && cd ../\n",
      "mkdir E15.5.input.rep1 && cd E15.5.input.rep1\n",
      "ln -sf ../0_raw_data/SRR3652069.fastq.gz E15.5.input.rep1_part1.fq.gz\n",
      "ln -sf ../0_raw_data/SRR3652070.fastq.gz E15.5.input.rep1_part2.fq.gz && cd ../\n",
      "cat E15.5.input.rep1/E15.5.input.rep1_part*.fq.gz > E15.5.input.rep1/E15.5.input.rep1.fq.gz && rm E15.5.input.rep1/E15.5.input.rep1_part*.fq.gz\n",
      "mkdir E15.5.input.rep2 && cd E15.5.input.rep2\n",
      "ln -sf ../0_raw_data/SRR3652071.fastq.gz E15.5.input.rep2_part1.fq.gz\n",
      "ln -sf ../0_raw_data/SRR3652072.fastq.gz E15.5.input.rep2_part2.fq.gz && cd ../\n",
      "cat E15.5.input.rep2/E15.5.input.rep2_part*.fq.gz > E15.5.input.rep2/E15.5.input.rep2.fq.gz && rm E15.5.input.rep2/E15.5.input.rep2_part*.fq.gz\n",
      "mkdir E16.5.input.rep1 && cd E16.5.input.rep1\n",
      "ln -sf ../0_raw_data/SRR3652779.fastq.gz E16.5.input.rep1.fq.gz && cd ../\n",
      "mkdir E16.5.input.rep2 && cd E16.5.input.rep2\n",
      "ln -sf ../0_raw_data/SRR3652780.fastq.gz E16.5.input.rep2.fq.gz && cd ../\n",
      "mkdir P0.input.rep1 && cd P0.input.rep1\n",
      "ln -sf ../0_raw_data/SRR3651884.fastq.gz P0.input.rep1.fq.gz && cd ../\n",
      "mkdir P0.input.rep2 && cd P0.input.rep2\n",
      "ln -sf ../0_raw_data/SRR3651885.fastq.gz P0.input.rep2.fq.gz && cd ../\n",
      "\n",
      "\n",
      "cd ~/CHMsInOtherContexts/CellStateTransition/LiverDevelopment\n",
      "mkdir E11.5.H3K9me3.rep1 && cd E11.5.H3K9me3.rep1\n",
      "ln -sf ../0_raw_data/SRR3653117.fastq.gz E11.5.H3K9me3.rep1_part1.fq.gz\n",
      "ln -sf ../0_raw_data/SRR3653118.fastq.gz E11.5.H3K9me3.rep1_part2.fq.gz && cd ../\n",
      "cat E11.5.H3K9me3.rep1/E11.5.H3K9me3.rep1_part*.fq.gz > E11.5.H3K9me3.rep1/E11.5.H3K9me3.rep1.fq.gz && rm E11.5.H3K9me3.rep1/E11.5.H3K9me3.rep1_part*.fq.gz\n",
      "mkdir E11.5.H3K9me3.rep2 && cd E11.5.H3K9me3.rep2\n",
      "ln -sf ../0_raw_data/SRR3653119.fastq.gz E11.5.H3K9me3.rep2_part1.fq.gz\n",
      "ln -sf ../0_raw_data/SRR3653120.fastq.gz E11.5.H3K9me3.rep2_part2.fq.gz && cd ../\n",
      "cat E11.5.H3K9me3.rep2/E11.5.H3K9me3.rep2_part*.fq.gz > E11.5.H3K9me3.rep2/E11.5.H3K9me3.rep2.fq.gz && rm E11.5.H3K9me3.rep2/E11.5.H3K9me3.rep2_part*.fq.gz\n",
      "mkdir E12.5.H3K9me3.rep1 && cd E12.5.H3K9me3.rep1\n",
      "ln -sf ../0_raw_data/SRR3651489.fastq.gz E12.5.H3K9me3.rep1.fq.gz && cd ../\n",
      "mkdir E12.5.H3K9me3.rep2 && cd E12.5.H3K9me3.rep2\n",
      "ln -sf ../0_raw_data/SRR3651490.fastq.gz E12.5.H3K9me3.rep2.fq.gz && cd ../\n",
      "mkdir E13.5.H3K9me3.rep1 && cd E13.5.H3K9me3.rep1\n",
      "ln -sf ../0_raw_data/SRR3651548.fastq.gz E13.5.H3K9me3.rep1.fq.gz && cd ../\n",
      "mkdir E13.5.H3K9me3.rep2 && cd E13.5.H3K9me3.rep2\n",
      "ln -sf ../0_raw_data/SRR3651549.fastq.gz E13.5.H3K9me3.rep2.fq.gz && cd ../\n",
      "mkdir E14.5.H3K9me3.rep1 && cd E14.5.H3K9me3.rep1\n",
      "ln -sf ../0_raw_data/SRR3651430.fastq.gz E14.5.H3K9me3.rep1_part1.fq.gz\n",
      "ln -sf ../0_raw_data/SRR3651431.fastq.gz E14.5.H3K9me3.rep1_part2.fq.gz\n",
      "ln -sf ../0_raw_data/SRR3651432.fastq.gz E14.5.H3K9me3.rep1_part3.fq.gz && cd ../\n",
      "cat E14.5.H3K9me3.rep1/E14.5.H3K9me3.rep1_part*.fq.gz > E14.5.H3K9me3.rep1/E14.5.H3K9me3.rep1.fq.gz && rm E14.5.H3K9me3.rep1/E14.5.H3K9me3.rep1_part*.fq.gz\n",
      "mkdir E14.5.H3K9me3.rep2 && cd E14.5.H3K9me3.rep2\n",
      "ln -sf ../0_raw_data/SRR3651433.fastq.gz E14.5.H3K9me3.rep2_part1.fq.gz\n",
      "ln -sf ../0_raw_data/SRR3651434.fastq.gz E14.5.H3K9me3.rep2_part2.fq.gz\n",
      "ln -sf ../0_raw_data/SRR3651435.fastq.gz E14.5.H3K9me3.rep2_part3.fq.gz && cd ../\n",
      "cat E14.5.H3K9me3.rep2/E14.5.H3K9me3.rep2_part*.fq.gz > E14.5.H3K9me3.rep2/E14.5.H3K9me3.rep2.fq.gz && rm E14.5.H3K9me3.rep2/E14.5.H3K9me3.rep2_part*.fq.gz\n",
      "mkdir E15.5.H3K9me3.rep1 && cd E15.5.H3K9me3.rep1\n",
      "ln -sf ../0_raw_data/SRR3653134.fastq.gz E15.5.H3K9me3.rep1.fq.gz && cd ../\n",
      "mkdir E15.5.H3K9me3.rep2 && cd E15.5.H3K9me3.rep2\n",
      "ln -sf ../0_raw_data/SRR3653135.fastq.gz E15.5.H3K9me3.rep2.fq.gz && cd ../\n",
      "mkdir E16.5.H3K9me3.rep1 && cd E16.5.H3K9me3.rep1\n",
      "ln -sf ../0_raw_data/SRR3651381.fastq.gz E16.5.H3K9me3.rep1.fq.gz && cd ../\n",
      "mkdir E16.5.H3K9me3.rep2 && cd E16.5.H3K9me3.rep2\n",
      "ln -sf ../0_raw_data/SRR3651382.fastq.gz E16.5.H3K9me3.rep2.fq.gz && cd ../\n",
      "mkdir P0.H3K9me3.rep1 && cd P0.H3K9me3.rep1\n",
      "ln -sf ../0_raw_data/SRR3651898.fastq.gz P0.H3K9me3.rep1_part1.fq.gz\n",
      "ln -sf ../0_raw_data/SRR3651899.fastq.gz P0.H3K9me3.rep1_part2.fq.gz && cd ../\n",
      "cat P0.H3K9me3.rep1/P0.H3K9me3.rep1_part*.fq.gz > P0.H3K9me3.rep1/P0.H3K9me3.rep1.fq.gz && rm P0.H3K9me3.rep1/P0.H3K9me3.rep1_part*.fq.gz\n",
      "mkdir P0.H3K9me3.rep2 && cd P0.H3K9me3.rep2\n",
      "ln -sf ../0_raw_data/SRR3651900.fastq.gz P0.H3K9me3.rep2_part1.fq.gz\n",
      "ln -sf ../0_raw_data/SRR3651901.fastq.gz P0.H3K9me3.rep2_part2.fq.gz\n",
      "ln -sf ../0_raw_data/SRR3651902.fastq.gz P0.H3K9me3.rep2_part3.fq.gz && cd ../\n",
      "cat P0.H3K9me3.rep2/P0.H3K9me3.rep2_part*.fq.gz > P0.H3K9me3.rep2/P0.H3K9me3.rep2.fq.gz && rm P0.H3K9me3.rep2/P0.H3K9me3.rep2_part*.fq.gz\n",
      "mkdir E11.5.input.rep1 && cd E11.5.input.rep1\n",
      "ln -sf ../0_raw_data/SRR3652454.fastq.gz E11.5.input.rep1.fq.gz && cd ../\n",
      "mkdir E11.5.input.rep2 && cd E11.5.input.rep2\n",
      "ln -sf ../0_raw_data/SRR3652455.fastq.gz E11.5.input.rep2.fq.gz && cd ../\n",
      "mkdir E12.5.input.rep1 && cd E12.5.input.rep1\n",
      "ln -sf ../0_raw_data/SRR3652762.fastq.gz E12.5.input.rep1_part1.fq.gz\n",
      "ln -sf ../0_raw_data/SRR3652763.fastq.gz E12.5.input.rep1_part2.fq.gz && cd ../\n",
      "cat E12.5.input.rep1/E12.5.input.rep1_part*.fq.gz > E12.5.input.rep1/E12.5.input.rep1.fq.gz && rm E12.5.input.rep1/E12.5.input.rep1_part*.fq.gz\n",
      "mkdir E12.5.input.rep2 && cd E12.5.input.rep2\n",
      "ln -sf ../0_raw_data/SRR3652764.fastq.gz E12.5.input.rep2_part1.fq.gz\n",
      "ln -sf ../0_raw_data/SRR3652765.fastq.gz E12.5.input.rep2_part2.fq.gz && cd ../\n",
      "cat E12.5.input.rep2/E12.5.input.rep2_part*.fq.gz > E12.5.input.rep2/E12.5.input.rep2.fq.gz && rm E12.5.input.rep2/E12.5.input.rep2_part*.fq.gz\n",
      "mkdir E13.5.input.rep1 && cd E13.5.input.rep1\n",
      "ln -sf ../0_raw_data/SRR3651584.fastq.gz E13.5.input.rep1_part1.fq.gz\n",
      "ln -sf ../0_raw_data/SRR3651585.fastq.gz E13.5.input.rep1_part2.fq.gz\n",
      "ln -sf ../0_raw_data/SRR3651586.fastq.gz E13.5.input.rep1_part3.fq.gz && cd ../\n",
      "cat E13.5.input.rep1/E13.5.input.rep1_part*.fq.gz > E13.5.input.rep1/E13.5.input.rep1.fq.gz && rm E13.5.input.rep1/E13.5.input.rep1_part*.fq.gz\n",
      "mkdir E13.5.input.rep2 && cd E13.5.input.rep2\n",
      "ln -sf ../0_raw_data/SRR3651587.fastq.gz E13.5.input.rep2_part1.fq.gz\n",
      "ln -sf ../0_raw_data/SRR3651588.fastq.gz E13.5.input.rep2_part2.fq.gz && cd ../\n",
      "cat E13.5.input.rep2/E13.5.input.rep2_part*.fq.gz > E13.5.input.rep2/E13.5.input.rep2.fq.gz && rm E13.5.input.rep2/E13.5.input.rep2_part*.fq.gz\n",
      "mkdir E14.5.input.rep1 && cd E14.5.input.rep1\n",
      "ln -sf ../0_raw_data/SRR3652945.fastq.gz E14.5.input.rep1.fq.gz && cd ../\n",
      "mkdir E14.5.input.rep2 && cd E14.5.input.rep2\n",
      "ln -sf ../0_raw_data/SRR3652946.fastq.gz E14.5.input.rep2.fq.gz && cd ../\n",
      "mkdir E15.5.input.rep1 && cd E15.5.input.rep1\n",
      "ln -sf ../0_raw_data/SRR3651754.fastq.gz E15.5.input.rep1_part1.fq.gz\n",
      "ln -sf ../0_raw_data/SRR3651755.fastq.gz E15.5.input.rep1_part2.fq.gz && cd ../\n",
      "cat E15.5.input.rep1/E15.5.input.rep1_part*.fq.gz > E15.5.input.rep1/E15.5.input.rep1.fq.gz && rm E15.5.input.rep1/E15.5.input.rep1_part*.fq.gz\n",
      "mkdir E15.5.input.rep2 && cd E15.5.input.rep2\n",
      "ln -sf ../0_raw_data/SRR3651756.fastq.gz E15.5.input.rep2_part1.fq.gz\n",
      "ln -sf ../0_raw_data/SRR3651757.fastq.gz E15.5.input.rep2_part2.fq.gz && cd ../\n",
      "cat E15.5.input.rep2/E15.5.input.rep2_part*.fq.gz > E15.5.input.rep2/E15.5.input.rep2.fq.gz && rm E15.5.input.rep2/E15.5.input.rep2_part*.fq.gz\n",
      "mkdir E16.5.input.rep1 && cd E16.5.input.rep1\n",
      "ln -sf ../0_raw_data/SRR3651335.fastq.gz E16.5.input.rep1.fq.gz && cd ../\n",
      "mkdir E16.5.input.rep2 && cd E16.5.input.rep2\n",
      "ln -sf ../0_raw_data/SRR3651336.fastq.gz E16.5.input.rep2.fq.gz && cd ../\n",
      "mkdir P0.input.rep1 && cd P0.input.rep1\n",
      "ln -sf ../0_raw_data/SRR3652553.fastq.gz P0.input.rep1.fq.gz && cd ../\n",
      "mkdir P0.input.rep2 && cd P0.input.rep2\n",
      "ln -sf ../0_raw_data/SRR3652554.fastq.gz P0.input.rep2.fq.gz && cd ../\n",
      "mkdir E11.5.WGBS.rep1 && cd E11.5.WGBS.rep1\n",
      "ln -sf ../0_raw_data/SRR3651222.fastq.gz E11.5.WGBS.rep1.fq.gz && cd ../\n",
      "mkdir E11.5.WGBS.rep2 && cd E11.5.WGBS.rep2\n",
      "ln -sf ../0_raw_data/SRR3651223.fastq.gz E11.5.WGBS.rep2.fq.gz && cd ../\n",
      "mkdir E11.5.WGBS.rep3 && cd E11.5.WGBS.rep3\n",
      "ln -sf ../0_raw_data/SRR3651224.fastq.gz E11.5.WGBS.rep3.fq.gz && cd ../\n",
      "mkdir E11.5.WGBS.rep4 && cd E11.5.WGBS.rep4\n",
      "ln -sf ../0_raw_data/SRR3651225.fastq.gz E11.5.WGBS.rep4.fq.gz && cd ../\n",
      "mkdir E12.5.WGBS.rep1 && cd E12.5.WGBS.rep1\n",
      "ln -sf ../0_raw_data/SRR3651878.fastq.gz E12.5.WGBS.rep1.fq.gz && cd ../\n",
      "mkdir E12.5.WGBS.rep2 && cd E12.5.WGBS.rep2\n",
      "ln -sf ../0_raw_data/SRR3651879.fastq.gz E12.5.WGBS.rep2.fq.gz && cd ../\n",
      "mkdir E12.5.WGBS.rep3 && cd E12.5.WGBS.rep3\n",
      "ln -sf ../0_raw_data/SRR3651880.fastq.gz E12.5.WGBS.rep3.fq.gz && cd ../\n",
      "mkdir E12.5.WGBS.rep4 && cd E12.5.WGBS.rep4\n",
      "ln -sf ../0_raw_data/SRR3651881.fastq.gz E12.5.WGBS.rep4.fq.gz && cd ../\n",
      "mkdir E13.5.WGBS.rep1 && cd E13.5.WGBS.rep1\n",
      "ln -sf ../0_raw_data/SRR3652697.fastq.gz E13.5.WGBS.rep1.fq.gz && cd ../\n",
      "mkdir E13.5.WGBS.rep2 && cd E13.5.WGBS.rep2\n",
      "ln -sf ../0_raw_data/SRR3652698.fastq.gz E13.5.WGBS.rep2.fq.gz && cd ../\n",
      "mkdir E13.5.WGBS.rep3 && cd E13.5.WGBS.rep3\n",
      "ln -sf ../0_raw_data/SRR3652699.fastq.gz E13.5.WGBS.rep3.fq.gz && cd ../\n",
      "mkdir E13.5.WGBS.rep4 && cd E13.5.WGBS.rep4\n",
      "ln -sf ../0_raw_data/SRR3652700.fastq.gz E13.5.WGBS.rep4.fq.gz && cd ../\n",
      "mkdir E14.5.WGBS.rep1 && cd E14.5.WGBS.rep1\n",
      "ln -sf ../0_raw_data/SRR3653025.fastq.gz E14.5.WGBS.rep1.fq.gz && cd ../\n",
      "mkdir E14.5.WGBS.rep2 && cd E14.5.WGBS.rep2\n",
      "ln -sf ../0_raw_data/SRR3653026.fastq.gz E14.5.WGBS.rep2.fq.gz && cd ../\n",
      "mkdir E14.5.WGBS.rep3 && cd E14.5.WGBS.rep3\n",
      "ln -sf ../0_raw_data/SRR3653027.fastq.gz E14.5.WGBS.rep3.fq.gz && cd ../\n",
      "mkdir E14.5.WGBS.rep4 && cd E14.5.WGBS.rep4\n",
      "ln -sf ../0_raw_data/SRR3653028.fastq.gz E14.5.WGBS.rep4.fq.gz && cd ../\n",
      "mkdir E14.5.WGBS.rep5 && cd E14.5.WGBS.rep5\n",
      "ln -sf ../0_raw_data/SRR3653029.fastq.gz E14.5.WGBS.rep5.fq.gz && cd ../\n",
      "mkdir E14.5.WGBS.rep6 && cd E14.5.WGBS.rep6\n",
      "ln -sf ../0_raw_data/SRR3653030.fastq.gz E14.5.WGBS.rep6.fq.gz && cd ../\n",
      "mkdir E15.5.WGBS.rep1 && cd E15.5.WGBS.rep1\n",
      "ln -sf ../0_raw_data/SRR3651847.fastq.gz E15.5.WGBS.rep1.fq.gz && cd ../\n",
      "mkdir E15.5.WGBS.rep2 && cd E15.5.WGBS.rep2\n",
      "ln -sf ../0_raw_data/SRR3651848.fastq.gz E15.5.WGBS.rep2.fq.gz && cd ../\n",
      "mkdir E15.5.WGBS.rep3 && cd E15.5.WGBS.rep3\n",
      "ln -sf ../0_raw_data/SRR3651849.fastq.gz E15.5.WGBS.rep3.fq.gz && cd ../\n",
      "mkdir E15.5.WGBS.rep4 && cd E15.5.WGBS.rep4\n",
      "ln -sf ../0_raw_data/SRR3651850.fastq.gz E15.5.WGBS.rep4.fq.gz && cd ../\n",
      "mkdir E16.5.WGBS.rep1 && cd E16.5.WGBS.rep1\n",
      "ln -sf ../0_raw_data/SRR3651452.fastq.gz E16.5.WGBS.rep1.fq.gz && cd ../\n",
      "mkdir E16.5.WGBS.rep2 && cd E16.5.WGBS.rep2\n",
      "ln -sf ../0_raw_data/SRR3651453.fastq.gz E16.5.WGBS.rep2.fq.gz && cd ../\n",
      "mkdir E16.5.WGBS.rep3 && cd E16.5.WGBS.rep3\n",
      "ln -sf ../0_raw_data/SRR3651454.fastq.gz E16.5.WGBS.rep3.fq.gz && cd ../\n",
      "mkdir E16.5.WGBS.rep4 && cd E16.5.WGBS.rep4\n",
      "ln -sf ../0_raw_data/SRR3651455.fastq.gz E16.5.WGBS.rep4.fq.gz && cd ../\n",
      "mkdir P0.WGBS.rep1 && cd P0.WGBS.rep1\n",
      "ln -sf ../0_raw_data/SRR3652380.fastq.gz P0.WGBS.rep1.fq.gz && cd ../\n",
      "mkdir P0.WGBS.rep2 && cd P0.WGBS.rep2\n",
      "ln -sf ../0_raw_data/SRR3652381.fastq.gz P0.WGBS.rep2.fq.gz && cd ../\n",
      "mkdir P0.WGBS.rep3 && cd P0.WGBS.rep3\n",
      "ln -sf ../0_raw_data/SRR3652382.fastq.gz P0.WGBS.rep3.fq.gz && cd ../\n",
      "mkdir P0.WGBS.rep4 && cd P0.WGBS.rep4\n",
      "ln -sf ../0_raw_data/SRR3652383.fastq.gz P0.WGBS.rep4.fq.gz && cd ../\n"
     ]
    }
   ],
   "source": [
    "## generate soft link commands\n",
    "\n",
    "for tissue in ['heart', 'liver'] :\n",
    "    print(\"\\n\")\n",
    "    cmd1 = f'cd ~/CHMsInOtherContexts/CellStateTransition/{tissue[0].upper()}{tissue[1:]}Development'\n",
    "    print(cmd1)\n",
    "    for gsm in geo_sra_merge_dict[tissue]['GSMID'].unique().tolist() :\n",
    "        layout = geo_sra_merge_dict[tissue].loc[geo_sra_merge_dict[tissue]['GSMID'] == gsm]['LibraryLayout'].unique()[0]\n",
    "        srr_ls = geo_sra_merge_dict[tissue].loc[geo_sra_merge_dict[tissue]['GSMID'] == gsm]['Run'].tolist()\n",
    "        sampleN = geo_sra_merge_dict[tissue].loc[geo_sra_merge_dict[tissue]['GSMID'] == gsm]['sample_name'].tolist()[0]\n",
    "        cmd2 = f'mkdir {sampleN} && cd {sampleN}'\n",
    "        print(cmd2)\n",
    "        if len(srr_ls) == 1 :\n",
    "            if layout == 'SINGLE' :\n",
    "                cmd3 = f\"ln -sf ../0_raw_data/{srr_ls[0]}.fastq.gz {sampleN}.fq.gz && cd ../\"\n",
    "            else :\n",
    "                cmd3 = f\"ln -sf ../0_raw_data/{srr_ls[0]}_1.fastq.gz {sampleN}_R1.fq.gz && ln -sf ../0_raw_data/{srr_ls[0]}_2.fastq.gz {sampleN}_R2.fq.gz && cd ../\"\n",
    "            print(cmd3)\n",
    "        else :\n",
    "            for i,srr in enumerate(srr_ls) :\n",
    "                if layout == 'SINGLE' :\n",
    "                    cmd3 = f\"ln -sf ../0_raw_data/{srr}.fastq.gz {sampleN}_part{i+1}.fq.gz\"\n",
    "                else :\n",
    "                    cmd3 = f\"ln -sf ../0_raw_data/{srr}_1.fastq.gz {sampleN}_part{i+1}_R1.fq.gz && ln -sf ../0_raw_data/{srr}_2.fastq.gz {sampleN}_part{i+1}_R2.fq.gz\"\n",
    "                cmd3 = f\"{cmd3} && cd ../\" if i == len(srr_ls)-1 else cmd3\n",
    "                print(cmd3)\n",
    "                \n",
    "            if layout == 'SINGLE' :\n",
    "                cmd4 = f'cat {sampleN}/{sampleN}_part*.fq.gz > {sampleN}/{sampleN}.fq.gz && rm {sampleN}/{sampleN}_part*.fq.gz'\n",
    "            else :\n",
    "                cmd4 = f'cat {sampleN}/{sampleN}_part*_R1.fq.gz > {sampleN}/{sampleN}_R1.fq.gz && cat {sampleN}/{sampleN}_part*_R2.fq.gz > {sampleN}/{sampleN}_R2.fq.gz && rm {sampleN}/{sampleN}_part*_R[12].fq.gz'\n",
    "                      \n",
    "            print(cmd4)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 79,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Stage</th>\n",
       "      <th>Data</th>\n",
       "      <th>GSMID</th>\n",
       "      <th>GSEID</th>\n",
       "      <th>Run</th>\n",
       "      <th>Assay Type</th>\n",
       "      <th>LibraryLayout</th>\n",
       "      <th>SRA Study</th>\n",
       "      <th>GEO_Accession (exp)</th>\n",
       "      <th>Tissue</th>\n",
       "      <th>dev_stage</th>\n",
       "      <th>sample_name</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>44</th>\n",
       "      <td>E11.5 Liver</td>\n",
       "      <td>WGBS</td>\n",
       "      <td>GSM2190976</td>\n",
       "      <td>GSE82367</td>\n",
       "      <td>SRR3651222</td>\n",
       "      <td>Bisulfite-Seq</td>\n",
       "      <td>SINGLE</td>\n",
       "      <td>SRP013703</td>\n",
       "      <td>GSM2190976</td>\n",
       "      <td>liver</td>\n",
       "      <td>embryonic</td>\n",
       "      <td>E11.5.WGBS.rep1</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "          Stage  Data       GSMID     GSEID         Run     Assay Type  \\\n",
       "44  E11.5 Liver  WGBS  GSM2190976  GSE82367  SRR3651222  Bisulfite-Seq   \n",
       "\n",
       "   LibraryLayout  SRA Study GEO_Accession (exp) Tissue  dev_stage  \\\n",
       "44        SINGLE  SRP013703          GSM2190976  liver  embryonic   \n",
       "\n",
       "        sample_name  \n",
       "44  E11.5.WGBS.rep1  "
      ]
     },
     "execution_count": 79,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "geo_sra_merge_dict['liver'].loc[geo_sra_merge_dict[tissue]['Run'] == 'SRR3651222'] "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 115,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "\n",
      "WGBS_processing_publicData.sh ~/CHMsInOtherContexts/CellStateTransition/HeartDevelopment GEO GSM2701081 SINGLE E10.5.WGBS.rep1 mm10\n",
      "WGBS_processing_publicData.sh ~/CHMsInOtherContexts/CellStateTransition/HeartDevelopment GEO GSM2701082 SINGLE E10.5.WGBS.rep2 mm10\n",
      "WGBS_processing_publicData.sh ~/CHMsInOtherContexts/CellStateTransition/HeartDevelopment GEO GSM2701083 SINGLE E10.5.WGBS.rep3 mm10\n",
      "WGBS_processing_publicData.sh ~/CHMsInOtherContexts/CellStateTransition/HeartDevelopment GEO GSM2701084 SINGLE E10.5.WGBS.rep4 mm10\n",
      "ChIPseq_processing_publicData.sh ~/CHMsInOtherContexts/CellStateTransition/HeartDevelopment GEO GSM2308518 SINGLE E10.5.H3K9me3.rep1 mm10\n",
      "ChIPseq_processing_publicData.sh ~/CHMsInOtherContexts/CellStateTransition/HeartDevelopment GEO GSM2308519 SINGLE E10.5.H3K9me3.rep2 mm10\n",
      "ChIPseq_processing_publicData.sh ~/CHMsInOtherContexts/CellStateTransition/HeartDevelopment GEO GSM2308570 SINGLE E10.5.input.rep1 mm10\n",
      "ChIPseq_processing_publicData.sh ~/CHMsInOtherContexts/CellStateTransition/HeartDevelopment GEO GSM2308571 SINGLE E10.5.input.rep2 mm10\n",
      "WGBS_processing_publicData.sh ~/CHMsInOtherContexts/CellStateTransition/HeartDevelopment GEO GSM2192039 SINGLE E11.5.WGBS.rep1 mm10\n",
      "WGBS_processing_publicData.sh ~/CHMsInOtherContexts/CellStateTransition/HeartDevelopment GEO GSM2192040 SINGLE E11.5.WGBS.rep2 mm10\n",
      "WGBS_processing_publicData.sh ~/CHMsInOtherContexts/CellStateTransition/HeartDevelopment GEO GSM2192041 SINGLE E11.5.WGBS.rep3 mm10\n",
      "WGBS_processing_publicData.sh ~/CHMsInOtherContexts/CellStateTransition/HeartDevelopment GEO GSM2192042 SINGLE E11.5.WGBS.rep4 mm10\n",
      "ChIPseq_processing_publicData.sh ~/CHMsInOtherContexts/CellStateTransition/HeartDevelopment GEO GSM2191779 SINGLE E11.5.H3K9me3.rep1 mm10\n",
      "ChIPseq_processing_publicData.sh ~/CHMsInOtherContexts/CellStateTransition/HeartDevelopment GEO GSM2191780 SINGLE E11.5.H3K9me3.rep2 mm10\n",
      "ChIPseq_processing_publicData.sh ~/CHMsInOtherContexts/CellStateTransition/HeartDevelopment GEO GSM2191518 SINGLE E11.5.input.rep1 mm10\n",
      "ChIPseq_processing_publicData.sh ~/CHMsInOtherContexts/CellStateTransition/HeartDevelopment GEO GSM2191519 SINGLE E11.5.input.rep2 mm10\n",
      "WGBS_processing_publicData.sh ~/CHMsInOtherContexts/CellStateTransition/HeartDevelopment GEO GSM2191410 SINGLE E12.5.WGBS.rep1 mm10\n",
      "WGBS_processing_publicData.sh ~/CHMsInOtherContexts/CellStateTransition/HeartDevelopment GEO GSM2191411 SINGLE E12.5.WGBS.rep2 mm10\n",
      "WGBS_processing_publicData.sh ~/CHMsInOtherContexts/CellStateTransition/HeartDevelopment GEO GSM2191412 SINGLE E12.5.WGBS.rep3 mm10\n",
      "WGBS_processing_publicData.sh ~/CHMsInOtherContexts/CellStateTransition/HeartDevelopment GEO GSM2191413 SINGLE E12.5.WGBS.rep4 mm10\n",
      "ChIPseq_processing_publicData.sh ~/CHMsInOtherContexts/CellStateTransition/HeartDevelopment GEO GSM2191937 SINGLE E12.5.H3K9me3.rep1 mm10\n",
      "ChIPseq_processing_publicData.sh ~/CHMsInOtherContexts/CellStateTransition/HeartDevelopment GEO GSM2191938 SINGLE E12.5.H3K9me3.rep2 mm10\n",
      "ChIPseq_processing_publicData.sh ~/CHMsInOtherContexts/CellStateTransition/HeartDevelopment GEO GSM2192077 SINGLE E12.5.input.rep1 mm10\n",
      "ChIPseq_processing_publicData.sh ~/CHMsInOtherContexts/CellStateTransition/HeartDevelopment GEO GSM2192078 SINGLE E12.5.input.rep2 mm10\n",
      "WGBS_processing_publicData.sh ~/CHMsInOtherContexts/CellStateTransition/HeartDevelopment GEO GSM2191244 SINGLE E13.5.WGBS.rep1 mm10\n",
      "WGBS_processing_publicData.sh ~/CHMsInOtherContexts/CellStateTransition/HeartDevelopment GEO GSM2191245 SINGLE E13.5.WGBS.rep2 mm10\n",
      "WGBS_processing_publicData.sh ~/CHMsInOtherContexts/CellStateTransition/HeartDevelopment GEO GSM2191246 SINGLE E13.5.WGBS.rep3 mm10\n",
      "WGBS_processing_publicData.sh ~/CHMsInOtherContexts/CellStateTransition/HeartDevelopment GEO GSM2191247 SINGLE E13.5.WGBS.rep4 mm10\n",
      "ChIPseq_processing_publicData.sh ~/CHMsInOtherContexts/CellStateTransition/HeartDevelopment GEO GSM2191149 SINGLE E13.5.H3K9me3.rep1 mm10\n",
      "ChIPseq_processing_publicData.sh ~/CHMsInOtherContexts/CellStateTransition/HeartDevelopment GEO GSM2191151 SINGLE E13.5.H3K9me3.rep2 mm10\n",
      "ChIPseq_processing_publicData.sh ~/CHMsInOtherContexts/CellStateTransition/HeartDevelopment GEO GSM2191068 SINGLE E13.5.input.rep1 mm10\n",
      "ChIPseq_processing_publicData.sh ~/CHMsInOtherContexts/CellStateTransition/HeartDevelopment GEO GSM2191069 SINGLE E13.5.input.rep2 mm10\n",
      "WGBS_processing_publicData.sh ~/CHMsInOtherContexts/CellStateTransition/HeartDevelopment GEO GSM2192063 SINGLE E14.5.WGBS.rep1 mm10\n",
      "WGBS_processing_publicData.sh ~/CHMsInOtherContexts/CellStateTransition/HeartDevelopment GEO GSM2192064 SINGLE E14.5.WGBS.rep2 mm10\n",
      "WGBS_processing_publicData.sh ~/CHMsInOtherContexts/CellStateTransition/HeartDevelopment GEO GSM2192065 SINGLE E14.5.WGBS.rep3 mm10\n",
      "WGBS_processing_publicData.sh ~/CHMsInOtherContexts/CellStateTransition/HeartDevelopment GEO GSM2192066 SINGLE E14.5.WGBS.rep4 mm10\n",
      "ChIPseq_processing_publicData.sh ~/CHMsInOtherContexts/CellStateTransition/HeartDevelopment GEO GSM2190928 SINGLE E14.5.H3K9me3.rep1 mm10\n",
      "ChIPseq_processing_publicData.sh ~/CHMsInOtherContexts/CellStateTransition/HeartDevelopment GEO GSM2190929 SINGLE E14.5.H3K9me3.rep2 mm10\n",
      "ChIPseq_processing_publicData.sh ~/CHMsInOtherContexts/CellStateTransition/HeartDevelopment GEO GSM2192333 SINGLE E14.5.input.rep1 mm10\n",
      "ChIPseq_processing_publicData.sh ~/CHMsInOtherContexts/CellStateTransition/HeartDevelopment GEO GSM2192334 SINGLE E14.5.input.rep2 mm10\n",
      "WGBS_processing_publicData.sh ~/CHMsInOtherContexts/CellStateTransition/HeartDevelopment GEO GSM2191434 SINGLE E15.5.WGBS.rep1 mm10\n",
      "WGBS_processing_publicData.sh ~/CHMsInOtherContexts/CellStateTransition/HeartDevelopment GEO GSM2191435 SINGLE E15.5.WGBS.rep2 mm10\n",
      "WGBS_processing_publicData.sh ~/CHMsInOtherContexts/CellStateTransition/HeartDevelopment GEO GSM2191436 SINGLE E15.5.WGBS.rep3 mm10\n",
      "WGBS_processing_publicData.sh ~/CHMsInOtherContexts/CellStateTransition/HeartDevelopment GEO GSM2191437 SINGLE E15.5.WGBS.rep4 mm10\n",
      "ChIPseq_processing_publicData.sh ~/CHMsInOtherContexts/CellStateTransition/HeartDevelopment GEO GSM2191060 SINGLE E15.5.H3K9me3.rep1 mm10\n",
      "ChIPseq_processing_publicData.sh ~/CHMsInOtherContexts/CellStateTransition/HeartDevelopment GEO GSM2191061 SINGLE E15.5.H3K9me3.rep2 mm10\n",
      "ChIPseq_processing_publicData.sh ~/CHMsInOtherContexts/CellStateTransition/HeartDevelopment GEO GSM2191707 SINGLE E15.5.input.rep1 mm10\n",
      "ChIPseq_processing_publicData.sh ~/CHMsInOtherContexts/CellStateTransition/HeartDevelopment GEO GSM2191708 SINGLE E15.5.input.rep2 mm10\n",
      "WGBS_processing_publicData.sh ~/CHMsInOtherContexts/CellStateTransition/HeartDevelopment GEO GSM2191004 SINGLE E16.5.WGBS.rep1 mm10\n",
      "WGBS_processing_publicData.sh ~/CHMsInOtherContexts/CellStateTransition/HeartDevelopment GEO GSM2191005 SINGLE E16.5.WGBS.rep2 mm10\n",
      "WGBS_processing_publicData.sh ~/CHMsInOtherContexts/CellStateTransition/HeartDevelopment GEO GSM2191006 SINGLE E16.5.WGBS.rep3 mm10\n",
      "WGBS_processing_publicData.sh ~/CHMsInOtherContexts/CellStateTransition/HeartDevelopment GEO GSM2191007 SINGLE E16.5.WGBS.rep4 mm10\n",
      "ChIPseq_processing_publicData.sh ~/CHMsInOtherContexts/CellStateTransition/HeartDevelopment GEO GSM2191192 SINGLE E16.5.H3K9me3.rep1 mm10\n",
      "ChIPseq_processing_publicData.sh ~/CHMsInOtherContexts/CellStateTransition/HeartDevelopment GEO GSM2191193 SINGLE E16.5.H3K9me3.rep2 mm10\n",
      "ChIPseq_processing_publicData.sh ~/CHMsInOtherContexts/CellStateTransition/HeartDevelopment GEO GSM2192139 SINGLE E16.5.input.rep1 mm10\n",
      "ChIPseq_processing_publicData.sh ~/CHMsInOtherContexts/CellStateTransition/HeartDevelopment GEO GSM2192140 SINGLE E16.5.input.rep2 mm10\n",
      "WGBS_processing_publicData.sh ~/CHMsInOtherContexts/CellStateTransition/HeartDevelopment GEO GSM2191651 SINGLE P0.WGBS.rep1 mm10\n",
      "WGBS_processing_publicData.sh ~/CHMsInOtherContexts/CellStateTransition/HeartDevelopment GEO GSM2191652 SINGLE P0.WGBS.rep2 mm10\n",
      "WGBS_processing_publicData.sh ~/CHMsInOtherContexts/CellStateTransition/HeartDevelopment GEO GSM2191653 SINGLE P0.WGBS.rep3 mm10\n",
      "WGBS_processing_publicData.sh ~/CHMsInOtherContexts/CellStateTransition/HeartDevelopment GEO GSM2191654 SINGLE P0.WGBS.rep4 mm10\n",
      "ChIPseq_processing_publicData.sh ~/CHMsInOtherContexts/CellStateTransition/HeartDevelopment GEO GSM2192519 SINGLE P0.H3K9me3.rep1 mm10\n",
      "ChIPseq_processing_publicData.sh ~/CHMsInOtherContexts/CellStateTransition/HeartDevelopment GEO GSM2192520 SINGLE P0.H3K9me3.rep2 mm10\n",
      "ChIPseq_processing_publicData.sh ~/CHMsInOtherContexts/CellStateTransition/HeartDevelopment GEO GSM2191552 SINGLE P0.input.rep1 mm10\n",
      "ChIPseq_processing_publicData.sh ~/CHMsInOtherContexts/CellStateTransition/HeartDevelopment GEO GSM2191553 SINGLE P0.input.rep2 mm10\n",
      "\n",
      "\n",
      "ChIPseq_processing_publicData.sh ~/CHMsInOtherContexts/CellStateTransition/LiverDevelopment GEO GSM2192347 SINGLE E11.5.H3K9me3.rep1 mm10\n",
      "ChIPseq_processing_publicData.sh ~/CHMsInOtherContexts/CellStateTransition/LiverDevelopment GEO GSM2192348 SINGLE E11.5.H3K9me3.rep2 mm10\n",
      "ChIPseq_processing_publicData.sh ~/CHMsInOtherContexts/CellStateTransition/LiverDevelopment GEO GSM2191963 SINGLE E11.5.input.rep1 mm10\n",
      "ChIPseq_processing_publicData.sh ~/CHMsInOtherContexts/CellStateTransition/LiverDevelopment GEO GSM2191964 SINGLE E11.5.input.rep2 mm10\n",
      "WGBS_processing_publicData.sh ~/CHMsInOtherContexts/CellStateTransition/LiverDevelopment GEO GSM2190976 SINGLE E11.5.WGBS.rep1 mm10\n",
      "WGBS_processing_publicData.sh ~/CHMsInOtherContexts/CellStateTransition/LiverDevelopment GEO GSM2190977 SINGLE E11.5.WGBS.rep2 mm10\n",
      "WGBS_processing_publicData.sh ~/CHMsInOtherContexts/CellStateTransition/LiverDevelopment GEO GSM2190978 SINGLE E11.5.WGBS.rep3 mm10\n",
      "WGBS_processing_publicData.sh ~/CHMsInOtherContexts/CellStateTransition/LiverDevelopment GEO GSM2190979 SINGLE E11.5.WGBS.rep4 mm10\n",
      "ChIPseq_processing_publicData.sh ~/CHMsInOtherContexts/CellStateTransition/LiverDevelopment GEO GSM2191248 SINGLE E12.5.H3K9me3.rep1 mm10\n",
      "ChIPseq_processing_publicData.sh ~/CHMsInOtherContexts/CellStateTransition/LiverDevelopment GEO GSM2191249 SINGLE E12.5.H3K9me3.rep2 mm10\n",
      "ChIPseq_processing_publicData.sh ~/CHMsInOtherContexts/CellStateTransition/LiverDevelopment GEO GSM2192133 SINGLE E12.5.input.rep1 mm10\n",
      "ChIPseq_processing_publicData.sh ~/CHMsInOtherContexts/CellStateTransition/LiverDevelopment GEO GSM2192134 SINGLE E12.5.input.rep2 mm10\n",
      "WGBS_processing_publicData.sh ~/CHMsInOtherContexts/CellStateTransition/LiverDevelopment GEO GSM2191548 SINGLE E12.5.WGBS.rep1 mm10\n",
      "WGBS_processing_publicData.sh ~/CHMsInOtherContexts/CellStateTransition/LiverDevelopment GEO GSM2191549 SINGLE E12.5.WGBS.rep2 mm10\n",
      "WGBS_processing_publicData.sh ~/CHMsInOtherContexts/CellStateTransition/LiverDevelopment GEO GSM2191550 SINGLE E12.5.WGBS.rep3 mm10\n",
      "WGBS_processing_publicData.sh ~/CHMsInOtherContexts/CellStateTransition/LiverDevelopment GEO GSM2191551 SINGLE E12.5.WGBS.rep4 mm10\n",
      "ChIPseq_processing_publicData.sh ~/CHMsInOtherContexts/CellStateTransition/LiverDevelopment GEO GSM2191298 SINGLE E13.5.H3K9me3.rep1 mm10\n",
      "ChIPseq_processing_publicData.sh ~/CHMsInOtherContexts/CellStateTransition/LiverDevelopment GEO GSM2191299 SINGLE E13.5.H3K9me3.rep2 mm10\n",
      "ChIPseq_processing_publicData.sh ~/CHMsInOtherContexts/CellStateTransition/LiverDevelopment GEO GSM2191328 SINGLE E13.5.input.rep1 mm10\n",
      "ChIPseq_processing_publicData.sh ~/CHMsInOtherContexts/CellStateTransition/LiverDevelopment GEO GSM2191329 SINGLE E13.5.input.rep2 mm10\n",
      "WGBS_processing_publicData.sh ~/CHMsInOtherContexts/CellStateTransition/LiverDevelopment GEO GSM2192103 SINGLE E13.5.WGBS.rep1 mm10\n",
      "WGBS_processing_publicData.sh ~/CHMsInOtherContexts/CellStateTransition/LiverDevelopment GEO GSM2192104 SINGLE E13.5.WGBS.rep2 mm10\n",
      "WGBS_processing_publicData.sh ~/CHMsInOtherContexts/CellStateTransition/LiverDevelopment GEO GSM2192105 SINGLE E13.5.WGBS.rep3 mm10\n",
      "WGBS_processing_publicData.sh ~/CHMsInOtherContexts/CellStateTransition/LiverDevelopment GEO GSM2192106 SINGLE E13.5.WGBS.rep4 mm10\n",
      "ChIPseq_processing_publicData.sh ~/CHMsInOtherContexts/CellStateTransition/LiverDevelopment GEO GSM2191198 SINGLE E14.5.H3K9me3.rep1 mm10\n",
      "ChIPseq_processing_publicData.sh ~/CHMsInOtherContexts/CellStateTransition/LiverDevelopment GEO GSM2191199 SINGLE E14.5.H3K9me3.rep2 mm10\n",
      "ChIPseq_processing_publicData.sh ~/CHMsInOtherContexts/CellStateTransition/LiverDevelopment GEO GSM2192213 SINGLE E14.5.input.rep1 mm10\n",
      "ChIPseq_processing_publicData.sh ~/CHMsInOtherContexts/CellStateTransition/LiverDevelopment GEO GSM2192214 SINGLE E14.5.input.rep2 mm10\n",
      "WGBS_processing_publicData.sh ~/CHMsInOtherContexts/CellStateTransition/LiverDevelopment GEO GSM2192279 SINGLE E14.5.WGBS.rep1 mm10\n",
      "WGBS_processing_publicData.sh ~/CHMsInOtherContexts/CellStateTransition/LiverDevelopment GEO GSM2192280 SINGLE E14.5.WGBS.rep2 mm10\n",
      "WGBS_processing_publicData.sh ~/CHMsInOtherContexts/CellStateTransition/LiverDevelopment GEO GSM2192281 SINGLE E14.5.WGBS.rep3 mm10\n",
      "WGBS_processing_publicData.sh ~/CHMsInOtherContexts/CellStateTransition/LiverDevelopment GEO GSM2192282 SINGLE E14.5.WGBS.rep4 mm10\n",
      "WGBS_processing_publicData.sh ~/CHMsInOtherContexts/CellStateTransition/LiverDevelopment GEO GSM2192283 SINGLE E14.5.WGBS.rep5 mm10\n",
      "WGBS_processing_publicData.sh ~/CHMsInOtherContexts/CellStateTransition/LiverDevelopment GEO GSM2192284 SINGLE E14.5.WGBS.rep6 mm10\n",
      "ChIPseq_processing_publicData.sh ~/CHMsInOtherContexts/CellStateTransition/LiverDevelopment GEO GSM2192353 SINGLE E15.5.H3K9me3.rep1 mm10\n",
      "ChIPseq_processing_publicData.sh ~/CHMsInOtherContexts/CellStateTransition/LiverDevelopment GEO GSM2192354 SINGLE E15.5.H3K9me3.rep2 mm10\n",
      "ChIPseq_processing_publicData.sh ~/CHMsInOtherContexts/CellStateTransition/LiverDevelopment GEO GSM2191450 SINGLE E15.5.input.rep1 mm10\n",
      "ChIPseq_processing_publicData.sh ~/CHMsInOtherContexts/CellStateTransition/LiverDevelopment GEO GSM2191451 SINGLE E15.5.input.rep2 mm10\n",
      "WGBS_processing_publicData.sh ~/CHMsInOtherContexts/CellStateTransition/LiverDevelopment GEO GSM2191530 SINGLE E15.5.WGBS.rep1 mm10\n",
      "WGBS_processing_publicData.sh ~/CHMsInOtherContexts/CellStateTransition/LiverDevelopment GEO GSM2191531 SINGLE E15.5.WGBS.rep2 mm10\n",
      "WGBS_processing_publicData.sh ~/CHMsInOtherContexts/CellStateTransition/LiverDevelopment GEO GSM2191532 SINGLE E15.5.WGBS.rep3 mm10\n",
      "WGBS_processing_publicData.sh ~/CHMsInOtherContexts/CellStateTransition/LiverDevelopment GEO GSM2191533 SINGLE E15.5.WGBS.rep4 mm10\n",
      "ChIPseq_processing_publicData.sh ~/CHMsInOtherContexts/CellStateTransition/LiverDevelopment GEO GSM2191141 SINGLE E16.5.H3K9me3.rep1 mm10\n",
      "ChIPseq_processing_publicData.sh ~/CHMsInOtherContexts/CellStateTransition/LiverDevelopment GEO GSM2191142 SINGLE E16.5.H3K9me3.rep2 mm10\n",
      "ChIPseq_processing_publicData.sh ~/CHMsInOtherContexts/CellStateTransition/LiverDevelopment GEO GSM2191074 SINGLE E16.5.input.rep1 mm10\n",
      "ChIPseq_processing_publicData.sh ~/CHMsInOtherContexts/CellStateTransition/LiverDevelopment GEO GSM2191075 SINGLE E16.5.input.rep2 mm10\n",
      "WGBS_processing_publicData.sh ~/CHMsInOtherContexts/CellStateTransition/LiverDevelopment GEO GSM2191212 SINGLE E16.5.WGBS.rep1 mm10\n",
      "WGBS_processing_publicData.sh ~/CHMsInOtherContexts/CellStateTransition/LiverDevelopment GEO GSM2191213 SINGLE E16.5.WGBS.rep2 mm10\n",
      "WGBS_processing_publicData.sh ~/CHMsInOtherContexts/CellStateTransition/LiverDevelopment GEO GSM2191214 SINGLE E16.5.WGBS.rep3 mm10\n",
      "WGBS_processing_publicData.sh ~/CHMsInOtherContexts/CellStateTransition/LiverDevelopment GEO GSM2191215 SINGLE E16.5.WGBS.rep4 mm10\n",
      "ChIPseq_processing_publicData.sh ~/CHMsInOtherContexts/CellStateTransition/LiverDevelopment GEO GSM2191568 SINGLE P0.H3K9me3.rep1 mm10\n",
      "ChIPseq_processing_publicData.sh ~/CHMsInOtherContexts/CellStateTransition/LiverDevelopment GEO GSM2191569 SINGLE P0.H3K9me3.rep2 mm10\n",
      "ChIPseq_processing_publicData.sh ~/CHMsInOtherContexts/CellStateTransition/LiverDevelopment GEO GSM2192021 SINGLE P0.input.rep1 mm10\n",
      "ChIPseq_processing_publicData.sh ~/CHMsInOtherContexts/CellStateTransition/LiverDevelopment GEO GSM2192022 SINGLE P0.input.rep2 mm10\n",
      "WGBS_processing_publicData.sh ~/CHMsInOtherContexts/CellStateTransition/LiverDevelopment GEO GSM2191921 SINGLE P0.WGBS.rep1 mm10\n",
      "WGBS_processing_publicData.sh ~/CHMsInOtherContexts/CellStateTransition/LiverDevelopment GEO GSM2191922 SINGLE P0.WGBS.rep2 mm10\n",
      "WGBS_processing_publicData.sh ~/CHMsInOtherContexts/CellStateTransition/LiverDevelopment GEO GSM2191923 SINGLE P0.WGBS.rep3 mm10\n",
      "WGBS_processing_publicData.sh ~/CHMsInOtherContexts/CellStateTransition/LiverDevelopment GEO GSM2191924 SINGLE P0.WGBS.rep4 mm10\n"
     ]
    }
   ],
   "source": [
    "for tissue in ['heart', 'liver'] :\n",
    "    print(\"\\n\")\n",
    "    for gsm in sample_name_dict[tissue].keys() :\n",
    "        sample_name = sample_name_dict[tissue][gsm]\n",
    "        datatype = sample_name.split('.')[-2]\n",
    "        if datatype == 'WGBS' :\n",
    "            cmd5 = f'WGBS_processing_publicData.sh ~/CHMsInOtherContexts/CellStateTransition/{tissue[0].upper()}{tissue[1:]}Development GEO {gsm} SINGLE {sample_name} mm10'\n",
    "            print(cmd5)\n",
    "        elif datatype == 'H3K9me3' or datatype == 'input' :\n",
    "            cmd5 = f'ChIPseq_processing_publicData.sh ~/CHMsInOtherContexts/CellStateTransition/{tissue[0].upper()}{tissue[1:]}Development GEO {gsm} SINGLE {sample_name} mm10'\n",
    "            print(cmd5)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python [conda env:root] *",
   "language": "python",
   "name": "conda-root-py"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.8.5"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 4
}
